{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# ABC model over GM12878 with same accessions as for BENGI benchmarks, but starting from ENCODE ccRE-dELS as candidate enhancers."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## How to use this notebook?\n",
    "First, make a copy of [this notebook](http://genoweb.toulouse.inra.fr/~thoellinger/stable/ipynb/ABC/BENGI_GM12878_from_ccRE_ELSs.ipynb) on your computer / cluster.\n",
    "\n",
    "Then, to use this notebook, one should only have to carefully modify the content of the \"Set working directory\" section, then to execute the notebook cell by cell, in the correct order. After execution of each cell, remember to check for errors before executing the next one."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "One can use the following to switch notebook theme:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# night theme\n",
    "#!jt -t monokai -f fira -fs 10 -nf ptsans -nfs 11 -N -kl -cursw 2 -cursc r -cellw 95% -T"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# standard theme\n",
    "#!jt -r"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Import required librairies"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os.path #Or see https://stackoverflow.com/a/82852 for an object-oriented approach\n",
    "from IPython.core.magic import register_line_cell_magic"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "@register_line_cell_magic\n",
    "def writetemplate(line, cell):\n",
    "    with open(line, 'w') as f:\n",
    "        f.write(cell.format(**globals()))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Set working directory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "mail_user = \"tristan.hoellinger@inserm.fr\" # slurm notifications\n",
    "version = \"hg19\" # used only in the current cell to compute some paths \n",
    "cell_type = \"gm12878\" # used all along the notebook\n",
    "specie = \"homo_sapiens/\"+version+\"/\" # used only in the current cell to compute some paths \n",
    "\n",
    "# Where to store run-specific references, scripts, intermediate files, predictions, etc\n",
    "work_dir = \"/work2/project/regenet/workspace/thoellinger/ABC-Enhancer-Gene-Prediction/BENGI_GM12878_from_ccRE_ELSs/\"\n",
    "\n",
    "scripts = work_dir+\"hi_slurm/\" # where to store scripts specific to this run\n",
    "results_dir = \"/work2/project/regenet/results/\" # used to compute paths where data for this cell type is\n",
    "                                                # stored / will be downloaded\n",
    "dnase_dir = results_dir+\"dnaseseq/\"+specie+cell_type+'/'\n",
    "chipseq_dir = results_dir+\"chipseq/h3k27ac/\"+specie+cell_type+'/'\n",
    "expression_dir = results_dir+\"rnaseq/\"+specie+cell_type+'/'\n",
    "blacklist_dir = \"/work2/project/regenet/data/species/\"+specie # where the ENCODE blacklist is (going to be) stored\n",
    "\n",
    "blacklist = blacklist_dir+version+\"-blacklist.v2.bed.gz\" # need not exist yet\n",
    "\n",
    "# Gene annotation in the gtf format and gene name / gene id table (1st column id, 2nd column name)\n",
    "gene_annotation = \"/work2/project/fragencode/data/species/homo_sapiens/hg19.gencv19/homo_sapiens.gtf\"\n",
    "gnid_gname = \"/work2/project/fragencode/data/species/homo_sapiens/hg19.gencv19/homo_sapiens.gnid.gnname.tsv\"\n",
    "\n",
    "# Accessions for the current run. Just put accession numbers here, no need to download these accessions \"by hand\".\n",
    "dnase_rep1 = \"ENCFF664VNB\" # exp ENCSR000COQ\n",
    "dnase_rep2 = \"ENCFF790WER\" # exp ENCSR000COQ\n",
    "h3k27_rep1 = \"ENCFF197QHX\" # exp ENCSR000AKC\n",
    "h3k27_rep2 = \"ENCFF882PRP\" # exp ENCSR000AKC\n",
    "rnaseq_rep1 = \"ENCFF587ALK\" # exp ENCSR000COQ\n",
    "rnaseq_rep2 = \"ENCFF766CPS\" # exp ENCSR000COQ\n",
    "\n",
    "# Where to download the blacklist. Will not be used unless blacklist not found yet.\n",
    "blacklist_link = \"https://github.com/Boyle-Lab/Blacklist/raw/master/lists/hg19-blacklist.v2.bed.gz\"\n",
    "\n",
    "dnase_file_rep1 = dnase_dir+dnase_rep1+\".bam\" # do not change\n",
    "dnase_file_rep2 = dnase_dir+dnase_rep2+\".bam\"  # do not change\n",
    "h3k27_file_rep1 = chipseq_dir+h3k27_rep1+\".bam\"  # do not change\n",
    "h3k27_file_rep2 = chipseq_dir+h3k27_rep2+\".bam\"  # do not change\n",
    "rnaseq_file_rep1 = expression_dir+rnaseq_rep1+\".tsv\"  # do not change\n",
    "rnaseq_file_rep2 = expression_dir+rnaseq_rep2+\".tsv\"  # do not change\n",
    "\n",
    "reference_dir = work_dir+\"reference/\" # do not change\n",
    "annotations_dir = reference_dir+\"gene_annotation/\" # do not change\n",
    "peaks = work_dir+\"ABC_output/Peaks/\" # do not change\n",
    "neighborhoods = work_dir+\"ABC_output/Neighborhoods/\" # do not change\n",
    "predictions = work_dir+\"ABC_output/Predictions/\" # do not change\n",
    "\n",
    "light_annotation = annotations_dir+\"gene_ids.bed\"\n",
    "\n",
    "# Ubiquitously expressed genes (gene names). If provided, `ubiquitously_expressed_genes` will be\n",
    "# automatically generated with gene ids instead of gene names. If not (left empty), \n",
    "# `ubiquitously_expressed_genes` must be provided.\n",
    "ubiquitous_gene_names = work_dir+\"../reference/UbiquitouslyExpressedGenesHG19.txt\"\n",
    "# Ubiquitously expressed genes (gene ids). Please provide either `ubiquitously_expressed_genes` or\n",
    "# `ubiquitous_gene_names`\n",
    "ubiquitously_expressed_genes = reference_dir+\"UbiquitouslyExpressedGenes_ids.txt\" # need not exist\n",
    "\n",
    "compute_mean_expression = work_dir+\"../compute_mean_expression.awk\"\n",
    "# compute_mean_expression.awk computes an expression table (<gene id> <expression>) taking the mean\n",
    "# TPM expression found in the 2 input polyA+ RNA-seq tsv.\n",
    "\n",
    "make_candidate_regions = work_dir+\"../src/makeCandidateRegions.py\"\n",
    "run_neighborhoods = work_dir+\"../src/run.neighborhoods.py\"\n",
    "predict = work_dir+\"../src/predict.py\"\n",
    "\n",
    "genome_file = reference_dir+\"chr_sizes\" # shall not exist yet\n",
    "\n",
    "gene_expression_table = reference_dir+\"expression/\"+cell_type+\".\"+rnaseq_rep1+\"_\"+rnaseq_rep2+\".mean.TPM.txt\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "if not os.path.isdir(reference_dir):\n",
    "    !mkdir $reference_dir\n",
    "if not os.path.isdir(scripts):\n",
    "    !mkdir $scripts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# One may only change what is between quotes here\n",
    "slurm_2 = scripts+\"step2.sh\"\n",
    "slurm_3 = scripts+\"step3.sh\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "ccREs_dir = results_dir+\"ccREs/\"+specie+\"V1/\"+cell_type+'/'\n",
    "\n",
    "ccREs_accession = \"ENCFF028SGJ\"\n",
    "TSS_list = \"/work2/project/regenet/workspace/thoellinger/BENGI/Benchmark/Annotations/GENCODEv19-TSSs.bed\"\n",
    "\n",
    "ccREs = ccREs_dir+ccREs_accession+\".bed\" # need not exist yet\n",
    "ccREs_ELS = ccREs_dir+cell_type+\"_ccRE_ELSs.bed\" # need not exist yet\n",
    "ccREs_dELS = ccREs_dir+cell_type+\"_ccRE_dELSs.bed\" # need not exist yet\n",
    "\n",
    "candidateRegions = reference_dir+\"candidateRegions.bed\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Data acquisition"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Chromatin accessibility (DNase-seq)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "if not os.path.isfile(dnase_file_rep1):\n",
    "    !wget https://www.encodeproject.org/files/$dnase_rep1/@@download/$dnase_rep1$\".bam\" -P $dnase_dir\n",
    "            \n",
    "if not os.path.isfile(dnase_file_rep2):\n",
    "    !wget https://www.encodeproject.org/files/$dnase_rep2/@@download/$dnase_rep2$\".bam\" -P $dnase_dir"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Blacklist"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "if not os.path.isfile(blacklist):\n",
    "    !wget $blacklist_link -P $blacklist_dir"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Histone mark H3K27ac ChIP-seq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "if not os.path.isfile(h3k27_file_rep1):\n",
    "    !wget https://www.encodeproject.org/files/$h3k27_rep1/@@download/$h3k27_rep1$\".bam\" -P $chipseq_dir\n",
    "            \n",
    "if not os.path.isfile(h3k27_file_rep2):\n",
    "    !wget https://www.encodeproject.org/files/$h3k27_rep2/@@download/$h3k27_rep2$\".bam\" -P $chipseq_dir"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Gene expression (polyA+ RNA-seq)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "if not os.path.isfile(rnaseq_file_rep1):\n",
    "    !wget https://www.encodeproject.org/files/$rnaseq_rep1/@@download/$rnaseq_rep1$\".bam\" -P $expression_dir\n",
    "            \n",
    "if not os.path.isfile(rnaseq_file_rep2):\n",
    "    !wget https://www.encodeproject.org/files/$rnaseq_rep2/@@download/$rnaseq_rep2$\".bam\" -P $expression_dir"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Ubiquitously expressed genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash -s \"$gnid_gname\" \"$ubiquitous_gene_names\" \"$ubiquitously_expressed_genes\"\n",
    "if [[ -f $2 ]] && [[ ! -f $3 ]]; then\n",
    "    awk 'BEGIN{sep=\"\\t\"} NR==FNR {id[$2]=$1; next} id[$1] {print id[$1]}' $1 $2 > $3\n",
    "fi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Genome file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash -s \"$work_dir\" \"$dnase_file_rep1\" \"$genome_file\"\n",
    "CONDA_BASE=$(conda info --base) && source $CONDA_BASE/etc/profile.d/conda.sh\n",
    "conda activate base && module load bioinfo/samtools-1.9\n",
    "samtools view -H $2 | grep SQ | cut -f 2,3 |cut -c 4- |awk 'BEGIN{FS=\"\\t\"} {split($2,locus,\":\"); if($1 ~ /^(chr)([0-9]{1,2}$)|(M$)|(X$)|(Y$)/){print($1\"\\t\"locus[2])}}' > $3\n",
    "awk '{print $1\"\\t\"0\"\\t\"$2}' $3 > $3\".bed\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### ccREs\n",
    "As candidate elements, we use ENCODE ccREs with distal enhancer-like signature (ccRE-dELS) for GM12878. Hence we  need to download ccREs with annotations for this cell type."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2021-01-07 12:19:58--  https://www.encodeproject.org/files/ENCFF028SGJ/@@download/ENCFF028SGJ.bed.gz\n",
      "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
      "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
      "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
      "Location: https://encode-public.s3.amazonaws.com/2017/08/21/f281e5ef-2e5d-4552-9716-a41bccbeea6a/ENCFF028SGJ.bed.gz?response-content-disposition=attachment%3B%20filename%3DENCFF028SGJ.bed.gz&AWSAccessKeyId=ASIATGZNGCNX23M77F6O&Signature=bBXj4VILUYZvYSml8FbwUAfdGVA%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEHsaCXVzLXdlc3QtMiJGMEQCIHPL2nO9fZlB01OMTFyBPq%2BAWvq9wuG%2B%2BWatPxVfheciAiBbFLXoWBz3E84MZvcSuuRNR2C5L2PbrSNg9n002iknOiq0AwhEEAAaDDIyMDc0ODcxNDg2MyIM0uGls%2BBaw6Z%2Fs4w9KpEDpEB18RAanxxcpOfloqMsXPBOAamsg32N6%2Borzy2nmXI9a3OlLccK35a4dzftO%2B4u7qQXw%2BmCOZgYBBnrjpZseQ6tX5XP3duXjuEmhOqY2VgMMR1M%2BmSJ5LILHNNMwiqj2i7Ld9oleRg9Mg9xPNzHGyJOWySoh8HJIVAB4nV1ZQ6V6Ky0uwu8rD185ahx%2BU%2BEUGD8mE8%2Bab8LD3KKfrmRh0InbFRnaNPv4ZhkF3dFD0LpMqqK9runudgZYT1JAHMZVKJ7x8PeoYpc6gGUFucaxcTUI0psdF9MP92%2FSAQob9VDnIV9mbOqk0wgqTOQIHTKs7b5VtLz%2FppBwBCfer1L1JvIbH4TA30ayx1J3L5jIRIbAiyX7p8X%2B1ekCnGYCi8ka57eljutwxv49ng3MgYuYU7gN%2B63dXUrZsmgxo7E1O9g9XO2Ha6vhBxKm5uwARO93kWKtbfdbpY6MtfmuhXHygE2sujmzPO5pg2MXfaAggW2gWjjOu5bEoj78YEiGbix%2FO1S6kimEpUY7AL4HrdqqM4wk9bb%2FwU67AH1DT3H7Kyftp1sKFWXQrlZj%2BtfeyKdtYSguln6VCf8jaUIyLq%2BNUmnqV7RstrbKDg7b3KWGTzyMWxO%2BNdv1CQxo8D7VmimQ58xganSXQXWT8Z7kLnPxzvnxhmyNCx3YdwFo2VPPiPINAZ%2Bh1EMqB6cAfGxsh03q3abmLGrWETQsdKul3NpFN6YhxH7Lp37Y0VsLHiwMc5%2BLzor83OmR5hfOG4NdbVElmyscq1mad5CuAcCFa%2F2eScFnMswZjxLW1DIFRGK219x33Brx2Ex8YQay8gnBTsvBXb%2BE04akm1txRiM5INk71wxSCgV0A%3D%3D&Expires=1610147999 [following]\n",
      "--2021-01-07 12:19:59--  https://encode-public.s3.amazonaws.com/2017/08/21/f281e5ef-2e5d-4552-9716-a41bccbeea6a/ENCFF028SGJ.bed.gz?response-content-disposition=attachment%3B%20filename%3DENCFF028SGJ.bed.gz&AWSAccessKeyId=ASIATGZNGCNX23M77F6O&Signature=bBXj4VILUYZvYSml8FbwUAfdGVA%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEHsaCXVzLXdlc3QtMiJGMEQCIHPL2nO9fZlB01OMTFyBPq%2BAWvq9wuG%2B%2BWatPxVfheciAiBbFLXoWBz3E84MZvcSuuRNR2C5L2PbrSNg9n002iknOiq0AwhEEAAaDDIyMDc0ODcxNDg2MyIM0uGls%2BBaw6Z%2Fs4w9KpEDpEB18RAanxxcpOfloqMsXPBOAamsg32N6%2Borzy2nmXI9a3OlLccK35a4dzftO%2B4u7qQXw%2BmCOZgYBBnrjpZseQ6tX5XP3duXjuEmhOqY2VgMMR1M%2BmSJ5LILHNNMwiqj2i7Ld9oleRg9Mg9xPNzHGyJOWySoh8HJIVAB4nV1ZQ6V6Ky0uwu8rD185ahx%2BU%2BEUGD8mE8%2Bab8LD3KKfrmRh0InbFRnaNPv4ZhkF3dFD0LpMqqK9runudgZYT1JAHMZVKJ7x8PeoYpc6gGUFucaxcTUI0psdF9MP92%2FSAQob9VDnIV9mbOqk0wgqTOQIHTKs7b5VtLz%2FppBwBCfer1L1JvIbH4TA30ayx1J3L5jIRIbAiyX7p8X%2B1ekCnGYCi8ka57eljutwxv49ng3MgYuYU7gN%2B63dXUrZsmgxo7E1O9g9XO2Ha6vhBxKm5uwARO93kWKtbfdbpY6MtfmuhXHygE2sujmzPO5pg2MXfaAggW2gWjjOu5bEoj78YEiGbix%2FO1S6kimEpUY7AL4HrdqqM4wk9bb%2FwU67AH1DT3H7Kyftp1sKFWXQrlZj%2BtfeyKdtYSguln6VCf8jaUIyLq%2BNUmnqV7RstrbKDg7b3KWGTzyMWxO%2BNdv1CQxo8D7VmimQ58xganSXQXWT8Z7kLnPxzvnxhmyNCx3YdwFo2VPPiPINAZ%2Bh1EMqB6cAfGxsh03q3abmLGrWETQsdKul3NpFN6YhxH7Lp37Y0VsLHiwMc5%2BLzor83OmR5hfOG4NdbVElmyscq1mad5CuAcCFa%2F2eScFnMswZjxLW1DIFRGK219x33Brx2Ex8YQay8gnBTsvBXb%2BE04akm1txRiM5INk71wxSCgV0A%3D%3D&Expires=1610147999\n",
      "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.218.241.178\n",
      "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.218.241.178|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 17995639 (17M) [binary/octet-stream]\n",
      "Saving to: ‘/work2/project/regenet/results/ccREs/homo_sapiens/hg19/V1/gm12878/ENCFF028SGJ.bed.gz’\n",
      "\n",
      "100%[======================================>] 17 995 639  3,35MB/s   in 15s    \n",
      "\n",
      "2021-01-07 12:20:14 (1,17 MB/s) - ‘/work2/project/regenet/results/ccREs/homo_sapiens/hg19/V1/gm12878/ENCFF028SGJ.bed.gz’ saved [17995639/17995639]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "if not os.path.isfile(ccREs+\".gz\"):\n",
    "    !wget https://www.encodeproject.org/files/$ccREs_accession/@@download/$ccREs_accession$\".bed.gz\" -P $ccREs_dir\n",
    "        \n",
    "if not os.path.isfile(ccREs):\n",
    "    !gzip -cd $ccREs\".gz\" > $ccREs"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Data reprocessing\n",
    "\n",
    "### Create expression table\n",
    "\n",
    "We use `compute_mean_expression.awk` to take as the reference TPM value, the mean value for the 2 replicates."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash -s \"$compute_mean_expression\" \"$reference_dir\" \"$rnaseq_file_rep1\" \"$rnaseq_file_rep2\" \"$cell_type\" \"$rnaseq_rep1\" \"$rnaseq_rep2\"\n",
    "if [[ ! -d $2\"expression/\" ]]; then mkdir $2\"expression/\"; fi\n",
    "awk -f $1 $3 $4 > $2\"expression/\"$5\".\"$6\"_\"$7\".mean.TPM.txt\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Filter gene annotation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash -s \"$work_dir\" \"$genome_file\" \"$annotations_dir\" \"$gene_annotation\" \"$light_annotation\"\n",
    "if [[ ! -d $3 ]]; then mkdir $3; fi\n",
    "CONDA_BASE=$(conda info --base) && source $CONDA_BASE/etc/profile.d/conda.sh\n",
    "conda activate base && module load bioinfo/bedtools-2.27.1\n",
    "awk '{if(NR==FNR){!genome[$1]++; next}; if(genome[$1]){if($3 ~ /(^gene$)/){print $1\"\\t\"$4\"\\t\"$5\"\\t\"substr($10,2,length($10)-3)\"\\t\"0\"\\t\"$7}}}' $2 $4 | bedtools sort -g $2 -i stdin > $5"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Running the ABC model"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Step 1: define candidate elements\n",
    "\n",
    "As candidate elements, we use ENCODE ccREs with distal enhancer-like signature (ccRE-dELS) for GM12878. First we extract only ccRE-ELS."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash -s \"$ccREs\" \"$ccREs_ELS\"\n",
    "if [[ ! -f $2 ]]; then\n",
    "    awk -v OFS='\\t' '$9 ~ /(^255,205,0)/ {print $1, $2, $3, \".\", $4, \"Enhancer-like\"}' $1 > $2\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "srun: job 22240197 queued and waiting for resources\n",
      "srun: job 22240197 has been allocated resources\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$ccREs_ELS\" \"$genome_file\"\n",
    "if [[ ! -f $1\".sorted\" ]]\n",
    "then\n",
    "    export V1=$1 V2=$2\n",
    "    CONDA_BASE=$(conda info --base) && source $CONDA_BASE/etc/profile.d/conda.sh\n",
    "    conda activate base && conda activate abcmodel\n",
    "    module load bioinfo/bedtools-2.27.1\n",
    "    srun bash\n",
    "    bedtools sort -faidx $V2 -i $V1 > $V1\".sorted\"\n",
    "fi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Then using `bedtools closest` we keep only distal ccRE-ELS (ie greater than 2 kb from an annotated TSS, GENCODE v19)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "srun: job 22240200 queued and waiting for resources\n",
      "srun: job 22240200 has been allocated resources\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$genome_file\" \"$ccREs_ELS\" \"$TSS_list\" \"$ccREs_dELS\" \"$genome_file\"\n",
    "if [[ ! -f $4 ]]\n",
    "then\n",
    "    export V1=$1 V2=$2 V3=$3 V4=$4 V5=$5\n",
    "    CONDA_BASE=$(conda info --base) && source $CONDA_BASE/etc/profile.d/conda.sh\n",
    "    conda activate base && conda activate abcmodel\n",
    "    module load bioinfo/bedtools-2.27.1\n",
    "    srun bash\n",
    "    awk '{if(NR==FNR){chr[$1]++; next}; if(chr[$1]){print $0}}' $V5 $V3 | bedtools sort -faidx $V5 -i stdin | bedtools closest -d -t \"first\" -g $V1 -a $V2\".sorted\" -b stdin | awk -v OFS='\\t' '$14 >= 2000 {print $1, $2, $3, $4, $5, \"distal-ELS\"}' > $V4\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash -s \"$ccREs_dELS\" \"$candidateRegions\"\n",
    "if [[ ! -f $2 ]]; then\n",
    "    awk -v OFS='\\t' '{print $1, $2, $3}' $1 > $2\n",
    "fi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Step 2: quantifying enhancer activity\n",
    "\n",
    "#### Use ABC `run.neighborhoods.py`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%writetemplate $slurm_2\n",
    "#!/bin/sh\n",
    "python {run_neighborhoods} \\\n",
    "--candidate_enhancer_regions {candidateRegions} \\\n",
    "--genes {annotations_dir}gene_ids.bed \\\n",
    "--H3K27ac {h3k27_file_rep1},{h3k27_file_rep1} \\\n",
    "--DHS {dnase_file_rep1},{dnase_file_rep2} \\\n",
    "--expression_table {gene_expression_table}  \\\n",
    "--chrom_sizes {genome_file} \\\n",
    "--ubiquitously_expressed_genes {ubiquitously_expressed_genes} \\\n",
    "--cellType {cell_type} \\\n",
    "--outdir {neighborhoods}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "WARNING: depending on the length of the inputs, this script may be very RAM-demanding. So we recommend to allocate at least 32 GB RAM to avoid \"out of memory\" errors."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Submitted batch job 22240256\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$work_dir\" \"$slurm_2\" \"$mail_user\"\n",
    "CONDA_BASE=$(conda info --base) && source $CONDA_BASE/etc/profile.d/conda.sh\n",
    "conda activate base && conda activate abcmodel\n",
    "module load bioinfo/samtools-1.9 bioinfo/bedtools-2.27.1 bioinfo/tabix-0.2.5 bioinfo/juicer-1.5.6\n",
    "sbatch --mem=16G --cpus-per-task=1 -J step2 --mail-user=$3 --mail-type=END,FAIL --workdir=$1 --export=ALL -p workq $2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job ID: 22240256\r\n",
      "Cluster: genobull\r\n",
      "User/Group: thoellinger/U1220\r\n",
      "State: COMPLETED (exit code 0)\r\n",
      "Cores: 1\r\n",
      "CPU Utilized: 02:48:52\r\n",
      "CPU Efficiency: 99.85% of 02:49:07 core-walltime\r\n",
      "Job Wall-clock time: 02:49:07\r\n",
      "Memory Utilized: 11.83 GB\r\n",
      "Memory Efficiency: 36.96% of 32.00 GB\r\n"
     ]
    }
   ],
   "source": [
    "!seff 22240256 #should take 2 hours => at 14h30 it will be ok"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Step 3: computing the ABC score\n",
    "\n",
    "If experimentally derived contact data is not available, one can run the ABC model using the powerlaw estimate only. In this case the ```--HiCdir``` argument should be excluded from ```predict.py``` and the ```--score_column powerlaw.Score``` argument should be included in ```predict.py```. In this case the ```ABC.Score``` column of the predictions file will be set to ```NaN```. The ```powerlaw.Score``` column of the output prediction files will be the relevant Score column to use."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%writetemplate $slurm_3\n",
    "#!/bin/sh\n",
    "python {predict} \\\n",
    "--enhancers {neighborhoods}EnhancerList.txt \\\n",
    "--genes {neighborhoods}GeneList.txt \\\n",
    "--score_column powerlaw.Score \\\n",
    "--hic_resolution 5000 \\\n",
    "--scale_hic_using_powerlaw \\\n",
    "--threshold .02 \\\n",
    "--cellType {cell_type} \\\n",
    "--outdir {predictions} \\\n",
    "--make_all_putative"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Submitted batch job 22242814\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$work_dir\" \"$slurm_3\" \"$mail_user\"\n",
    "CONDA_BASE=$(conda info --base) && source $CONDA_BASE/etc/profile.d/conda.sh\n",
    "conda activate base && conda activate abcmodel\n",
    "module load bioinfo/samtools-1.9 bioinfo/bedtools-2.27.1 bioinfo/tabix-0.2.5 bioinfo/juicer-1.5.6\n",
    "sbatch --mem=8G --cpus-per-task=1 -J step3 --mail-user=$3 --mail-type=END,FAIL --workdir=$1 --export=ALL -p workq $2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job ID: 22242814\r\n",
      "Cluster: genobull\r\n",
      "User/Group: thoellinger/U1220\r\n",
      "State: COMPLETED (exit code 0)\r\n",
      "Cores: 1\r\n",
      "CPU Utilized: 00:07:49\r\n",
      "CPU Efficiency: 100.00% of 00:07:49 core-walltime\r\n",
      "Job Wall-clock time: 00:07:49\r\n",
      "Memory Utilized: 3.47 GB\r\n",
      "Memory Efficiency: 43.36% of 8.00 GB\r\n"
     ]
    }
   ],
   "source": [
    "!seff 22242814"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Evaluation of ABC predictions on BENGI datasets"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Data processing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [],
   "source": [
    "predictions_non_expressed = predictions+\"EnhancerPredictionsAllPutativeNonExpressedGenes.txt\"\n",
    "predictions_expressed = predictions+\"EnhancerPredictionsAllPutative.txt\"\n",
    "all_predictions = predictions+\"AllPredictions.bedpe\" # does not exist yet\n",
    "all_predictions_sorted = all_predictions+\".sorted\"\n",
    "all_predictions_ccre_els = all_predictions_sorted+\".ccRE_intersect.txt\"\n",
    "all_predictions_ccre_els_bedpe = all_predictions_sorted+\".ccRE_intersect.bedpe\"\n",
    "\n",
    "bengi = \"/work2/project/regenet/workspace/thoellinger/BENGI/Benchmark/All-Pairs.Natural-Ratio/\"\n",
    "nb_benchmarks = 6\n",
    "file_names = list()\n",
    "# benchmark file names without extensions\n",
    "file_names.append(\"GM12878.CHiC-Benchmark.v3\")\n",
    "file_names.append(\"GM12878.CTCF-ChIAPET-Benchmark.v3\")\n",
    "file_names.append(\"GM12878.GEUVADIS-Benchmark.v3\")\n",
    "file_names.append(\"GM12878.GTEx-Benchmark.v3\")\n",
    "file_names.append(\"GM12878.HiC-Benchmark.v3\")\n",
    "file_names.append(\"GM12878.RNAPII-ChIAPET-Benchmark.v3\")\n",
    "\n",
    "# short custom names for benchmarks, same order as above\n",
    "names = [\"CHiC\", \"CTCF\", \"GEUVADIS\", \"GTEx\", \"HiC\", \"RNAPII\"]\n",
    "\n",
    "files_dict = {}\n",
    "for k in range(nb_benchmarks):\n",
    "    files_dict[names[k]] = bengi+file_names[k]+\".new.bedpe\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Unzip ABC predictions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [],
   "source": [
    "if not os.path.isfile(predictions_non_expressed):\n",
    "    !gzip -cd $predictions_non_expressed\".gz\" > $predictions_non_expressed\n",
    "if not os.path.isfile(predictions_expressed):\n",
    "    !gzip -cd $predictions_expressed\".gz\" > $predictions_expressed    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Merge ABC predictions for expressed and non expressed genes, keep only columns of interest and sort the result."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash -s \"$all_predictions\" \"$predictions_expressed\" \"$predictions_non_expressed\"\n",
    "if [[ ! -f $1 ]]; then\n",
    "    awk '{print $1\"\\t\"$2\"\\t\"$3\"\\t\"$1\"\\t\"$10\"\\t\"$10\"\\t\"$9\"\\t\"$20\"\\t.\\t.\\t\"$13\"\\t\"$11}' $2 > $1;\n",
    "    tail -n+2 $3 | awk '{print $1\"\\t\"$2\"\\t\"$3\"\\t\"$1\"\\t\"$10\"\\t\"$10\"\\t\"$9\"\\t\"$20\"\\t.\\t.\\t\"$13\"\\t\"$11}' >> $1;\n",
    "fi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The following is very RAM-demanding, we recommend using at least 64GB of ram to avoid \"out of memory\" errors."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "srun: job 22242943 queued and waiting for resources\n",
      "srun: job 22242943 has been allocated resources\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$all_predictions\" \"$all_predictions_sorted\" \"$genome_file\"\n",
    "export V1=$1 V2=$2 V3=$3\n",
    "CONDA_BASE=$(conda info --base) && source $CONDA_BASE/etc/profile.d/conda.sh\n",
    "conda activate base && conda activate abcmodel\n",
    "module load bioinfo/bedtools-2.27.1\n",
    "srun --mem=64G bash\n",
    "if [[ ! -f $V2 ]]; then\n",
    "    tail -n+2 $V1 | bedtools sort -faidx $V3 -i stdin  > $V2\n",
    "fi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Intersect ABC candidate regions with ccRE-dELS."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "srun: job 22242951 queued and waiting for resources\n",
      "srun: job 22242951 has been allocated resources\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$ccREs_dELS\" \"$all_predictions_sorted\" \"$all_predictions_ccre_els\"\n",
    "export V1=$1 V2=$2 V3=$3\n",
    "CONDA_BASE=$(conda info --base) && source $CONDA_BASE/etc/profile.d/conda.sh\n",
    "conda activate base && conda activate abcmodel\n",
    "module load bioinfo/bedtools-2.27.1\n",
    "srun --mem=32G bash\n",
    "if [[ ! -f $V3 ]]; then\n",
    "    bedtools intersect -sorted -wo -a $V1 -b $V2 > $V3\n",
    "fi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "When multiple ABC candidate enhancers for a given gene overlap the same ccRE-ELS, we keep only the one that maximize the ABC score (the \"powerlaw.Score\" if ABC ran without 3D data). Here it did not matter as there were no duplicates ; although in general we may not want to keep only the max, if at the end of the day it results in a smaller intersection between ABC predictions and BENGI datasets."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "srun: job 22242953 queued and waiting for resources\n",
      "srun: job 22242953 has been allocated resources\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$light_annotation\" \"$all_predictions_ccre_els\" \"$all_predictions_ccre_els_bedpe\"\n",
    "export V1=$1 V2=$2 V3=$3\n",
    "CONDA_BASE=$(conda info --base) && source $CONDA_BASE/etc/profile.d/conda.sh\n",
    "conda activate base && conda activate abcmodel\n",
    "module load bioinfo/bedtools-2.27.1\n",
    "srun --mem=32G bash\n",
    "awk '{ \\\n",
    "    if(NR==FNR){ \\\n",
    "        start[$4]=$2; \\\n",
    "        end[$4]=$3; \\\n",
    "        next; \\\n",
    "    }; \\\n",
    "    if(!pair[$5$13]){ \\\n",
    "        pair[$5$13]=$1\"\\t\"start[$13]\"\\t\"end[$13]\"\\t\"$1\"\\t\"$2\"\\t\"$3\"\\t\"$13\":\"$5\"\\t\"$14\"\\t\"$15\"\\t\"$16\"\\t\"$2\"\\t\"$19\"\\t\"$17\"\\t\"$18; \\\n",
    "        next}; \\\n",
    "    split(pair[$5$13],current_pair,\"\\t\"); \\\n",
    "    if($14>current_pair[$14]){ \\\n",
    "        pair[$5$13]=$1\"\\t\"start[$13]\"\\t\"end[$13]\"\\t\"$1\"\\t\"$2\"\\t\"$3\"\\t\"$13\":\"$5\"\\t\"$14\"\\t\"$15\"\\t\"$16\"\\t\"$2\"\\t\"$19\"\\t\"$17\"\\t\"$18; \\\n",
    "    }} \\\n",
    "END{for(u in pair){ \\\n",
    "    print pair[u] \\\n",
    "    }}' $V1 $V2 | bedtools sort -i stdin > $V3"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Intersection between ABC predictions / BENGI"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Import files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [],
   "source": [
    "col_names = [\"chr1\", \"start1\", \"end1\", \"chr2\", \"start2\", \"end2\", \"name\", \"is.positive\", \"strand1\", \"strand2\", \"TSSs\"] # column names in bedpe benchmarks\n",
    "col_names_abc = [\"chr1\", \"start1\", \"end1\", \"chr2\", \"start2\", \"end2\", \"name\", \"score\", \"strand1\", \"strand2\", \"TSS\", \"overlap\", \"is.expressed\", \"expression\"] # column names in abc bedpe\n",
    "benchmarks = {}\n",
    "for k in range(nb_benchmarks):\n",
    "    benchmarks[names[k]] = pd.read_csv(files_dict[names[k]], sep='\\t', header=None, usecols=[0, 1, 2, 4, 5, 6, 7, 10], names=col_names, dtype='str', engine='c')\n",
    "\n",
    "abc = pd.read_csv(all_predictions_ccre_els_bedpe, sep='\\t', header=None, usecols=[0, 1, 2, 4, 5, 6, 7, 10, 11], names=col_names_abc, dtype='str', engine='c')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [],
   "source": [
    "intersection = {}\n",
    "for k in range(nb_benchmarks):\n",
    "    intersection[names[k]] = benchmarks[names[k]].copy().merge(abc.iloc[:,[5,6,8]], how='inner', left_on=col_names[6], right_on=col_names_abc[6])\n",
    "    \n",
    "col_summary = ['ABC.length', 'BENGI.length', 'intersection.length', 'intersection.length (%)', 'BENGI.positives', 'intersection.positives', 'intersection.positives (%)']\n",
    "summary = pd.DataFrame(columns=col_summary)\n",
    "for k in range(nb_benchmarks):\n",
    "    len_abc = len(abc)\n",
    "    len_bengi = len(benchmarks[names[k]])\n",
    "    len_inter = len(intersection[names[k]])\n",
    "    ratio = len_inter/len_bengi*100\n",
    "    bengi_positives = sum(benchmarks[names[k]][col_names[7]] == \"1\")\n",
    "    inter_positives = sum(intersection[names[k]][col_names[7]] == \"1\")\n",
    "    ratio_positives = inter_positives/bengi_positives*100\n",
    "    summary = summary.append(pd.DataFrame([[len_abc, len_bengi, len_inter, ratio, bengi_positives, inter_positives, ratio_positives]], columns=col_summary), ignore_index=True)\n",
    "summary.index = names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ABC.length</th>\n",
       "      <th>BENGI.length</th>\n",
       "      <th>intersection.length</th>\n",
       "      <th>intersection.length (%)</th>\n",
       "      <th>BENGI.positives</th>\n",
       "      <th>intersection.positives</th>\n",
       "      <th>intersection.positives (%)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CHiC</th>\n",
       "      <td>5879612</td>\n",
       "      <td>375728</td>\n",
       "      <td>375728</td>\n",
       "      <td>100.0</td>\n",
       "      <td>88245</td>\n",
       "      <td>88245</td>\n",
       "      <td>100.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CTCF</th>\n",
       "      <td>5879612</td>\n",
       "      <td>105016</td>\n",
       "      <td>105016</td>\n",
       "      <td>100.0</td>\n",
       "      <td>7591</td>\n",
       "      <td>7591</td>\n",
       "      <td>100.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GEUVADIS</th>\n",
       "      <td>5879612</td>\n",
       "      <td>50999</td>\n",
       "      <td>50999</td>\n",
       "      <td>100.0</td>\n",
       "      <td>2073</td>\n",
       "      <td>2073</td>\n",
       "      <td>100.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GTEx</th>\n",
       "      <td>5879612</td>\n",
       "      <td>38200</td>\n",
       "      <td>38200</td>\n",
       "      <td>100.0</td>\n",
       "      <td>1301</td>\n",
       "      <td>1301</td>\n",
       "      <td>100.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HiC</th>\n",
       "      <td>5879612</td>\n",
       "      <td>153739</td>\n",
       "      <td>153739</td>\n",
       "      <td>100.0</td>\n",
       "      <td>3404</td>\n",
       "      <td>3404</td>\n",
       "      <td>100.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RNAPII</th>\n",
       "      <td>5879612</td>\n",
       "      <td>157235</td>\n",
       "      <td>157235</td>\n",
       "      <td>100.0</td>\n",
       "      <td>23699</td>\n",
       "      <td>23699</td>\n",
       "      <td>100.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         ABC.length BENGI.length intersection.length  intersection.length (%)  \\\n",
       "CHiC        5879612       375728              375728                    100.0   \n",
       "CTCF        5879612       105016              105016                    100.0   \n",
       "GEUVADIS    5879612        50999               50999                    100.0   \n",
       "GTEx        5879612        38200               38200                    100.0   \n",
       "HiC         5879612       153739              153739                    100.0   \n",
       "RNAPII      5879612       157235              157235                    100.0   \n",
       "\n",
       "         BENGI.positives intersection.positives  intersection.positives (%)  \n",
       "CHiC               88245                  88245                       100.0  \n",
       "CTCF                7591                   7591                       100.0  \n",
       "GEUVADIS            2073                   2073                       100.0  \n",
       "GTEx                1301                   1301                       100.0  \n",
       "HiC                 3404                   3404                       100.0  \n",
       "RNAPII             23699                  23699                       100.0  "
      ]
     },
     "execution_count": 79,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "summary"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Export files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [],
   "source": [
    "if not os.path.isdir(predictions+\"BENGI_intersect/\"):\n",
    "    !mkdir $predictions\"BENGI_intersect/\"\n",
    "for k in range(nb_benchmarks):\n",
    "    intersection[names[k]].drop(intersection[names[1]].columns[-1], axis=1).to_csv(predictions+\"BENGI_intersect/AllPredictions.intersect.\"+file_names[k]+\".bedpe\",sep='\\t',header=False,index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
