
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
        <meta name="author" content="Tristan Hoellinger">
      
      
        <link rel="canonical" href="https://eg.github.io/haemochromatosis/ABC/data_processing.html">
      
      
        <link rel="prev" href="../CHiC/go_enrichment.html">
      
      
        <link rel="next" href="preliminary_analysis_v9.html">
      
      <link rel="icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.4.2, mkdocs-material-9.0.11">
    
    
      
        <title>Preprocessing - Genome-wide identification of E/G interactions</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.0d440cfe.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.2505c338.min.css">
      
      

    
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../stylesheets/extra.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="black" data-md-color-accent="">
  
    
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#explore-and-prepare-data-for-further-analysis-of-e-g-network-with-genes-involved-in-hemochromatosis" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Genome-wide identification of E/G interactions" class="md-header__button md-logo" aria-label="Genome-wide identification of E/G interactions" data-md-component="logo">
      
  <img src="../../logo/logo-inserm-blc.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Genome-wide identification of E/G interactions
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Preprocessing
            
          </span>
        </div>
      </div>
    </div>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/hoellin/eg" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
  </div>
  <div class="md-source__repository">
    hoellin/eg
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../guidebooks/introduction.html" class="md-tabs__link">
        Guidebooks
      </a>
    </li>
  

      
        
  
  
    
  


  
  
  
    <li class="md-tabs__item">
      <a href="../introduction.html" class="md-tabs__link md-tabs__link--active">
        Haemochromatosis
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../notes_BENGI/introduction.html" class="md-tabs__link">
        BENGI
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../notes_ABC/introduction.html" class="md-tabs__link">
        ABC
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../index.html" class="md-tabs__link">
        About
      </a>
    </li>
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Genome-wide identification of E/G interactions" class="md-nav__button md-logo" aria-label="Genome-wide identification of E/G interactions" data-md-component="logo">
      
  <img src="../../logo/logo-inserm-blc.png" alt="logo">

    </a>
    Genome-wide identification of E/G interactions
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/hoellin/eg" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
  </div>
  <div class="md-source__repository">
    hoellin/eg
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1" >
      
      
      
        <label class="md-nav__link" for="__nav_1" id="__nav_1_label" tabindex="0">
          Guidebooks
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_1_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_1">
          <span class="md-nav__icon md-icon"></span>
          Guidebooks
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../guidebooks/introduction.html" class="md-nav__link">
        Introduction
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../guidebooks/script_bedpe_summary/script_bedpe_summary.html" class="md-nav__link">
        Compute summary stats of a bedpe
      </a>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_3" >
      
      
      
        <label class="md-nav__link" for="__nav_1_3" id="__nav_1_3_label" tabindex="0">
          Basics
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_1_3">
          <span class="md-nav__icon md-icon"></span>
          Basics
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../guidebooks/shell.html" class="md-nav__link">
        Shell
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../guidebooks/conda_environments.html" class="md-nav__link">
        Conda basics
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../guidebooks/git.html" class="md-nav__link">
        Git
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../guidebooks/genotoul.html" class="md-nav__link">
        Genotoul cluster
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../guidebooks/notebooks_ssh.html" class="md-nav__link">
        How to use Jupyter through ssh tunneling?
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_4" >
      
      
      
        <label class="md-nav__link" for="__nav_1_4" id="__nav_1_4_label" tabindex="0">
          Misc
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_4_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_1_4">
          <span class="md-nav__icon md-icon"></span>
          Misc
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../guidebooks/compute_correlations.html" class="md-nav__link">
        Compute correlations
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" checked>
      
      
      
        <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
          Haemochromatosis
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="true">
        <label class="md-nav__title" for="__nav_2">
          <span class="md-nav__icon md-icon"></span>
          Haemochromatosis
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../introduction.html" class="md-nav__link">
        Introduction
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../data_format.html" class="md-nav__link">
        Data format
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="custom_go.html" class="md-nav__link">
        Custom script GO
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../main_results.html" class="md-nav__link">
        Main results
      </a>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2_5" >
      
      
      
        <label class="md-nav__link" for="__nav_2_5" id="__nav_2_5_label" tabindex="0">
          CHiC data
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_2_5_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_2_5">
          <span class="md-nav__icon md-icon"></span>
          CHiC data
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../CHiC/preprocess_data.html" class="md-nav__link">
        Preprocessing
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../CHiC/preliminary_analysis_chic_v2.html" class="md-nav__link">
        Networks
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../CHiC/go_enrichment.html" class="md-nav__link">
        GO enrichment
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
    
  
  
    
      
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2_6" checked>
      
      
      
        <label class="md-nav__link" for="__nav_2_6" id="__nav_2_6_label" tabindex="0">
          ABC data
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_2_6_label" aria-expanded="true">
        <label class="md-nav__title" for="__nav_2_6">
          <span class="md-nav__icon md-icon"></span>
          ABC data
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          Preprocessing
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="data_processing.html" class="md-nav__link md-nav__link--active">
        Preprocessing
      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#requirements" class="md-nav__link">
    Requirements
  </a>
  
    <nav class="md-nav" aria-label="Requirements">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#data-availability" class="md-nav__link">
    Data availability
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#output" class="md-nav__link">
    Output
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#bedtools" class="md-nav__link">
    bedtools
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#main-remarks" class="md-nav__link">
    Main remarks
  </a>
  
    <nav class="md-nav" aria-label="Main remarks">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#on-merges-of-enhancers" class="md-nav__link">
    On merges of enhancers
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#extracting-data-of-interest" class="md-nav__link">
    Extracting data of interest
  </a>
  
    <nav class="md-nav" aria-label="Extracting data of interest">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#main-filters" class="md-nav__link">
    Main filters
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#select-biosamples" class="md-nav__link">
    Select biosamples
  </a>
  
    <nav class="md-nav" aria-label="Select biosamples">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#all-liver-and-intestine-biosamples" class="md-nav__link">
    All liver and intestine biosamples
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#all-liver-biosamples" class="md-nav__link">
    All liver biosamples
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#all-intestine-biosamples" class="md-nav__link">
    All intestine biosamples
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#all-5-separate-biosamples-of-interest" class="md-nav__link">
    All 5 separate biosamples of interest
  </a>
  
    <nav class="md-nav" aria-label="All 5 separate biosamples of interest">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#hepatocyte-encode" class="md-nav__link">
    hepatocyte-ENCODE
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#liver-encode" class="md-nav__link">
    liver-ENCODE
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hepg2-roadmap" class="md-nav__link">
    HepG2-Roadmap
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#large_intestine_fetal-roadmap" class="md-nav__link">
    large_intestine_fetal-Roadmap
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#small_intestine_fetal-roadmap" class="md-nav__link">
    small_intestine_fetal-Roadmap
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#all-biosamples" class="md-nav__link">
    All biosamples
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#investigate-putative-regulatory-elements" class="md-nav__link">
    Investigate putative regulatory elements
  </a>
  
    <nav class="md-nav" aria-label="Investigate putative regulatory elements">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#small-r-script-to-compute-summary-statistics" class="md-nav__link">
    Small R script to compute summary statistics
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ccre" class="md-nav__link">
    ccRE
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#merged-enhancers-of-all-131-biosamples" class="md-nav__link">
    Merged enhancers of all 131 biosamples
  </a>
  
    <nav class="md-nav" aria-label="Merged enhancers of all 131 biosamples">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#original-enhancers" class="md-nav__link">
    Original enhancers
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#merged-enhancers" class="md-nav__link">
    Merged enhancers
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#overlap-with-ccre-els" class="md-nav__link">
    Overlap with ccRE-ELS
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#overlap-with-all-ccre" class="md-nav__link">
    Overlap with all ccRE
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#merged-enhancers-of-all-liver-and-intestine-biosamples" class="md-nav__link">
    Merged enhancers of all liver and intestine biosamples
  </a>
  
    <nav class="md-nav" aria-label="Merged enhancers of all liver and intestine biosamples">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#original-enhancers_1" class="md-nav__link">
    Original enhancers
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#merged-enhancers-merged-across-131-biosamples" class="md-nav__link">
    Merged enhancers (merged across 131 biosamples)
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#merged-enhancers-merged-across-5-liver-and-intestine-biosamples" class="md-nav__link">
    Merged enhancers (merged across 5 liver and intestine biosamples)
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#overlap-with-ccre-els_1" class="md-nav__link">
    Overlap with ccRE-ELS
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#merged-enhancers-of-all-liver-biosamples" class="md-nav__link">
    Merged enhancers of all liver biosamples
  </a>
  
    <nav class="md-nav" aria-label="Merged enhancers of all liver biosamples">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#original-enhancers_2" class="md-nav__link">
    Original enhancers
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#merged-enhancers-merged-across-131-biosamples_1" class="md-nav__link">
    Merged enhancers (merged across 131 biosamples)
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#merged-enhancers-across-all-3-liver-biosamples" class="md-nav__link">
    Merged enhancers (across all 3 liver biosamples)
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#overlap-with-ccre-els_2" class="md-nav__link">
    Overlap with ccRE-ELS
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#merged-enhancers-of-all-intestine-biosamples" class="md-nav__link">
    Merged enhancers of all intestine biosamples
  </a>
  
    <nav class="md-nav" aria-label="Merged enhancers of all intestine biosamples">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#original-enhancers_3" class="md-nav__link">
    Original enhancers
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#merged-enhancers-merged-across-131-biosamples_2" class="md-nav__link">
    Merged enhancers (merged across 131 biosamples)
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#merged-enhancers-across-all-2-intestine-biosamples" class="md-nav__link">
    Merged enhancers (across all 2 intestine biosamples)
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="preliminary_analysis_v9.html" class="md-nav__link">
        Networks
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="summary_statistics_eg_pairs.html" class="md-nav__link">
        Summary statistics enhancers
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="go_enrichment.html" class="md-nav__link">
        GO enrichment
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="gorilla.html" class="md-nav__link">
        GOrilla
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="compute_avg_nb_of_connections_per_gene.html" class="md-nav__link">
        Avg nb connections per gene
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="gwas.html" class="md-nav__link">
        GWAS (13 initial genes)
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="gwas_10.html" class="md-nav__link">
        GWAS (10 initial genes)
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" >
      
      
      
        <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
          BENGI
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3">
          <span class="md-nav__icon md-icon"></span>
          BENGI
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/introduction.html" class="md-nav__link">
        Introduction
      </a>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_2" >
      
      
      
        <label class="md-nav__link" for="__nav_3_2" id="__nav_3_2_label" tabindex="0">
          Notebooks
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_2_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_2">
          <span class="md-nav__icon md-icon"></span>
          Notebooks
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/BENGI_to_custom_bedpe.html" class="md-nav__link">
        Convert BENGI datasets to custom bedpe
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_3" >
      
      
      
        <label class="md-nav__link" for="__nav_3_3" id="__nav_3_3_label" tabindex="0">
          Summary statistics
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_3">
          <span class="md-nav__icon md-icon"></span>
          Summary statistics
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/summary_statistics/summary_statistics.html" class="md-nav__link">
        Summary statistics on BENGI GM12878 benchmarks
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/summary_statistics_BENGI/summary_BENGI_bedpe.html" class="md-nav__link">
        (R code) Compute summary statistics over BENGI
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_4" >
      
      
      
        <label class="md-nav__link" for="__nav_3_4" id="__nav_3_4_label" tabindex="0">
          Evaluation of a few methods over BENGI GM12878
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_4_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_4">
          <span class="md-nav__icon md-icon"></span>
          Evaluation of a few methods over BENGI GM12878
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_4_1" >
      
      
      
        <label class="md-nav__link" for="__nav_3_4_1" id="__nav_3_4_1_label" tabindex="0">
          Distance method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_4_1_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_4_1">
          <span class="md-nav__icon md-icon"></span>
          Distance method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/distance_method/distance_evaluation_with_code.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/distance_method/summary_distance_evaluation.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_4_2" >
      
      
      
        <label class="md-nav__link" for="__nav_3_4_2" id="__nav_3_4_2_label" tabindex="0">
          DNase-expression correlation method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_4_2_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_4_2">
          <span class="md-nav__icon md-icon"></span>
          DNase-expression correlation method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/dnase_expression_correlation/correlation_method_with_code.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/dnase_expression_correlation/summary_correlation_method.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_4_3" >
      
      
      
        <label class="md-nav__link" for="__nav_3_4_3" id="__nav_3_4_3_label" tabindex="0">
          Average rank method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_4_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_4_3">
          <span class="md-nav__icon md-icon"></span>
          Average rank method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/avg_rank_method/avg_rank_method_with_code.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/avg_rank_method/summary_avg_rank_method.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_5" >
      
      
      
        <label class="md-nav__link" for="__nav_3_5" id="__nav_3_5_label" tabindex="0">
          Evaluation of a few methods over Fulco et al 2019 CRISPRi-FlowFISH
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_5_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_5">
          <span class="md-nav__icon md-icon"></span>
          Evaluation of a few methods over Fulco et al 2019 CRISPRi-FlowFISH
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_5_1" >
      
      
      
        <label class="md-nav__link" for="__nav_3_5_1" id="__nav_3_5_1_label" tabindex="0">
          Distance method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_5_1_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_5_1">
          <span class="md-nav__icon md-icon"></span>
          Distance method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/CRISPRi_FlowFISH/distance_method/distance_over_fulco_et_al_crispri.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/CRISPRi_FlowFISH/distance_method/summary_distance_over_fulco_et_al_crispri.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_5_2" >
      
      
      
        <label class="md-nav__link" for="__nav_3_5_2" id="__nav_3_5_2_label" tabindex="0">
          DNase-expression correlation method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_5_2_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_5_2">
          <span class="md-nav__icon md-icon"></span>
          DNase-expression correlation method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/CRISPRi_FlowFISH/dnase_expression_correlation/correlation_method_with_code.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/CRISPRi_FlowFISH/dnase_expression_correlation/summary_correlation_method.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_5_3" >
      
      
      
        <label class="md-nav__link" for="__nav_3_5_3" id="__nav_3_5_3_label" tabindex="0">
          Average rank method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_5_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_5_3">
          <span class="md-nav__icon md-icon"></span>
          Average rank method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/CRISPRi_FlowFISH/avg_rank_method/avg_rank_method_with_code.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/CRISPRi_FlowFISH/avg_rank_method/summary_avg_rank.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" >
      
      
      
        <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
          ABC
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_4">
          <span class="md-nav__icon md-icon"></span>
          ABC
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/introduction.html" class="md-nav__link">
        Introduction
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/BENGI/ABC_BENGI_results.html" class="md-nav__link">
        ABC over BENGI for GM12878, main results
      </a>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_3" >
      
      
      
        <label class="md-nav__link" for="__nav_4_3" id="__nav_4_3_label" tabindex="0">
          CRISPRi-FlowFISH
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_4_3">
          <span class="md-nav__icon md-icon"></span>
          CRISPRi-FlowFISH
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/K562/ABC_K562_CRISPRi_FlowFISH.html" class="md-nav__link">
        Original ABC predictions over K562
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/K562/recap_investigations_ABC_over_CRiFF.html" class="md-nav__link">
        Summary ABC over CRiFF
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_4" >
      
      
      
        <label class="md-nav__link" for="__nav_4_4" id="__nav_4_4_label" tabindex="0">
          Notebooks
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_4_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_4_4">
          <span class="md-nav__icon md-icon"></span>
          Notebooks
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/K562/april_K562_56_genes/april_K562_56_genes.html" class="md-nav__link">
        ABC over K562 cCRiFF
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/K562/april_K562_59_genes/april_K562_59_genes.html" class="md-nav__link">
        ABC over K562 CRiFF
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/K562/april_K562_candidates_not_in_whitelist_56_genes/april_K562_candidates_NOT_in_whitelist.html" class="md-nav__link">
        ABC over K562 cCRiFF, no whitelist
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/generic_notebooks/turnkey_notebook_to_run_ABC_with_example_over_GM12878.html" class="md-nav__link">
        Generic notebook to run ABC method & Example over GM12878
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/BENGI/notebook_ABC_over_BENGI_GM12878.html" class="md-nav__link">
        ABC over GM12878 with same accessions as for BENGI
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/BENGI/notebook_ABC_over_BENGI_GM12878_from_ccRE_ELSs.html" class="md-nav__link">
        ABC over GM12878 with same accessions as for BENGI, starting with ccRE-dELS
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_5" >
      
      
      
        <label class="md-nav__link" for="__nav_4_5" id="__nav_4_5_label" tabindex="0">
          Full codes
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_5_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_4_5">
          <span class="md-nav__icon md-icon"></span>
          Full codes
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/test_abc_pipeline.html" class="md-nav__link">
        Test of ABC pipeline over chr22
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/BENGI/ABC_over_BENGI_GM12878.html" class="md-nav__link">
        ABC over GM12878 with same accessions as for BENGI
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/differences_example_chr22.html" class="md-nav__link">
        Meld of results over chr22 vs authors' results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" >
      
      
      
        <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
          About
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_5">
          <span class="md-nav__icon md-icon"></span>
          About
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../index.html" class="md-nav__link">
        Getting started
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../about_me.html" class="md-nav__link">
        About me
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#requirements" class="md-nav__link">
    Requirements
  </a>
  
    <nav class="md-nav" aria-label="Requirements">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#data-availability" class="md-nav__link">
    Data availability
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#output" class="md-nav__link">
    Output
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#bedtools" class="md-nav__link">
    bedtools
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#main-remarks" class="md-nav__link">
    Main remarks
  </a>
  
    <nav class="md-nav" aria-label="Main remarks">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#on-merges-of-enhancers" class="md-nav__link">
    On merges of enhancers
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#extracting-data-of-interest" class="md-nav__link">
    Extracting data of interest
  </a>
  
    <nav class="md-nav" aria-label="Extracting data of interest">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#main-filters" class="md-nav__link">
    Main filters
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#select-biosamples" class="md-nav__link">
    Select biosamples
  </a>
  
    <nav class="md-nav" aria-label="Select biosamples">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#all-liver-and-intestine-biosamples" class="md-nav__link">
    All liver and intestine biosamples
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#all-liver-biosamples" class="md-nav__link">
    All liver biosamples
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#all-intestine-biosamples" class="md-nav__link">
    All intestine biosamples
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#all-5-separate-biosamples-of-interest" class="md-nav__link">
    All 5 separate biosamples of interest
  </a>
  
    <nav class="md-nav" aria-label="All 5 separate biosamples of interest">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#hepatocyte-encode" class="md-nav__link">
    hepatocyte-ENCODE
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#liver-encode" class="md-nav__link">
    liver-ENCODE
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hepg2-roadmap" class="md-nav__link">
    HepG2-Roadmap
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#large_intestine_fetal-roadmap" class="md-nav__link">
    large_intestine_fetal-Roadmap
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#small_intestine_fetal-roadmap" class="md-nav__link">
    small_intestine_fetal-Roadmap
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#all-biosamples" class="md-nav__link">
    All biosamples
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#investigate-putative-regulatory-elements" class="md-nav__link">
    Investigate putative regulatory elements
  </a>
  
    <nav class="md-nav" aria-label="Investigate putative regulatory elements">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#small-r-script-to-compute-summary-statistics" class="md-nav__link">
    Small R script to compute summary statistics
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ccre" class="md-nav__link">
    ccRE
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#merged-enhancers-of-all-131-biosamples" class="md-nav__link">
    Merged enhancers of all 131 biosamples
  </a>
  
    <nav class="md-nav" aria-label="Merged enhancers of all 131 biosamples">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#original-enhancers" class="md-nav__link">
    Original enhancers
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#merged-enhancers" class="md-nav__link">
    Merged enhancers
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#overlap-with-ccre-els" class="md-nav__link">
    Overlap with ccRE-ELS
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#overlap-with-all-ccre" class="md-nav__link">
    Overlap with all ccRE
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#merged-enhancers-of-all-liver-and-intestine-biosamples" class="md-nav__link">
    Merged enhancers of all liver and intestine biosamples
  </a>
  
    <nav class="md-nav" aria-label="Merged enhancers of all liver and intestine biosamples">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#original-enhancers_1" class="md-nav__link">
    Original enhancers
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#merged-enhancers-merged-across-131-biosamples" class="md-nav__link">
    Merged enhancers (merged across 131 biosamples)
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#merged-enhancers-merged-across-5-liver-and-intestine-biosamples" class="md-nav__link">
    Merged enhancers (merged across 5 liver and intestine biosamples)
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#overlap-with-ccre-els_1" class="md-nav__link">
    Overlap with ccRE-ELS
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#merged-enhancers-of-all-liver-biosamples" class="md-nav__link">
    Merged enhancers of all liver biosamples
  </a>
  
    <nav class="md-nav" aria-label="Merged enhancers of all liver biosamples">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#original-enhancers_2" class="md-nav__link">
    Original enhancers
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#merged-enhancers-merged-across-131-biosamples_1" class="md-nav__link">
    Merged enhancers (merged across 131 biosamples)
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#merged-enhancers-across-all-3-liver-biosamples" class="md-nav__link">
    Merged enhancers (across all 3 liver biosamples)
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#overlap-with-ccre-els_2" class="md-nav__link">
    Overlap with ccRE-ELS
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#merged-enhancers-of-all-intestine-biosamples" class="md-nav__link">
    Merged enhancers of all intestine biosamples
  </a>
  
    <nav class="md-nav" aria-label="Merged enhancers of all intestine biosamples">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#original-enhancers_3" class="md-nav__link">
    Original enhancers
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#merged-enhancers-merged-across-131-biosamples_2" class="md-nav__link">
    Merged enhancers (merged across 131 biosamples)
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#merged-enhancers-across-all-2-intestine-biosamples" class="md-nav__link">
    Merged enhancers (across all 2 intestine biosamples)
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  

  
  



<h1 id="explore-and-prepare-data-for-further-analysis-of-e-g-network-with-genes-involved-in-hemochromatosis">Explore and prepare data for further analysis of E-G network with genes involved in hemochromatosis<a class="headerlink" href="#explore-and-prepare-data-for-further-analysis-of-e-g-network-with-genes-involved-in-hemochromatosis" title="Permanent link">&para;</a></h1>
<h2 id="requirements">Requirements<a class="headerlink" href="#requirements" title="Permanent link">&para;</a></h2>
<h3 id="data-availability">Data availability<a class="headerlink" href="#data-availability" title="Permanent link">&para;</a></h3>
<p>All data are downloaded from <a href="https://www.engreitzlab.org/resources/">Nasser et al. 2021</a> and available in <code>/work2/project/regenet/results/multi/abc.model/Nasser2021</code> on Genotoul.</p>
<p><code>AllPredictions.AvgHiC.ABC0.015.minus150.ForABCPaperV3.txt</code> contains <a href="https://www.engreitzlab.org/resources/">predictions made with the ABC model over 131 biosamples from Nasser et al. 2021</a>.</p>
<blockquote>
<div class="highlight"><pre><span></span><code>wget<span class="w"> </span>ftp://ftp.broadinstitute.org/outgoing/lincRNA/ABC/AllPredictions.AvgHiC.ABC0.015.minus150.ForABCPaperV3.txt.gz<span class="w"> </span>-P<span class="w"> </span>/work2/project/regenet/results/multi/abc.model/Nasser2021/
</code></pre></div>
</blockquote>
<p>List of biosamples linked to liver:</p>
<div class="highlight"><pre><span></span><code>hepatocyte-ENCODE
HepG2-Roadmap<span class="w"> </span><span class="c1"># human liver cancer cell line</span>
liver-ENCODE
</code></pre></div>
<p>List of biosamples linked to intestine:</p>
<div class="highlight"><pre><span></span><code>large_intestine_fetal-Roadmap
small_intestine_fetal-Roadmap
</code></pre></div>
<p>List of biosamples linked to colon:</p>
<div class="highlight"><pre><span></span><code><span class="c1">#HCT116-ENCODE # colon</span>
<span class="c1">#HT29 # colon</span>
<span class="c1">#sigmoid_colon-ENCODE # sigmoid colon</span>
<span class="c1">#transverse_colon-ENCODE # transverse colon</span>
</code></pre></div>
<h3 id="output">Output<a class="headerlink" href="#output" title="Permanent link">&para;</a></h3>
<p>At the end of the day, we want to compute <code>bedpe</code> files the columns of which should be as follows:</p>
<div class="highlight"><pre><span></span><code><span class="s">&quot;chrom1&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;start1&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;end1&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;chrom2&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;start2&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;end2&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;name&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;ABC.score&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;strand1&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;strand2&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;biosample&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;gene&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;distance&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;tissue&quot;</span><span class="p">(,</span><span class="w"> </span><span class="kc">...</span><span class="p">)</span>
</code></pre></div>
<p>where "1" refers to enhancers and "2" refers to genes (either its TSS or the most upstream base of its most probable promoter according to what we want to do with the bedpe).</p>
<h3 id="bedtools"><code>bedtools</code><a class="headerlink" href="#bedtools" title="Permanent link">&para;</a></h3>
<p>One can load the appropriate bedtools module using:</p>
<div class="highlight"><pre><span></span><code>conda<span class="w"> </span>activate<span class="w"> </span>base<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>module<span class="w"> </span>load<span class="w"> </span>bioinfo/bedtools-2.27.1<span class="w">  </span>
</code></pre></div>
<h2 id="main-remarks">Main remarks<a class="headerlink" href="#main-remarks" title="Permanent link">&para;</a></h2>
<h3 id="on-merges-of-enhancers">On merges of enhancers<a class="headerlink" href="#on-merges-of-enhancers" title="Permanent link">&para;</a></h3>
<ul>
<li>When, for a single biosample, we replace the enhancers by the "merged" enhancers (whatever they are) and we merge the resulting E-G pairs in which the merged enhancers overlap, we will have to merge ABC predictions eventually. This raises the following question: how to combine the ABC scores? Well, strictly speaking, there are no easy rigorous way to combine two ABC scores such that the score obtained is an ABC score too, even if we take the overlap into account (see next paragraph) because it could happen that the denominator in the two ABC scores are different, and would be different also if one looked at the merged enhancers from the beginning.</li>
</ul>
<p><strong><em>Here we simply take the mean of the two ABC scores.</em></strong> </p>
<p>A slightly better approach could be to do as follows (do a drawing to re-understand when reading again): let E1 and E3 be the two enhancers, such that the intersection of them is F2 ; and denote F1 = E1-F2, F3 = E3-F2 and F = F1+F2+F3 = union of E1 and E3. Consider the following ratios, the sum of which is 1: r1 = F1/F, r2 = F2/F, r3 = F3/F. Then, the new ABC score could read as follows: ABC = r1 x ABC1 + r3 x ABC3 + r2 x (ABC1+ABC3)/2.</p>
<h2 id="extracting-data-of-interest">Extracting data of interest<a class="headerlink" href="#extracting-data-of-interest" title="Permanent link">&para;</a></h2>
<blockquote>
<div class="highlight"><pre><span></span><code>head<span class="w"> </span>-n<span class="w"> </span><span class="m">1</span><span class="w"> </span>AllPredictions.AvgHiC.ABC0.015.minus150.ForABCPaperV3.txt
</code></pre></div>
<div class="highlight"><pre><span></span><code><span class="m">1</span><span class="w"> </span>chr
<span class="m">2</span><span class="w"> </span>start
<span class="m">3</span><span class="w"> </span>end
<span class="m">4</span><span class="w"> </span>name
<span class="m">5</span><span class="w"> </span>class
<span class="m">6</span><span class="w"> </span>activity_base
<span class="m">7</span><span class="w"> </span>TargetGene
<span class="m">8</span><span class="w"> </span>TargetGeneTSS
<span class="m">9</span><span class="w"> </span>TargetGeneExpression
<span class="m">10</span><span class="w"> </span>TargetGenePromoterActivityQuantile
<span class="m">11</span><span class="w"> </span>TargetGeneIsExpressed
<span class="m">12</span><span class="w"> </span>distance
<span class="m">13</span><span class="w"> </span>isSelfPromoter
<span class="m">14</span><span class="w"> </span>hic_contact
<span class="m">15</span><span class="w"> </span>powerlaw_contact
<span class="m">16</span><span class="w"> </span>powerlaw_contact_reference
<span class="m">17</span><span class="w"> </span>hic_contact_pl_scaled
<span class="m">18</span><span class="w"> </span>hic_pseudocount
<span class="m">19</span><span class="w"> </span>hic_contact_pl_scaled_adj
<span class="m">20</span><span class="w"> </span>ABC.Score.Numerator
<span class="m">21</span><span class="w"> </span>ABC.Score
<span class="m">22</span><span class="w"> </span>powerlaw.Score.Numerator
<span class="m">23</span><span class="w"> </span>powerlaw.Score
<span class="m">24</span><span class="w"> </span>CellType
</code></pre></div>
<div class="highlight"><pre><span></span><code>wc<span class="w"> </span>-l<span class="w"> </span>AllPredictions.AvgHiC.ABC0.015.minus150.ForABCPaperV3.txt
</code></pre></div>
<p>7,717,393</p>
</blockquote>
<h3 id="main-filters">Main filters<a class="headerlink" href="#main-filters" title="Permanent link">&para;</a></h3>
<p>First we verified that all genes in <code>AllPredictions.AvgHiC.ABC0.015.minus150.ForABCPaperV3.txt</code> are expressed.</p>
<blockquote>
<div class="highlight"><pre><span></span><code>tail<span class="w"> </span>-n+2<span class="w"> </span>AllPredictions.AvgHiC.ABC0.015.minus150.ForABCPaperV3.txt<span class="w"> </span><span class="p">|</span>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;{_[$11]++} END{for(u in _){print u, _[u]}}&#39;</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>True<span class="w">  </span><span class="m">7717392</span>
</code></pre></div>
</blockquote>
<p>OK.</p>
<p>Now we verify that all ABC scores are above 0.015 as expected:</p>
<blockquote>
<div class="highlight"><pre><span></span><code>tail<span class="w"> </span>-n+2<span class="w"> </span>AllPredictions.AvgHiC.ABC0.015.minus150.ForABCPaperV3.txt<span class="w"> </span><span class="p">|</span>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;{if($21&gt;=0.015){print $0}}&#39;</span><span class="w"> </span><span class="p">|</span>wc<span class="w"> </span>-l
</code></pre></div>
<p>7,717,392</p>
</blockquote>
<p>OK.</p>
<p>We see that in a lot (&gt; 1 million) of those element-gene pairs, the element is actually the promoter of the considered gene:</p>
<blockquote>
<div class="highlight"><pre><span></span><code>tail<span class="w"> </span>-n+2<span class="w"> </span>AllPredictions.AvgHiC.ABC0.015.minus150.ForABCPaperV3.txt<span class="w"> </span><span class="p">|</span>awk<span class="w"> </span><span class="s1">&#39;BEGIN{FS=&quot;\t&quot;; OFS=&quot;\t&quot;} {_[$13]++} END{for(u in _){print u, _[u]}}&#39;</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>True<span class="w">  </span><span class="m">1177021</span>
False<span class="w"> </span><span class="m">6540371</span>
</code></pre></div>
</blockquote>
<p>Let's keep only the rows corresponding to enhancers and store the results, with the fields of interest, in a bedpe file.</p>
<div class="highlight"><pre><span></span><code>conda<span class="w"> </span>activate<span class="w"> </span>base<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>module<span class="w"> </span>load<span class="w"> </span>bioinfo/bedtools-2.27.1<span class="w">  </span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>tail<span class="w"> </span>-n<span class="w"> </span>+2<span class="w"> </span>AllPredictions.AvgHiC.ABC0.015.minus150.ForABCPaperV3.txt<span class="w"> </span><span class="p">|</span>awk<span class="w"> </span><span class="s1">&#39;BEGIN{FS=&quot;\t&quot;; OFS=&quot;\t&quot;} {if($13==&quot;False&quot;){print $1, $2, $3, $1, $8, $8, $4&quot;::&quot;$7, $21, &quot;.&quot;, &quot;.&quot;, $24, $7, $12}}&#39;</span><span class="w"> </span><span class="p">|</span>bedtools<span class="w"> </span>sort<span class="w"> </span>-faidx<span class="w"> </span>chr_sizes<span class="w"> </span>-i<span class="w"> </span>stdin<span class="w"> </span>&gt;<span class="w"> </span>Nasser2021ABCPredictions.all_biosamples.all_putative_enhancers.sorted.bedpe
</code></pre></div>
<blockquote>
<div class="highlight"><pre><span></span><code>wc<span class="w"> </span>-l<span class="w"> </span>Nasser2021ABCPredictions.all_biosamples.all_putative_enhancers.bedpe
</code></pre></div>
<p>6,540,371</p>
</blockquote>
<p>OK, this is the expected length.</p>
<p>In the resulting <code>Nasser2021ABCPredictions.all_biosamples.all_putative_enhancers.sorted.bedpe</code> bedpe file, there are no more <strong><em>self</em></strong>-promoters, and the candidate elements are distributed as follows (a promoter of a gene G1 can be an enhancer for gene G2):</p>
<blockquote>
<div class="highlight"><pre><span></span><code>tail<span class="w"> </span>-n<span class="w"> </span>+2<span class="w"> </span>AllPredictions.AvgHiC.ABC0.015.minus150.ForABCPaperV3.txt<span class="w"> </span><span class="p">|</span>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;{if($13==&quot;False&quot;){_[$5]++}} END{for(u in _){print u, _[u]}}&#39;</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>intergenic<span class="w"> </span><span class="m">2758413</span>
promoter<span class="w"> </span><span class="m">224350</span>
genic<span class="w"> </span><span class="m">3557608</span>
</code></pre></div>
</blockquote>
<p>Now we extract the list of all enhancers involved in the predictions of interest for all 131 biosamples:</p>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span><span class="s1">&#39;BEGIN{FS=&quot;\t&quot;; OFS=&quot;\t&quot;} {print $1, $2, $3}&#39;</span><span class="w"> </span>Nasser2021ABCPredictions.all_biosamples.all_putative_enhancers.sorted.bedpe<span class="w"> </span><span class="p">|</span>uniq<span class="w"> </span>&gt;<span class="w"> </span>list_all_enhancers.bed
</code></pre></div>
<p>Contains 2,463,310 enhancers.</p>
<p>And we define the list of merged enhancers among those 131 biosamples, so that there are no overlaps in the resulting list:</p>
<div class="highlight"><pre><span></span><code>bedtools<span class="w"> </span>merge<span class="w"> </span>-i<span class="w"> </span>list_all_enhancers.bed<span class="w"> </span>&gt;<span class="w"> </span>list_all_enhancers.merged.bed
</code></pre></div>
<p>Contains 269,254 enhancers.</p>
<p>Now let's replace all enhancers coordinates in the result, by the coordinates of the corresponding merged enhancers in <code>list_all_enhancers.merged.bed</code>.</p>
<p>WARNING: it is <strong>absolutely necessary</strong> to have <strong>at least</strong> 32 GB of memory to run the code below. If the memory is not sufficient, the arrays indexed by <code>$11":::"$7</code> are not going to be stored but no error message will be yielded, which could make it pretty hard to debug the code.</p>
<div class="highlight"><pre><span></span><code>srun<span class="w"> </span>--mem<span class="o">=</span>32G<span class="w"> </span>--pty<span class="w"> </span>bash
conda<span class="w"> </span>activate<span class="w"> </span>base<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>module<span class="w"> </span>load<span class="w"> </span>bioinfo/bedtools-2.27.1
</code></pre></div>
<p>Note that the "." we add at the very last column is for compatibility with scrips used in downstream analysis, as when we select particular biosamples, this fields will contain the list of tissues of interest.</p>
<div class="highlight"><pre><span></span><code>bedtools<span class="w"> </span>intersect<span class="w"> </span>-wb<span class="w"> </span>-a<span class="w"> </span>Nasser2021ABCPredictions.all_biosamples.all_putative_enhancers.sorted.bedpe<span class="w"> </span>-b<span class="w"> </span>list_all_enhancers.merged.bed<span class="w"> </span><span class="p">|</span>awk<span class="w"> </span><span class="s1">&#39;BEGIN{FS=&quot;\t&quot;; OFS=&quot;\t&quot;} {split($7,l1,&quot;:&quot;); print $14, $15, $16, $4, $5, $6, $14&quot;:&quot;$15&quot;-&quot;$16&quot;::&quot;l1[4], $8, $9, $10, $11, $12, $13}&#39;</span><span class="w"> </span><span class="p">|</span>awk<span class="w"> </span><span class="s1">&#39;BEGIN{FS=&quot;\t&quot;; OFS=&quot;\t&quot;} {lines[$11&quot;:::&quot;$7]=$0; score[$11&quot;:::&quot;$7]+=$8; nb[$11&quot;:::&quot;$7]++} END{for(u in lines){split(u,parts,&quot;:::&quot;); split(lines[u],l,&quot;\t&quot;); print l[1], l[2], l[3], l[4], l[5], l[6], parts[2], score[u]/nb[u], l[9], l[10], l[11], l[12], l[13], &quot;.&quot;}}&#39;</span><span class="w"> </span><span class="p">|</span>bedtools<span class="w"> </span>sort<span class="w"> </span>-faidx<span class="w"> </span>chr_sizes<span class="w"> </span>-i<span class="w"> </span>stdin<span class="w"> </span>&gt;<span class="w"> </span>Nasser2021ABCPredictions.all_biosamples.all_putative_enhancers.merged_enhancers.sorted.bedpe
</code></pre></div>
<p>After a pretty long time:</p>
<blockquote>
<div class="highlight"><pre><span></span><code>wc<span class="w"> </span>-l<span class="w"> </span>Nasser2021ABCPredictions.all_biosamples.all_putative_enhancers.merged_enhancers.sorted.new.bedpe
</code></pre></div>
<p>6,086,345</p>
<p>To ensure that the Awk array has not been "cropped out" for insufficient memory reasons, we executed  the same code with twice as much max memory - eg 64 GB - to see if there is a difference (as it simply gives nothing with a small max memory) and it gave the same result =&gt; OK
</p>
</blockquote>
<p>Note that by construction, <code>Nasser2021ABCPredictions.all_biosamples.all_putative_enhancers.merged_enhancers.sorted.bedpe</code> contains almost duplicated rows when identical E-G are found in different biosamples, the only difference being the ABC score, the biosample and the distance fields.</p>
<p>We will also need the corresponding a similar files where all unique E-G correspond to unique lines with $11 is the list of biosamples, $13 the list of initial E-TSS distances (before replacing the enhancers by merged enhancers) and $14 the list of tissues + other additional fields are defined, see header of the result.</p>
<p>For that purpose, we use the <code>pandas</code> library from Python, which is more efficient. We use the script <code>reduce_df.py</code>, the content of which is:</p>
<div class="highlight"><pre><span></span><code><span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">import</span> <span class="nn">numpy</span> <span class="k">as</span> <span class="nn">np</span>
<span class="kn">import</span> <span class="nn">argparse</span>
<span class="kn">import</span> <span class="nn">sys</span>

<span class="c1"># Create the parser</span>
<span class="n">parser</span> <span class="o">=</span> <span class="n">argparse</span><span class="o">.</span><span class="n">ArgumentParser</span><span class="p">(</span><span class="n">description</span><span class="o">=</span><span class="s2">&quot;&lt;input file&gt;, &lt;output file&gt;&quot;</span><span class="p">)</span>

<span class="c1"># Add the arguments</span>
<span class="n">parser</span><span class="o">.</span><span class="n">add_argument</span><span class="p">(</span><span class="s2">&quot;-i&quot;</span><span class="p">,</span>
                       <span class="n">metavar</span><span class="o">=</span><span class="s2">&quot;path&quot;</span><span class="p">,</span>
                       <span class="nb">type</span><span class="o">=</span><span class="nb">str</span><span class="p">,</span>
                       <span class="n">required</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                       <span class="n">help</span><span class="o">=</span><span class="s2">&quot;path to input file&quot;</span><span class="p">)</span>
<span class="n">parser</span><span class="o">.</span><span class="n">add_argument</span><span class="p">(</span><span class="s2">&quot;-o&quot;</span><span class="p">,</span>
                       <span class="n">metavar</span><span class="o">=</span><span class="s2">&quot;path&quot;</span><span class="p">,</span>
                       <span class="nb">type</span><span class="o">=</span><span class="nb">str</span><span class="p">,</span>
                       <span class="n">required</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
                       <span class="n">help</span><span class="o">=</span><span class="s2">&quot;path to output file&quot;</span><span class="p">)</span>

<span class="n">args</span> <span class="o">=</span> <span class="n">parser</span><span class="o">.</span><span class="n">parse_args</span><span class="p">()</span>

<span class="n">input_path</span> <span class="o">=</span> <span class="n">args</span><span class="o">.</span><span class="n">i</span>
<span class="n">output_path</span> <span class="o">=</span> <span class="n">args</span><span class="o">.</span><span class="n">o</span>

<span class="n">colnames</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;chrom1&quot;</span><span class="p">,</span> <span class="s2">&quot;start1&quot;</span><span class="p">,</span> <span class="s2">&quot;end1&quot;</span><span class="p">,</span> <span class="s2">&quot;chrom2&quot;</span><span class="p">,</span> <span class="s2">&quot;start2&quot;</span><span class="p">,</span> <span class="s2">&quot;end2&quot;</span><span class="p">,</span> <span class="s2">&quot;name&quot;</span><span class="p">,</span>
            <span class="s2">&quot;ABC.score&quot;</span><span class="p">,</span> <span class="s2">&quot;strand1&quot;</span><span class="p">,</span> <span class="s2">&quot;strand2&quot;</span><span class="p">,</span> <span class="s2">&quot;biosample&quot;</span><span class="p">,</span> <span class="s2">&quot;gene&quot;</span><span class="p">,</span> <span class="s2">&quot;original_distance&quot;</span><span class="p">,</span> <span class="s2">&quot;tissue&quot;</span><span class="p">]</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Loading input...&quot;</span><span class="p">)</span>
<span class="n">eg_pairs</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">input_path</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">dtype</span><span class="o">=</span><span class="s1">&#39;str&#39;</span><span class="p">,</span> <span class="n">engine</span><span class="o">=</span><span class="s1">&#39;c&#39;</span><span class="p">,</span> <span class="n">names</span> <span class="o">=</span> <span class="n">colnames</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Done.&quot;</span><span class="p">)</span>
<span class="n">eg_pairs</span><span class="p">[[</span><span class="s2">&quot;start1&quot;</span><span class="p">,</span> <span class="s2">&quot;end1&quot;</span><span class="p">,</span> <span class="s2">&quot;start2&quot;</span><span class="p">,</span> <span class="s2">&quot;end2&quot;</span><span class="p">,</span> <span class="s2">&quot;ABC.score&quot;</span><span class="p">,</span> <span class="s2">&quot;original_distance&quot;</span><span class="p">]]</span> <span class="o">=</span> <span class="n">eg_pairs</span><span class="p">[[</span><span class="s2">&quot;start1&quot;</span><span class="p">,</span> <span class="s2">&quot;end1&quot;</span><span class="p">,</span> <span class="s2">&quot;start2&quot;</span><span class="p">,</span> <span class="s2">&quot;end2&quot;</span><span class="p">,</span> <span class="s2">&quot;ABC.score&quot;</span><span class="p">,</span> <span class="s2">&quot;original_distance&quot;</span><span class="p">]]</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="n">pd</span><span class="o">.</span><span class="n">to_numeric</span><span class="p">)</span>
<span class="n">eg_pairs</span><span class="p">[[</span><span class="s2">&quot;ABC.score&quot;</span><span class="p">,</span> <span class="s2">&quot;original_distance&quot;</span><span class="p">]]</span><span class="o">.</span><span class="n">astype</span><span class="p">(</span><span class="nb">float</span><span class="p">,</span> <span class="n">copy</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>

<span class="c1"># For testing purposes we keep only a few rows</span>
<span class="c1">#eg_pairs = eg_pairs.iloc[:10000,:]</span>

<span class="n">column_identical_for_unique_EG_pair</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;chrom1&quot;</span><span class="p">,</span> <span class="s2">&quot;start1&quot;</span><span class="p">,</span> <span class="s2">&quot;end1&quot;</span><span class="p">,</span> <span class="s2">&quot;chrom2&quot;</span><span class="p">,</span> <span class="s2">&quot;start2&quot;</span><span class="p">,</span> <span class="s2">&quot;end2&quot;</span><span class="p">,</span> <span class="s2">&quot;name&quot;</span><span class="p">,</span>
                                       <span class="s2">&quot;strand1&quot;</span><span class="p">,</span> <span class="s2">&quot;strand2&quot;</span><span class="p">,</span> <span class="s2">&quot;gene&quot;</span><span class="p">]</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Reducing input (this may take a while)...&quot;</span><span class="p">)</span>
<span class="n">eg_reduced</span> <span class="o">=</span> <span class="p">(</span>
    <span class="n">eg_pairs</span><span class="o">.</span><span class="n">groupby</span><span class="p">(</span><span class="n">column_identical_for_unique_EG_pair</span><span class="p">)</span>
    <span class="o">.</span><span class="n">agg</span><span class="p">({</span><span class="s2">&quot;ABC.score&quot;</span><span class="p">:</span> <span class="nb">list</span><span class="p">,</span> <span class="s2">&quot;biosample&quot;</span><span class="p">:</span> <span class="nb">list</span><span class="p">,</span> <span class="s2">&quot;original_distance&quot;</span><span class="p">:</span> <span class="nb">list</span><span class="p">,</span> <span class="s2">&quot;tissue&quot;</span><span class="p">:</span> <span class="nb">list</span><span class="p">})</span>
    <span class="o">.</span><span class="n">reset_index</span><span class="p">()</span>
    <span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Done.&quot;</span><span class="p">)</span>
<span class="n">eg_reduced</span><span class="o">.</span><span class="n">rename</span><span class="p">({</span><span class="s2">&quot;ABC.score&quot;</span><span class="p">:</span> <span class="s2">&quot;ABC.scores&quot;</span><span class="p">,</span> <span class="s2">&quot;biosample&quot;</span><span class="p">:</span> <span class="s2">&quot;biosamples&quot;</span><span class="p">,</span> <span class="s2">&quot;original_distance&quot;</span><span class="p">:</span> <span class="s2">&quot;original_distances&quot;</span><span class="p">,</span> <span class="s2">&quot;tissue&quot;</span><span class="p">:</span> <span class="s2">&quot;tissues&quot;</span><span class="p">},</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>

<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Adding more columns...&quot;</span><span class="p">)</span>
<span class="n">eg_reduced</span><span class="p">[</span><span class="s2">&quot;ABC.mean&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="n">eg_reduced</span><span class="p">[</span><span class="s2">&quot;ABC.scores&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">mean</span><span class="p">)</span>
<span class="n">eg_reduced</span><span class="p">[</span><span class="s2">&quot;ABC.min&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="n">eg_reduced</span><span class="p">[</span><span class="s2">&quot;ABC.scores&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">min</span><span class="p">)</span>
<span class="n">eg_reduced</span><span class="p">[</span><span class="s2">&quot;ABC.max&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="n">eg_reduced</span><span class="p">[</span><span class="s2">&quot;ABC.scores&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">max</span><span class="p">)</span>

<span class="n">eg_reduced</span><span class="p">[</span><span class="s2">&quot;original_distance.mean&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="n">eg_reduced</span><span class="p">[</span><span class="s2">&quot;original_distances&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">mean</span><span class="p">)</span>
<span class="n">eg_reduced</span><span class="p">[</span><span class="s2">&quot;original_distance.min&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="n">eg_reduced</span><span class="p">[</span><span class="s2">&quot;original_distances&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">min</span><span class="p">)</span>
<span class="n">eg_reduced</span><span class="p">[</span><span class="s2">&quot;original_distance.max&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="n">eg_reduced</span><span class="p">[</span><span class="s2">&quot;original_distances&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">max</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Done.&quot;</span><span class="p">)</span>

<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Preparing output...&quot;</span><span class="p">)</span>
<span class="n">eg_reduced</span><span class="p">[</span><span class="s2">&quot;ABC.scores&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="n">eg_reduced</span><span class="p">[</span><span class="s2">&quot;ABC.scores&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="s1">&#39;,&#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="nb">list</span><span class="p">(</span><span class="nb">map</span><span class="p">(</span><span class="nb">str</span><span class="p">,</span> <span class="n">x</span><span class="p">))))</span>
<span class="n">eg_reduced</span><span class="p">[</span><span class="s2">&quot;original_distances&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="n">eg_reduced</span><span class="p">[</span><span class="s2">&quot;original_distances&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="s1">&#39;,&#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="nb">list</span><span class="p">(</span><span class="nb">map</span><span class="p">(</span><span class="nb">str</span><span class="p">,</span> <span class="n">x</span><span class="p">))))</span>
<span class="n">eg_reduced</span><span class="p">[</span><span class="s2">&quot;biosamples&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="n">eg_reduced</span><span class="p">[</span><span class="s2">&quot;biosamples&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="s1">&#39;,&#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">x</span><span class="p">))</span>
<span class="n">eg_reduced</span><span class="p">[</span><span class="s2">&quot;tissues&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="n">eg_reduced</span><span class="p">[</span><span class="s2">&quot;tissues&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="s1">&#39;,&#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">x</span><span class="p">))</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Done&quot;</span><span class="p">)</span>

<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Writing output... Don&#39;t interrupt...&quot;</span><span class="p">)</span>
<span class="n">eg_reduced</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="n">output_path</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">index</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Done.&quot;</span><span class="p">)</span>
<span class="n">sys</span><span class="o">.</span><span class="n">exit</span><span class="p">(</span><span class="mi">0</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code><span class="c1"># requires a lot of memory, better use at least 32GB</span>
python<span class="w"> </span>reduce_df.py<span class="w"> </span>-i<span class="w"> </span>Nasser2021ABCPredictions.all_biosamples.all_putative_enhancers.merged_enhancers.sorted.bedpe<span class="w"> </span>-o<span class="w"> </span>Nasser2021ABCPredictions.all_biosamples.all_putative_enhancers.merged_enhancers.sorted.uniques_eg.bedpe
</code></pre></div>
<blockquote>
<div class="highlight"><pre><span></span><code>wc<span class="w"> </span>-l<span class="w"> </span>Nasser2021ABCPredictions.all_biosamples.all_putative_enhancers.merged_enhancers.sorted.bedpe
</code></pre></div>
<p>6,086,345</p>
<div class="highlight"><pre><span></span><code>wc<span class="w"> </span>-l<span class="w"> </span>Nasser2021ABCPredictions.all_biosamples.all_putative_enhancers.merged_enhancers.sorted.uniques_eg.bedpe
</code></pre></div>
<p>1,041,155</p>
</blockquote>
<h3 id="select-biosamples">Select biosamples<a class="headerlink" href="#select-biosamples" title="Permanent link">&para;</a></h3>
<h4 id="all-liver-and-intestine-biosamples">All liver and intestine biosamples<a class="headerlink" href="#all-liver-and-intestine-biosamples" title="Permanent link">&para;</a></h4>
<p>The five biosamples of interest are:</p>
<blockquote>
<div class="highlight"><pre><span></span><code>hepatocyte-ENCODE
HepG2-Roadmap<span class="w"> </span><span class="c1"># human liver cancer cell line</span>
liver-ENCODE
large_intestine_fetal-Roadmap
small_intestine_fetal-Roadmap
</code></pre></div>
</blockquote>
<p>We extract corresponding data:</p>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span><span class="s1">&#39;BEGIN{FS=&quot;\t&quot;; OFS=&quot;\t&quot;} {bool=0; if($11~/^(hepatocyte-ENCODE|HepG2-Roadmap|liver-ENCODE)$/){bool=1; tissue=&quot;liver&quot;} else if ($11~/^(large_intestine_fetal-Roadmap|small_intestine_fetal-Roadmap)$/){bool=1; tissue=&quot;intestine&quot;}; if(bool){print $1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12, $13, tissue}}&#39;</span><span class="w"> </span>Nasser2021ABCPredictions.all_biosamples.all_putative_enhancers.merged_enhancers.sorted.bedpe<span class="w"> </span>&gt;<span class="w"> </span>Nasser2021ABCPredictions.liver_and_intestine.all_putative_enhancers.merged_enhancers.sorted.bedpe
</code></pre></div>
<blockquote>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;{tissues[$14]++} END{for(u in tissues){print u, tissues[u]}}&#39;</span><span class="w"> </span>Nasser2021ABCPredictions.liver_and_intestine.all_putative_enhancers.merged_enhancers.sorted.bedpe
</code></pre></div>
<div class="highlight"><pre><span></span><code>liver<span class="w"> </span><span class="m">145132</span>
intestine<span class="w"> </span><span class="m">101488</span>
</code></pre></div>
</blockquote>
<p>Note that by construction, <code>Nasser2021ABCPredictions.liver_and_intestine.all_putative_enhancers.merged_enhancers.sorted.bedpe</code> contains almost duplicated rows when identical E-G are found in different biosamples, the only difference being the ABC score, the biosample and the distance fields.</p>
<p>Just as before, we use <code>reduce_df.py</code> to compute a bedpe in which all E-G pairs correspond to unique lines with $11 is the list of biosamples, $13 the list of initial E-TSS distances (before replacing the enhancers by merged enhancers) and $14 the list of tissues + other additional fields are defined, see header of the result.</p>
<div class="highlight"><pre><span></span><code>python<span class="w"> </span>reduce_df.py<span class="w"> </span>-i<span class="w"> </span>Nasser2021ABCPredictions.liver_and_intestine.all_putative_enhancers.merged_enhancers.sorted.bedpe<span class="w"> </span>-o<span class="w"> </span>Nasser2021ABCPredictions.liver_and_intestine.all_putative_enhancers.merged_enhancers.sorted.uniques_eg.bedpe
</code></pre></div>
<blockquote>
<div class="highlight"><pre><span></span><code>wc<span class="w"> </span>-l<span class="w"> </span>Nasser2021ABCPredictions.liver_and_intestine.all_putative_enhancers.merged_enhancers.sorted.bedpe
</code></pre></div>
<p>246,620</p>
<div class="highlight"><pre><span></span><code>wc<span class="w"> </span>-l<span class="w"> </span>Nasser2021ABCPredictions.liver_and_intestine.all_putative_enhancers.merged_enhancers.sorted.uniques_eg.bedpe
</code></pre></div>
<p>147,748</p>
</blockquote>
<p>We do the same with unmerged (original) enhancers:</p>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span><span class="s1">&#39;BEGIN{FS=&quot;\t&quot;; OFS=&quot;\t&quot;} {bool=0; if($11~/^(hepatocyte-ENCODE|HepG2-Roadmap|liver-ENCODE)$/){bool=1; tissue=&quot;liver&quot;} else if ($11~/^(large_intestine_fetal-Roadmap|small_intestine_fetal-Roadmap)$/){bool=1; tissue=&quot;intestine&quot;}; if(bool){print $1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12, $13, tissue}}&#39;</span><span class="w"> </span>Nasser2021ABCPredictions.all_biosamples.all_putative_enhancers.sorted.bedpe<span class="w"> </span>&gt;<span class="w"> </span>Nasser2021ABCPredictions.liver_and_intestine.all_putative_enhancers.sorted.bedpe
</code></pre></div>
<blockquote>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;{tissues[$14]++} END{for(u in tissues){print u, tissues[u]}}&#39;</span><span class="w"> </span>Nasser2021ABCPredictions.liver_and_intestine.all_putative_enhancers.sorted.bedpe
</code></pre></div>
<div class="highlight"><pre><span></span><code>liver<span class="w"> </span><span class="m">157901</span>
intestine<span class="w"> </span><span class="m">108221</span>
</code></pre></div>
</blockquote>
<p>Just as before, we use <code>reduce_df.py</code> to compute a bedpe in which all E-G pairs correspond to unique lines with $11 is the list of biosamples, $13 the list of initial E-TSS distances (before replacing the enhancers by merged enhancers) and $14 the list of tissues + other additional fields are defined, see header of the result.</p>
<div class="highlight"><pre><span></span><code>python<span class="w"> </span>reduce_df.py<span class="w"> </span>-i<span class="w"> </span>Nasser2021ABCPredictions.liver_and_intestine.all_putative_enhancers.sorted.bedpe<span class="w"> </span>-o<span class="w"> </span>Nasser2021ABCPredictions.liver_and_intestine.all_putative_enhancers.sorted.uniques_eg.bedpe
</code></pre></div>
<blockquote>
<div class="highlight"><pre><span></span><code>wc<span class="w"> </span>-l<span class="w"> </span>Nasser2021ABCPredictions.liver_and_intestine.all_putative_enhancers.sorted.bedpe
</code></pre></div>
<p>266,122</p>
<div class="highlight"><pre><span></span><code>wc<span class="w"> </span>-l<span class="w"> </span>Nasser2021ABCPredictions.liver_and_intestine.all_putative_enhancers.sorted.uniques_eg.bedpe
</code></pre></div>
<p>265,975</p>
</blockquote>
<p>And finally, we do the same again with enhancers merged only over the selected biosamples, ie the 5 liver and intestine biosamples:</p>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span><span class="s1">&#39;BEGIN{FS=&quot;\t&quot;; OFS=&quot;\t&quot;} {print $1, $2, $3}&#39;</span><span class="w"> </span>Nasser2021ABCPredictions.liver_and_intestine.all_putative_enhancers.sorted.bedpe<span class="w"> </span><span class="p">|</span>uniq<span class="w"> </span>&gt;<span class="w"> </span>list_enhancers.liver_and_intestine.bed
</code></pre></div>
<p>Contains 102,996 enhancers.</p>
<div class="highlight"><pre><span></span><code>bedtools<span class="w"> </span>merge<span class="w"> </span>-i<span class="w"> </span>list_enhancers.liver_and_intestine.bed<span class="w"> </span>&gt;<span class="w"> </span>list_enhancers.liver_and_intestine.merged.bed
</code></pre></div>
<p>Contains 57,398 enhancers.</p>
<p>So we replace all enhancers coordinates in the result, by that of the corresponding merged enhancers in <code>list_enhancers.liver_and_intestine.merged.bed</code>.</p>
<div class="highlight"><pre><span></span><code>bedtools<span class="w"> </span>intersect<span class="w"> </span>-wb<span class="w"> </span>-a<span class="w"> </span>Nasser2021ABCPredictions.liver_and_intestine.all_putative_enhancers.sorted.bedpe<span class="w"> </span>-b<span class="w"> </span>list_enhancers.liver_and_intestine.merged.bed<span class="w"> </span><span class="p">|</span>awk<span class="w"> </span><span class="s1">&#39;BEGIN{FS=&quot;\t&quot;; OFS=&quot;\t&quot;} {split($7,l1,&quot;:&quot;); print $15, $16, $17, $4, $5, $6, $15&quot;:&quot;$16&quot;-&quot;$17&quot;::&quot;l1[4], $8, $9, $10, $11, $12, $13, $14}&#39;</span><span class="w"> </span><span class="p">|</span>awk<span class="w"> </span><span class="s1">&#39;BEGIN{FS=&quot;\t&quot;; OFS=&quot;\t&quot;} {lines[$11&quot;:::&quot;$7]=$0; score[$11&quot;:::&quot;$7]+=$8; nb[$11&quot;:::&quot;$7]++} END{for(u in lines){split(u,parts,&quot;:::&quot;); split(lines[u],l,&quot;\t&quot;); print l[1], l[2], l[3], l[4], l[5], l[6], parts[2], score[u]/nb[u], l[9], l[10], l[11], l[12], l[13], l[14]}}&#39;</span><span class="w"> </span><span class="p">|</span>bedtools<span class="w"> </span>sort<span class="w"> </span>-faidx<span class="w"> </span>chr_sizes<span class="w"> </span>-i<span class="w"> </span>stdin<span class="w"> </span>&gt;<span class="w"> </span>Nasser2021ABCPredictions.liver_and_intestine.all_putative_enhancers.merged_liver_and_intestine_enhancers.sorted.bedpe
</code></pre></div>
<blockquote>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;{tissues[$14]++} END{for(u in tissues){print u, tissues[u]}}&#39;</span><span class="w"> </span>Nasser2021ABCPredictions.liver_and_intestine.all_putative_enhancers.merged_liver_and_intestine_enhancers.sorted.bedpe
</code></pre></div>
<div class="highlight"><pre><span></span><code>liver<span class="w"> </span><span class="m">153404</span>
intestine<span class="w"> </span><span class="m">105800</span>
</code></pre></div>
</blockquote>
<p>Just as before, we use <code>reduce_df.py</code> to compute a bedpe in which all E-G pairs correspond to unique lines with $11 is the list of biosamples, $13 the list of initial E-TSS distances (before replacing the enhancers by merged enhancers) and $14 the list of tissues + other additional fields are defined, see header of the result.</p>
<div class="highlight"><pre><span></span><code>python<span class="w"> </span>reduce_df.py<span class="w"> </span>-i<span class="w"> </span>Nasser2021ABCPredictions.liver_and_intestine.all_putative_enhancers.merged_liver_and_intestine_enhancers.sorted.bedpe<span class="w"> </span>-o<span class="w"> </span>Nasser2021ABCPredictions.liver_and_intestine.all_putative_enhancers.merged_liver_and_intestine_enhancers.sorted.uniques_eg.bedpe
</code></pre></div>
<blockquote>
<div class="highlight"><pre><span></span><code>wc<span class="w"> </span>-l<span class="w"> </span>Nasser2021ABCPredictions.liver_and_intestine.all_putative_enhancers.merged_liver_and_intestine_enhancers.sorted.bedp
</code></pre></div>
<p>259,204</p>
<div class="highlight"><pre><span></span><code>wc<span class="w"> </span>-l<span class="w"> </span>Nasser2021ABCPredictions.liver_and_intestine.all_putative_enhancers.merged_liver_and_intestine_enhancers.sorted.uniques_eg.bedpe
</code></pre></div>
<p>159,675</p>
</blockquote>
<h4 id="all-liver-biosamples">All liver biosamples<a class="headerlink" href="#all-liver-biosamples" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span><span class="s1">&#39;$14==&quot;liver&quot;&#39;</span><span class="w"> </span>Nasser2021ABCPredictions.liver_and_intestine.all_putative_enhancers.merged_enhancers.sorted.bedpe<span class="w"> </span>&gt;<span class="w"> </span>Nasser2021ABCPredictions.liver.all_putative_enhancers.merged_enhancers.sorted.bedpe
</code></pre></div>
<p>OK.</p>
<p>Just as before, we use <code>reduce_df.py</code> to compute a bedpe in which all E-G pairs correspond to unique lines with $11 is the list of biosamples, $13 the list of initial E-TSS distances (before replacing the enhancers by merged enhancers) and $14 the list of tissues + other additional fields are defined, see header of the result.</p>
<div class="highlight"><pre><span></span><code>python<span class="w"> </span>reduce_df.py<span class="w"> </span>-i<span class="w"> </span>Nasser2021ABCPredictions.liver.all_putative_enhancers.merged_enhancers.sorted.bedpe<span class="w"> </span>-o<span class="w"> </span>Nasser2021ABCPredictions.liver.all_putative_enhancers.merged_enhancers.sorted.uniques_egbedpe
</code></pre></div>
<p>We do the same with unmerged (original) enhancers:</p>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span><span class="s1">&#39;$14==&quot;liver&quot;&#39;</span><span class="w"> </span>Nasser2021ABCPredictions.liver_and_intestine.all_putative_enhancers.sorted.bedpe<span class="w"> </span>&gt;<span class="w"> </span>Nasser2021ABCPredictions.liver.all_putative_enhancers.sorted.bedpe
</code></pre></div>
<p>Just as before, we use <code>reduce_df.py</code> to compute a bedpe in which all E-G pairs correspond to unique lines with $11 is the list of biosamples, $13 the list of initial E-TSS distances (before replacing the enhancers by merged enhancers) and $14 the list of tissues + other additional fields are defined, see header of the result.</p>
<div class="highlight"><pre><span></span><code>python<span class="w"> </span>reduce_df.py<span class="w"> </span>-i<span class="w"> </span>Nasser2021ABCPredictions.liver.all_putative_enhancers.sorted.bedpe<span class="w"> </span>-o<span class="w"> </span>Nasser2021ABCPredictions.liver.all_putative_enhancers.sorted.uniques_eg.bedpe
</code></pre></div>
<p>And finally, we do the same again with enhancers merged only over the selected biosamples, ie the 3 liver biosamples:</p>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span><span class="s1">&#39;BEGIN{FS=&quot;\t&quot;; OFS=&quot;\t&quot;} {print $1, $2, $3}&#39;</span><span class="w"> </span>Nasser2021ABCPredictions.liver.all_putative_enhancers.sorted.bedpe<span class="w"> </span><span class="p">|</span>uniq<span class="w"> </span>&gt;<span class="w"> </span>list_enhancers.liver.bed
</code></pre></div>
<p>Contains 61,838 enhancers.</p>
<div class="highlight"><pre><span></span><code>bedtools<span class="w"> </span>merge<span class="w"> </span>-i<span class="w"> </span>list_enhancers.liver.bed<span class="w"> </span>&gt;<span class="w"> </span>list_enhancers.liver.merged.bed
</code></pre></div>
<p>Contains 44,979 enhancers.</p>
<p>So we replace all enhancers coordinates in the result, by that of the corresponding merged enhancers in <code>list_enhancers.liver.merged.bed</code>.</p>
<div class="highlight"><pre><span></span><code>bedtools<span class="w"> </span>intersect<span class="w"> </span>-wb<span class="w"> </span>-a<span class="w"> </span>Nasser2021ABCPredictions.liver.all_putative_enhancers.sorted.bedpe<span class="w"> </span>-b<span class="w"> </span>list_enhancers.liver.merged.bed<span class="w"> </span><span class="p">|</span>awk<span class="w"> </span><span class="s1">&#39;BEGIN{FS=&quot;\t&quot;; OFS=&quot;\t&quot;} {split($7,l1,&quot;:&quot;); print $15, $16, $17, $4, $5, $6, $15&quot;:&quot;$16&quot;-&quot;$17&quot;::&quot;l1[4], $8, $9, $10, $11, $12, $13, $14}&#39;</span><span class="w"> </span><span class="p">|</span>awk<span class="w"> </span><span class="s1">&#39;BEGIN{FS=&quot;\t&quot;; OFS=&quot;\t&quot;} {lines[$11&quot;:::&quot;$7]=$0; score[$11&quot;:::&quot;$7]+=$8; nb[$11&quot;:::&quot;$7]++} END{for(u in lines){split(u,parts,&quot;:::&quot;); split(lines[u],l,&quot;\t&quot;); print l[1], l[2], l[3], l[4], l[5], l[6], parts[2], score[u]/nb[u], l[9], l[10], l[11], l[12], l[13], l[14]}}&#39;</span><span class="w"> </span><span class="p">|</span>bedtools<span class="w"> </span>sort<span class="w"> </span>-faidx<span class="w"> </span>chr_sizes<span class="w"> </span>-i<span class="w"> </span>stdin<span class="w"> </span>&gt;<span class="w"> </span>Nasser2021ABCPredictions.liver.all_putative_enhancers.merged_liver_enhancers.sorted.bedpe
</code></pre></div>
<blockquote>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;{tissues[$14]++} END{for(u in tissues){print u, tissues[u]}}&#39;</span><span class="w"> </span>Nasser2021ABCPredictions.liver.all_putative_enhancers.merged_liver_enhancers.sorted.bedpe
</code></pre></div>
<div class="highlight"><pre><span></span><code>liver<span class="w"> </span><span class="m">155293</span>
</code></pre></div>
</blockquote>
<p>Just as before, we use <code>reduce_df.py</code> to compute a bedpe in which all E-G pairs correspond to unique lines with $11 is the list of biosamples, $13 the list of initial E-TSS distances (before replacing the enhancers by merged enhancers) and $14 the list of tissues + other additional fields are defined, see header of the result.</p>
<div class="highlight"><pre><span></span><code>python<span class="w"> </span>reduce_df.py<span class="w"> </span>-i<span class="w"> </span>Nasser2021ABCPredictions.liver.all_putative_enhancers.merged_liver_enhancers.sorted.bedpe<span class="w"> </span>-o<span class="w"> </span>Nasser2021ABCPredictions.liver.all_putative_enhancers.merged_liver_enhancers.sorted.uniques_eg.bedpe
</code></pre></div>
<h4 id="all-intestine-biosamples">All intestine biosamples<a class="headerlink" href="#all-intestine-biosamples" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span><span class="s1">&#39;$14==&quot;intestine&quot;&#39;</span><span class="w"> </span>Nasser2021ABCPredictions.liver_and_intestine.all_putative_enhancers.merged_enhancers.sorted.bedpe<span class="w"> </span>&gt;<span class="w"> </span>Nasser2021ABCPredictions.intestine.all_putative_enhancers.merged_enhancers.sorted.bedpe
</code></pre></div>
<p>OK.</p>
<p>Just as before, we use <code>reduce_df.py</code> to compute a bedpe in which all E-G pairs correspond to unique lines with $11 is the list of biosamples, $13 the list of initial E-TSS distances (before replacing the enhancers by merged enhancers) and $14 the list of tissues + other additional fields are defined, see header of the result.</p>
<div class="highlight"><pre><span></span><code>python<span class="w"> </span>reduce_df.py<span class="w"> </span>-i<span class="w"> </span>Nasser2021ABCPredictions.intestine.all_putative_enhancers.merged_enhancers.sorted.bedpe<span class="w"> </span>-o<span class="w"> </span>Nasser2021ABCPredictions.intestine.all_putative_enhancers.merged_enhancers.sorted.uniques_eg.bedpe
</code></pre></div>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span><span class="s1">&#39;$14==&quot;intestine&quot;&#39;</span><span class="w"> </span>Nasser2021ABCPredictions.liver_and_intestine.all_putative_enhancers.sorted.bedpe<span class="w"> </span>&gt;<span class="w"> </span>Nasser2021ABCPredictions.intestine.all_putative_enhancers.sorted.bedpe
</code></pre></div>
<p>Just as before, we use <code>reduce_df.py</code> to compute a bedpe in which all E-G pairs correspond to unique lines with $11 is the list of biosamples, $13 the list of initial E-TSS distances (before replacing the enhancers by merged enhancers) and $14 the list of tissues + other additional fields are defined, see header of the result.</p>
<div class="highlight"><pre><span></span><code>python<span class="w"> </span>reduce_df.py<span class="w"> </span>-i<span class="w"> </span>Nasser2021ABCPredictions.intestine.all_putative_enhancers.sorted.bedpe<span class="w"> </span>-o<span class="w"> </span>Nasser2021ABCPredictions.intestine.all_putative_enhancers.sorted.uniques_eg.bedpe
</code></pre></div>
<p>And finally, we do the same again with enhancers merged only over the selected biosamples, ie the 2 intestine biosamples:</p>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span><span class="s1">&#39;BEGIN{FS=&quot;\t&quot;; OFS=&quot;\t&quot;} {print $1, $2, $3}&#39;</span><span class="w"> </span>Nasser2021ABCPredictions.intestine.all_putative_enhancers.sorted.bedpe<span class="w"> </span><span class="p">|</span>uniq<span class="w"> </span>&gt;<span class="w"> </span>list_enhancers.intestine.bed
</code></pre></div>
<p>Contains 41,102 enhancers.</p>
<div class="highlight"><pre><span></span><code>bedtools<span class="w"> </span>merge<span class="w"> </span>-i<span class="w"> </span>list_enhancers.intestine.bed<span class="w"> </span>&gt;<span class="w"> </span>list_enhancers.intestine.merged.bed
</code></pre></div>
<p>Contains 27,102 enhancers.</p>
<p>So we replace all enhancers coordinates in the result, by that of the corresponding merged enhancers in <code>list_enhancers.liver.merged.bed</code>.</p>
<div class="highlight"><pre><span></span><code>bedtools<span class="w"> </span>intersect<span class="w"> </span>-wb<span class="w"> </span>-a<span class="w"> </span>Nasser2021ABCPredictions.intestine.all_putative_enhancers.sorted.bedpe<span class="w"> </span>-b<span class="w"> </span>list_enhancers.intestine.merged.bed<span class="w"> </span><span class="p">|</span>awk<span class="w"> </span><span class="s1">&#39;BEGIN{FS=&quot;\t&quot;; OFS=&quot;\t&quot;} {split($7,l1,&quot;:&quot;); print $15, $16, $17, $4, $5, $6, $15&quot;:&quot;$16&quot;-&quot;$17&quot;::&quot;l1[4], $8, $9, $10, $11, $12, $13, $14}&#39;</span><span class="w"> </span><span class="p">|</span>awk<span class="w"> </span><span class="s1">&#39;BEGIN{FS=&quot;\t&quot;; OFS=&quot;\t&quot;} {lines[$11&quot;:::&quot;$7]=$0; score[$11&quot;:::&quot;$7]+=$8; nb[$11&quot;:::&quot;$7]++} END{for(u in lines){split(u,parts,&quot;:::&quot;); split(lines[u],l,&quot;\t&quot;); print l[1], l[2], l[3], l[4], l[5], l[6], parts[2], score[u]/nb[u], l[9], l[10], l[11], l[12], l[13], l[14]}}&#39;</span><span class="w"> </span><span class="p">|</span>bedtools<span class="w"> </span>sort<span class="w"> </span>-faidx<span class="w"> </span>chr_sizes<span class="w"> </span>-i<span class="w"> </span>stdin<span class="w"> </span>&gt;<span class="w"> </span>Nasser2021ABCPredictions.intestine.all_putative_enhancers.merged_intestine_enhancers.sorted.bedpe
</code></pre></div>
<blockquote>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;{tissues[$14]++} END{for(u in tissues){print u, tissues[u]}}&#39;</span><span class="w"> </span>Nasser2021ABCPredictions.intestine.all_putative_enhancers.merged_intestine_enhancers.sorted.bedpe
</code></pre></div>
<div class="highlight"><pre><span></span><code>intestine<span class="w"> </span><span class="m">106883</span>
</code></pre></div>
</blockquote>
<p>Just as before, we use <code>reduce_df.py</code> to compute a bedpe in which all E-G pairs correspond to unique lines with $11 is the list of biosamples, $13 the list of initial E-TSS distances (before replacing the enhancers by merged enhancers) and $14 the list of tissues + other additional fields are defined, see header of the result.</p>
<div class="highlight"><pre><span></span><code>python<span class="w"> </span>reduce_df.py<span class="w"> </span>-i<span class="w"> </span>Nasser2021ABCPredictions.intestine.all_putative_enhancers.merged_intestine_enhancers.sorted.bedpe<span class="w"> </span>-o<span class="w"> </span>Nasser2021ABCPredictions.intestine.all_putative_enhancers.merged_intestine_enhancers.sorted.uniques_eg.bedpe
</code></pre></div>
<h4 id="all-5-separate-biosamples-of-interest">All 5 <strong><em>separate</em></strong> biosamples of interest<a class="headerlink" href="#all-5-separate-biosamples-of-interest" title="Permanent link">&para;</a></h4>
<h5 id="hepatocyte-encode">hepatocyte-ENCODE<a class="headerlink" href="#hepatocyte-encode" title="Permanent link">&para;</a></h5>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span><span class="s1">&#39;$11==&quot;hepatocyte-ENCODE&quot;&#39;</span><span class="w"> </span>Nasser2021ABCPredictions.liver_and_intestine.all_putative_enhancers.merged_enhancers.sorted.bedpe<span class="w"> </span>&gt;<span class="w"> </span>Nasser2021ABCPredictions.hepatocyte-ENCODE.all_putative_enhancers.merged_enhancers.sorted.bedpe
</code></pre></div>
<p>OK.</p>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span><span class="s1">&#39;$11==&quot;hepatocyte-ENCODE&quot;&#39;</span><span class="w"> </span>Nasser2021ABCPredictions.liver_and_intestine.all_putative_enhancers.sorted.bedpe<span class="w"> </span>&gt;<span class="w"> </span>Nasser2021ABCPredictions.hepatocyte-ENCODE.all_putative_enhancers.sorted.bedpe
</code></pre></div>
<p>And finally, we do the same again with enhancers merged only over the selected biosample:</p>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span><span class="s1">&#39;BEGIN{FS=&quot;\t&quot;; OFS=&quot;\t&quot;} {print $1, $2, $3}&#39;</span><span class="w"> </span>Nasser2021ABCPredictions.hepatocyte-ENCODE.all_putative_enhancers.sorted.bedpe<span class="w"> </span><span class="p">|</span>uniq<span class="w"> </span>&gt;<span class="w"> </span>list_enhancers.hepatocyte-ENCODE.bed
</code></pre></div>
<p>Contains 23,436 enhancers.</p>
<div class="highlight"><pre><span></span><code>bedtools<span class="w"> </span>merge<span class="w"> </span>-i<span class="w"> </span>list_enhancers.hepatocyte-ENCODE.bed<span class="w"> </span>&gt;<span class="w"> </span>list_enhancers.hepatocyte-ENCODE.merged.bed
</code></pre></div>
<p>Contains 23,436 enhancers =&gt; indeed, it seems natural that enhancers in one biosample only do not overlap each other.</p>
<p>So we do not need to replace all enhancers coordinates in the result, by that of the corresponding merged enhancers in <code>list_enhancers.hepatocyte-ENCODE.merged.bed</code>.</p>
<h5 id="liver-encode">liver-ENCODE<a class="headerlink" href="#liver-encode" title="Permanent link">&para;</a></h5>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span><span class="s1">&#39;$11==&quot;liver-ENCODE&quot;&#39;</span><span class="w"> </span>Nasser2021ABCPredictions.liver_and_intestine.all_putative_enhancers.merged_enhancers.sorted.bedpe<span class="w"> </span>&gt;<span class="w"> </span>Nasser2021ABCPredictions.liver-ENCODE.all_putative_enhancers.merged_enhancers.sorted.bedpe
</code></pre></div>
<p>OK.</p>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span><span class="s1">&#39;$11==&quot;liver-ENCODE&quot;&#39;</span><span class="w"> </span>Nasser2021ABCPredictions.liver_and_intestine.all_putative_enhancers.sorted.bedpe<span class="w"> </span>&gt;<span class="w"> </span>Nasser2021ABCPredictions.liver-ENCODE.all_putative_enhancers.sorted.bedpe
</code></pre></div>
<p>And finally, we do the same again with enhancers merged only over the selected biosample:</p>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span><span class="s1">&#39;BEGIN{FS=&quot;\t&quot;; OFS=&quot;\t&quot;} {print $1, $2, $3}&#39;</span><span class="w"> </span>Nasser2021ABCPredictions.liver-ENCODE.all_putative_enhancers.sorted.bedpe<span class="w"> </span><span class="p">|</span>uniq<span class="w"> </span>&gt;<span class="w"> </span>list_enhancers.liver-ENCODE.bed
</code></pre></div>
<p>Contains 18,623 enhancers.</p>
<div class="highlight"><pre><span></span><code>bedtools<span class="w"> </span>merge<span class="w"> </span>-i<span class="w"> </span>list_enhancers.liver-ENCODE.bed<span class="w"> </span>&gt;<span class="w"> </span>list_enhancers.liver-ENCODE.merged.bed
</code></pre></div>
<p>Contains 18,623 enhancers =&gt; indeed, it seems natural that enhancers in one biosample only do not overlap each other.</p>
<p>So we do not need to replace all enhancers coordinates in the result, by that of the corresponding merged enhancers in <code>list_enhancers.liver-ENCODE.merged.bed</code>.</p>
<h5 id="hepg2-roadmap">HepG2-Roadmap<a class="headerlink" href="#hepg2-roadmap" title="Permanent link">&para;</a></h5>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span><span class="s1">&#39;$11==&quot;HepG2-Roadmap&quot;&#39;</span><span class="w"> </span>Nasser2021ABCPredictions.liver_and_intestine.all_putative_enhancers.merged_enhancers.sorted.bedpe<span class="w"> </span>&gt;<span class="w"> </span>Nasser2021ABCPredictions.HepG2-Roadmap.all_putative_enhancers.merged_enhancers.sorted.bedpe
</code></pre></div>
<p>OK.</p>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span><span class="s1">&#39;$11==&quot;HepG2-Roadmap&quot;&#39;</span><span class="w"> </span>Nasser2021ABCPredictions.liver_and_intestine.all_putative_enhancers.sorted.bedpe<span class="w"> </span>&gt;<span class="w"> </span>Nasser2021ABCPredictions.HepG2-Roadmap.all_putative_enhancers.sorted.bedpe
</code></pre></div>
<p>And finally, we do the same again with enhancers merged only over the selected biosample:</p>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span><span class="s1">&#39;BEGIN{FS=&quot;\t&quot;; OFS=&quot;\t&quot;} {print $1, $2, $3}&#39;</span><span class="w"> </span>Nasser2021ABCPredictions.HepG2-Roadmap.all_putative_enhancers.sorted.bedpe<span class="w"> </span><span class="p">|</span>uniq<span class="w"> </span>&gt;<span class="w"> </span>list_enhancers.HepG2-Roadmap.bed
</code></pre></div>
<p>Contains 19,766 enhancers.</p>
<div class="highlight"><pre><span></span><code>bedtools<span class="w"> </span>merge<span class="w"> </span>-i<span class="w"> </span>list_enhancers.HepG2-Roadmap.bed<span class="w"> </span>&gt;<span class="w"> </span>list_enhancers.HepG2-Roadmap.merged.bed
</code></pre></div>
<p>Contains 19,766 enhancers =&gt; indeed, it seems natural that enhancers in one biosample only do not overlap each other.</p>
<p>So we do not need to replace all enhancers coordinates in the result, by that of the corresponding merged enhancers in <code>list_enhancers.HepG2-Roadmap.merged.bed</code>.</p>
<h5 id="large_intestine_fetal-roadmap">large_intestine_fetal-Roadmap<a class="headerlink" href="#large_intestine_fetal-roadmap" title="Permanent link">&para;</a></h5>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span><span class="s1">&#39;$11==&quot;large_intestine_fetal-Roadmap&quot;&#39;</span><span class="w"> </span>Nasser2021ABCPredictions.liver_and_intestine.all_putative_enhancers.merged_enhancers.sorted.bedpe<span class="w"> </span>&gt;<span class="w"> </span>Nasser2021ABCPredictions.large_intestine_fetal-Roadmap.all_putative_enhancers.merged_enhancers.sorted.bedpe
</code></pre></div>
<p>OK.</p>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span><span class="s1">&#39;$11==&quot;large_intestine_fetal-Roadmap&quot;&#39;</span><span class="w"> </span>Nasser2021ABCPredictions.liver_and_intestine.all_putative_enhancers.sorted.bedpe<span class="w"> </span>&gt;<span class="w"> </span>Nasser2021ABCPredictions.large_intestine_fetal-Roadmap.all_putative_enhancers.sorted.bedpe
</code></pre></div>
<p>And finally, we do the same again with enhancers merged only over the selected biosample:</p>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span><span class="s1">&#39;BEGIN{FS=&quot;\t&quot;; OFS=&quot;\t&quot;} {print $1, $2, $3}&#39;</span><span class="w"> </span>Nasser2021ABCPredictions.large_intestine_fetal-Roadmap.all_putative_enhancers.sorted.bedpe<span class="w"> </span><span class="p">|</span>uniq<span class="w"> </span>&gt;<span class="w"> </span>list_enhancers.large_intestine_fetal-Roadmap.bed
</code></pre></div>
<p>Contains 20,517 enhancers.</p>
<div class="highlight"><pre><span></span><code>bedtools<span class="w"> </span>merge<span class="w"> </span>-i<span class="w"> </span>list_enhancers.large_intestine_fetal-Roadmap.bed<span class="w"> </span>&gt;<span class="w"> </span>list_enhancers.large_intestine_fetal-Roadmap.merged.bed
</code></pre></div>
<p>Contains 20,517 enhancers =&gt; indeed, it seems natural that enhancers in one biosample only do not overlap each other.</p>
<p>So we do not need to replace all enhancers coordinates in the result, by that of the corresponding merged enhancers in <code>list_enhancers.large_intestine_fetal-Roadmap.merged.bed</code>.</p>
<h5 id="small_intestine_fetal-roadmap">small_intestine_fetal-Roadmap<a class="headerlink" href="#small_intestine_fetal-roadmap" title="Permanent link">&para;</a></h5>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span><span class="s1">&#39;$11==&quot;small_intestine_fetal-Roadmap&quot;&#39;</span><span class="w"> </span>Nasser2021ABCPredictions.liver_and_intestine.all_putative_enhancers.merged_enhancers.sorted.bedpe<span class="w"> </span>&gt;<span class="w"> </span>Nasser2021ABCPredictions.small_intestine_fetal-Roadmap.all_putative_enhancers.merged_enhancers.sorted.bedpe
</code></pre></div>
<p>OK.</p>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span><span class="s1">&#39;$11==&quot;small_intestine_fetal-Roadmap&quot;&#39;</span><span class="w"> </span>Nasser2021ABCPredictions.liver_and_intestine.all_putative_enhancers.sorted.bedpe<span class="w"> </span>&gt;<span class="w"> </span>Nasser2021ABCPredictions.small_intestine_fetal-Roadmap.all_putative_enhancers.sorted.bedpe
</code></pre></div>
<p>And finally, we do the same again with enhancers merged only over the selected biosample:</p>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span><span class="s1">&#39;BEGIN{FS=&quot;\t&quot;; OFS=&quot;\t&quot;} {print $1, $2, $3}&#39;</span><span class="w"> </span>Nasser2021ABCPredictions.small_intestine_fetal-Roadmap.all_putative_enhancers.sorted.bedpe<span class="w"> </span><span class="p">|</span>uniq<span class="w"> </span>&gt;<span class="w"> </span>list_enhancers.small_intestine_fetal-Roadmap.bed
</code></pre></div>
<p>Contains 20,555 enhancers.</p>
<div class="highlight"><pre><span></span><code>bedtools<span class="w"> </span>merge<span class="w"> </span>-i<span class="w"> </span>list_enhancers.small_intestine_fetal-Roadmap.bed<span class="w"> </span>&gt;<span class="w"> </span>list_enhancers.small_intestine_fetal-Roadmap.merged.bed
</code></pre></div>
<p>Contains 20,555 enhancers =&gt; indeed, it seems natural that enhancers in one biosample only do not overlap each other.</p>
<p>So we do not need to replace all enhancers coordinates in the result, by that of the corresponding merged enhancers in <code>list_enhancers.small_intestine_fetal-Roadmap.merged.bed</code>.</p>
<h4 id="all-biosamples">All biosamples<a class="headerlink" href="#all-biosamples" title="Permanent link">&para;</a></h4>
<p>We already did it in section <a href="#main-filters">Main filters</a>.</p>
<ul>
<li><code>Nasser2021ABCPredictions.all_biosamples.all_putative_enhancers.merged_enhancers.sorted.bedpe</code> for merged enhancers</li>
<li><code>Nasser2021ABCPredictions.all_biosamples.all_putative_enhancers.merged_enhancers.sorted.uniques_eg.bedpe</code> for merged enhancers (with 2 rows = 2 distinct unique E-G pairs)</li>
<li><code>Nasser2021ABCPredictions.all_biosamples.all_putative_enhancers.sorted.bedpe</code> for unmerged (original) enhancers</li>
</ul>
<h2 id="investigate-putative-regulatory-elements">Investigate putative regulatory elements<a class="headerlink" href="#investigate-putative-regulatory-elements" title="Permanent link">&para;</a></h2>
<h3 id="small-r-script-to-compute-summary-statistics">Small R script to compute summary statistics<a class="headerlink" href="#small-r-script-to-compute-summary-statistics" title="Permanent link">&para;</a></h3>
<p>See <code>compute_summary_stats_enhancer_list.R</code>. Content:</p>
<blockquote>
<p><div class="highlight"><pre><span></span><code><span class="n">options</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">commandArgs</span><span class="p">(</span><span class="n">trailingOnly</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span>

<span class="n">enhancers</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">as.data.frame</span><span class="p">(</span><span class="nf">read.table</span><span class="p">(</span><span class="n">options</span><span class="p">[</span><span class="m">1</span><span class="p">],</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;\t&quot;</span><span class="p">))</span>
<span class="n">enhancers</span><span class="o">$</span><span class="n">length</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">abs</span><span class="p">(</span><span class="n">enhancers</span><span class="o">$</span><span class="n">V3</span><span class="o">-</span><span class="n">enhancers</span><span class="o">$</span><span class="n">V2</span><span class="p">)</span>
<span class="nf">summary</span><span class="p">(</span><span class="n">enhancers</span><span class="o">$</span><span class="n">length</span><span class="p">)</span>
</code></pre></div>
</p>
</blockquote>
<p>We need to load a module containing R to be able to launch the script:</p>
<div class="highlight"><pre><span></span><code>module<span class="w"> </span>load<span class="w"> </span>system/R-4.1.1_gcc-9.3.0
</code></pre></div>
<p>We can execute the R script with:</p>
<div class="highlight"><pre><span></span><code>Rscript<span class="w"> </span>compute_summary_stats_enhancer_list.R<span class="w"> </span>&lt;relative_path_to_enhancers_list&gt;
</code></pre></div>
<h3 id="ccre">ccRE<a class="headerlink" href="#ccre" title="Permanent link">&para;</a></h3>
<p>We take the ccRE from <code>/work2/project/regenet/results/multi/bengi/Annotations/hg19-cCREs.bed</code>.</p>
<div class="highlight"><pre><span></span><code>cp<span class="w"> </span>/work2/project/regenet/results/multi/bengi/Annotations/hg19-cCREs.bed<span class="w"> </span>.
</code></pre></div>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;$6~/(^Enhancer-like$)/&#39;</span><span class="w"> </span>hg19-cCREs.bed<span class="w"> </span>&gt;<span class="w"> </span>/work2/project/regenet/results/multi/abc.model/Nasser2021/ccRE-ELS.bed
</code></pre></div>
<p>989,712 ccRE-ELS</p>
<div class="highlight"><pre><span></span><code><span class="w">   </span><span class="n">Min.</span><span class="w"> </span><span class="m">1</span><span class="n">st</span><span class="w"> </span><span class="n">Qu.</span><span class="w">  </span><span class="n">Median</span><span class="w">    </span><span class="n">Mean</span><span class="w"> </span><span class="m">3</span><span class="n">rd</span><span class="w"> </span><span class="n">Qu.</span><span class="w">    </span><span class="n">Max.</span><span class="w"> </span>
<span class="w">   </span><span class="m">50.0</span><span class="w">   </span><span class="m">247.0</span><span class="w">   </span><span class="m">352.0</span><span class="w">   </span><span class="m">423.4</span><span class="w">   </span><span class="m">519.0</span><span class="w"> </span><span class="m">16633.0</span>
</code></pre></div>
<h3 id="merged-enhancers-of-all-131-biosamples">Merged enhancers of all 131 biosamples<a class="headerlink" href="#merged-enhancers-of-all-131-biosamples" title="Permanent link">&para;</a></h3>
<h4 id="original-enhancers">Original enhancers<a class="headerlink" href="#original-enhancers" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code><span class="w">   </span><span class="n">Min.</span><span class="w"> </span><span class="m">1</span><span class="n">st</span><span class="w"> </span><span class="n">Qu.</span><span class="w">  </span><span class="n">Median</span><span class="w">    </span><span class="n">Mean</span><span class="w"> </span><span class="m">3</span><span class="n">rd</span><span class="w"> </span><span class="n">Qu.</span><span class="w">    </span><span class="n">Max.</span><span class="w"> </span>
<span class="w">  </span><span class="m">200.0</span><span class="w">   </span><span class="m">200.0</span><span class="w">   </span><span class="m">308.0</span><span class="w">   </span><span class="m">481.7</span><span class="w">   </span><span class="m">637.0</span><span class="w">  </span><span class="m">6991.0</span>
</code></pre></div>
<h4 id="merged-enhancers">Merged enhancers<a class="headerlink" href="#merged-enhancers" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code><span class="w">   </span><span class="n">Min.</span><span class="w"> </span><span class="m">1</span><span class="n">st</span><span class="w"> </span><span class="n">Qu.</span><span class="w">  </span><span class="n">Median</span><span class="w">    </span><span class="n">Mean</span><span class="w"> </span><span class="m">3</span><span class="n">rd</span><span class="w"> </span><span class="n">Qu.</span><span class="w">    </span><span class="n">Max.</span><span class="w"> </span>
<span class="w">  </span><span class="m">200.0</span><span class="w">   </span><span class="m">288.0</span><span class="w">   </span><span class="m">631.0</span><span class="w">   </span><span class="m">807.8</span><span class="w">  </span><span class="m">1078.0</span><span class="w"> </span><span class="m">11616.0</span>
</code></pre></div>
<h4 id="overlap-with-ccre-els">Overlap with ccRE-ELS<a class="headerlink" href="#overlap-with-ccre-els" title="Permanent link">&para;</a></h4>
<blockquote>
<div class="highlight"><pre><span></span><code>wc<span class="w"> </span>-l<span class="w"> </span>list_all_enhancers.merged.bed
</code></pre></div>
<p>269,254</p>
</blockquote>
<div class="highlight"><pre><span></span><code>bedtools<span class="w"> </span>intersect<span class="w"> </span>-c<span class="w"> </span>-a<span class="w"> </span>list_all_enhancers.merged.bed<span class="w"> </span>-b<span class="w"> </span>ccRE-ELS.bed<span class="w"> </span><span class="p">|</span>awk<span class="w"> </span><span class="s1">&#39;BEGIN{FS=&quot;\t&quot;; OFS=&quot;\t&quot;} {nbs[$4]++} END{for(u in nbs){print u, nbs[u]}}&#39;</span><span class="w"> </span><span class="p">|</span>sort<span class="w"> </span>-nrk2,2<span class="w">    </span>
</code></pre></div>
<blockquote>
<div class="highlight"><pre><span></span><code><span class="m">1</span><span class="w"> </span><span class="m">112356</span>
<span class="m">0</span><span class="w"> </span><span class="m">85937</span>
<span class="m">2</span><span class="w"> </span><span class="m">44230</span>
<span class="m">3</span><span class="w"> </span><span class="m">16616</span>
<span class="m">4</span><span class="w"> </span><span class="m">6135</span>
<span class="m">5</span><span class="w"> </span><span class="m">2319</span>
<span class="m">6</span><span class="w"> </span><span class="m">934</span>
<span class="m">7</span><span class="w"> </span><span class="m">388</span>
<span class="m">8</span><span class="w"> </span><span class="m">164</span>
<span class="m">9</span><span class="w"> </span><span class="m">78</span>
<span class="m">10</span><span class="w">    </span><span class="m">43</span>
<span class="m">11</span><span class="w">    </span><span class="m">24</span>
<span class="m">12</span><span class="w">    </span><span class="m">14</span>
<span class="m">13</span><span class="w">    </span><span class="m">7</span>
<span class="m">14</span><span class="w">    </span><span class="m">4</span>
<span class="m">17</span><span class="w">    </span><span class="m">3</span>
<span class="m">16</span><span class="w">    </span><span class="m">2</span>
</code></pre></div>
</blockquote>
<p>85,937 (32%) merged putative enhancers across all 131 biosamples, do not match any ccRE-ELS.</p>
<blockquote>
<div class="highlight"><pre><span></span><code>wc<span class="w"> </span>-l<span class="w"> </span>list_all_enhancers.bed
</code></pre></div>
<p>2,463,310</p>
</blockquote>
<div class="highlight"><pre><span></span><code>bedtools<span class="w"> </span>intersect<span class="w"> </span>-c<span class="w"> </span>-a<span class="w"> </span>list_all_enhancers.bed<span class="w"> </span>-b<span class="w"> </span>ccRE-ELS.bed<span class="w"> </span><span class="p">|</span>awk<span class="w"> </span><span class="s1">&#39;BEGIN{FS=&quot;\t&quot;; OFS=&quot;\t&quot;} {nbs[$4]++} END{for(u in nbs){print u, nbs[u]}}&#39;</span><span class="w"> </span><span class="p">|</span>sort<span class="w"> </span>-nrk2,2<span class="w">   </span>
</code></pre></div>
<blockquote>
<div class="highlight"><pre><span></span><code><span class="m">0</span><span class="w"> </span><span class="m">1117560</span>
<span class="m">1</span><span class="w"> </span><span class="m">1054002</span>
<span class="m">2</span><span class="w"> </span><span class="m">221538</span>
<span class="m">3</span><span class="w"> </span><span class="m">52355</span>
<span class="m">4</span><span class="w"> </span><span class="m">12610</span>
<span class="m">5</span><span class="w"> </span><span class="m">3471</span>
<span class="m">6</span><span class="w"> </span><span class="m">1025</span>
<span class="m">7</span><span class="w"> </span><span class="m">422</span>
<span class="m">8</span><span class="w"> </span><span class="m">180</span>
<span class="m">9</span><span class="w"> </span><span class="m">69</span>
<span class="m">10</span><span class="w">    </span><span class="m">38</span>
<span class="m">11</span><span class="w">    </span><span class="m">26</span>
<span class="m">12</span><span class="w">    </span><span class="m">9</span>
<span class="m">13</span><span class="w">    </span><span class="m">2</span>
<span class="m">17</span><span class="w">    </span><span class="m">1</span>
<span class="m">16</span><span class="w">    </span><span class="m">1</span>
<span class="m">14</span><span class="w">    </span><span class="m">1</span>
</code></pre></div>
</blockquote>
<h4 id="overlap-with-all-ccre">Overlap with all ccRE<a class="headerlink" href="#overlap-with-all-ccre" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code>bedtools<span class="w"> </span>intersect<span class="w"> </span>-c<span class="w"> </span>-a<span class="w"> </span>list_all_enhancers.merged.bed<span class="w"> </span>-b<span class="w"> </span>hg19-cCREs.bed<span class="w"> </span><span class="p">|</span>awk<span class="w"> </span><span class="s1">&#39;BEGIN{FS=&quot;\t&quot;; OFS=&quot;\t&quot;} {nbs[$4]++} END{for(u in nbs){print u, nbs[u]}}&#39;</span><span class="w"> </span><span class="p">|</span>sort<span class="w"> </span>-nrk2,2
</code></pre></div>
<blockquote>
<div class="highlight"><pre><span></span><code><span class="m">1</span><span class="w"> </span><span class="m">144690</span>
<span class="m">2</span><span class="w"> </span><span class="m">58313</span>
<span class="m">0</span><span class="w"> </span><span class="m">25709</span>
<span class="m">3</span><span class="w"> </span><span class="m">23341</span>
<span class="m">4</span><span class="w"> </span><span class="m">9498</span>
<span class="m">5</span><span class="w"> </span><span class="m">4016</span>
<span class="m">6</span><span class="w"> </span><span class="m">1854</span>
<span class="m">7</span><span class="w"> </span><span class="m">899</span>
<span class="m">8</span><span class="w"> </span><span class="m">422</span>
<span class="m">9</span><span class="w"> </span><span class="m">235</span>
<span class="m">10</span><span class="w">    </span><span class="m">114</span>
<span class="m">11</span><span class="w">    </span><span class="m">63</span>
<span class="m">12</span><span class="w">    </span><span class="m">41</span>
<span class="m">13</span><span class="w">    </span><span class="m">24</span>
<span class="m">14</span><span class="w">    </span><span class="m">11</span>
<span class="m">15</span><span class="w">    </span><span class="m">6</span>
<span class="m">17</span><span class="w">    </span><span class="m">5</span>
<span class="m">16</span><span class="w">    </span><span class="m">5</span>
<span class="m">20</span><span class="w">    </span><span class="m">3</span>
<span class="m">18</span><span class="w">    </span><span class="m">3</span>
<span class="m">22</span><span class="w">    </span><span class="m">1</span>
<span class="m">19</span><span class="w">    </span><span class="m">1</span>
</code></pre></div>
</blockquote>
<p>Only 25,709 (9%) merged putative enhancers across all 131 biosamples, do not match any ccRE (vs 32% against ccRE-ELS), suggesting that a lot of those "putative enhancers" are actually other types of regulatory element as defined by Moore/ENCODE.</p>
<h3 id="merged-enhancers-of-all-liver-and-intestine-biosamples">Merged enhancers of all liver and intestine biosamples<a class="headerlink" href="#merged-enhancers-of-all-liver-and-intestine-biosamples" title="Permanent link">&para;</a></h3>
<h4 id="original-enhancers_1">Original enhancers<a class="headerlink" href="#original-enhancers_1" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code><span class="w">   </span><span class="n">Min.</span><span class="w"> </span><span class="m">1</span><span class="n">st</span><span class="w"> </span><span class="n">Qu.</span><span class="w">  </span><span class="n">Median</span><span class="w">    </span><span class="n">Mean</span><span class="w"> </span><span class="m">3</span><span class="n">rd</span><span class="w"> </span><span class="n">Qu.</span><span class="w">    </span><span class="n">Max.</span><span class="w"> </span>
<span class="w">  </span><span class="m">200.0</span><span class="w">   </span><span class="m">200.0</span><span class="w">   </span><span class="m">353.0</span><span class="w">   </span><span class="m">490.7</span><span class="w">   </span><span class="m">644.0</span><span class="w">  </span><span class="m">4626.0</span>
</code></pre></div>
<h4 id="merged-enhancers-merged-across-131-biosamples">Merged enhancers (merged across 131 biosamples)<a class="headerlink" href="#merged-enhancers-merged-across-131-biosamples" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code>bedtools<span class="w"> </span>intersect<span class="w"> </span>-wb<span class="w"> </span>-a<span class="w"> </span>list_enhancers.liver_and_intestine.bed<span class="w"> </span>-b<span class="w"> </span>list_all_enhancers.merged.bed<span class="w"> </span><span class="p">|</span>awk<span class="w"> </span><span class="s1">&#39;BEGIN{FS=&quot;\t&quot;; OFS=&quot;\t&quot;} {print $4, $5, $6}&#39;</span><span class="w"> </span><span class="p">|</span>bedtools<span class="w"> </span>sort<span class="w"> </span>-faidx<span class="w"> </span>chr_sizes<span class="w"> </span>-i<span class="w"> </span>stdin<span class="w"> </span><span class="p">|</span>uniq<span class="w"> </span>&gt;<span class="w"> </span>list_enhancers.liver_and_intestine.merged131.bed
</code></pre></div>
<blockquote>
<div class="highlight"><pre><span></span><code>wc<span class="w"> </span>-l<span class="w"> </span>list_enhancers.liver_and_intestine.bed
<span class="m">102996</span>
wc<span class="w"> </span>-l<span class="w"> </span>list_enhancers.liver_and_intestine.merged131.bed
<span class="m">50881</span>
</code></pre></div>
</blockquote>
<div class="highlight"><pre><span></span><code><span class="w">   </span><span class="n">Min.</span><span class="w"> </span><span class="m">1</span><span class="n">st</span><span class="w"> </span><span class="n">Qu.</span><span class="w">  </span><span class="n">Median</span><span class="w">    </span><span class="n">Mean</span><span class="w"> </span><span class="m">3</span><span class="n">rd</span><span class="w"> </span><span class="n">Qu.</span><span class="w">    </span><span class="n">Max.</span><span class="w"> </span>
<span class="w">    </span><span class="m">200</span><span class="w">     </span><span class="m">520</span><span class="w">     </span><span class="m">982</span><span class="w">    </span><span class="m">1231</span><span class="w">    </span><span class="m">1668</span><span class="w">   </span><span class="m">11616</span>
</code></pre></div>
<h4 id="merged-enhancers-merged-across-5-liver-and-intestine-biosamples">Merged enhancers (merged across 5 liver and intestine biosamples)<a class="headerlink" href="#merged-enhancers-merged-across-5-liver-and-intestine-biosamples" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code><span class="w">   </span><span class="n">Min.</span><span class="w"> </span><span class="m">1</span><span class="n">st</span><span class="w"> </span><span class="n">Qu.</span><span class="w">  </span><span class="n">Median</span><span class="w">    </span><span class="n">Mean</span><span class="w"> </span><span class="m">3</span><span class="n">rd</span><span class="w"> </span><span class="n">Qu.</span><span class="w">    </span><span class="n">Max.</span><span class="w"> </span>
<span class="w">  </span><span class="m">200.0</span><span class="w">   </span><span class="m">200.0</span><span class="w">   </span><span class="m">423.0</span><span class="w">   </span><span class="m">574.6</span><span class="w">   </span><span class="m">756.0</span><span class="w">  </span><span class="m">6298.0</span><span class="w"> </span>
</code></pre></div>
<h4 id="overlap-with-ccre-els_1">Overlap with ccRE-ELS<a class="headerlink" href="#overlap-with-ccre-els_1" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code>bedtools<span class="w"> </span>intersect<span class="w"> </span>-c<span class="w"> </span>-a<span class="w"> </span>list_enhancers.liver_and_intestine.merged.bed<span class="w"> </span>-b<span class="w"> </span>ccRE-ELS.bed<span class="w"> </span><span class="p">|</span>awk<span class="w"> </span><span class="s1">&#39;BEGIN{FS=&quot;\t&quot;; OFS=&quot;\t&quot;} {nbs[$4]++} END{for(u in nbs){print u, nbs[u]}}&#39;</span><span class="w"> </span><span class="p">|</span>sort<span class="w"> </span>-nrk2,2<span class="w">    </span>
</code></pre></div>
<blockquote>
<div class="highlight"><pre><span></span><code><span class="m">0</span><span class="w"> </span><span class="m">26768</span>
<span class="m">1</span><span class="w"> </span><span class="m">22443</span>
<span class="m">2</span><span class="w"> </span><span class="m">6037</span>
<span class="m">3</span><span class="w"> </span><span class="m">1564</span>
<span class="m">4</span><span class="w"> </span><span class="m">398</span>
<span class="m">5</span><span class="w"> </span><span class="m">135</span>
<span class="m">6</span><span class="w"> </span><span class="m">33</span>
<span class="m">7</span><span class="w"> </span><span class="m">10</span>
<span class="m">9</span><span class="w"> </span><span class="m">6</span>
<span class="m">10</span><span class="w">    </span><span class="m">2</span>
<span class="m">8</span><span class="w"> </span><span class="m">1</span>
<span class="m">16</span><span class="w">    </span><span class="m">1</span>
</code></pre></div>
</blockquote>
<p>47% do not overlap any ccRE-ELS !</p>
<h3 id="merged-enhancers-of-all-liver-biosamples">Merged enhancers of all liver biosamples<a class="headerlink" href="#merged-enhancers-of-all-liver-biosamples" title="Permanent link">&para;</a></h3>
<h4 id="original-enhancers_2">Original enhancers<a class="headerlink" href="#original-enhancers_2" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code><span class="w">   </span><span class="n">Min.</span><span class="w"> </span><span class="m">1</span><span class="n">st</span><span class="w"> </span><span class="n">Qu.</span><span class="w">  </span><span class="n">Median</span><span class="w">    </span><span class="n">Mean</span><span class="w"> </span><span class="m">3</span><span class="n">rd</span><span class="w"> </span><span class="n">Qu.</span><span class="w">    </span><span class="n">Max.</span><span class="w"> </span>
<span class="w">  </span><span class="m">200.0</span><span class="w">   </span><span class="m">200.0</span><span class="w">   </span><span class="m">296.0</span><span class="w">   </span><span class="m">455.3</span><span class="w">   </span><span class="m">585.0</span><span class="w">  </span><span class="m">4626.0</span>
</code></pre></div>
<h4 id="merged-enhancers-merged-across-131-biosamples_1">Merged enhancers (merged across 131 biosamples)<a class="headerlink" href="#merged-enhancers-merged-across-131-biosamples_1" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code>bedtools<span class="w"> </span>intersect<span class="w"> </span>-wb<span class="w"> </span>-a<span class="w"> </span>list_enhancers.liver.bed<span class="w"> </span>-b<span class="w"> </span>list_all_enhancers.merged.bed<span class="w"> </span><span class="p">|</span>awk<span class="w"> </span><span class="s1">&#39;BEGIN{FS=&quot;\t&quot;; OFS=&quot;\t&quot;} {print $4, $5, $6}&#39;</span><span class="w"> </span><span class="p">|</span>bedtools<span class="w"> </span>sort<span class="w"> </span>-faidx<span class="w"> </span>chr_sizes<span class="w"> </span>-i<span class="w"> </span>stdin<span class="w"> </span><span class="p">|</span>uniq<span class="w"> </span>&gt;<span class="w"> </span>list_enhancers.liver.merged131.bed
</code></pre></div>
<blockquote>
<div class="highlight"><pre><span></span><code>wc<span class="w"> </span>-l<span class="w"> </span>list_enhancers.liver.bed
<span class="m">61838</span>
wc<span class="w"> </span>-l<span class="w"> </span>list_enhancers.liver.merged131.bed
<span class="m">39477</span>
</code></pre></div>
</blockquote>
<div class="highlight"><pre><span></span><code><span class="w">   </span><span class="n">Min.</span><span class="w"> </span><span class="m">1</span><span class="n">st</span><span class="w"> </span><span class="n">Qu.</span><span class="w">  </span><span class="n">Median</span><span class="w">    </span><span class="n">Mean</span><span class="w"> </span><span class="m">3</span><span class="n">rd</span><span class="w"> </span><span class="n">Qu.</span><span class="w">    </span><span class="n">Max.</span><span class="w"> </span>
<span class="w">    </span><span class="m">200</span><span class="w">     </span><span class="m">528</span><span class="w">    </span><span class="m">1027</span><span class="w">    </span><span class="m">1292</span><span class="w">    </span><span class="m">1780</span><span class="w">   </span><span class="m">11616</span>
</code></pre></div>
<h4 id="merged-enhancers-across-all-3-liver-biosamples">Merged enhancers (across all 3 liver biosamples)<a class="headerlink" href="#merged-enhancers-across-all-3-liver-biosamples" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code><span class="w">   </span><span class="n">Min.</span><span class="w"> </span><span class="m">1</span><span class="n">st</span><span class="w"> </span><span class="n">Qu.</span><span class="w">  </span><span class="n">Median</span><span class="w">    </span><span class="n">Mean</span><span class="w"> </span><span class="m">3</span><span class="n">rd</span><span class="w"> </span><span class="n">Qu.</span><span class="w">    </span><span class="n">Max.</span><span class="w"> </span>
<span class="w">  </span><span class="m">200.0</span><span class="w">   </span><span class="m">200.0</span><span class="w">   </span><span class="m">348.0</span><span class="w">   </span><span class="m">507.1</span><span class="w">   </span><span class="m">650.0</span><span class="w">  </span><span class="m">4737.0</span>
</code></pre></div>
<h4 id="overlap-with-ccre-els_2">Overlap with ccRE-ELS<a class="headerlink" href="#overlap-with-ccre-els_2" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code>bedtools<span class="w"> </span>intersect<span class="w"> </span>-c<span class="w"> </span>-a<span class="w"> </span>list_enhancers.liver.merged.bed<span class="w"> </span>-b<span class="w"> </span>ccRE-ELS.bed<span class="w"> </span><span class="p">|</span>awk<span class="w"> </span><span class="s1">&#39;BEGIN{FS=&quot;\t&quot;; OFS=&quot;\t&quot;} {nbs[$4]++} END{for(u in nbs){print u, nbs[u]}}&#39;</span><span class="w"> </span><span class="p">|</span>sort<span class="w"> </span>-nrk2,2
</code></pre></div>
<blockquote>
<div class="highlight"><pre><span></span><code><span class="m">0</span><span class="w"> </span><span class="m">22717</span>
<span class="m">1</span><span class="w"> </span><span class="m">17340</span>
<span class="m">2</span><span class="w"> </span><span class="m">3802</span>
<span class="m">3</span><span class="w"> </span><span class="m">851</span>
<span class="m">4</span><span class="w"> </span><span class="m">184</span>
<span class="m">5</span><span class="w"> </span><span class="m">64</span>
<span class="m">6</span><span class="w"> </span><span class="m">11</span>
<span class="m">9</span><span class="w"> </span><span class="m">5</span>
<span class="m">7</span><span class="w"> </span><span class="m">3</span>
<span class="m">8</span><span class="w"> </span><span class="m">1</span>
<span class="m">10</span><span class="w">    </span><span class="m">1</span>
</code></pre></div>
</blockquote>
<p>51% do not overlap any ccRE-ELS !</p>
<h3 id="merged-enhancers-of-all-intestine-biosamples">Merged enhancers of all intestine biosamples<a class="headerlink" href="#merged-enhancers-of-all-intestine-biosamples" title="Permanent link">&para;</a></h3>
<h4 id="original-enhancers_3">Original enhancers<a class="headerlink" href="#original-enhancers_3" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code><span class="w">   </span><span class="n">Min.</span><span class="w"> </span><span class="m">1</span><span class="n">st</span><span class="w"> </span><span class="n">Qu.</span><span class="w">  </span><span class="n">Median</span><span class="w">    </span><span class="n">Mean</span><span class="w"> </span><span class="m">3</span><span class="n">rd</span><span class="w"> </span><span class="n">Qu.</span><span class="w">    </span><span class="n">Max.</span><span class="w"> </span>
<span class="w">    </span><span class="m">200</span><span class="w">     </span><span class="m">200</span><span class="w">     </span><span class="m">428</span><span class="w">     </span><span class="m">543</span><span class="w">     </span><span class="m">724</span><span class="w">    </span><span class="m">3679</span>
</code></pre></div>
<h4 id="merged-enhancers-merged-across-131-biosamples_2">Merged enhancers (merged across 131 biosamples)<a class="headerlink" href="#merged-enhancers-merged-across-131-biosamples_2" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code>bedtools<span class="w"> </span>intersect<span class="w"> </span>-wb<span class="w"> </span>-a<span class="w"> </span>list_enhancers.intestine.bed<span class="w"> </span>-b<span class="w"> </span>list_all_enhancers.merged.bed<span class="w"> </span><span class="p">|</span>awk<span class="w"> </span><span class="s1">&#39;BEGIN{FS=&quot;\t&quot;; OFS=&quot;\t&quot;} {print $4, $5, $6}&#39;</span><span class="w"> </span><span class="p">|</span>bedtools<span class="w"> </span>sort<span class="w"> </span>-faidx<span class="w"> </span>chr_sizes<span class="w"> </span>-i<span class="w"> </span>stdin<span class="w"> </span><span class="p">|</span>uniq<span class="w"> </span>&gt;<span class="w"> </span>list_enhancers.intestine.merged131.bed
</code></pre></div>
<blockquote>
<div class="highlight"><pre><span></span><code>wc<span class="w"> </span>-l<span class="w"> </span>list_enhancers.intestine.bed
<span class="m">41102</span>
wc<span class="w"> </span>-l<span class="w"> </span>list_enhancers.intestine.merged131.bed
<span class="m">24492</span>
</code></pre></div>
</blockquote>
<div class="highlight"><pre><span></span><code><span class="w">   </span><span class="n">Min.</span><span class="w"> </span><span class="m">1</span><span class="n">st</span><span class="w"> </span><span class="n">Qu.</span><span class="w">  </span><span class="n">Median</span><span class="w">    </span><span class="n">Mean</span><span class="w"> </span><span class="m">3</span><span class="n">rd</span><span class="w"> </span><span class="n">Qu.</span><span class="w">    </span><span class="n">Max.</span><span class="w"> </span>
<span class="w">    </span><span class="m">200</span><span class="w">     </span><span class="m">710</span><span class="w">    </span><span class="m">1209</span><span class="w">    </span><span class="m">1469</span><span class="w">    </span><span class="m">1978</span><span class="w">   </span><span class="m">11073</span>
</code></pre></div>
<h4 id="merged-enhancers-across-all-2-intestine-biosamples">Merged enhancers (across all 2 intestine biosamples)<a class="headerlink" href="#merged-enhancers-across-all-2-intestine-biosamples" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code><span class="w">   </span><span class="n">Min.</span><span class="w"> </span><span class="m">1</span><span class="n">st</span><span class="w"> </span><span class="n">Qu.</span><span class="w">  </span><span class="n">Median</span><span class="w">    </span><span class="n">Mean</span><span class="w"> </span><span class="m">3</span><span class="n">rd</span><span class="w"> </span><span class="n">Qu.</span><span class="w">    </span><span class="n">Max.</span><span class="w"> </span>
<span class="w">  </span><span class="m">200.0</span><span class="w">   </span><span class="m">221.0</span><span class="w">   </span><span class="m">488.0</span><span class="w">   </span><span class="m">606.8</span><span class="w">   </span><span class="m">814.0</span><span class="w">  </span><span class="m">4461.0</span>
</code></pre></div>





                
              </article>
            </div>
          
          
        </div>
        
          <a href="#" class="md-top md-icon" data-md-component="top" hidden>
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg>
            Back to top
          </a>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      2020 - 2022 Tristan Hoellinger
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
        <div class="md-social">
  
    
    
    
    
      
      
    
    <a href="https://twitter.com/TristanHoellin1" target="_blank" rel="noopener" title="twitter.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"/></svg>
    </a>
  
    
    
    
    
      
      
    
    <a href="https://fr.linkedin.com/in/tristan-hoellinger" target="_blank" rel="noopener" title="fr.linkedin.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"/></svg>
    </a>
  
    
    
    
    
    <a href="mailto:sarah.djebali@inserm.fr" target="_blank" rel="noopener" title="" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M498.1 5.6c10.1 7 15.4 19.1 13.5 31.2l-64 416c-1.5 9.7-7.4 18.2-16 23s-18.9 5.4-28 1.6L284 427.7l-68.5 74.1c-8.9 9.7-22.9 12.9-35.2 8.1S160 493.2 160 480v-83.6c0-4 1.5-7.8 4.2-10.7l167.6-182.9c5.8-6.3 5.6-16-.4-22s-15.7-6.4-22-.7L106 360.8l-88.3-44.2C7.1 311.3.3 300.7 0 288.9s5.9-22.8 16.1-28.7l448-256c10.7-6.1 23.9-5.5 34 1.4z"/></svg>
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    <script id="__config" type="application/json">{"base": "../..", "features": ["navigation.instant", "navigation.tabs", "navigation.sections", "navigation.top"], "search": "../../assets/javascripts/workers/search.db81ec45.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../../assets/javascripts/bundle.6df46069.min.js"></script>
      
    
  </body>
</html>