
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
        <meta name="author" content="Tristan Hoellinger">
      
      
        <link rel="canonical" href="https://eg.github.io/haemochromatosis/ABC/preliminary_analysis_v9.html">
      
      
        <link rel="prev" href="data_processing.html">
      
      
        <link rel="next" href="summary_statistics_eg_pairs.html">
      
      <link rel="icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.4.2, mkdocs-material-9.0.11">
    
    
      
        <title>Analysis of E-G Networks starting from genes involved in hemochromatosis / iron metabolism - Genome-wide identification of E/G interactions</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.0d440cfe.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.2505c338.min.css">
      
      

    
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../stylesheets/extra.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="black" data-md-color-accent="">
  
    
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#notice" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Genome-wide identification of E/G interactions" class="md-header__button md-logo" aria-label="Genome-wide identification of E/G interactions" data-md-component="logo">
      
  <img src="../../logo/logo-inserm-blc.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Genome-wide identification of E/G interactions
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Analysis of E-G Networks starting from genes involved in hemochromatosis / iron metabolism
            
          </span>
        </div>
      </div>
    </div>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/hoellin/eg" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
  </div>
  <div class="md-source__repository">
    hoellin/eg
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../guidebooks/introduction.html" class="md-tabs__link">
        Guidebooks
      </a>
    </li>
  

      
        
  
  
    
  


  
  
  
    <li class="md-tabs__item">
      <a href="../introduction.html" class="md-tabs__link md-tabs__link--active">
        Haemochromatosis
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../notes_BENGI/introduction.html" class="md-tabs__link">
        BENGI
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../notes_ABC/introduction.html" class="md-tabs__link">
        ABC
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../index.html" class="md-tabs__link">
        About
      </a>
    </li>
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Genome-wide identification of E/G interactions" class="md-nav__button md-logo" aria-label="Genome-wide identification of E/G interactions" data-md-component="logo">
      
  <img src="../../logo/logo-inserm-blc.png" alt="logo">

    </a>
    Genome-wide identification of E/G interactions
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/hoellin/eg" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
  </div>
  <div class="md-source__repository">
    hoellin/eg
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1" >
      
      
      
        <label class="md-nav__link" for="__nav_1" id="__nav_1_label" tabindex="0">
          Guidebooks
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_1_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_1">
          <span class="md-nav__icon md-icon"></span>
          Guidebooks
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../guidebooks/introduction.html" class="md-nav__link">
        Introduction
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../guidebooks/script_bedpe_summary/script_bedpe_summary.html" class="md-nav__link">
        Compute summary stats of a bedpe
      </a>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_3" >
      
      
      
        <label class="md-nav__link" for="__nav_1_3" id="__nav_1_3_label" tabindex="0">
          Basics
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_1_3">
          <span class="md-nav__icon md-icon"></span>
          Basics
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../guidebooks/shell.html" class="md-nav__link">
        Shell
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../guidebooks/conda_environments.html" class="md-nav__link">
        Conda basics
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../guidebooks/git.html" class="md-nav__link">
        Git
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../guidebooks/genotoul.html" class="md-nav__link">
        Genotoul cluster
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../guidebooks/notebooks_ssh.html" class="md-nav__link">
        How to use Jupyter through ssh tunneling?
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_4" >
      
      
      
        <label class="md-nav__link" for="__nav_1_4" id="__nav_1_4_label" tabindex="0">
          Misc
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_4_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_1_4">
          <span class="md-nav__icon md-icon"></span>
          Misc
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../guidebooks/compute_correlations.html" class="md-nav__link">
        Compute correlations
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" checked>
      
      
      
        <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
          Haemochromatosis
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="true">
        <label class="md-nav__title" for="__nav_2">
          <span class="md-nav__icon md-icon"></span>
          Haemochromatosis
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../introduction.html" class="md-nav__link">
        Introduction
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../data_format.html" class="md-nav__link">
        Data format
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="custom_go.html" class="md-nav__link">
        Custom script GO
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../main_results.html" class="md-nav__link">
        Main results
      </a>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2_5" >
      
      
      
        <label class="md-nav__link" for="__nav_2_5" id="__nav_2_5_label" tabindex="0">
          CHiC data
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_2_5_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_2_5">
          <span class="md-nav__icon md-icon"></span>
          CHiC data
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../CHiC/preprocess_data.html" class="md-nav__link">
        Preprocessing
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../CHiC/preliminary_analysis_chic_v2.html" class="md-nav__link">
        Networks
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../CHiC/go_enrichment.html" class="md-nav__link">
        GO enrichment
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
    
  
  
    
      
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2_6" checked>
      
      
      
        <label class="md-nav__link" for="__nav_2_6" id="__nav_2_6_label" tabindex="0">
          ABC data
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_2_6_label" aria-expanded="true">
        <label class="md-nav__title" for="__nav_2_6">
          <span class="md-nav__icon md-icon"></span>
          ABC data
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="data_processing.html" class="md-nav__link">
        Preprocessing
      </a>
    </li>
  

            
          
            
              
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
      <a href="preliminary_analysis_v9.html" class="md-nav__link md-nav__link--active">
        Networks
      </a>
      
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="summary_statistics_eg_pairs.html" class="md-nav__link">
        Summary statistics enhancers
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="go_enrichment.html" class="md-nav__link">
        GO enrichment
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="gorilla.html" class="md-nav__link">
        GOrilla
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="compute_avg_nb_of_connections_per_gene.html" class="md-nav__link">
        Avg nb connections per gene
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="gwas.html" class="md-nav__link">
        GWAS (13 initial genes)
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="gwas_10.html" class="md-nav__link">
        GWAS (10 initial genes)
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" >
      
      
      
        <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
          BENGI
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3">
          <span class="md-nav__icon md-icon"></span>
          BENGI
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/introduction.html" class="md-nav__link">
        Introduction
      </a>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_2" >
      
      
      
        <label class="md-nav__link" for="__nav_3_2" id="__nav_3_2_label" tabindex="0">
          Notebooks
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_2_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_2">
          <span class="md-nav__icon md-icon"></span>
          Notebooks
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/BENGI_to_custom_bedpe.html" class="md-nav__link">
        Convert BENGI datasets to custom bedpe
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_3" >
      
      
      
        <label class="md-nav__link" for="__nav_3_3" id="__nav_3_3_label" tabindex="0">
          Summary statistics
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_3">
          <span class="md-nav__icon md-icon"></span>
          Summary statistics
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/summary_statistics/summary_statistics.html" class="md-nav__link">
        Summary statistics on BENGI GM12878 benchmarks
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/summary_statistics_BENGI/summary_BENGI_bedpe.html" class="md-nav__link">
        (R code) Compute summary statistics over BENGI
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_4" >
      
      
      
        <label class="md-nav__link" for="__nav_3_4" id="__nav_3_4_label" tabindex="0">
          Evaluation of a few methods over BENGI GM12878
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_4_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_4">
          <span class="md-nav__icon md-icon"></span>
          Evaluation of a few methods over BENGI GM12878
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_4_1" >
      
      
      
        <label class="md-nav__link" for="__nav_3_4_1" id="__nav_3_4_1_label" tabindex="0">
          Distance method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_4_1_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_4_1">
          <span class="md-nav__icon md-icon"></span>
          Distance method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/distance_method/distance_evaluation_with_code.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/distance_method/summary_distance_evaluation.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_4_2" >
      
      
      
        <label class="md-nav__link" for="__nav_3_4_2" id="__nav_3_4_2_label" tabindex="0">
          DNase-expression correlation method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_4_2_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_4_2">
          <span class="md-nav__icon md-icon"></span>
          DNase-expression correlation method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/dnase_expression_correlation/correlation_method_with_code.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/dnase_expression_correlation/summary_correlation_method.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_4_3" >
      
      
      
        <label class="md-nav__link" for="__nav_3_4_3" id="__nav_3_4_3_label" tabindex="0">
          Average rank method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_4_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_4_3">
          <span class="md-nav__icon md-icon"></span>
          Average rank method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/avg_rank_method/avg_rank_method_with_code.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/avg_rank_method/summary_avg_rank_method.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_5" >
      
      
      
        <label class="md-nav__link" for="__nav_3_5" id="__nav_3_5_label" tabindex="0">
          Evaluation of a few methods over Fulco et al 2019 CRISPRi-FlowFISH
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_5_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_5">
          <span class="md-nav__icon md-icon"></span>
          Evaluation of a few methods over Fulco et al 2019 CRISPRi-FlowFISH
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_5_1" >
      
      
      
        <label class="md-nav__link" for="__nav_3_5_1" id="__nav_3_5_1_label" tabindex="0">
          Distance method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_5_1_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_5_1">
          <span class="md-nav__icon md-icon"></span>
          Distance method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/CRISPRi_FlowFISH/distance_method/distance_over_fulco_et_al_crispri.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/CRISPRi_FlowFISH/distance_method/summary_distance_over_fulco_et_al_crispri.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_5_2" >
      
      
      
        <label class="md-nav__link" for="__nav_3_5_2" id="__nav_3_5_2_label" tabindex="0">
          DNase-expression correlation method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_5_2_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_5_2">
          <span class="md-nav__icon md-icon"></span>
          DNase-expression correlation method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/CRISPRi_FlowFISH/dnase_expression_correlation/correlation_method_with_code.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/CRISPRi_FlowFISH/dnase_expression_correlation/summary_correlation_method.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_5_3" >
      
      
      
        <label class="md-nav__link" for="__nav_3_5_3" id="__nav_3_5_3_label" tabindex="0">
          Average rank method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_5_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_5_3">
          <span class="md-nav__icon md-icon"></span>
          Average rank method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/CRISPRi_FlowFISH/avg_rank_method/avg_rank_method_with_code.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/CRISPRi_FlowFISH/avg_rank_method/summary_avg_rank.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" >
      
      
      
        <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
          ABC
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_4">
          <span class="md-nav__icon md-icon"></span>
          ABC
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/introduction.html" class="md-nav__link">
        Introduction
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/BENGI/ABC_BENGI_results.html" class="md-nav__link">
        ABC over BENGI for GM12878, main results
      </a>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_3" >
      
      
      
        <label class="md-nav__link" for="__nav_4_3" id="__nav_4_3_label" tabindex="0">
          CRISPRi-FlowFISH
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_4_3">
          <span class="md-nav__icon md-icon"></span>
          CRISPRi-FlowFISH
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/K562/ABC_K562_CRISPRi_FlowFISH.html" class="md-nav__link">
        Original ABC predictions over K562
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/K562/recap_investigations_ABC_over_CRiFF.html" class="md-nav__link">
        Summary ABC over CRiFF
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_4" >
      
      
      
        <label class="md-nav__link" for="__nav_4_4" id="__nav_4_4_label" tabindex="0">
          Notebooks
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_4_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_4_4">
          <span class="md-nav__icon md-icon"></span>
          Notebooks
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/K562/april_K562_56_genes/april_K562_56_genes.html" class="md-nav__link">
        ABC over K562 cCRiFF
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/K562/april_K562_59_genes/april_K562_59_genes.html" class="md-nav__link">
        ABC over K562 CRiFF
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/K562/april_K562_candidates_not_in_whitelist_56_genes/april_K562_candidates_NOT_in_whitelist.html" class="md-nav__link">
        ABC over K562 cCRiFF, no whitelist
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/generic_notebooks/turnkey_notebook_to_run_ABC_with_example_over_GM12878.html" class="md-nav__link">
        Generic notebook to run ABC method & Example over GM12878
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/BENGI/notebook_ABC_over_BENGI_GM12878.html" class="md-nav__link">
        ABC over GM12878 with same accessions as for BENGI
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/BENGI/notebook_ABC_over_BENGI_GM12878_from_ccRE_ELSs.html" class="md-nav__link">
        ABC over GM12878 with same accessions as for BENGI, starting with ccRE-dELS
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_5" >
      
      
      
        <label class="md-nav__link" for="__nav_4_5" id="__nav_4_5_label" tabindex="0">
          Full codes
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_5_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_4_5">
          <span class="md-nav__icon md-icon"></span>
          Full codes
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/test_abc_pipeline.html" class="md-nav__link">
        Test of ABC pipeline over chr22
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/BENGI/ABC_over_BENGI_GM12878.html" class="md-nav__link">
        ABC over GM12878 with same accessions as for BENGI
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/differences_example_chr22.html" class="md-nav__link">
        Meld of results over chr22 vs authors' results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" >
      
      
      
        <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
          About
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_5">
          <span class="md-nav__icon md-icon"></span>
          About
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../index.html" class="md-nav__link">
        Getting started
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../about_me.html" class="md-nav__link">
        About me
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  

  
  



<style type="text/css">
.badCode {
background-color: #C9DDE4;
}
</style>

<h1 id="notice">Notice<a class="headerlink" href="#notice" title="Permanent link">&para;</a></h1>
<p>The current page essentially corresponds to a markdown knitted from a Rmd document. It does not render very well because of html widgets not handled, so one should better refer to the correct rendering available in <a href="../../../data_and_results/haemochromatosis/networks_hemochromatosis/preliminary_analysis_v9.html">preliminary_analysis_v9.html</a>.</p>
<h1 id="libraries-version">Libraries &amp; Version<a class="headerlink" href="#libraries-version" title="Permanent link">&para;</a></h1>
<div class="badCode highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">tidyverse</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">visNetwork</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.1 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0

locale:
 [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_GB.UTF-8        LC_COLLATE=en_GB.UTF-8    
 [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_GB.UTF-8   
 [7] LC_PAPER=en_GB.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] data.table_1.14.2 visNetwork_2.1.0  forcats_0.5.1     stringr_1.4.0    
 [5] dplyr_1.0.7       purrr_0.3.4       readr_2.0.2       tidyr_1.1.4      
 [9] tibble_3.1.6      ggplot2_3.3.5     tidyverse_1.3.1   knitr_1.36       

loaded via a namespace (and not attached):
 [1] tidyselect_1.1.1  xfun_0.28         bslib_0.3.1       haven_2.4.3      
 [5] colorspace_2.0-2  vctrs_0.3.8       generics_0.1.1    htmltools_0.5.2  
 [9] yaml_2.2.1        utf8_1.2.2        rlang_0.4.12      jquerylib_0.1.4  
[13] pillar_1.6.4      withr_2.4.2       glue_1.5.0        DBI_1.1.1        
[17] dbplyr_2.1.1      modelr_0.1.8      readxl_1.3.1      lifecycle_1.0.1  
[21] munsell_0.5.0     gtable_0.3.0      cellranger_1.1.0  rvest_1.0.2      
[25] htmlwidgets_1.5.4 evaluate_0.14     tzdb_0.2.0        fastmap_1.1.0    
[29] fansi_0.5.0       broom_0.7.10      Rcpp_1.0.7        backports_1.3.0  
[33] scales_1.1.1      formatR_1.11      jsonlite_1.7.2    fs_1.5.0         
[37] hms_1.1.1         digest_0.6.28     stringi_1.7.5     grid_4.0.3       
[41] cli_3.1.0         tools_4.0.3       magrittr_2.0.1    sass_0.4.0       
[45] crayon_1.4.2      pkgconfig_2.0.3   ellipsis_0.3.2    xml2_1.3.2       
[49] reprex_2.0.1      lubridate_1.8.0   rstudioapi_0.13   assertthat_0.2.1 
[53] rmarkdown_2.11    httr_1.4.2        R6_2.5.1          compiler_4.0.3   
</code></pre></div>
<h1 id="preliminary-work">Preliminary work<a class="headerlink" href="#preliminary-work" title="Permanent link">&para;</a></h1>
<p>Documentation on <code>visNetwork</code> can be found <a href="https://datastorm-open.github.io/visNetwork/nodes.html">here</a>.</p>
<h2 id="data-importation">Data importation<a class="headerlink" href="#data-importation" title="Permanent link">&para;</a></h2>
<p>All the files imported here can be found on Genotoul, in <code>/work2/project/regenet/results/multi/abc.model/Nasser2021</code>.</p>
<p>The full code itself is available here: <a href="../../../data_and_results/haemochromatosis/networks_hemochromatosis/preliminary_analysis_v10.Rmd"><code>preliminary_analysis_v10.Rmd</code></a>.</p>
<div class="badCode highlight"><pre><span></span><code><span class="nf">rm</span><span class="p">(</span><span class="n">list</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">ls</span><span class="p">())</span>

<span class="c1"># wd = &#39;data/&#39; wd =</span>
<span class="c1"># &#39;/home/hoellinger/Documents/INSERM/shared/automne_2021/networks_hemochromatosis/data/&#39;</span>
<span class="n">wd</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;/home/thoellinger/Documents/shared/automne_2021/networks_hemochromatosis/data/&quot;</span>
<span class="n">egfile</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste</span><span class="p">(</span><span class="n">wd</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;Nasser2021ABCPredictions.liver_and_intestine.all_putative_enhancers.merged_enhancers.sorted.uniques_eg.bedpe&quot;</span><span class="p">,</span>
<span class="w">    </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;&quot;</span><span class="p">)</span>
<span class="n">unmerged_efile</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste</span><span class="p">(</span><span class="n">wd</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;list_all_enhancers.bed&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;&quot;</span><span class="p">)</span>
<span class="n">merged_efile</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste</span><span class="p">(</span><span class="n">wd</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;list_all_enhancers.merged.bed&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;&quot;</span><span class="p">)</span>

<span class="c1">############# Enhancers # The only reasons all enhancers are loaded is to</span>
<span class="c1">############# compute a few summary statistics.</span>
<span class="n">e</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">as.data.frame</span><span class="p">(</span><span class="nf">read.table</span><span class="p">(</span><span class="n">unmerged_efile</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;\t&quot;</span><span class="p">))</span>
<span class="n">me</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">as.data.frame</span><span class="p">(</span><span class="nf">read.table</span><span class="p">(</span><span class="n">merged_efile</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;\t&quot;</span><span class="p">))</span>

<span class="c1">############ E-G list #</span>

<span class="n">eg</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">as.data.frame</span><span class="p">(</span><span class="nf">read.table</span><span class="p">(</span><span class="n">egfile</span><span class="p">,</span><span class="w"> </span><span class="n">header</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">T</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;\t&quot;</span><span class="p">))</span>

<span class="c1">############### known genes #</span>
<span class="n">gene_list</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;HFE&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;TFR2&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;HFE2&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;HAMP&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;SLC40A1&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;BMP6&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;TMPRSS6&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;TFRC&quot;</span><span class="p">,</span>
<span class="w">    </span><span class="s">&quot;SLC11A2&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;CYBRD1&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;NEO1&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;CIAPIN1&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;SLC39A14&quot;</span><span class="p">)</span>

<span class="c1"># genes known to be causally involved in hemochromatosis: &#39;HFE&#39;, &#39;TFR2&#39;,</span>
<span class="c1"># &#39;HFE2&#39;, &#39;HAMP&#39;, &#39;SLC40A1&#39;, &#39;BMP6&#39; (the other genes are involved in iron</span>
<span class="c1"># metabolism regulation) genes expressed in the liver: &#39;HFE&#39;, &#39;TFR2&#39;, &#39;HFE2&#39;,</span>
<span class="c1"># &#39;HAMP&#39;, &#39;SLC40A1&#39;, &#39;BMP6&#39;, &#39;TMPRSS6&#39;, &#39;TFRC&#39; genes expressed in intestine:</span>
<span class="c1"># &#39;SLC40A1&#39;, &#39;SLC11A2&#39;, &#39;CYBRD1&#39;, &#39;NEO1&#39;, &#39;CIAPIN1&#39;</span>
</code></pre></div>
<h2 id="data-preprocessing">Data preprocessing<a class="headerlink" href="#data-preprocessing" title="Permanent link">&para;</a></h2>
<h3 id="new-columns">New columns<a class="headerlink" href="#new-columns" title="Permanent link">&para;</a></h3>
<p>We will need two more columns containing the following:</p>
<div class="badCode highlight"><pre><span></span><code><span class="n">eg</span><span class="o">$</span><span class="n">biosamples.uniq</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">unlist</span><span class="p">(</span><span class="nf">lapply</span><span class="p">(</span><span class="n">eg</span><span class="o">$</span><span class="n">biosamples</span><span class="p">,</span><span class="w"> </span><span class="nf">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="w"> </span><span class="nf">paste</span><span class="p">(</span><span class="nf">unlist</span><span class="p">(</span><span class="nf">sort</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="nf">strsplit</span><span class="p">(</span><span class="n">x</span><span class="p">,</span>
<span class="w">    </span><span class="s">&quot;,&quot;</span><span class="p">)[[</span><span class="m">1</span><span class="p">]]))),</span><span class="w"> </span><span class="n">collapse</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;,&quot;</span><span class="p">)))</span>
<span class="n">eg</span><span class="o">$</span><span class="n">tissues.uniq</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">unlist</span><span class="p">(</span><span class="nf">lapply</span><span class="p">(</span><span class="n">eg</span><span class="o">$</span><span class="n">tissues</span><span class="p">,</span><span class="w"> </span><span class="nf">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="w"> </span><span class="nf">paste</span><span class="p">(</span><span class="nf">unlist</span><span class="p">(</span><span class="nf">sort</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="nf">strsplit</span><span class="p">(</span><span class="n">x</span><span class="p">,</span>
<span class="w">    </span><span class="s">&quot;,&quot;</span><span class="p">)[[</span><span class="m">1</span><span class="p">]]))),</span><span class="w"> </span><span class="n">collapse</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;,&quot;</span><span class="p">)))</span>
</code></pre></div>
<h3 id="conversion-to-factors">Conversion to factors<a class="headerlink" href="#conversion-to-factors" title="Permanent link">&para;</a></h3>
<p>This has to be done AFTER the creation of new columns done above:</p>
<div class="badCode highlight"><pre><span></span><code><span class="n">to_factor_cols</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;chrom1&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;chrom2&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;name&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;strand1&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;strand2&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;gene&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;biosamples&quot;</span><span class="p">,</span>
<span class="w">    </span><span class="s">&quot;tissues&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;biosamples.uniq&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;tissues.uniq&quot;</span><span class="p">)</span>
<span class="n">eg</span><span class="p">[</span><span class="n">to_factor_cols</span><span class="p">]</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">lapply</span><span class="p">(</span><span class="n">eg</span><span class="p">[</span><span class="n">to_factor_cols</span><span class="p">],</span><span class="w"> </span><span class="n">factor</span><span class="p">)</span>
</code></pre></div>
<h2 id="exploration">Exploration<a class="headerlink" href="#exploration" title="Permanent link">&para;</a></h2>
<h3 id="summary-statistics-on-enhancer-lists">Summary statistics on enhancer lists<a class="headerlink" href="#summary-statistics-on-enhancer-lists" title="Permanent link">&para;</a></h3>
<div class="badCode highlight"><pre><span></span><code><span class="nf">length</span><span class="p">(</span><span class="n">e</span><span class="p">[,</span><span class="w"> </span><span class="m">1</span><span class="p">])</span><span class="w">  </span><span class="c1"># list of all enhancers</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>[1] 2463310
</code></pre></div>
<div class="badCode highlight"><pre><span></span><code><span class="nf">length</span><span class="p">(</span><span class="n">me</span><span class="p">[,</span><span class="w"> </span><span class="m">1</span><span class="p">])</span><span class="w">  </span><span class="c1"># list of all merged enhancers (such that none of those merged enhancers are overlapping)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>[1] 269254
</code></pre></div>
<p>Warning: we shall pay attention to the fact that among those 269,254 merged putative enhancers, 85,937 (32%)  <strong>do not overlap any ccRE-ELS</strong>, and only 112,356 enhancers (42%) overlap exactly one ccRE-ELS (but the latter behavior is expected because of the merging process). Contrariwise, only 25,709 of those enhancers (9%) do not overlap any ccRE (ie candidate regulatory element not necessarily with Enhancer-Like-Signature), suggesting that many of those 269,254 merged putative enhancers might not be real enhancers (ie with both high DNase and high H3K27ac signal in one or more of the ENCODE biosamples used to defined ccRE), but CTCT-only, promoters or DNase-H3K4me2 regions.</p>
<p>Nevertheless, we chose to use the list of 269,254 putative enhancers for consistency in our analysis when it comes to compare results when removing or adding a new biosamples / tissues (so that the list of enhancers does not change in the process). In the future, we might take some time to compare what we would have obtained otherwise.</p>
<div class="badCode highlight"><pre><span></span><code><span class="n">e</span><span class="o">$</span><span class="n">length</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">abs</span><span class="p">(</span><span class="n">e</span><span class="o">$</span><span class="n">V3</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="n">e</span><span class="o">$</span><span class="n">V2</span><span class="p">)</span>
<span class="n">me</span><span class="o">$</span><span class="n">length</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">abs</span><span class="p">(</span><span class="n">me</span><span class="o">$</span><span class="n">V3</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="n">me</span><span class="o">$</span><span class="n">V2</span><span class="p">)</span>

<span class="nf">summary</span><span class="p">(</span><span class="n">e</span><span class="o">$</span><span class="n">length</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  200.0   200.0   308.0   481.7   637.0  6991.0 
</code></pre></div>
<div class="badCode highlight"><pre><span></span><code><span class="nf">summary</span><span class="p">(</span><span class="n">me</span><span class="o">$</span><span class="n">length</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  200.0   288.0   631.0   807.8  1078.0 11616.0 
</code></pre></div>
<h3 id="e-g-pairs">E-G pairs<a class="headerlink" href="#e-g-pairs" title="Permanent link">&para;</a></h3>
<p>We extract the subsample of the E-G bedpe input list, where genes are contained in our list of genes involved either directly in hemochromatosis or in iron metabolism. In the variable name, "dist0" stands for "distance is 0 between the genes in <code>eg_dist0</code> and the list of initial genes".</p>
<div class="badCode highlight"><pre><span></span><code><span class="n">eg_dist0</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">eg</span><span class="p">[</span><span class="n">eg</span><span class="o">$</span><span class="n">gene</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">gene_list</span><span class="p">,</span><span class="w"> </span><span class="p">]</span>
</code></pre></div>
<div class="badCode highlight"><pre><span></span><code><span class="nf">length</span><span class="p">(</span><span class="n">eg_dist0</span><span class="o">$</span><span class="n">name</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>[1] 137
</code></pre></div>
<div class="badCode highlight"><pre><span></span><code><span class="nf">length</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="n">eg_dist0</span><span class="o">$</span><span class="n">gene</span><span class="p">))</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>[1] 13
</code></pre></div>
<div class="badCode highlight"><pre><span></span><code><span class="nf">table</span><span class="p">(</span><span class="n">eg_dist0</span><span class="o">$</span><span class="n">tissues.uniq</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>      intestine intestine,liver           liver 
             36              21              80 
</code></pre></div>
<h3 id="genes">Genes<a class="headerlink" href="#genes" title="Permanent link">&para;</a></h3>
<p>Chromosomes where the genes are located:</p>
<div class="highlight"><pre><span></span><code>     HFE     TFR2     HFE2     HAMP  SLC40A1     BMP6  TMPRSS6     TFRC 
  &quot;chr6&quot;   &quot;chr7&quot;   &quot;chr1&quot;  &quot;chr19&quot;   &quot;chr2&quot;   &quot;chr6&quot;  &quot;chr22&quot;   &quot;chr3&quot; 
 SLC11A2   CYBRD1     NEO1  CIAPIN1 SLC39A14 
 &quot;chr12&quot;   &quot;chr2&quot;  &quot;chr15&quot;  &quot;chr16&quot;   &quot;chr8&quot; 
</code></pre></div>
<h1 id="networks">Networks<a class="headerlink" href="#networks" title="Permanent link">&para;</a></h1>
<p>Note: in all subsequent graphs, size of nodes of type "gene" (and not "known_gene", for which the size is fixed) is proportional to the number of distinct enhancers regulating them.</p>
<h2 id="find-all-genes-regulated-by-the-initial-enhancers">Find all genes regulated by the initial enhancers<a class="headerlink" href="#find-all-genes-regulated-by-the-initial-enhancers" title="Permanent link">&para;</a></h2>
<p>The "initial enhancers" are the enhancers involved in eg_dist0, ie all the enhancers regulating the initial genes in <code>gene_list</code>.</p>
<p><code>genes</code> is the same as <code>gene_list</code> (strictly speaking, <code>genes</code> is the subset of <code>gene_list</code> for which we have data), and <code>enhancers</code> is the list of enhancers regulating those initial known genes.</p>
<div class="badCode highlight"><pre><span></span><code><span class="n">enhancers</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">unique</span><span class="p">(</span><span class="nf">paste</span><span class="p">(</span><span class="n">eg_dist0</span><span class="o">$</span><span class="n">chrom1</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;:&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">eg_dist0</span><span class="o">$</span><span class="n">start1</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;-&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">eg_dist0</span><span class="o">$</span><span class="n">end1</span><span class="p">,</span>
<span class="w">    </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;&quot;</span><span class="p">))</span>
<span class="n">genes</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">unique</span><span class="p">(</span><span class="n">eg_dist0</span><span class="o">$</span><span class="n">gene</span><span class="p">)</span>
</code></pre></div>
<p>Now we compute the list of all genes regulated by enhancers in <code>eg_dist0</code>. Specifically, we extract, from the full E-G list <code>eg</code>, the list <code>eg_dist1</code> containing only the enhancers-genes pairs for which the gene G is regulated by any of the enhancers regulating a gene in <code>gene_list</code> ("dist1" stands for "distance is at most 1 between the genes in <code>eg_dist1</code> and the list of initial genes").</p>
<div class="badCode highlight"><pre><span></span><code><span class="n">eg_enhancers_id</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">data.frame</span><span class="p">(</span><span class="n">source</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste</span><span class="p">(</span><span class="n">eg</span><span class="o">$</span><span class="n">chrom1</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;:&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">eg</span><span class="o">$</span><span class="n">start1</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;-&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">eg</span><span class="o">$</span><span class="n">end1</span><span class="p">,</span>
<span class="w">    </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;&quot;</span><span class="p">),</span><span class="w"> </span><span class="n">eg</span><span class="p">[,</span><span class="w"> </span><span class="m">4</span><span class="o">:</span><span class="m">22</span><span class="p">])</span><span class="w">  </span><span class="c1"># same as eg but columns 1-3 have been concatenated to make unique enhancers id</span>
<span class="n">eg_dist1</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">eg_enhancers_id</span><span class="p">[</span><span class="n">eg_enhancers_id</span><span class="o">$</span><span class="n">source</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="nf">paste</span><span class="p">(</span><span class="n">eg_dist0</span><span class="o">$</span><span class="n">chrom1</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;:&quot;</span><span class="p">,</span>
<span class="w">    </span><span class="n">eg_dist0</span><span class="o">$</span><span class="n">start1</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;-&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">eg_dist0</span><span class="o">$</span><span class="n">end1</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;&quot;</span><span class="p">),</span><span class="w"> </span><span class="p">]</span>
<span class="n">eg_dist1</span><span class="o">$</span><span class="n">from</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">lapply</span><span class="p">(</span><span class="n">eg_dist1</span><span class="o">$</span><span class="n">source</span><span class="p">,</span><span class="w"> </span><span class="nf">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="w"> </span><span class="nf">unique</span><span class="p">(</span><span class="nf">as.character</span><span class="p">(</span><span class="n">eg_dist0</span><span class="p">[</span><span class="nf">paste</span><span class="p">(</span><span class="n">eg_dist0</span><span class="o">$</span><span class="n">chrom1</span><span class="p">,</span>
<span class="w">    </span><span class="s">&quot;:&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">eg_dist0</span><span class="o">$</span><span class="n">start1</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;-&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">eg_dist0</span><span class="o">$</span><span class="n">end1</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">x</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;gene&quot;</span><span class="p">])))</span>
<span class="n">eg_dist1</span><span class="o">$</span><span class="n">ABC.IE</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">left_join</span><span class="p">(</span><span class="nf">data.frame</span><span class="p">(</span><span class="n">name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste</span><span class="p">(</span><span class="n">eg_dist1</span><span class="o">$</span><span class="n">source</span><span class="p">,</span><span class="w"> </span><span class="n">eg_dist1</span><span class="o">$</span><span class="n">from</span><span class="p">,</span>
<span class="w">    </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;::&quot;</span><span class="p">)),</span><span class="w"> </span><span class="n">eg_enhancers_id</span><span class="p">,</span><span class="w"> </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;name&quot;</span><span class="p">)</span><span class="o">$</span><span class="n">ABC.max</span><span class="w">  </span><span class="c1"># max ABC score of the I-E pair (initialGene-Enhancer) corresponding to the E-G pair</span>
<span class="n">genes_dist1</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">unique</span><span class="p">(</span><span class="n">eg_dist1</span><span class="o">$</span><span class="n">gene</span><span class="p">)</span>
</code></pre></div>
<p><code>genes_dist1</code> is the list of genes regulated by <code>enhancers</code>.</p>
<p>Compute "ABC product", ie the product of the ABC scores of:
- the initial gene - enhancer pair (I-E)
- the enhancer - gene pair (E-G)</p>
<div class="badCode highlight"><pre><span></span><code><span class="n">eg_dist1</span><span class="o">$</span><span class="n">ABC.product</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">eg_dist1</span><span class="o">$</span><span class="n">ABC.max</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="n">eg_dist1</span><span class="o">$</span><span class="n">ABC.IE</span>
<span class="n">eg_dist1</span><span class="o">$</span><span class="n">ABC.product</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">eg_dist1</span><span class="o">$</span><span class="n">ABC.product</span><span class="o">/</span><span class="nf">max</span><span class="p">(</span><span class="n">eg_dist1</span><span class="o">$</span><span class="n">ABC.product</span><span class="p">)</span>
</code></pre></div>
<div class="badCode highlight"><pre><span></span><code><span class="nf">print</span><span class="p">(</span><span class="nf">min</span><span class="p">(</span><span class="n">eg_dist1</span><span class="o">$</span><span class="n">ABC.product</span><span class="p">))</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>[1] 0.003458878
</code></pre></div>
<div class="badCode highlight"><pre><span></span><code><span class="nf">print</span><span class="p">(</span><span class="nf">median</span><span class="p">(</span><span class="n">eg_dist1</span><span class="o">$</span><span class="n">ABC.product</span><span class="p">))</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>[1] 0.01070413
</code></pre></div>
<div class="badCode highlight"><pre><span></span><code><span class="nf">print</span><span class="p">(</span><span class="nf">mean</span><span class="p">(</span><span class="n">eg_dist1</span><span class="o">$</span><span class="n">ABC.product</span><span class="p">))</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>[1] 0.02172209
</code></pre></div>
<div class="badCode highlight"><pre><span></span><code><span class="nf">print</span><span class="p">(</span><span class="nf">quantile</span><span class="p">(</span><span class="n">eg_dist1</span><span class="o">$</span><span class="n">ABC.product</span><span class="p">,</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">0.1</span><span class="p">,</span><span class="w"> </span><span class="m">0.6</span><span class="p">,</span><span class="w"> </span><span class="m">0.8</span><span class="p">,</span><span class="w"> </span><span class="m">0.9</span><span class="p">)))</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>        10%         60%         80%         90% 
0.004902703 0.013383311 0.027607616 0.044685971 
</code></pre></div>
<div class="badCode highlight"><pre><span></span><code><span class="nf">print</span><span class="p">(</span><span class="nf">max</span><span class="p">(</span><span class="n">eg_dist1</span><span class="o">$</span><span class="n">ABC.product</span><span class="p">))</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>[1] 1
</code></pre></div>
<div class="badCode highlight"><pre><span></span><code><span class="n">eg_dist1</span><span class="o">$</span><span class="n">ABC.product.label</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span>
<span class="n">eg_dist1</span><span class="p">[</span><span class="n">eg_dist1</span><span class="o">$</span><span class="n">ABC.product</span><span class="w"> </span><span class="o">&gt;=</span><span class="w"> </span><span class="nf">median</span><span class="p">(</span><span class="n">eg_dist1</span><span class="o">$</span><span class="n">ABC.product</span><span class="p">),</span><span class="w"> </span><span class="p">]</span><span class="o">$</span><span class="n">ABC.product.label</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">2</span>
<span class="n">eg_dist1</span><span class="p">[</span><span class="n">eg_dist1</span><span class="o">$</span><span class="n">ABC.product</span><span class="w"> </span><span class="o">&gt;=</span><span class="w"> </span><span class="nf">quantile</span><span class="p">(</span><span class="n">eg_dist1</span><span class="o">$</span><span class="n">ABC.product</span><span class="p">,</span><span class="w"> </span><span class="m">0.9</span><span class="p">)[[</span><span class="m">1</span><span class="p">]],</span><span class="w"> </span><span class="p">]</span><span class="o">$</span><span class="n">ABC.product.label</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">3</span>
</code></pre></div>
<div class="badCode highlight"><pre><span></span><code><span class="nf">table</span><span class="p">(</span><span class="n">eg_dist1</span><span class="o">$</span><span class="n">ABC.product.label</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>  1   2   3 
673 539 135 
</code></pre></div>
<p>In the following cell we create <code>genes_dist1.more</code> which is in a well-suited format for further "concatenation" with inferences made with other type of data (CHiC or QTL -based).</p>
<div class="badCode highlight"><pre><span></span><code><span class="n">genes_dist1.more</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">eg_dist1</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">group_by</span><span class="p">(</span><span class="n">gene</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">mutate</span><span class="p">(</span><span class="n">ABC.sources</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="n">source</span><span class="p">,</span><span class="w"> </span><span class="n">collapse</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;,&quot;</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">mutate</span><span class="p">(</span><span class="n">ABC.count</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">length</span><span class="p">(</span><span class="nf">str_split</span><span class="p">(</span><span class="n">ABC.sources</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;,&quot;</span><span class="p">)[[</span><span class="m">1</span><span class="p">]]))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">group_by</span><span class="p">(</span><span class="n">gene</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">slice</span><span class="p">(</span><span class="nf">which.max</span><span class="p">(</span><span class="n">ABC.product.label</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="c1"># slice_max(ABC.product.label) %&gt;%</span>
<span class="nf">select</span><span class="p">(</span><span class="n">gene</span><span class="p">,</span><span class="w"> </span><span class="n">ABC.sources</span><span class="p">,</span><span class="w"> </span><span class="n">ABC.product.label</span><span class="p">,</span><span class="w"> </span><span class="n">ABC.count</span><span class="p">,</span><span class="w"> </span><span class="n">from</span><span class="p">)</span>
<span class="n">genes_dist1.more</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">subset</span><span class="p">(</span><span class="n">genes_dist1.more</span><span class="p">,</span><span class="w"> </span><span class="o">!</span><span class="p">(</span><span class="n">genes_dist1.more</span><span class="o">$</span><span class="n">gene</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">genes</span><span class="p">))</span>
<span class="n">genes_dist1.more</span><span class="o">$</span><span class="n">from</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">as.character</span><span class="p">(</span><span class="n">genes_dist1.more</span><span class="o">$</span><span class="n">from</span><span class="p">)</span>
<span class="n">genes_dist1.more</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code># A tibble: 444 × 5
# Groups:   gene [444]
   gene     ABC.sources                        ABC.product.lab… ABC.count from  
   &lt;fct&gt;    &lt;chr&gt;                                         &lt;dbl&gt;     &lt;int&gt; &lt;chr&gt; 
 1 AAAS     chr12:53318068-53321694                           1         1 SLC11…
 2 ABT1     chr6:24719303-24722493                            1         1 HFE   
 3 ACOT13   chr6:24719303-24722493,chr6:25991…                3         2 HFE   
 4 ACTL6B   chr7:100166895-100168233,chr7:100…                2         2 TFR2  
 5 ACVR1B   chr12:53318068-53321694                           2         1 SLC11…
 6 ACVRL1   chr12:53318068-53321694                           2         1 SLC11…
 7 ADAM28   chr8:22417293-22420339                            2         1 SLC39…
 8 ADAMDEC1 chr8:22417293-22420339                            2         1 SLC39…
 9 ADGRG1   chr16:56639960-56645831,chr16:573…                3         7 CIAPI…
10 ADGRG3   chr16:56639960-56645831,chr16:573…                2         4 CIAPI…
# … with 434 more rows
</code></pre></div>
<p>We re-arrange <code>eg_dist1</code> as an edge list <code>edges_list_dist1</code>, which will be more suitable to later construct the edges list for visualization as a graph.</p>
<div class="badCode highlight"><pre><span></span><code><span class="n">edges_list_dist1</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">data.frame</span><span class="p">(</span><span class="n">source</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">eg_dist1</span><span class="o">$</span><span class="n">source</span><span class="p">,</span><span class="w"> </span><span class="n">target</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">eg_dist1</span><span class="o">$</span><span class="n">gene</span><span class="p">,</span><span class="w"> </span><span class="n">ABC.mean.x100</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">floor</span><span class="p">(</span><span class="m">100</span><span class="w"> </span><span class="o">*</span>
<span class="w">    </span><span class="n">eg_dist1</span><span class="o">$</span><span class="n">ABC.mean</span><span class="p">),</span><span class="w"> </span><span class="n">ABC.max.x100</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">floor</span><span class="p">(</span><span class="m">100</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="n">eg_dist1</span><span class="o">$</span><span class="n">ABC.max</span><span class="p">),</span><span class="w"> </span><span class="n">tissues</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">eg_dist1</span><span class="o">$</span><span class="n">tissues.uniq</span><span class="p">,</span>
<span class="w">    </span><span class="n">distance_kb</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">floor</span><span class="p">(</span><span class="n">eg_dist1</span><span class="o">$</span><span class="n">original_distance.mean</span><span class="o">/</span><span class="m">1000</span><span class="p">),</span><span class="w"> </span><span class="n">inv_dist</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="o">/</span><span class="p">(</span><span class="n">eg_dist1</span><span class="o">$</span><span class="n">original_distance.mean</span><span class="w"> </span><span class="o">+</span>
<span class="w">        </span><span class="m">1</span><span class="p">),</span><span class="w"> </span><span class="n">rescaled_log_inv_dist</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="nf">min</span><span class="p">(</span><span class="nf">log</span><span class="p">(</span><span class="m">1</span><span class="o">/</span><span class="p">(</span><span class="n">eg_dist1</span><span class="o">$</span><span class="n">original_distance.mean</span><span class="w"> </span><span class="o">+</span>
<span class="w">        </span><span class="m">1</span><span class="p">)))</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="nf">log</span><span class="p">(</span><span class="m">1</span><span class="o">/</span><span class="p">(</span><span class="n">eg_dist1</span><span class="o">$</span><span class="n">original_distance.mean</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="m">1</span><span class="p">)))</span>
<span class="n">edges_list_dist1</span>
</code></pre></div>
<p>Below we compute the list of (colored) nodes required to compute our graphs, a node corresponding either to an enhancer or a gene. To that purpose, we need first to construct 2 tables:
- <code>gname.tissue</code> contains, for each unique gene involved in <code>eg_dist1</code>, the comma-separated list of tissues in which it appears in <code>eg_dist1</code>
- <code>enhancer.tissue</code> contains, for each unique enhancer involved in <code>eg_dist1</code>, the comma-separated list of tissues in which it appears in <code>eg_dist1</code></p>
<p>For more details on the <code>Reduce</code> function and its application to our case, see for instance: https://stackoverflow.com/questions/60592775/how-to-apply-reduce-to-groups-based-on-columns-of-a-data-frame</p>
<div class="badCode highlight"><pre><span></span><code><span class="n">gname.tissue</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">unique</span><span class="p">(</span><span class="n">eg_dist1</span><span class="p">[</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;gene&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;tissues.uniq&quot;</span><span class="p">)])</span>
<span class="n">gname.tissue</span><span class="o">$</span><span class="n">tissues.uniq</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">as.character</span><span class="p">(</span><span class="n">gname.tissue</span><span class="o">$</span><span class="n">tissues.uniq</span><span class="p">)</span>
<span class="n">gname.tissue</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">gname.tissue</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">group_by</span><span class="p">(</span><span class="n">gene</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">mutate</span><span class="p">(</span><span class="n">tissues</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">Reduce</span><span class="p">(</span><span class="nf">function</span><span class="p">(</span><span class="n">x</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">        </span><span class="nf">paste</span><span class="p">(</span><span class="nf">unlist</span><span class="p">(</span><span class="nf">sort</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="nf">strsplit</span><span class="p">(</span><span class="n">x</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;,&quot;</span><span class="p">)[[</span><span class="m">1</span><span class="p">]],</span><span class="w"> </span><span class="nf">strsplit</span><span class="p">(</span><span class="n">y</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;,&quot;</span><span class="p">)[[</span><span class="m">1</span><span class="p">]])))),</span>
<span class="w">            </span><span class="n">collapse</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;,&quot;</span><span class="p">)</span>
<span class="w">    </span><span class="p">},</span><span class="w"> </span><span class="n">tissues.uniq</span><span class="p">))</span>
<span class="n">gname.tissue</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">unique</span><span class="p">(</span><span class="n">gname.tissue</span><span class="p">[</span><span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="m">3</span><span class="p">)])</span>

<span class="n">enhancer.tissue</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">unique</span><span class="p">(</span><span class="n">eg_dist1</span><span class="p">[</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;source&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;tissues.uniq&quot;</span><span class="p">)])</span>
<span class="n">enhancer.tissue</span><span class="o">$</span><span class="n">tissues.uniq</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">as.character</span><span class="p">(</span><span class="n">enhancer.tissue</span><span class="o">$</span><span class="n">tissues.uniq</span><span class="p">)</span>
<span class="n">enhancer.tissue</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">enhancer.tissue</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">group_by</span><span class="p">(</span><span class="n">source</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">mutate</span><span class="p">(</span><span class="n">tissues</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">Reduce</span><span class="p">(</span><span class="nf">function</span><span class="p">(</span><span class="n">x</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">        </span><span class="nf">paste</span><span class="p">(</span><span class="nf">unlist</span><span class="p">(</span><span class="nf">sort</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="nf">strsplit</span><span class="p">(</span><span class="n">x</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;,&quot;</span><span class="p">)[[</span><span class="m">1</span><span class="p">]],</span><span class="w"> </span><span class="nf">strsplit</span><span class="p">(</span><span class="n">y</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;,&quot;</span><span class="p">)[[</span><span class="m">1</span><span class="p">]])))),</span>
<span class="w">            </span><span class="n">collapse</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;,&quot;</span><span class="p">)</span>
<span class="w">    </span><span class="p">},</span><span class="w"> </span><span class="n">tissues.uniq</span><span class="p">))</span>
<span class="n">enhancer.tissue</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">unique</span><span class="p">(</span><span class="n">enhancer.tissue</span><span class="p">[</span><span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="m">3</span><span class="p">)])</span>

<span class="n">gname.tissue</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code># A tibble: 457 × 2
# Groups:   gene [457]
   gene         tissues        
   &lt;fct&gt;        &lt;chr&gt;          
 1 PDE4DIP      intestine,liver
 2 LOC100996724 intestine,liver
 3 SEC22B       intestine,liver
 4 NOTCH2NL     intestine,liver
 5 NBPF10       intestine,liver
 6 LINC01719    intestine,liver
 7 HFE2         liver          
 8 TXNIP        intestine,liver
 9 POLR3GL      intestine,liver
10 ANKRD34A     intestine,liver
# … with 447 more rows
</code></pre></div>
<div class="badCode highlight"><pre><span></span><code><span class="n">enhancer.tissue</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code># A tibble: 137 × 2
# Groups:   source [137]
   source                   tissues        
   &lt;chr&gt;                    &lt;chr&gt;          
 1 chr1:144930398-144933456 intestine,liver
 2 chr1:145394876-145400153 intestine,liver
 3 chr1:145413923-145415673 liver          
 4 chr1:145421224-145422239 liver          
 5 chr1:145426950-145428636 intestine,liver
 6 chr1:145434950-145435719 intestine,liver
 7 chr1:145442141-145445254 intestine,liver
 8 chr1:145451402-145451862 liver          
 9 chr1:145454003-145457954 intestine,liver
10 chr1:145473958-145474912 intestine,liver
# … with 127 more rows
</code></pre></div>
<p>So now we can compute the list <code>nodes_dist1</code> of (colored) nodes required for our graphs. There are 3 types of nodes: <code>enhancer</code>, <code>known_gene</code> and (unknown) <code>gene</code>.</p>
<div class="badCode highlight"><pre><span></span><code><span class="n">nodes_dist1</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">full_join</span><span class="p">(</span><span class="nf">data.frame</span><span class="p">(</span><span class="n">label</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">enhancer.tissue</span><span class="o">$</span><span class="n">source</span><span class="p">,</span><span class="w"> </span><span class="n">sample</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">enhancer.tissue</span><span class="o">$</span><span class="n">tissues</span><span class="p">,</span>
<span class="w">    </span><span class="n">group</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;enhancer&quot;</span><span class="p">),</span><span class="w"> </span><span class="nf">data.frame</span><span class="p">(</span><span class="n">label</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">gname.tissue</span><span class="o">$</span><span class="n">gene</span><span class="p">,</span><span class="w"> </span><span class="n">sample</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">gname.tissue</span><span class="o">$</span><span class="n">tissues</span><span class="p">,</span>
<span class="w">    </span><span class="n">group</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;gene&quot;</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">rowid_to_column</span><span class="p">(</span><span class="s">&quot;id&quot;</span><span class="p">)</span>
<span class="n">nodes_dist1</span><span class="p">[</span><span class="n">nodes_dist1</span><span class="o">$</span><span class="n">label</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">genes</span><span class="p">,</span><span class="w"> </span><span class="p">]</span><span class="o">$</span><span class="n">group</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;known_gene&quot;</span>
</code></pre></div>
<div class="badCode highlight"><pre><span></span><code><span class="nf">table</span><span class="p">(</span><span class="n">nodes_dist1</span><span class="o">$</span><span class="n">sample</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>      intestine intestine,liver           liver 
            135             290             169 
</code></pre></div>
<div class="badCode highlight"><pre><span></span><code><span class="nf">table</span><span class="p">(</span><span class="n">nodes_dist1</span><span class="o">$</span><span class="n">group</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>  enhancer       gene known_gene 
       137        444         13 
</code></pre></div>
<p>In the list of edges of the graph, <code>edges_dist1</code>, the <code>sample</code> column indicates in which family of tissues (liver, intestine or both) each E-G pair has been found.</p>
<div class="badCode highlight"><pre><span></span><code><span class="nf">table</span><span class="p">(</span><span class="n">edges_dist1</span><span class="o">$</span><span class="n">sample</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>      intestine intestine,liver           liver 
            420             321             606 
</code></pre></div>
<p>We see that 1221 unique E-G pairs found, 46% are specific to liver, 29% are specific to intestine and 24% are shared between intestine and liver.</p>
<p>We add to <code>nodes_dist1</code> a column <code>d_in</code> for plotting purpose: it contains 1 for each node of type <code>enhancer</code>, the number of incoming enhancers for each node of type <code>gene</code>, and the max of the latter for each node of type <code>known_gene</code>.</p>
<h2 id="edge-weight-based-on-abc-score">Edge weight based on ABC score<a class="headerlink" href="#edge-weight-based-on-abc-score" title="Permanent link">&para;</a></h2>
<h3 id="abcmean">ABC.mean<a class="headerlink" href="#abcmean" title="Permanent link">&para;</a></h3>
<p>In this graph, the weights of the edges are proportional to the corresponding mean ABC scores (averaged over all the instances of the E-G pairs found in the different biosamples. Not that, for a given biosample, the "ABC.score" column contains ABC scores that have already been averaged once when merging the enhancers).</p>
<p><a href="https://genoweb.toulouse.inra.fr/~thoellinger/2022/ABC_inferred_genes/preliminary_analysis_v9.html#321_ABCmean"><img alt="" src="images_Rmd/ABC.mean.png" /></a></p>
<blockquote>
<p>We see that none of the genes connected through enhancers to different genes among the initial list of 13 genes ; are connected to more than 1 of them ; ie we still have 13 connected compounds. Yet, most of the initial genes are on different chromosomes, so this is a completely expected behavior.</p>
<p>Also note that one may observe that when selecting a sample, all edges connected to enhancers included in the selected groups are colored (ie are in darker grey than the others), and not all edges connected to genes included in the selected group. The actual reason is that the graph is considered as an oriented graph, resulting in a distinction between entering and outgoing edges. Namely, for each selected nodes, all outgoing edges are colored ; whereas regarding entering edges, only those coming from another selected nodes are colored. In our case, all edges connected to any enhancer are considered as outgoing edges, and all edges connected to any gene are considered as entering edges ; which explains what we observe. Unfortunately there is nothing we can do with the <code>visNetwork</code> package to change this behavior.</p>
</blockquote>
<h3 id="abcmax">ABC.max<a class="headerlink" href="#abcmax" title="Permanent link">&para;</a></h3>
<p>In this graph, the weights of the edges are proportional to the corresponding max ABC scores (over all the instances of the E-G pairs found in the different biosamples. Not that, for a given biosample, the "ABC.score" column contains ABC scores that have already been averaged once when merging the enhancers).</p>
<p><a href="https://genoweb.toulouse.inra.fr/~thoellinger/2022/ABC_inferred_genes/preliminary_analysis_v9.html#322_ABCmax"><img alt="" src="images_Rmd/ABC.max.png" /></a></p>
<h2 id="edge-weight-based-on-distance">Edge weight based on distance<a class="headerlink" href="#edge-weight-based-on-distance" title="Permanent link">&para;</a></h2>
<p>Width proportional to mean distanc. For each enhancer-gene pair $E$-$G$, the distance indicated in the <code>eg</code> dataframe as <code>original_distance.mean</code> is the average distance $E_b$-$G$ over the biosamples $b$ (in base pairs), where $E_b$ is the original enhancer in biosample $b$ which has been replaced by its overlapping merged enhancer $E$. Here, <code>distance_kb</code> is the very same quantity but expressed in <code>kb</code>.</p>
<p><a href="https://genoweb.toulouse.inra.fr/~thoellinger/2022/ABC_inferred_genes/preliminary_analysis_v9.html#33_Edge_weight_based_on_distance"><img alt="" src="images_Rmd/distance.png" /></a></p>
<p>Width proportional to inverse distance:</p>
<p><a href="https://genoweb.toulouse.inra.fr/~thoellinger/2022/ABC_inferred_genes/preliminary_analysis_v9.html#33_Edge_weight_based_on_distance"><img alt="" src="images_Rmd/inverse_distance.png" /></a></p>
<p>Width proportional to rescaled and translated log inverse distance:</p>
<p><a href="https://genoweb.toulouse.inra.fr/~thoellinger/2022/ABC_inferred_genes/preliminary_analysis_v9.html#33_Edge_weight_based_on_distance"><img alt="" src="images_Rmd/rescaled_inverse_distance.png" /></a></p>
<h1 id="results">Results<a class="headerlink" href="#results" title="Permanent link">&para;</a></h1>
<p>The list of the initial genes + all genes regulated by enhancers regulating the 13 initial genes, can be found here: <code>/work2/project/regenet/workspace/thoellinger/shared/2022/networks_hemochromatosis/results/new_genes_v7.list</code>.</p>
<div class="badCode highlight"><pre><span></span><code><span class="nf">write.table</span><span class="p">(</span><span class="n">genes_dist1</span><span class="p">,</span><span class="w"> </span><span class="n">file</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;results/new_genes_v7.list&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">quote</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;\t&quot;</span><span class="p">,</span>
<span class="w">    </span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">F</span><span class="p">,</span><span class="w"> </span><span class="n">col.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">F</span><span class="p">)</span>
</code></pre></div>
<p>The list all genes regulated by enhancers regulating the 13 initial genes, with useful info, can be found here: <code>/work2/project/regenet/workspace/thoellinger/shared/2022/networks_hemochromatosis/results/new_genes_abc_v9_more_info.list</code>.</p>
<div class="badCode highlight"><pre><span></span><code><span class="nf">write.table</span><span class="p">(</span><span class="n">genes_dist1.more</span><span class="p">,</span><span class="w"> </span><span class="n">file</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;results/new_genes_abc_v9_more_info.list&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">quote</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span>
<span class="w">    </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">F</span><span class="p">,</span><span class="w"> </span><span class="n">col.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">T</span><span class="p">)</span>
</code></pre></div>
<p>For each one of the 13 initial genes, the list of genes regulated by enhancers regulating that initial gene, can be found here:</p>
<div class="badCode highlight"><pre><span></span><code><span class="nf">for </span><span class="p">(</span><span class="n">gene</span><span class="w"> </span><span class="n">in</span><span class="w"> </span><span class="n">genes</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="n">current_gene_dist1</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">unique</span><span class="p">(</span><span class="n">eg_dist1</span><span class="p">[</span><span class="n">eg_dist1</span><span class="o">$</span><span class="n">from</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">gene</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;gene&quot;</span><span class="p">])</span>
<span class="w">    </span><span class="nf">write.table</span><span class="p">(</span><span class="n">current_gene_dist1</span><span class="p">,</span><span class="w"> </span><span class="n">file</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste</span><span class="p">(</span><span class="s">&quot;results/separate/&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">gene</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;.list&quot;</span><span class="p">,</span>
<span class="w">        </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;&quot;</span><span class="p">),</span><span class="w"> </span><span class="n">quote</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">F</span><span class="p">,</span><span class="w"> </span><span class="n">col.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">F</span><span class="p">)</span>
<span class="p">}</span>
</code></pre></div>
<p>The list of the enhancers regulating the 13 initial genes, can be found here: <code>/work2/project/regenet/workspace/thoellinger/shared/2022/networks_hemochromatosis/results/enhancers.list</code></p>
<div class="badCode highlight"><pre><span></span><code><span class="nf">write.table</span><span class="p">(</span><span class="n">enhancers</span><span class="p">,</span><span class="w"> </span><span class="n">file</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;results/enhancers.list&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">quote</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;\t&quot;</span><span class="p">,</span>
<span class="w">    </span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">F</span><span class="p">,</span><span class="w"> </span><span class="n">col.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">F</span><span class="p">)</span>
</code></pre></div>
<p>The list of E-G pairs involving the 13 initial genes, can be found here: <code>/work2/project/regenet/workspace/thoellinger/shared/2022/networks_hemochromatosis/results/eg_dist0.bedpe</code></p>
<div class="badCode highlight"><pre><span></span><code><span class="nf">write.table</span><span class="p">(</span><span class="n">eg_dist0</span><span class="p">,</span><span class="w"> </span><span class="n">file</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;results/eg_dist0.bedpe&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">quote</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;\t&quot;</span><span class="p">,</span>
<span class="w">    </span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">F</span><span class="p">,</span><span class="w"> </span><span class="n">col.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">F</span><span class="p">)</span>
</code></pre></div>
<p>The list of the enhancers regulating the 457 initial+new genes, can be found here: <code>/work2/project/regenet/workspace/thoellinger/shared/2022/networks_hemochromatosis/results/enhancers.new_genes.list</code></p>
<div class="badCode highlight"><pre><span></span><code><span class="n">eg_dist2</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">eg</span><span class="p">[</span><span class="n">eg</span><span class="o">$</span><span class="n">gene</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">genes_dist1</span><span class="p">,</span><span class="w"> </span><span class="p">]</span>
<span class="n">enhancers.new_genes</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">unique</span><span class="p">(</span><span class="nf">paste</span><span class="p">(</span><span class="n">eg_dist2</span><span class="o">$</span><span class="n">chrom1</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;:&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">eg_dist2</span><span class="o">$</span><span class="n">start1</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;-&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">eg_dist2</span><span class="o">$</span><span class="n">end1</span><span class="p">,</span>
<span class="w">    </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;&quot;</span><span class="p">))</span>
<span class="nf">write.table</span><span class="p">(</span><span class="n">enhancers.new_genes</span><span class="p">,</span><span class="w"> </span><span class="n">file</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;results/enhancers.new_genes.list&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">quote</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span>
<span class="w">    </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">F</span><span class="p">,</span><span class="w"> </span><span class="n">col.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">F</span><span class="p">)</span>
</code></pre></div>
<p>The list of E-G pairs involving the 457 initial+new genes, can be found here: <code>/work2/project/regenet/workspace/thoellinger/shared/2022/networks_hemochromatosis/results/eg_dist2.bedpe</code></p>
<div class="badCode highlight"><pre><span></span><code><span class="nf">write.table</span><span class="p">(</span><span class="n">eg_dist2</span><span class="p">,</span><span class="w"> </span><span class="n">file</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;results/eg_dist2.bedpe&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">quote</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;\t&quot;</span><span class="p">,</span>
<span class="w">    </span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">F</span><span class="p">,</span><span class="w"> </span><span class="n">col.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">F</span><span class="p">)</span>
</code></pre></div>
<h1 id="appendix">Appendix<a class="headerlink" href="#appendix" title="Permanent link">&para;</a></h1>
<h2 id="tables">Tables<a class="headerlink" href="#tables" title="Permanent link">&para;</a></h2>
<h3 id="table-of-the-initial-abc-predicted-e-g-pairs-involving-the-initial-genes">Table of the initial ABC-predicted E-G pairs involving the initial genes<a class="headerlink" href="#table-of-the-initial-abc-predicted-e-g-pairs-involving-the-initial-genes" title="Permanent link">&para;</a></h3>
<div class="badCode highlight"><pre><span></span><code><span class="nf">options</span><span class="p">(</span><span class="n">max.print</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1000</span><span class="p">)</span>
<span class="n">eg_dist0</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>      chrom1    start1      end1 chrom2    start2      end2
7686    chr1 144930398 144933456   chr1 145413190 145413190
7781    chr1 145394876 145400153   chr1 145413190 145413190
7797    chr1 145413923 145415673   chr1 145413190 145413190
7798    chr1 145421224 145422239   chr1 145413190 145413190
7799    chr1 145426950 145428636   chr1 145413190 145413190
7801    chr1 145434950 145435719   chr1 145413190 145413190
7815    chr1 145442141 145445254   chr1 145413190 145413190
7831    chr1 145451402 145451862   chr1 145413190 145413190
7846    chr1 145454003 145457954   chr1 145413190 145413190
7867    chr1 145473958 145474912   chr1 145413190 145413190
7894    chr1 145541765 145543880   chr1 145413190 145413190
30929  chr12  51158462  51159633  chr12  51420199  51420199
30935  chr12  51326262  51329574  chr12  51420199  51420199
30939  chr12  51417885  51419543  chr12  51420199  51420199
30946  chr12  51421716  51423284  chr12  51420199  51420199
31061  chr12  53318068  53321694  chr12  51420199  51420199
46386  chr15  73074997  73077510  chr15  73344824  73344824
46391  chr15  73320830  73321661  chr15  73344824  73344824
46392  chr15  73326079  73327042  chr15  73344824  73344824
46393  chr15  73345478  73345779  chr15  73344824  73344824
46394  chr15  73346002  73346202  chr15  73344824  73344824
46395  chr15  73346837  73347246  chr15  73344824  73344824
46398  chr15  73401919  73403010  chr15  73344824  73344824
46488  chr15  74906565  74909246  chr15  73344824  73344824
51946  chr16  56639960  56645831  chr16  57481440  57481440
52046  chr16  57298869  57301500  chr16  57481440  57481440
52075  chr16  57302558  57303592  chr16  57481440  57481440
52123  chr16  57333210  57335328  chr16  57481440  57481440
52149  chr16  57337937  57338465  chr16  57481440  57481440
52156  chr16  57417430  57418771  chr16  57481440  57481440
52160  chr16  57451676  57452165  chr16  57481440  57481440
52164  chr16  57452493  57453437  chr16  57481440  57481440
52166  chr16  57466172  57466712  chr16  57481440  57481440
52168  chr16  57477983  57478427  chr16  57481440  57481440
52171  chr16  57504477  57506928  chr16  57481440  57481440
52177  chr16  57512427  57514785  chr16  57481440  57481440
52193  chr16  57553700  57554263  chr16  57481440  57481440
52215  chr16  57653300  57655255  chr16  57481440  57481440
52238  chr16  57669027  57670023  chr16  57481440  57481440
52283  chr16  57830970  57833098  chr16  57481440  57481440
52345  chr16  57925806  57928939  chr16  57481440  57481440
68172  chr19  35757791  35760841  chr19  35773248  35773248
68173  chr19  35768288  35769137  chr19  35773248  35773248
68174  chr19  35769477  35770158  chr19  35773248  35773248
68175  chr19  35770384  35772541  chr19  35773248  35773248
                                  name strand1 strand2    gene
7686    chr1:144930398-144933456::HFE2       .       .    HFE2
7781    chr1:145394876-145400153::HFE2       .       .    HFE2
7797    chr1:145413923-145415673::HFE2       .       .    HFE2
7798    chr1:145421224-145422239::HFE2       .       .    HFE2
7799    chr1:145426950-145428636::HFE2       .       .    HFE2
7801    chr1:145434950-145435719::HFE2       .       .    HFE2
7815    chr1:145442141-145445254::HFE2       .       .    HFE2
7831    chr1:145451402-145451862::HFE2       .       .    HFE2
7846    chr1:145454003-145457954::HFE2       .       .    HFE2
7867    chr1:145473958-145474912::HFE2       .       .    HFE2
7894    chr1:145541765-145543880::HFE2       .       .    HFE2
30929 chr12:51158462-51159633::SLC11A2       .       . SLC11A2
30935 chr12:51326262-51329574::SLC11A2       .       . SLC11A2
30939 chr12:51417885-51419543::SLC11A2       .       . SLC11A2
30946 chr12:51421716-51423284::SLC11A2       .       . SLC11A2
31061 chr12:53318068-53321694::SLC11A2       .       . SLC11A2
46386    chr15:73074997-73077510::NEO1       .       .    NEO1
46391    chr15:73320830-73321661::NEO1       .       .    NEO1
46392    chr15:73326079-73327042::NEO1       .       .    NEO1
46393    chr15:73345478-73345779::NEO1       .       .    NEO1
46394    chr15:73346002-73346202::NEO1       .       .    NEO1
46395    chr15:73346837-73347246::NEO1       .       .    NEO1
46398    chr15:73401919-73403010::NEO1       .       .    NEO1
46488    chr15:74906565-74909246::NEO1       .       .    NEO1
51946 chr16:56639960-56645831::CIAPIN1       .       . CIAPIN1
52046 chr16:57298869-57301500::CIAPIN1       .       . CIAPIN1
52075 chr16:57302558-57303592::CIAPIN1       .       . CIAPIN1
52123 chr16:57333210-57335328::CIAPIN1       .       . CIAPIN1
52149 chr16:57337937-57338465::CIAPIN1       .       . CIAPIN1
52156 chr16:57417430-57418771::CIAPIN1       .       . CIAPIN1
52160 chr16:57451676-57452165::CIAPIN1       .       . CIAPIN1
52164 chr16:57452493-57453437::CIAPIN1       .       . CIAPIN1
52166 chr16:57466172-57466712::CIAPIN1       .       . CIAPIN1
52168 chr16:57477983-57478427::CIAPIN1       .       . CIAPIN1
52171 chr16:57504477-57506928::CIAPIN1       .       . CIAPIN1
52177 chr16:57512427-57514785::CIAPIN1       .       . CIAPIN1
52193 chr16:57553700-57554263::CIAPIN1       .       . CIAPIN1
52215 chr16:57653300-57655255::CIAPIN1       .       . CIAPIN1
52238 chr16:57669027-57670023::CIAPIN1       .       . CIAPIN1
52283 chr16:57830970-57833098::CIAPIN1       .       . CIAPIN1
52345 chr16:57925806-57928939::CIAPIN1       .       . CIAPIN1
68172    chr19:35757791-35760841::HAMP       .       .    HAMP
68173    chr19:35768288-35769137::HAMP       .       .    HAMP
68174    chr19:35769477-35770158::HAMP       .       .    HAMP
68175    chr19:35770384-35772541::HAMP       .       .    HAMP
                                                       ABC.scores
7686                                 0.028761000000000002,0.02026
7781                                                     0.044185
7797                                                     0.062645
7798                                                     0.015668
7799                                                      0.02574
7801                    0.028239999999999998,0.015686000000000002
7815                                                     0.062419
7831                                         0.021353999999999998
7846                               0.028807999999999997,0.0489073
7867                                            0.039256,0.021347
7894                                             0.045182,0.01541
30929                                                    0.018211
30935                               0.015196000000000001,0.022472
30939  0.05407000000000001,0.055278999999999995,0.065178,0.062988
30946                                           0.138303,0.145403
31061                                        0.018961000000000002
46386                                                    0.024743
46391                                         0.05282100000000001
46392                                                    0.042369
46393                                        0.022713999999999998
46394                                        0.015252000000000002
46395                                            0.02547,0.029026
46398                                           0.020356,0.027807
46488                                                     0.01845
51946                                                    0.018703
52046                                                   0.0205155
52075                                0.030206999999999998,0.03244
52123  0.052578,0.088226,0.024952000000000002,0.06717999999999999
52149                                                    0.018795
52156                                                      0.0161
52160                                        0.020324000000000002
52164                                        0.015609999999999999
52166                                           0.016451,0.015881
52168                                                    0.034379
52171   0.016694,0.0385055,0.021072999999999998,0.022296,0.024221
52177 0.025434000000000002,0.031333,0.024978999999999998,0.022293
52193                                                    0.016816
52215                                           0.017281,0.024939
52238                               0.017934000000000002,0.017163
52283                                                    0.017979
52345                   0.017008000000000002,0.015305000000000001
68172                                        0.020316999999999998
68173                                                    0.017357
68174                                        0.015806999999999998
68175                                   0.06381,0.023445,0.065302
                                                                                                    biosamples
7686                                                                            liver-ENCODE,hepatocyte-ENCODE
7781                                                                                              liver-ENCODE
7797                                                                                         hepatocyte-ENCODE
7798                                                                                         hepatocyte-ENCODE
7799                                                                                              liver-ENCODE
7801                                                                            liver-ENCODE,hepatocyte-ENCODE
7815                                                                                              liver-ENCODE
7831                                                                                              liver-ENCODE
7846                                                                            hepatocyte-ENCODE,liver-ENCODE
7867                                                                            hepatocyte-ENCODE,liver-ENCODE
7894                                                                            hepatocyte-ENCODE,liver-ENCODE
30929                                                                                        hepatocyte-ENCODE
30935                                                                               liver-ENCODE,HepG2-Roadmap
30939                               small_intestine_fetal-Roadmap,HepG2-Roadmap,hepatocyte-ENCODE,liver-ENCODE
30946                                              small_intestine_fetal-Roadmap,large_intestine_fetal-Roadmap
31061                                                                            large_intestine_fetal-Roadmap
46386                                                                                        hepatocyte-ENCODE
46391                                                                            large_intestine_fetal-Roadmap
46392                                                                            large_intestine_fetal-Roadmap
46393                                                                                            HepG2-Roadmap
46394                                                                            small_intestine_fetal-Roadmap
46395                                              large_intestine_fetal-Roadmap,small_intestine_fetal-Roadmap
46398                                              small_intestine_fetal-Roadmap,large_intestine_fetal-Roadmap
46488                                                                                        hepatocyte-ENCODE
51946                                                                                             liver-ENCODE
52046                                                                            small_intestine_fetal-Roadmap
52075                                              large_intestine_fetal-Roadmap,small_intestine_fetal-Roadmap
52123                   liver-ENCODE,large_intestine_fetal-Roadmap,HepG2-Roadmap,small_intestine_fetal-Roadmap
52149                                                                                             liver-ENCODE
52156                                                                            large_intestine_fetal-Roadmap
52160                                                                                             liver-ENCODE
52164                                                                                             liver-ENCODE
52166                                              large_intestine_fetal-Roadmap,small_intestine_fetal-Roadmap
52168                                                                                        hepatocyte-ENCODE
52171 small_intestine_fetal-Roadmap,hepatocyte-ENCODE,large_intestine_fetal-Roadmap,liver-ENCODE,HepG2-Roadmap
52177              small_intestine_fetal-Roadmap,hepatocyte-ENCODE,HepG2-Roadmap,large_intestine_fetal-Roadmap
52193                                                                                        hepatocyte-ENCODE
52215                                              small_intestine_fetal-Roadmap,large_intestine_fetal-Roadmap
52238                                              large_intestine_fetal-Roadmap,small_intestine_fetal-Roadmap
52283                                                                                             liver-ENCODE
52345                                                                               HepG2-Roadmap,liver-ENCODE
68172                                                                                             liver-ENCODE
68173                                                                                        hepatocyte-ENCODE
68174                                                                                            HepG2-Roadmap
68175                                                             HepG2-Roadmap,liver-ENCODE,hepatocyte-ENCODE
                           original_distances
7686                        481207.0,480853.0
7781                                  15386.0
7797                                   1252.0
7798                                   8577.0
7799                                  14427.0
7801                          22160.0,22318.0
7815                                  30687.0
7831                                  38465.5
7846                          44235.5,44088.0
7867                          61135.0,61186.0
7894                        129732.5,129661.0
30929                                261301.5
30935                         91637.0,92887.0
30939             1340.5,1461.0,1895.5,1741.0
30946                           2216.0,2198.0
31061                               1900020.0
46386                                267544.0
46391                                 23578.5
46392                                 18400.5
46393                                   855.0
46394                                  1278.0
46395                           2205.0,2198.0
46398                         57556.0,57790.0
46488                               1563131.0
51946                                840504.5
52046                                181096.0
52075                       178475.5,178416.5
52123     147262.0,147349.0,147375.0,147212.0
52149                                143184.5
52156                                 63738.0
52160                                 29562.0
52164                                 28344.5
52166                         14998.0,14968.0
52168                                  3235.0
52171 24545.0,24996.5,24548.0,23673.5,24795.5
52177         32869.0,32909.0,32686.5,32869.0
52193                                 72489.5
52215                       172594.5,172494.0
52238                       188158.5,188051.0
52283                                350469.5
52345                       446763.0,446265.0
68172                                 14913.0
68173                                  4328.0
68174                                  3303.0
68175                    2157.0,1050.0,1856.5
                                    tissues   ABC.mean   ABC.min   ABC.max
7686                            liver,liver 0.02451050 0.0202600 0.0287610
7781                                  liver 0.04418500 0.0441850 0.0441850
7797                                  liver 0.06264500 0.0626450 0.0626450
7798                                  liver 0.01566800 0.0156680 0.0156680
7799                                  liver 0.02574000 0.0257400 0.0257400
7801                            liver,liver 0.02196300 0.0156860 0.0282400
7815                                  liver 0.06241900 0.0624190 0.0624190
7831                                  liver 0.02135400 0.0213540 0.0213540
7846                            liver,liver 0.03885765 0.0288080 0.0489073
7867                            liver,liver 0.03030150 0.0213470 0.0392560
7894                            liver,liver 0.03029600 0.0154100 0.0451820
30929                                 liver 0.01821100 0.0182110 0.0182110
30935                           liver,liver 0.01883400 0.0151960 0.0224720
30939           intestine,liver,liver,liver 0.05937875 0.0540700 0.0651780
30946                   intestine,intestine 0.14185300 0.1383030 0.1454030
31061                             intestine 0.01896100 0.0189610 0.0189610
46386                                 liver 0.02474300 0.0247430 0.0247430
46391                             intestine 0.05282100 0.0528210 0.0528210
46392                             intestine 0.04236900 0.0423690 0.0423690
46393                                 liver 0.02271400 0.0227140 0.0227140
46394                             intestine 0.01525200 0.0152520 0.0152520
46395                   intestine,intestine 0.02724800 0.0254700 0.0290260
46398                   intestine,intestine 0.02408150 0.0203560 0.0278070
46488                                 liver 0.01845000 0.0184500 0.0184500
51946                                 liver 0.01870300 0.0187030 0.0187030
52046                             intestine 0.02051550 0.0205155 0.0205155
52075                   intestine,intestine 0.03132350 0.0302070 0.0324400
52123       liver,intestine,liver,intestine 0.05823400 0.0249520 0.0882260
52149                                 liver 0.01879500 0.0187950 0.0187950
52156                             intestine 0.01610000 0.0161000 0.0161000
52160                                 liver 0.02032400 0.0203240 0.0203240
52164                                 liver 0.01561000 0.0156100 0.0156100
52166                   intestine,intestine 0.01616600 0.0158810 0.0164510
52168                                 liver 0.03437900 0.0343790 0.0343790
52171 intestine,liver,intestine,liver,liver 0.02455790 0.0166940 0.0385055
52177       intestine,liver,liver,intestine 0.02600975 0.0222930 0.0313330
52193                                 liver 0.01681600 0.0168160 0.0168160
52215                   intestine,intestine 0.02111000 0.0172810 0.0249390
52238                   intestine,intestine 0.01754850 0.0171630 0.0179340
52283                                 liver 0.01797900 0.0179790 0.0179790
52345                           liver,liver 0.01615650 0.0153050 0.0170080
68172                                 liver 0.02031700 0.0203170 0.0203170
68173                                 liver 0.01735700 0.0173570 0.0173570
68174                                 liver 0.01580700 0.0158070 0.0158070
68175                     liver,liver,liver 0.05085233 0.0234450 0.0653020
      original_distance.mean original_distance.min original_distance.max
7686              481030.000              480853.0              481207.0
7781               15386.000               15386.0               15386.0
7797                1252.000                1252.0                1252.0
7798                8577.000                8577.0                8577.0
7799               14427.000               14427.0               14427.0
7801               22239.000               22160.0               22318.0
7815               30687.000               30687.0               30687.0
7831               38465.500               38465.5               38465.5
7846               44161.750               44088.0               44235.5
7867               61160.500               61135.0               61186.0
7894              129696.750              129661.0              129732.5
30929             261301.500              261301.5              261301.5
30935              92262.000               91637.0               92887.0
30939               1609.500                1340.5                1895.5
30946               2207.000                2198.0                2216.0
31061            1900020.000             1900020.0             1900020.0
46386             267544.000              267544.0              267544.0
46391              23578.500               23578.5               23578.5
46392              18400.500               18400.5               18400.5
46393                855.000                 855.0                 855.0
46394               1278.000                1278.0                1278.0
46395               2201.500                2198.0                2205.0
46398              57673.000               57556.0               57790.0
46488            1563131.000             1563131.0             1563131.0
51946             840504.500              840504.5              840504.5
52046             181096.000              181096.0              181096.0
52075             178446.000              178416.5              178475.5
52123             147299.500              147212.0              147375.0
52149             143184.500              143184.5              143184.5
52156              63738.000               63738.0               63738.0
52160              29562.000               29562.0               29562.0
52164              28344.500               28344.5               28344.5
52166              14983.000               14968.0               14998.0
52168               3235.000                3235.0                3235.0
52171              24511.700               23673.5               24996.5
52177              32833.375               32686.5               32909.0
52193              72489.500               72489.5               72489.5
52215             172544.250              172494.0              172594.5
52238             188104.750              188051.0              188158.5
52283             350469.500              350469.5              350469.5
52345             446514.000              446265.0              446763.0
68172              14913.000               14913.0               14913.0
68173               4328.000                4328.0                4328.0
68174               3303.000                3303.0                3303.0
68175               1687.833                1050.0                2157.0
                                                                                               biosamples.uniq
7686                                                                            hepatocyte-ENCODE,liver-ENCODE
7781                                                                                              liver-ENCODE
7797                                                                                         hepatocyte-ENCODE
7798                                                                                         hepatocyte-ENCODE
7799                                                                                              liver-ENCODE
7801                                                                            hepatocyte-ENCODE,liver-ENCODE
7815                                                                                              liver-ENCODE
7831                                                                                              liver-ENCODE
7846                                                                            hepatocyte-ENCODE,liver-ENCODE
7867                                                                            hepatocyte-ENCODE,liver-ENCODE
7894                                                                            hepatocyte-ENCODE,liver-ENCODE
30929                                                                                        hepatocyte-ENCODE
30935                                                                               HepG2-Roadmap,liver-ENCODE
30939                               hepatocyte-ENCODE,HepG2-Roadmap,liver-ENCODE,small_intestine_fetal-Roadmap
30946                                              large_intestine_fetal-Roadmap,small_intestine_fetal-Roadmap
31061                                                                            large_intestine_fetal-Roadmap
46386                                                                                        hepatocyte-ENCODE
46391                                                                            large_intestine_fetal-Roadmap
46392                                                                            large_intestine_fetal-Roadmap
46393                                                                                            HepG2-Roadmap
46394                                                                            small_intestine_fetal-Roadmap
46395                                              large_intestine_fetal-Roadmap,small_intestine_fetal-Roadmap
46398                                              large_intestine_fetal-Roadmap,small_intestine_fetal-Roadmap
46488                                                                                        hepatocyte-ENCODE
51946                                                                                             liver-ENCODE
52046                                                                            small_intestine_fetal-Roadmap
52075                                              large_intestine_fetal-Roadmap,small_intestine_fetal-Roadmap
52123                   HepG2-Roadmap,large_intestine_fetal-Roadmap,liver-ENCODE,small_intestine_fetal-Roadmap
52149                                                                                             liver-ENCODE
52156                                                                            large_intestine_fetal-Roadmap
52160                                                                                             liver-ENCODE
52164                                                                                             liver-ENCODE
52166                                              large_intestine_fetal-Roadmap,small_intestine_fetal-Roadmap
52168                                                                                        hepatocyte-ENCODE
52171 hepatocyte-ENCODE,HepG2-Roadmap,large_intestine_fetal-Roadmap,liver-ENCODE,small_intestine_fetal-Roadmap
52177              hepatocyte-ENCODE,HepG2-Roadmap,large_intestine_fetal-Roadmap,small_intestine_fetal-Roadmap
52193                                                                                        hepatocyte-ENCODE
52215                                              large_intestine_fetal-Roadmap,small_intestine_fetal-Roadmap
52238                                              large_intestine_fetal-Roadmap,small_intestine_fetal-Roadmap
52283                                                                                             liver-ENCODE
52345                                                                               HepG2-Roadmap,liver-ENCODE
68172                                                                                             liver-ENCODE
68173                                                                                        hepatocyte-ENCODE
68174                                                                                            HepG2-Roadmap
68175                                                             hepatocyte-ENCODE,HepG2-Roadmap,liver-ENCODE
         tissues.uniq
7686            liver
7781            liver
7797            liver
7798            liver
7799            liver
7801            liver
7815            liver
7831            liver
7846            liver
7867            liver
7894            liver
30929           liver
30935           liver
30939 intestine,liver
30946       intestine
31061       intestine
46386           liver
46391       intestine
46392       intestine
46393           liver
46394       intestine
46395       intestine
46398       intestine
46488           liver
51946           liver
52046       intestine
52075       intestine
52123 intestine,liver
52149           liver
52156       intestine
52160           liver
52164           liver
52166       intestine
52168           liver
52171 intestine,liver
52177 intestine,liver
52193           liver
52215       intestine
52238       intestine
52283           liver
52345           liver
68172           liver
68173           liver
68174           liver
68175           liver
 [ reached &#39;max&#39; / getOption(&quot;max.print&quot;) -- omitted 92 rows ]
</code></pre></div>
<div class="badCode highlight"><pre><span></span><code><span class="nf">options</span><span class="p">(</span><span class="n">max.print</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">75</span><span class="p">)</span>
</code></pre></div>
<h3 id="list-of-the-infered-new-genes">List of the infered new genes<a class="headerlink" href="#list-of-the-infered-new-genes" title="Permanent link">&para;</a></h3>
<div class="badCode highlight"><pre><span></span><code><span class="nf">options</span><span class="p">(</span><span class="n">max.print</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1000</span><span class="p">)</span>
<span class="n">genes_dist1</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>  [1] PDE4DIP         LOC100996724    SEC22B          NOTCH2NL       
  [5] NBPF10          LINC01719       HFE2            TXNIP          
  [9] POLR3GL         ANKRD34A        LIX1L           RBM8A          
 [13] GNRHR2          PEX11B          ANKRD35         PIAS3          
 [17] NUDT17          RNF115          POLR3C          PDZK1          
 [21] GPR89A          PRKAB2          PDIA3P1         FMO5           
 [25] CHD1L           ITGA10          BCL9            DIP2B          
 [29] ATF1            METTL7A         SLC11A2         TMPRSS12       
 [33] LETMD1          CSRNP2          TFCP2           POU6F1         
 [37] DAZAP2          SMAGP           BIN2            GALNT6         
 [41] SLC4A8          ANKRD33         ACVRL1          ACVR1B         
 [45] GRASP           NR4A1           ATG101          KRT80          
 [49] KRT7            KRT18           KRT8            EIF4B          
 [53] TNS2            LOC283335       SPRYD3          IGFBP6         
 [57] SOAT2           CSAD            ZNF740          ITGB7          
 [61] RARG            MFSD5           ESPL1           PFDN5          
 [65] C12orf10        AAAS            AMHR2           PRR13          
 [69] PCBP2           MAP3K12         TARBP2          ATF7           
 [73] ATP5G2          CALCOCO1        CISTR           HOXC10         
 [77] HOXC-AS3        HOXC-AS1        HOXC9           HOXC8          
 [81] HOXC4           HOXC6           HOXC5           LOC100240735   
 [85] SMUG1           CBX5            HNRNPA1         HNRNPA1P10     
 [89] NFE2            COPZ1           LOC102724050    ITGA5          
 [93] NCKAP1L         PDE1B           PPP1R1A         HIGD2B         
 [97] BBS4            ADPGK-AS1       ADPGK           NEO1           
[101] NPTN            CD276           LOXL1           LOXL1-AS1      
[105] STOML1          PML             ISLR            STRA6          
[109] CCDC33          SEMA7A          UBL7            UBL7-AS1       
[113] ARID3B          EDC3            CYP1A1          CSK            
[117] ULK3            SCAMP2          MPI             FAM219B        
[121] COX5A           RPP25           SCAMP5          PPCDC          
[125] C15orf39        COMMD4          NEIL1           MAN2C1         
[129] PTPN9           CSPG4           ODF3L1          UBE2Q2         
[133] FBXO22          TMEM266         CRNDE           IRX5           
[137] MMP2            LPCAT2          CES1            LOC283856      
[141] GNAO1           DKFZP434H168    AMFR            NUDT21         
[145] OGFOD1          BBS2            MT2A            MT1L           
[149] MT1E            MT1M            MT1A            MT1DP          
[153] MT1B            MT1F            MT1G            MT1H           
[157] MT1X            NUP93           HERPUD1         CPNE2          
[161] FAM192A         RSPRY1          ARL2BP          PLLP           
[165] CX3CL1          COQ9            CIAPIN1         POLR2C         
[169] DOK4            CCDC102A        ADGRG1          ADGRG3         
[173] DRC7            KATNB1          CSNK2A2         CCDC113        
[177] GINS3           SETD6           CES1P1          KIFC3          
[181] ZNF319          USB1            SLC38A7         GOT2           
[185] LOC388282       MMP15           CFAP20          NDRG4          
[189] ADGRG5          CNOT1           FXYD5           LSR            
[193] USF2            HAMP            MAG             LINC01531      
[197] FFAR2           WTIP            ZNF792          HPN            
[201] LGI4            FXYD1           TMEM147         TMEM147-AS1    
[205] HAUS5           RBM42           ETV2            COX6B1         
[209] UPK1A-AS1       KMT2B           IGFLR1          U2AF1L4        
[213] PSENEN          LIN37           HSPB6           PROSER3        
[217] ARHGAP33        PRODH2          APLP1           NFKBID         
[221] HCST            LRFN3           SDHAF1          ALKBH6         
[225] LOC101927572    BBS5            KLHL41          FASTKD1        
[229] CCDC173         PHOSPHO2-KLHL23 PHOSPHO2        LINC01124      
[233] SP5             ERICH2          LOC101926913    GAD1           
[237] CYBRD1          GORASP2         TLK1            METTL8         
[241] DCAF17          DYNC1I2         SLC25A12        HAT1           
[245] METAP1D         DLX1            DLX2            DLX2-AS1       
[249] ITGA6           PDK1            CDCA7           RAPGEF4-AS1    
[253] RAPGEF4         ZAK             LINC01960       OLA1           
[257] CIR1            SCRN3           GPR155          CHN1           
[261] GULP1           COL3A1          COL5A2          WDR75          
[265] SLC40A1         ASNSD1          ANKAR           OSGEPL1-AS1    
[269] OSGEPL1         PMS1            ORMDL1          C2orf88        
[273] HIBCH           MFSD6           NEMP2           INPP1          
[277] NAB1            STAT4           MYO1B           GLS            
[281] STAT1           LOC105373805    NABP1           LOC105747689   
[285] SDPR            TMEFF2          FOXRED2         EIF3D          
[289] IFT27           NCF4            CSF2RB          TST            
[293] MPST            KCTD17          TMPRSS6         IL2RB          
[297] C1QTNF6         SSTR3           RAC2            CYTH4          
[301] SDHAP1          TFRC            SLC51A          LINC00885      
[305] ZDHHC19         TCTEX1D2        TM4SF19-AS1     UBXN7-AS1      
[309] UBXN7           RNF168          LY86            RREB1          
[313] SSR1            CAGE1           RIOK1           SNRNP48        
[317] BMP6            BLOC1S5         BLOC1S5-TXNDC5  EEF1E1         
[321] EEF1E1-BLOC1S5  LOC101927972    DSP             TXNDC5         
[325] SLC35B3         LOC100506207    HULC            NRSN1          
[329] KAAG1           DCDC2           MRS2            GPLD1          
[333] ALDH5A1         KIAA0319        TDP2            ACOT13         
[337] C6orf62         GMNN            C6orf229        RIPOR2         
[341] CMAHP           CARMIL1         SCGN            HIST1H2AA      
[345] HIST1H2APS1     SLC17A4         SLC17A1         TRIM38         
[349] HIST1H3A        HIST1H4A        HIST1H4B        HIST1H3B       
[353] HIST1H2AB       HIST1H1C        HFE             HIST1H4C       
[357] HIST1H2BC       HIST1H2AC       HIST1H1D        HIST1H4F       
[361] HIST1H3F        HIST1H2BH       HIST1H3G        HIST1H2BI      
[365] HIST1H4H        BTN3A2          BTN2A2          BTN3A1         
[369] BTN2A3P         BTN3A3          BTN2A1          ABT1           
[373] ZNF322          HIST1H1A        HIST1H2BB       HIST1H3C       
[377] LOC105374988    LOC100270746    LOC108783645    HIST1H1T       
[381] HIST1H1E        HIST1H2BD       HIST1H2BE       HIST1H4D       
[385] HIST1H2AD       HIST1H3D        HIST1H2BF       HIST1H4E       
[389] ZSCAN25         LAMTOR4         C7orf43         GPC2           
[393] STAG3           GATS            PILRA           PPP1R35        
[397] TSC22D4         NYAP1           AGFG2           SAP25          
[401] ZASP            LRCH4           FBXO24          PCOLCE         
[405] PCOLCE-AS1      MOSPD3          TFR2            GNB2           
[409] POP7            EPO             EPHB4           SLC12A9        
[413] TRIP6           SRRT            MUC3A           LOC102724094   
[417] VGF             LOC101927746    GIGYF1          ACTL6B         
[421] LOC105375429    UFSP1           DOK2            XPO7           
[425] NPM2            DMTN            FAM160B2        NUDT18         
[429] HR              REEP4           BMP1            PHYHIP         
[433] POLR3D          SLC39A14        PPP3CC          SORBS3         
[437] PDLIM2          C8orf58         CCAR2           BIN3-IT1       
[441] BIN3            RHOBTB2         LOC286059       TNFRSF10B      
[445] TNFRSF10D       TNFRSF10A       LOC389641       R3HCC1         
[449] LOC100507156    LOXL2           ENTPD4          SLC25A37       
[453] EGR3            LOC254896       ADAM28          ADAMDEC1       
[457] PEBP4          
17527 Levels: A1BG A1BG-AS1 A1CF A2M A2M-AS1 A3GALT2 A4GALT AAAS AACS ... ZZZ3
</code></pre></div>
<div class="badCode highlight"><pre><span></span><code><span class="nf">length</span><span class="p">(</span><span class="n">genes_dist1</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>[1] 457
</code></pre></div>
<div class="badCode highlight"><pre><span></span><code><span class="nf">options</span><span class="p">(</span><span class="n">max.print</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">75</span><span class="p">)</span>
</code></pre></div>





                
              </article>
            </div>
          
          
        </div>
        
          <a href="#" class="md-top md-icon" data-md-component="top" hidden>
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg>
            Back to top
          </a>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      2020 - 2022 Tristan Hoellinger
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
        <div class="md-social">
  
    
    
    
    
      
      
    
    <a href="https://twitter.com/TristanHoellin1" target="_blank" rel="noopener" title="twitter.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"/></svg>
    </a>
  
    
    
    
    
      
      
    
    <a href="https://fr.linkedin.com/in/tristan-hoellinger" target="_blank" rel="noopener" title="fr.linkedin.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"/></svg>
    </a>
  
    
    
    
    
    <a href="mailto:sarah.djebali@inserm.fr" target="_blank" rel="noopener" title="" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M498.1 5.6c10.1 7 15.4 19.1 13.5 31.2l-64 416c-1.5 9.7-7.4 18.2-16 23s-18.9 5.4-28 1.6L284 427.7l-68.5 74.1c-8.9 9.7-22.9 12.9-35.2 8.1S160 493.2 160 480v-83.6c0-4 1.5-7.8 4.2-10.7l167.6-182.9c5.8-6.3 5.6-16-.4-22s-15.7-6.4-22-.7L106 360.8l-88.3-44.2C7.1 311.3.3 300.7 0 288.9s5.9-22.8 16.1-28.7l448-256c10.7-6.1 23.9-5.5 34 1.4z"/></svg>
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    <script id="__config" type="application/json">{"base": "../..", "features": ["navigation.instant", "navigation.tabs", "navigation.sections", "navigation.top"], "search": "../../assets/javascripts/workers/search.db81ec45.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../../assets/javascripts/bundle.6df46069.min.js"></script>
      
    
  </body>
</html>