
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
        <meta name="author" content="Tristan Hoellinger">
      
      
        <link rel="canonical" href="https://eg.github.io/notes_ABC/BENGI/ABC_over_BENGI_GM12878.html">
      
      
        <link rel="prev" href="../test_abc_pipeline.html">
      
      
        <link rel="next" href="../differences_example_chr22.html">
      
      <link rel="icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.4.2, mkdocs-material-9.0.11">
    
    
      
        <title>ABC over GM12878 with same accessions as for BENGI - Genome-wide identification of E/G interactions</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.0d440cfe.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.2505c338.min.css">
      
      

    
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../stylesheets/extra.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="black" data-md-color-accent="">
  
    
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#abc-over-gm12878-with-same-accessions-as-for-bengi-gm12878-benchmarks" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Genome-wide identification of E/G interactions" class="md-header__button md-logo" aria-label="Genome-wide identification of E/G interactions" data-md-component="logo">
      
  <img src="../../logo/logo-inserm-blc.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Genome-wide identification of E/G interactions
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              ABC over GM12878 with same accessions as for BENGI
            
          </span>
        </div>
      </div>
    </div>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/hoellin/eg" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
  </div>
  <div class="md-source__repository">
    hoellin/eg
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../guidebooks/introduction.html" class="md-tabs__link">
        Guidebooks
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../haemochromatosis/introduction.html" class="md-tabs__link">
        Haemochromatosis
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../notes_BENGI/introduction.html" class="md-tabs__link">
        BENGI
      </a>
    </li>
  

      
        
  
  
    
  


  
  
  
    <li class="md-tabs__item">
      <a href="../introduction.html" class="md-tabs__link md-tabs__link--active">
        ABC
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../index.html" class="md-tabs__link">
        About
      </a>
    </li>
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Genome-wide identification of E/G interactions" class="md-nav__button md-logo" aria-label="Genome-wide identification of E/G interactions" data-md-component="logo">
      
  <img src="../../logo/logo-inserm-blc.png" alt="logo">

    </a>
    Genome-wide identification of E/G interactions
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/hoellin/eg" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
  </div>
  <div class="md-source__repository">
    hoellin/eg
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1" >
      
      
      
        <label class="md-nav__link" for="__nav_1" id="__nav_1_label" tabindex="0">
          Guidebooks
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_1_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_1">
          <span class="md-nav__icon md-icon"></span>
          Guidebooks
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../guidebooks/introduction.html" class="md-nav__link">
        Introduction
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../guidebooks/script_bedpe_summary/script_bedpe_summary.html" class="md-nav__link">
        Compute summary stats of a bedpe
      </a>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_3" >
      
      
      
        <label class="md-nav__link" for="__nav_1_3" id="__nav_1_3_label" tabindex="0">
          Basics
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_1_3">
          <span class="md-nav__icon md-icon"></span>
          Basics
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../guidebooks/shell.html" class="md-nav__link">
        Shell
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../guidebooks/conda_environments.html" class="md-nav__link">
        Conda basics
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../guidebooks/git.html" class="md-nav__link">
        Git
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../guidebooks/genotoul.html" class="md-nav__link">
        Genotoul cluster
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../guidebooks/notebooks_ssh.html" class="md-nav__link">
        How to use Jupyter through ssh tunneling?
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_4" >
      
      
      
        <label class="md-nav__link" for="__nav_1_4" id="__nav_1_4_label" tabindex="0">
          Misc
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_4_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_1_4">
          <span class="md-nav__icon md-icon"></span>
          Misc
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../guidebooks/compute_correlations.html" class="md-nav__link">
        Compute correlations
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" >
      
      
      
        <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
          Haemochromatosis
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_2">
          <span class="md-nav__icon md-icon"></span>
          Haemochromatosis
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../haemochromatosis/introduction.html" class="md-nav__link">
        Introduction
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../haemochromatosis/data_format.html" class="md-nav__link">
        Data format
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../haemochromatosis/ABC/custom_go.html" class="md-nav__link">
        Custom script GO
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../haemochromatosis/main_results.html" class="md-nav__link">
        Main results
      </a>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2_5" >
      
      
      
        <label class="md-nav__link" for="__nav_2_5" id="__nav_2_5_label" tabindex="0">
          CHiC data
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_2_5_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_2_5">
          <span class="md-nav__icon md-icon"></span>
          CHiC data
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../haemochromatosis/CHiC/preprocess_data.html" class="md-nav__link">
        Preprocessing
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../haemochromatosis/CHiC/preliminary_analysis_chic_v2.html" class="md-nav__link">
        Networks
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../haemochromatosis/CHiC/go_enrichment.html" class="md-nav__link">
        GO enrichment
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2_6" >
      
      
      
        <label class="md-nav__link" for="__nav_2_6" id="__nav_2_6_label" tabindex="0">
          ABC data
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_2_6_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_2_6">
          <span class="md-nav__icon md-icon"></span>
          ABC data
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../haemochromatosis/ABC/data_processing.html" class="md-nav__link">
        Preprocessing
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../haemochromatosis/ABC/preliminary_analysis_v9.html" class="md-nav__link">
        Networks
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../haemochromatosis/ABC/summary_statistics_eg_pairs.html" class="md-nav__link">
        Summary statistics enhancers
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../haemochromatosis/ABC/go_enrichment.html" class="md-nav__link">
        GO enrichment
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../haemochromatosis/ABC/gorilla.html" class="md-nav__link">
        GOrilla
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../haemochromatosis/ABC/compute_avg_nb_of_connections_per_gene.html" class="md-nav__link">
        Avg nb connections per gene
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../haemochromatosis/ABC/gwas.html" class="md-nav__link">
        GWAS (13 initial genes)
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../haemochromatosis/ABC/gwas_10.html" class="md-nav__link">
        GWAS (10 initial genes)
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" >
      
      
      
        <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
          BENGI
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3">
          <span class="md-nav__icon md-icon"></span>
          BENGI
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/introduction.html" class="md-nav__link">
        Introduction
      </a>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_2" >
      
      
      
        <label class="md-nav__link" for="__nav_3_2" id="__nav_3_2_label" tabindex="0">
          Notebooks
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_2_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_2">
          <span class="md-nav__icon md-icon"></span>
          Notebooks
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/BENGI_to_custom_bedpe.html" class="md-nav__link">
        Convert BENGI datasets to custom bedpe
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_3" >
      
      
      
        <label class="md-nav__link" for="__nav_3_3" id="__nav_3_3_label" tabindex="0">
          Summary statistics
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_3">
          <span class="md-nav__icon md-icon"></span>
          Summary statistics
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/summary_statistics/summary_statistics.html" class="md-nav__link">
        Summary statistics on BENGI GM12878 benchmarks
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/summary_statistics_BENGI/summary_BENGI_bedpe.html" class="md-nav__link">
        (R code) Compute summary statistics over BENGI
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_4" >
      
      
      
        <label class="md-nav__link" for="__nav_3_4" id="__nav_3_4_label" tabindex="0">
          Evaluation of a few methods over BENGI GM12878
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_4_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_4">
          <span class="md-nav__icon md-icon"></span>
          Evaluation of a few methods over BENGI GM12878
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_4_1" >
      
      
      
        <label class="md-nav__link" for="__nav_3_4_1" id="__nav_3_4_1_label" tabindex="0">
          Distance method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_4_1_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_4_1">
          <span class="md-nav__icon md-icon"></span>
          Distance method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/distance_method/distance_evaluation_with_code.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/distance_method/summary_distance_evaluation.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_4_2" >
      
      
      
        <label class="md-nav__link" for="__nav_3_4_2" id="__nav_3_4_2_label" tabindex="0">
          DNase-expression correlation method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_4_2_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_4_2">
          <span class="md-nav__icon md-icon"></span>
          DNase-expression correlation method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/dnase_expression_correlation/correlation_method_with_code.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/dnase_expression_correlation/summary_correlation_method.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_4_3" >
      
      
      
        <label class="md-nav__link" for="__nav_3_4_3" id="__nav_3_4_3_label" tabindex="0">
          Average rank method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_4_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_4_3">
          <span class="md-nav__icon md-icon"></span>
          Average rank method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/avg_rank_method/avg_rank_method_with_code.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/avg_rank_method/summary_avg_rank_method.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_5" >
      
      
      
        <label class="md-nav__link" for="__nav_3_5" id="__nav_3_5_label" tabindex="0">
          Evaluation of a few methods over Fulco et al 2019 CRISPRi-FlowFISH
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_5_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_5">
          <span class="md-nav__icon md-icon"></span>
          Evaluation of a few methods over Fulco et al 2019 CRISPRi-FlowFISH
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_5_1" >
      
      
      
        <label class="md-nav__link" for="__nav_3_5_1" id="__nav_3_5_1_label" tabindex="0">
          Distance method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_5_1_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_5_1">
          <span class="md-nav__icon md-icon"></span>
          Distance method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/CRISPRi_FlowFISH/distance_method/distance_over_fulco_et_al_crispri.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/CRISPRi_FlowFISH/distance_method/summary_distance_over_fulco_et_al_crispri.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_5_2" >
      
      
      
        <label class="md-nav__link" for="__nav_3_5_2" id="__nav_3_5_2_label" tabindex="0">
          DNase-expression correlation method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_5_2_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_5_2">
          <span class="md-nav__icon md-icon"></span>
          DNase-expression correlation method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/CRISPRi_FlowFISH/dnase_expression_correlation/correlation_method_with_code.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/CRISPRi_FlowFISH/dnase_expression_correlation/summary_correlation_method.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_5_3" >
      
      
      
        <label class="md-nav__link" for="__nav_3_5_3" id="__nav_3_5_3_label" tabindex="0">
          Average rank method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_5_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_5_3">
          <span class="md-nav__icon md-icon"></span>
          Average rank method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/CRISPRi_FlowFISH/avg_rank_method/avg_rank_method_with_code.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/CRISPRi_FlowFISH/avg_rank_method/summary_avg_rank.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" checked>
      
      
      
        <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
          ABC
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="true">
        <label class="md-nav__title" for="__nav_4">
          <span class="md-nav__icon md-icon"></span>
          ABC
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../introduction.html" class="md-nav__link">
        Introduction
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="ABC_BENGI_results.html" class="md-nav__link">
        ABC over BENGI for GM12878, main results
      </a>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_3" >
      
      
      
        <label class="md-nav__link" for="__nav_4_3" id="__nav_4_3_label" tabindex="0">
          CRISPRi-FlowFISH
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_4_3">
          <span class="md-nav__icon md-icon"></span>
          CRISPRi-FlowFISH
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../K562/ABC_K562_CRISPRi_FlowFISH.html" class="md-nav__link">
        Original ABC predictions over K562
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../K562/recap_investigations_ABC_over_CRiFF.html" class="md-nav__link">
        Summary ABC over CRiFF
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_4" >
      
      
      
        <label class="md-nav__link" for="__nav_4_4" id="__nav_4_4_label" tabindex="0">
          Notebooks
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_4_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_4_4">
          <span class="md-nav__icon md-icon"></span>
          Notebooks
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../K562/april_K562_56_genes/april_K562_56_genes.html" class="md-nav__link">
        ABC over K562 cCRiFF
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../K562/april_K562_59_genes/april_K562_59_genes.html" class="md-nav__link">
        ABC over K562 CRiFF
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../K562/april_K562_candidates_not_in_whitelist_56_genes/april_K562_candidates_NOT_in_whitelist.html" class="md-nav__link">
        ABC over K562 cCRiFF, no whitelist
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../generic_notebooks/turnkey_notebook_to_run_ABC_with_example_over_GM12878.html" class="md-nav__link">
        Generic notebook to run ABC method & Example over GM12878
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="notebook_ABC_over_BENGI_GM12878.html" class="md-nav__link">
        ABC over GM12878 with same accessions as for BENGI
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="notebook_ABC_over_BENGI_GM12878_from_ccRE_ELSs.html" class="md-nav__link">
        ABC over GM12878 with same accessions as for BENGI, starting with ccRE-dELS
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
    
  
  
    
      
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_5" checked>
      
      
      
        <label class="md-nav__link" for="__nav_4_5" id="__nav_4_5_label" tabindex="0">
          Full codes
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_5_label" aria-expanded="true">
        <label class="md-nav__title" for="__nav_4_5">
          <span class="md-nav__icon md-icon"></span>
          Full codes
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../test_abc_pipeline.html" class="md-nav__link">
        Test of ABC pipeline over chr22
      </a>
    </li>
  

            
          
            
              
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          ABC over GM12878 with same accessions as for BENGI
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="ABC_over_BENGI_GM12878.html" class="md-nav__link md-nav__link--active">
        ABC over GM12878 with same accessions as for BENGI
      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#requirements" class="md-nav__link">
    Requirements
  </a>
  
    <nav class="md-nav" aria-label="Requirements">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#chromatin-accessibility-dnase-seq" class="md-nav__link">
    Chromatin accessibility (DNase-seq)
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#blacklist" class="md-nav__link">
    Blacklist
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#histone-mark-h3k27ac-chip-seq" class="md-nav__link">
    Histone mark H3K27ac ChIP-seq
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gene-expression-polya-rna-seq" class="md-nav__link">
    Gene expression (polyA+ RNA-seq)
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gene-annotation" class="md-nav__link">
    Gene annotation
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hi-c" class="md-nav__link">
    Hi-C
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ubiquitously-expressed-genes" class="md-nav__link">
    Ubiquitously expressed genes
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#genome-file" class="md-nav__link">
    Genome file
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#data-processing" class="md-nav__link">
    Data processing
  </a>
  
    <nav class="md-nav" aria-label="Data processing">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#create-expression-table" class="md-nav__link">
    Create expression table
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#filter-gene-annotation" class="md-nav__link">
    Filter gene annotation
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ubiquitously-expressed-genes_1" class="md-nav__link">
    Ubiquitously expressed genes
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#running-the-abc-model" class="md-nav__link">
    Running the ABC model
  </a>
  
    <nav class="md-nav" aria-label="Running the ABC model">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#step-1-define-candidate-elements" class="md-nav__link">
    Step 1: define candidate elements
  </a>
  
    <nav class="md-nav" aria-label="Step 1: define candidate elements">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#call-peaks-with-macs2" class="md-nav__link">
    Call peaks with macs2
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#use-abc-makecandidateregionspy-to-define-candidate-regions" class="md-nav__link">
    Use ABC makeCandidateRegions.py to define candidate regions
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-2-quantifying-enhancer-activity" class="md-nav__link">
    Step 2: quantifying enhancer activity
  </a>
  
    <nav class="md-nav" aria-label="Step 2: quantifying enhancer activity">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#use-abc-runneighborhoodspy" class="md-nav__link">
    Use ABC run.neighborhoods.py
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-3-computing-the-abc-score" class="md-nav__link">
    Step 3: computing the ABC score
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#results" class="md-nav__link">
    Results
  </a>
  
    <nav class="md-nav" aria-label="Results">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#summary-statistics" class="md-nav__link">
    Summary statistics
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#evaluation-of-abc-predictions-for-gm12878-on-bengi-datasets" class="md-nav__link">
    Evaluation of ABC predictions for GM12878 on BENGI datasets
  </a>
  
    <nav class="md-nav" aria-label="Evaluation of ABC predictions for GM12878 on BENGI datasets">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#intersection-between-abc-prediction-bengi" class="md-nav__link">
    Intersection between ABC prediction / BENGI
  </a>
  
    <nav class="md-nav" aria-label="Intersection between ABC prediction / BENGI">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#standard-abc" class="md-nav__link">
    Standard ABC
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#standard-abc-with-more-peaks" class="md-nav__link">
    Standard ABC with more peaks
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#abc-starting-from-ccre-delss-as-candidate-enhancers" class="md-nav__link">
    ABC starting from ccRE-dELSs as candidate enhancers
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#precision-recall-curve" class="md-nav__link">
    Precision / recall curve
  </a>
  
    <nav class="md-nav" aria-label="Precision / recall curve">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#abc-starting-from-ccre-delss-as-candidate-enhancers_1" class="md-nav__link">
    ABC starting from ccRE-dELSs as candidate enhancers
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#distance-method" class="md-nav__link">
    Distance method
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#standard-abc_1" class="md-nav__link">
    Standard ABC
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#standard-abc-with-more-peaks_1" class="md-nav__link">
    Standard ABC with more peaks
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../differences_example_chr22.html" class="md-nav__link">
        Meld of results over chr22 vs authors' results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" >
      
      
      
        <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
          About
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_5">
          <span class="md-nav__icon md-icon"></span>
          About
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../index.html" class="md-nav__link">
        Getting started
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../about_me.html" class="md-nav__link">
        About me
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#requirements" class="md-nav__link">
    Requirements
  </a>
  
    <nav class="md-nav" aria-label="Requirements">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#chromatin-accessibility-dnase-seq" class="md-nav__link">
    Chromatin accessibility (DNase-seq)
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#blacklist" class="md-nav__link">
    Blacklist
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#histone-mark-h3k27ac-chip-seq" class="md-nav__link">
    Histone mark H3K27ac ChIP-seq
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gene-expression-polya-rna-seq" class="md-nav__link">
    Gene expression (polyA+ RNA-seq)
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gene-annotation" class="md-nav__link">
    Gene annotation
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hi-c" class="md-nav__link">
    Hi-C
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ubiquitously-expressed-genes" class="md-nav__link">
    Ubiquitously expressed genes
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#genome-file" class="md-nav__link">
    Genome file
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#data-processing" class="md-nav__link">
    Data processing
  </a>
  
    <nav class="md-nav" aria-label="Data processing">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#create-expression-table" class="md-nav__link">
    Create expression table
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#filter-gene-annotation" class="md-nav__link">
    Filter gene annotation
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ubiquitously-expressed-genes_1" class="md-nav__link">
    Ubiquitously expressed genes
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#running-the-abc-model" class="md-nav__link">
    Running the ABC model
  </a>
  
    <nav class="md-nav" aria-label="Running the ABC model">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#step-1-define-candidate-elements" class="md-nav__link">
    Step 1: define candidate elements
  </a>
  
    <nav class="md-nav" aria-label="Step 1: define candidate elements">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#call-peaks-with-macs2" class="md-nav__link">
    Call peaks with macs2
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#use-abc-makecandidateregionspy-to-define-candidate-regions" class="md-nav__link">
    Use ABC makeCandidateRegions.py to define candidate regions
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-2-quantifying-enhancer-activity" class="md-nav__link">
    Step 2: quantifying enhancer activity
  </a>
  
    <nav class="md-nav" aria-label="Step 2: quantifying enhancer activity">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#use-abc-runneighborhoodspy" class="md-nav__link">
    Use ABC run.neighborhoods.py
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-3-computing-the-abc-score" class="md-nav__link">
    Step 3: computing the ABC score
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#results" class="md-nav__link">
    Results
  </a>
  
    <nav class="md-nav" aria-label="Results">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#summary-statistics" class="md-nav__link">
    Summary statistics
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#evaluation-of-abc-predictions-for-gm12878-on-bengi-datasets" class="md-nav__link">
    Evaluation of ABC predictions for GM12878 on BENGI datasets
  </a>
  
    <nav class="md-nav" aria-label="Evaluation of ABC predictions for GM12878 on BENGI datasets">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#intersection-between-abc-prediction-bengi" class="md-nav__link">
    Intersection between ABC prediction / BENGI
  </a>
  
    <nav class="md-nav" aria-label="Intersection between ABC prediction / BENGI">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#standard-abc" class="md-nav__link">
    Standard ABC
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#standard-abc-with-more-peaks" class="md-nav__link">
    Standard ABC with more peaks
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#abc-starting-from-ccre-delss-as-candidate-enhancers" class="md-nav__link">
    ABC starting from ccRE-dELSs as candidate enhancers
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#precision-recall-curve" class="md-nav__link">
    Precision / recall curve
  </a>
  
    <nav class="md-nav" aria-label="Precision / recall curve">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#abc-starting-from-ccre-delss-as-candidate-enhancers_1" class="md-nav__link">
    ABC starting from ccRE-dELSs as candidate enhancers
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#distance-method" class="md-nav__link">
    Distance method
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#standard-abc_1" class="md-nav__link">
    Standard ABC
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#standard-abc-with-more-peaks_1" class="md-nav__link">
    Standard ABC with more peaks
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  

  
  



<h1 id="abc-over-gm12878-with-same-accessions-as-for-bengi-gm12878-benchmarks">ABC over GM12878 with same accessions as for BENGI GM12878 benchmarks<a class="headerlink" href="#abc-over-gm12878-with-same-accessions-as-for-bengi-gm12878-benchmarks" title="Permanent link">&para;</a></h1>
<h2 id="requirements">Requirements<a class="headerlink" href="#requirements" title="Permanent link">&para;</a></h2>
<h3 id="chromatin-accessibility-dnase-seq">Chromatin accessibility (DNase-seq)<a class="headerlink" href="#chromatin-accessibility-dnase-seq" title="Permanent link">&para;</a></h3>
<p>We use the same DNase-seq data as Moore et al. 2020 for DNase-DNase correlation method - that is, the data used by <a href="https://www.nature.com/articles/nature11232">Thurman et al. 2012</a>.</p>
<p>Experiment: <code>ENCSR000COQ</code> https://www.encodeproject.org/experiments/ENCSR000COQ/</p>
<p>Replicate 1: <code>ENCFF664VNB.bam</code> ; Replicate 2: <code>ENCFF790WER.bam</code></p>
<div class="highlight"><pre><span></span><code>wget<span class="w"> </span>https://www.encodeproject.org/files/ENCFF664VNB/@@download/ENCFF664VNB.bam<span class="w"> </span>https://www.encodeproject.org/files/ENCFF790WER/@@download/ENCFF790WER.bam<span class="w"> </span>-P<span class="w"> </span>/work2/project/regenet/results/dnaseseq/homo_sapiens/hg19/gm12878/
</code></pre></div>
<h3 id="blacklist">Blacklist<a class="headerlink" href="#blacklist" title="Permanent link">&para;</a></h3>
<p>We use <a href="https://github.com/Boyle-Lab/Blacklist">the ENCODE Blacklist</a> for Human GRCh19 available <a href="https://github.com/Boyle-Lab/Blacklist/blob/master/lists/hg19-blacklist.v2.bed.gz">here</a>.</p>
<div class="highlight"><pre><span></span><code>wget<span class="w"> </span>https://github.com/Boyle-Lab/Blacklist/raw/master/lists/hg19-blacklist.v2.bed.gz<span class="w"> </span>-P<span class="w"> </span>/work2/project/regenet/data/species/homo_sapiens/hg19/
</code></pre></div>
<h3 id="histone-mark-h3k27ac-chip-seq">Histone mark H3K27ac ChIP-seq<a class="headerlink" href="#histone-mark-h3k27ac-chip-seq" title="Permanent link">&para;</a></h3>
<p>In Moore et al. 2020, these histone marks data are used:
* to build the ccREs
* for the training of Target Finder</p>
<p>For the sake of comparability between methods evaluation, if ever it appears that H3K27ac ChIP-seq accessions are not the same for ccREs and Target Finder training, we should use here the same as the ones used for Target Finder training. We shall verify later whether the accessions are the same or not.</p>
<p>Quotation from Moore et al. 2020:</p>
<blockquote>
<p><strong><em>TargetFinder</em></strong>
We reimplemented TargetFinder to run on our cCRE-gene pairs with chromCV. For features, we used the identical datasets described by Whalen et al. for each cell type.</p>
</blockquote>
<p>The accessions of H3k27ac ChIP-seq files used are available in supplementary Table S2 of <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4910881/">Whalen et al.</a>:</p>
<table>
<thead>
<tr>
<th align="center">Cell line</th>
<th align="center">K562</th>
<th>GM12878</th>
<th>HeLa-S3</th>
<th align="center">HUVEC</th>
<th align="center">IMR90</th>
<th align="center">NHEK</th>
</tr>
</thead>
<tbody>
<tr>
<td align="center">ChIP-seq (H3K27ac)</td>
<td align="center">GSE29611</td>
<td>GSE29611</td>
<td>GSE29611</td>
<td align="center">GSE29611</td>
<td align="center">GSE16256</td>
<td align="center">GSE29611</td>
</tr>
</tbody>
</table>
<p>So either we download the big file (&gt; 330 gb) for the <code>GSE29611</code> accession on GEO, or only the info file:</p>
<div class="highlight"><pre><span></span><code>wget<span class="w"> </span>ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE29nnn/GSE29611/suppl/GSE29611_run_info.txt.gz<span class="w"> </span>-P<span class="w"> </span>/work2/project/regenet/results/chipseq/h3k27ac/homo_sapiens/hg19/gm12878/
</code></pre></div>
<p>in which we find the precise accession for H3K27ac ChIP-seq, which is: <code>GSM733771</code>: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM733771</p>
<p>So now:</p>
<div class="highlight"><pre><span></span><code>wget<span class="w"> </span>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM733nnn/GSM733771/suppl/GSM733771_hg19_wgEncodeBroadHistoneGm12878H3k27acStdAlnRep1.bam<span class="w"> </span>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM733nnn/GSM733771/suppl/GSM733771_hg19_wgEncodeBroadHistoneGm12878H3k27acStdAlnRep2.bam<span class="w"> </span>-P<span class="w"> </span>/work2/project/regenet/results/chipseq/h3k27ac/homo_sapiens/hg19/gm12878/
</code></pre></div>
<blockquote>
<div class="highlight"><pre><span></span><code>/work2/project/regenet/results/chipseq/h3k27ac/homo_sapiens/hg19/gm12878/
├──<span class="w"> </span>GSE29611_run_info.txt.gz
├──<span class="w"> </span>GSM733771_hg19_wgEncodeBroadHistoneGm12878H3k27acStdAlnRep1.bam
└──<span class="w"> </span>GSM733771_hg19_wgEncodeBroadHistoneGm12878H3k27acStdAlnRep2.bam
</code></pre></div>
</blockquote>
<p>Note that we could have download the corresponding accessions from <a href="https://www.encodeproject.org">encodeproject.org</a>. That is NOT what we have done for this first run, but that is what we have done in the second run from the Jupyter notebook - which by the way is the only difference and should not have much impact on the results.</p>
<p>Experiment: <code>ENCSR000AKC</code> https://www.encodeproject.org/experiments/ENCSR000AKC/</p>
<p>Replicate 1: <code>ENCFF197QHX.bam</code> ; Replicate 2: <code>ENCFF882PRP.bam</code></p>
<div class="highlight"><pre><span></span><code><span class="c1"># This IS NOT what we have done for this run (of which results are in `GEO_BENGI_GM12878`).</span>
<span class="c1"># These are the accessions used to obtain the results in `BENGI_GM12878`.</span>
wget<span class="w"> </span>https://www.encodeproject.org/files/ENCFF197QHX/@@download/ENCFF197QHX.bam<span class="w"> </span>https://www.encodeproject.org/files/ENCFF882PRP/@@download/ENCFF882PRP.bam<span class="w"> </span>-P<span class="w"> </span>/work2/project/regenet/results/dnaseseq/homo_sapiens/hg19/gm12878/
</code></pre></div>
<h3 id="gene-expression-polya-rna-seq">Gene expression (polyA+ RNA-seq)<a class="headerlink" href="#gene-expression-polya-rna-seq" title="Permanent link">&para;</a></h3>
<p>We use the two replicates provided by BENGI Moore et al. 2020 in Additional File 1 Table S1b.</p>
<p>Experiment: <code>ENCSR000COQ</code> Thomas Gingeras, CSHL https://www.encodeproject.org/experiments/ENCSR000COQ/</p>
<p>Replicate 1: <code>ENCFF587ALK</code> ; Replicate 2: <code>ENCFF766CPS</code></p>
<div class="highlight"><pre><span></span><code>wget<span class="w"> </span>https://www.encodeproject.org/files/ENCFF587ALK/@@download/ENCFF587ALK.tsv<span class="w"> </span>https://www.encodeproject.org/files/ENCFF766CPS/@@download/ENCFF766CPS.tsv<span class="w"> </span>-P<span class="w"> </span>/work2/project/regenet/results/rnaseq/homo_sapiens/hg19/gm12878/
</code></pre></div>
<h3 id="gene-annotation">Gene annotation<a class="headerlink" href="#gene-annotation" title="Permanent link">&para;</a></h3>
<p>We use the encode version 19 of hg19: <code>/work2/project/fragencode/data/species/homo_sapiens/hg19.gencv19/homo_sapiens.gtf</code></p>
<p>We also need a files with 2 columns making the association between gene names / gene ids. We use the following: <code>/work2/project/fragencode/data/species/homo_sapiens/hg19.gencv19/homo_sapiens.gnid.gnname.tsv</code></p>
<h3 id="hi-c">Hi-C<a class="headerlink" href="#hi-c" title="Permanent link">&para;</a></h3>
<p>First we try without Hi-C data, but later on we may retry with the Hi-C data used to build BENGI datasets.</p>
<h3 id="ubiquitously-expressed-genes">Ubiquitously expressed genes<a class="headerlink" href="#ubiquitously-expressed-genes" title="Permanent link">&para;</a></h3>
<p>We use the <code>UbiquitouslyExpressedGenesHG19.txt</code> list of 847 ubiquitously expressed genes used by ABC authors.</p>
<h3 id="genome-file">Genome file<a class="headerlink" href="#genome-file" title="Permanent link">&para;</a></h3>
<p>We extract chromosomes sizes from the header of <code>ENCFF664VNB.bam</code> (DNase-seq).</p>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span><span class="w"> </span>/work2/project/regenet/workspace/thoellinger/ABC-Enhancer-Gene-Prediction/BENGI_GM12878
conda<span class="w"> </span>activate<span class="w"> </span>base<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>module<span class="w"> </span>load<span class="w"> </span>bioinfo/samtools-1.9
samtools<span class="w"> </span>view<span class="w"> </span>-H<span class="w"> </span>/work2/project/regenet/results/dnaseseq/homo_sapiens/hg19/gm12878/ENCFF664VNB.bam<span class="w"> </span><span class="p">|</span><span class="w"> </span>grep<span class="w"> </span>SQ<span class="w"> </span><span class="p">|</span><span class="w"> </span>cut<span class="w"> </span>-f<span class="w"> </span><span class="m">2</span>,3<span class="w"> </span><span class="p">|</span>cut<span class="w"> </span>-c<span class="w"> </span><span class="m">4</span>-<span class="w"> </span><span class="p">|</span>awk<span class="w"> </span><span class="s1">&#39;BEGIN{FS=&quot;\t&quot;} {split($2,locus,&quot;:&quot;); if($1 ~ /^(chr)([0-9]{1,2}$)|(M$)|(X$)|(Y$)/){print($1&quot;\t&quot;locus[2])}}&#39;</span><span class="w"> </span>&gt;<span class="w"> </span>reference/chr_sizes
</code></pre></div>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span><span class="s1">&#39;{print $1&quot;\t&quot;0&quot;\t&quot;$2}&#39;</span><span class="w"> </span>reference/chr_sizes<span class="w"> </span>&gt;<span class="w"> </span>reference/chr_sizes.bed
</code></pre></div>
<h2 id="data-processing">Data processing<a class="headerlink" href="#data-processing" title="Permanent link">&para;</a></h2>
<h3 id="create-expression-table">Create expression table<a class="headerlink" href="#create-expression-table" title="Permanent link">&para;</a></h3>
<p>We use <code>ABC-Enhancer-Gene-Prediction/compute_mean_expression.awk</code> to take as the reference TPM value, the mean value for the 2 replicates. If needed, see content of the script in <a href="http://genoweb.toulouse.inra.fr/~thoellinger/notes_perso/ABC_model/OK/ABC_on_GM12878.html"><code>ABC_on_GM12818.md</code></a>.</p>
<div class="highlight"><pre><span></span><code>mkdir<span class="w"> </span>reference/expression/
</code></pre></div>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span>-f<span class="w"> </span>../compute_mean_expression.awk<span class="w"> </span>/work2/project/fragencode/data/species/homo_sapiens/hg19.gencv19/homo_sapiens.gnid.gnname.tsv<span class="w"> </span>/work2/project/regenet/results/rnaseq/homo_sapiens/hg19/gm12878/ENCFF587ALK.tsv<span class="w"> </span>/work2/project/regenet/results/rnaseq/homo_sapiens/hg19/gm12878/ENCFF766CPS.tsv<span class="w"> </span>&gt;<span class="w"> </span>reference/expression/GM12878.ENCFF587ALK_ENCFF766CPS.mean.TPM.txt
</code></pre></div>
<p>WARNING: for this run, we first computed mean expression for gene names (as it what was done initially on ABC examples), but to later convert it to expression for gene ids at the beginning of step 2. That was a mistake, as we lost a lot of gene doing so. We shall not reproduce this later! (Note that <code>compute_mean_expression.awk</code> now directly computes mean expression for gene ids so these guidelines are correct).</p>
<h3 id="filter-gene-annotation">Filter gene annotation<a class="headerlink" href="#filter-gene-annotation" title="Permanent link">&para;</a></h3>
<p>We keep genes only, and select appropriate columns to fit the format of <code>reference/RefSeqCurated.170308.bed.CollapsedGeneBounds.bed</code> available on <a href="https://github.com/broadinstitute/ABC-Enhancer-Gene-Prediction">ABC repository on Github</a>, that is as follows:</p>
<blockquote>
<div class="highlight"><pre><span></span><code>chrX<span class="w">  </span><span class="m">99883667</span><span class="w">    </span><span class="m">99894988</span><span class="w">    </span>ENSG00000000003.10<span class="w">  </span><span class="m">0</span><span class="w">   </span>-
chrX<span class="w">  </span><span class="m">99839799</span><span class="w">    </span><span class="m">99854882</span><span class="w">    </span>ENSG00000000005.5<span class="w">   </span><span class="m">0</span><span class="w">   </span>+
chr20<span class="w"> </span><span class="m">49551404</span><span class="w">    </span><span class="m">49575092</span><span class="w">    </span>ENSG00000000419.8<span class="w">   </span><span class="m">0</span><span class="w">   </span>-
</code></pre></div>
</blockquote>
<p>WARNING: ABC assumes gene annotation is sorted at step 2, so we NEED to sort it, otherwise step 2 won't work.</p>
<p>Moreover, we must keep only genes that lie on a chromosome listed in our curated <code>genome_file</code>.</p>
<div class="highlight"><pre><span></span><code>mkdir<span class="w"> </span>reference/genes_annotation
</code></pre></div>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span><span class="s1">&#39;$3 ~ /(^gene$)/ {print $1&quot;\t&quot;$4&quot;\t&quot;$5&quot;\t&quot;substr($10,2,length($10)-3)&quot;\t&quot;0&quot;\t&quot;$7}&#39;</span><span class="w"> </span>/work2/project/fragencode/data/species/homo_sapiens/hg19.gencv19/homo_sapiens.gtf<span class="w"> </span><span class="p">|</span><span class="w"> </span>bedtools<span class="w"> </span>sort<span class="w"> </span>-g<span class="w"> </span>/work2/project/regenet/workspace/thoellinger/ABC-Enhancer-Gene-Prediction/bisBENGI_GM12878/reference/chr_sizes<span class="w"> </span>-i<span class="w"> </span>stdin<span class="w"> </span>&gt;<span class="w"> </span>reference/genes_annotation/gene_ids.bed
</code></pre></div>
<h3 id="ubiquitously-expressed-genes_1">Ubiquitously expressed genes<a class="headerlink" href="#ubiquitously-expressed-genes_1" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span><span class="s1">&#39;BEGIN{sep=&quot;\t&quot;} NR==FNR {id[$2]=$1; next} id[$1] {print id[$1]}&#39;</span><span class="w"> </span>/work2/project/fragencode/data/species/homo_sapiens/hg19.gencv19/homo_sapiens.gnid.gnname.tsv<span class="w"> </span>../reference/UbiquitouslyExpressedGenesHG19.txt<span class="w"> </span>&gt;<span class="w"> </span>reference/UbiquitouslyExpressedGenes_ids.txt
</code></pre></div>
<blockquote>
<div class="highlight"><pre><span></span><code>$<span class="w"> </span>wc<span class="w"> </span>-l<span class="w"> </span>reference/UbiquitouslyExpressedGenes_ids.txt<span class="w"> </span>
<span class="m">843</span>
<span class="nv">$wc</span><span class="w"> </span>-l<span class="w"> </span>../reference/UbiquitouslyExpressedGenesHG19.txt<span class="w"> </span>
<span class="m">847</span>
</code></pre></div>
</blockquote>
<p>We lost 4 ubiquitously expressed genes along the way.</p>
<h2 id="running-the-abc-model">Running the ABC model<a class="headerlink" href="#running-the-abc-model" title="Permanent link">&para;</a></h2>
<h3 id="step-1-define-candidate-elements">Step 1: define candidate elements<a class="headerlink" href="#step-1-define-candidate-elements" title="Permanent link">&para;</a></h3>
<h4 id="call-peaks-with-macs2">Call peaks with <code>macs2</code><a class="headerlink" href="#call-peaks-with-macs2" title="Permanent link">&para;</a></h4>
<p>Write the following in <code>BENGI_GM12878/hi_slurm/step1.1.sh</code>:</p>
<div class="highlight"><pre><span></span><code><span class="ch">#!/bin/sh</span>
<span class="c1"># dependencies: python2</span>
macs2<span class="w"> </span>callpeak<span class="w"> </span><span class="se">\</span>
-t<span class="w"> </span>/work2/project/regenet/results/dnaseseq/homo_sapiens/hg19/gm12878/ENCFF664VNB.bam<span class="w"> </span>/work2/project/regenet/results/dnaseseq/homo_sapiens/hg19/gm12878/ENCFF790WER.bam<span class="w"> </span><span class="se">\</span>
-n<span class="w"> </span>ENCFF664VNB_ENCFF790WER.macs2<span class="w"> </span><span class="se">\</span>
-f<span class="w"> </span>BAM<span class="w"> </span><span class="se">\</span>
-g<span class="w"> </span>hs<span class="w"> </span><span class="se">\</span>
-p<span class="w"> </span>.1<span class="w"> </span><span class="se">\</span>
--call-summits<span class="w"> </span><span class="se">\</span>
--outdir<span class="w"> </span>/work2/project/regenet/workspace/thoellinger/ABC-Enhancer-Gene-Prediction/BENGI_GM12878/ABC_output/Peaks/
</code></pre></div>
<p>Now:</p>
<div class="highlight"><pre><span></span><code>conda<span class="w"> </span>activate<span class="w"> </span>py2<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>module<span class="w"> </span>load<span class="w"> </span>system/Python-2.7.2
</code></pre></div>
<div class="highlight"><pre><span></span><code>sbatch<span class="w"> </span>--mem<span class="o">=</span>2G<span class="w"> </span>--cpus-per-task<span class="o">=</span><span class="m">1</span><span class="w"> </span>-J<span class="w"> </span>step1.1<span class="w"> </span>--mail-user<span class="o">=</span>tristan.hoellinger@inserm.fr<span class="w"> </span>--mail-type<span class="o">=</span>END,FAIL<span class="w"> </span>--workdir<span class="o">=</span><span class="nv">$PWD</span><span class="w"> </span>--export<span class="o">=</span>ALL<span class="w"> </span>-p<span class="w"> </span>workq<span class="w"> </span>/work2/project/regenet/workspace/thoellinger/ABC-Enhancer-Gene-Prediction/BENGI_GM12878/hi_slurm/step1.1.sh
</code></pre></div>
<p>We stopped here.</p>
<div class="highlight"><pre><span></span><code>conda<span class="w"> </span>deactivate<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>module<span class="w"> </span>unload<span class="w"> </span>system/Python-2.7.2
</code></pre></div>
<h4 id="use-abc-makecandidateregionspy-to-define-candidate-regions">Use ABC <code>makeCandidateRegions.py</code> to define candidate regions<a class="headerlink" href="#use-abc-makecandidateregionspy-to-define-candidate-regions" title="Permanent link">&para;</a></h4>
<p>Once and for all:</p>
<div class="highlight"><pre><span></span><code>conda<span class="w"> </span>activate<span class="w"> </span>base<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>conda<span class="w"> </span>activate<span class="w"> </span>abcmodel<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>module<span class="w"> </span>load<span class="w"> </span>bioinfo/samtools-1.9<span class="w"> </span>bioinfo/bedtools-2.27.1<span class="w"> </span>bioinfo/tabix-0.2.5<span class="w"> </span>bioinfo/juicer-1.5.6
</code></pre></div>
<p>As there are peaks over 92 scaffolds listed in <code>ENCFF664VNB_ENCFF790WER.macs2_peaks.narrowPeak</code>, we modified it and <code>ENCFF587ALK_ENCFF766CPS.macs2_peaks.xls</code> to keep chromosomes only (otherwise <code>bedtools sort</code> below would not work). <strong>But we have to keep in mind that <code>ENCFF587ALK_ENCFF766CPS.macs2_model.r</code> will still corresponds to DNase peaks over the 92 scaffolds</strong>.</p>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span><span class="s1">&#39;$1 ~ /^(chr)([0-9]{1,2}$)|(M$)|(X$)|(Y$)/ {print $0}&#39;</span><span class="w"> </span>ABC_output/Peaks/ENCFF664VNB_ENCFF790WER.macs2_peaks.narrowPeak<span class="w"> </span>&gt;<span class="w"> </span>ABC_output/Peaks/ENCFF664VNB_ENCFF790WER.chromosomes_only.macs2_peaks.narrowPeak
</code></pre></div>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span><span class="s1">&#39;NR &lt;= 28 || $1 ~ /^(chr)([0-9]{1,2}$)|(M$)|(X$)|(Y$)/ {print $0}&#39;</span><span class="w"> </span>ABC_output/Peaks/ENCFF664VNB_ENCFF790WER.macs2_peaks.xls<span class="w"> </span>&gt;<span class="w"> </span>ABC_output/Peaks/ENCFF664VNB_ENCFF790WER.chromosomes_only.macs2_peaks.xls
</code></pre></div>
<p>We sort the results:</p>
<div class="highlight"><pre><span></span><code>srun<span class="w"> </span>--pty<span class="w"> </span>bash
</code></pre></div>
<div class="highlight"><pre><span></span><code>bedtools<span class="w"> </span>sort<span class="w"> </span>-faidx<span class="w"> </span>reference/chr_sizes<span class="w"> </span>-i<span class="w"> </span>ABC_output/Peaks/ENCFF664VNB_ENCFF790WER.chromosomes_only.macs2_peaks.narrowPeak<span class="w"> </span>&gt;<span class="w"> </span>ABC_output/Peaks/ENCFF664VNB_ENCFF790WER.macs2_peaks.narrowPeak.sorted
</code></pre></div>
<p>Now as <code>../src/makeCandidateRegions.py</code> does not support multiple <code>bam</code> inputs, we first need to concatenate the bam of the 2 replicates.  Indeed, <code>makeCandidateRegions.py</code> will use the bam input to count DNase reads on each peak of <code>ENCFF664VNB_ENCFF790WER.macs2_peaks.narrowPeak.sorted</code> found with <code>macs2</code>.</p>
<p>To that purpose, write the following in <code>hi_slurm/step1.2.sh</code>:</p>
<p>(WARNING: depending on the size of the input bam, which may be much variable, the execution on 1 thread only can take from a few minutes to several hours, so better use at least 8 threads)</p>
<div class="highlight"><pre><span></span><code><span class="ch">#!/bin/sh</span>
<span class="c1"># dependencies: bioinfo/samtools-1.9 bioinfo/bedtools-2.27.1</span>
<span class="k">if</span><span class="w"> </span><span class="o">[[</span><span class="w"> </span>!<span class="w"> </span>-f<span class="w"> </span>/work2/project/regenet/results/dnaseseq/homo_sapiens/hg19/gm12878/ENCFF664VNB_ENCFF790WER.bam<span class="w"> </span><span class="o">]]</span>
<span class="k">then</span>
<span class="w">    </span><span class="k">if</span><span class="w"> </span><span class="o">[[</span><span class="w"> </span>!<span class="w"> </span>-f<span class="w"> </span>/work2/project/regenet/results/dnaseseq/homo_sapiens/hg19/gm12878/ENCFF664VNB_ENCFF790WER.sam<span class="w"> </span><span class="o">]]</span>
<span class="w">    </span><span class="k">then</span>
<span class="w">        </span>samtools<span class="w"> </span>view<span class="w"> </span>-@<span class="w"> </span><span class="m">8</span><span class="w"> </span>-h<span class="w"> </span>/work2/project/regenet/results/dnaseseq/homo_sapiens/hg19/gm12878/ENCFF664VNB.bam<span class="w"> </span>&gt;<span class="w"> </span>/work2/project/regenet/results/dnaseseq/homo_sapiens/hg19/gm12878/ENCFF664VNB_ENCFF790WER.sam
<span class="w">        </span>samtools<span class="w"> </span>view<span class="w"> </span>-@<span class="w"> </span><span class="m">8</span><span class="w"> </span>/work2/project/regenet/results/dnaseseq/homo_sapiens/hg19/gm12878/ENCFF790WER.bam<span class="w"> </span>&gt;&gt;<span class="w"> </span>/work2/project/regenet/results/dnaseseq/homo_sapiens/hg19/gm12878/ENCFF664VNB_ENCFF790WER.sam
<span class="w">        </span>samtools<span class="w"> </span>view<span class="w"> </span>-@<span class="w"> </span><span class="m">8</span><span class="w"> </span>-S<span class="w"> </span>-b<span class="w"> </span>/work2/project/regenet/results/dnaseseq/homo_sapiens/hg19/gm12878/ENCFF664VNB_ENCFF790WER.sam<span class="w"> </span>&gt;<span class="w"> </span>/work2/project/regenet/results/dnaseseq/homo_sapiens/hg19/gm12878/ENCFF664VNB_ENCFF790WER.bam
<span class="w">        </span>rm<span class="w"> </span>/work2/project/regenet/results/dnaseseq/homo_sapiens/hg19/gm12878/ENCFF664VNB_ENCFF790WER.sam
<span class="w">    </span><span class="k">else</span>
<span class="w">        </span>samtools<span class="w"> </span>view<span class="w"> </span>-@<span class="w"> </span><span class="m">8</span><span class="w"> </span>-S<span class="w"> </span>-b<span class="w"> </span>/work2/project/regenet/results/dnaseseq/homo_sapiens/hg19/gm12878/ENCFF664VNB_ENCFF790WER.sam<span class="w"> </span>&gt;<span class="w"> </span>/work2/project/regenet/results/dnaseseq/homo_sapiens/hg19/gm12878/ENCFF664VNB_ENCFF790WER.bam
<span class="w">    </span><span class="k">fi</span>
<span class="k">fi</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>sbatch<span class="w"> </span>--mem<span class="o">=</span>2G<span class="w"> </span>--cpus-per-task<span class="o">=</span><span class="m">8</span><span class="w"> </span>-J<span class="w"> </span>step1.2<span class="w"> </span>--mail-user<span class="o">=</span>tristan.hoellinger@inserm.fr<span class="w"> </span>--mail-type<span class="o">=</span>END,FAIL<span class="w"> </span>--workdir<span class="o">=</span><span class="nv">$PWD</span><span class="w"> </span>--export<span class="o">=</span>ALL<span class="w"> </span>-p<span class="w"> </span>workq<span class="w"> </span>/work2/project/regenet/workspace/thoellinger/ABC-Enhancer-Gene-Prediction/BENGI_GM12878/hi_slurm/step1.2.sh
</code></pre></div>
<blockquote>
<div class="highlight"><pre><span></span><code>$<span class="w"> </span>seff<span class="w"> </span><span class="m">21980507</span>
CPU<span class="w"> </span>Efficiency:<span class="w"> </span><span class="m">60</span>.91%<span class="w"> </span>of<span class="w"> </span><span class="m">03</span>:50:08<span class="w"> </span>core-walltime
Job<span class="w"> </span>Wall-clock<span class="w"> </span>time:<span class="w"> </span><span class="m">00</span>:28:46
Memory<span class="w"> </span>Utilized:<span class="w"> </span><span class="m">8</span>.98<span class="w"> </span>MB
</code></pre></div>
</blockquote>
<p>Now write the following in <code>GM12878/hi_slurm/step1.3.sh</code></p>
<div class="highlight"><pre><span></span><code><span class="ch">#!/bin/sh</span>
<span class="c1"># dependencies: python3 bioinfo/samtools-1.9 bioinfo/bedtools-2.27.1 bioinfo/tabix-0.2.5 bioinfo/juicer-1.5.6</span>
python<span class="w"> </span>../src/makeCandidateRegions.py<span class="w"> </span><span class="se">\</span>
--narrowPeak<span class="w"> </span>/work2/project/regenet/workspace/thoellinger/ABC-Enhancer-Gene-Prediction/BENGI_GM12878/ABC_output/Peaks/ENCFF664VNB_ENCFF790WER.macs2_peaks.narrowPeak.sorted<span class="w"> </span><span class="se">\</span>
--bam<span class="w"> </span>/work2/project/regenet/results/dnaseseq/homo_sapiens/hg19/gm12878/ENCFF664VNB_ENCFF790WER.bam<span class="w"> </span><span class="se">\</span>
--outDir<span class="w"> </span>/work2/project/regenet/workspace/thoellinger/ABC-Enhancer-Gene-Prediction/BENGI_GM12878/ABC_output/Peaks/<span class="w"> </span><span class="se">\</span>
--chrom_sizes<span class="w"> </span>/work2/project/regenet/workspace/thoellinger/ABC-Enhancer-Gene-Prediction/BENGI_GM12878/reference/chr_sizes<span class="w"> </span><span class="se">\</span>
--regions_blacklist<span class="w"> </span>/work2/project/regenet/data/species/homo_sapiens/hg19/hg19-blacklist.v2.bed.gz<span class="w"> </span><span class="se">\</span>
--peakExtendFromSummit<span class="w"> </span><span class="m">250</span><span class="w"> </span><span class="se">\</span>
--nStrongestPeaks<span class="w"> </span><span class="m">150000</span>
<span class="c1">#Expected output: params.txt, foo.macs2_peaks.narrowPeak.sorted.candidateRegions.bed, foo.macs2_peaks.narrowPeak.sorted.foo.bam.Counts.bed</span>
</code></pre></div>
<p>WARNING: depending on the length of the inputs, this script may be very RAM-demanding. So we recommend to allocate at least 16 GB RAM to avoid "out of memory" errors.</p>
<div class="highlight"><pre><span></span><code>sbatch<span class="w"> </span>--mem<span class="o">=</span>16G<span class="w"> </span>--cpus-per-task<span class="o">=</span><span class="m">1</span><span class="w"> </span>-J<span class="w"> </span>step1.3<span class="w"> </span>--mail-user<span class="o">=</span>tristan.hoellinger@inserm.fr<span class="w"> </span>--mail-type<span class="o">=</span>END,FAIL<span class="w"> </span>--workdir<span class="o">=</span><span class="nv">$PWD</span><span class="w"> </span>--export<span class="o">=</span>ALL<span class="w"> </span>-p<span class="w"> </span>workq<span class="w"> </span>/work2/project/regenet/workspace/thoellinger/ABC-Enhancer-Gene-Prediction/BENGI_GM12878/hi_slurm/step1.3.sh
</code></pre></div>
<h3 id="step-2-quantifying-enhancer-activity">Step 2: quantifying enhancer activity<a class="headerlink" href="#step-2-quantifying-enhancer-activity" title="Permanent link">&para;</a></h3>
<h4 id="use-abc-runneighborhoodspy">Use ABC <code>run.neighborhoods.py</code><a class="headerlink" href="#use-abc-runneighborhoodspy" title="Permanent link">&para;</a></h4>
<p>Write the following in <code>BENGI_GM12878/hi_slurm/step2.sh</code>.</p>
<div class="highlight"><pre><span></span><code><span class="ch">#!/bin/sh</span>
python<span class="w"> </span>../src/run.neighborhoods.py<span class="w"> </span><span class="se">\</span>
--candidate_enhancer_regions<span class="w"> </span>/work2/project/regenet/workspace/thoellinger/ABC-Enhancer-Gene-Prediction/BENGI_GM12878/ABC_output/Peaks/ENCFF664VNB_ENCFF790WER.macs2_peaks.narrowPeak.sorted.candidateRegions.bed<span class="w"> </span><span class="se">\</span>
--genes<span class="w"> </span>/work2/project/regenet/workspace/thoellinger/ABC-Enhancer-Gene-Prediction/BENGI_GM12878/reference/genes_annotation/gene_ids.bed<span class="w"> </span><span class="se">\</span>
--H3K27ac<span class="w"> </span>/work2/project/regenet/results/chipseq/h3k27ac/homo_sapiens/hg19/gm12878/GSM733771_hg19_wgEncodeBroadHistoneGm12878H3k27acStdAlnRep1.bam,/work2/project/regenet/results/chipseq/h3k27ac/homo_sapiens/hg19/gm12878/GSM733771_hg19_wgEncodeBroadHistoneGm12878H3k27acStdAlnRep2.bam<span class="w"> </span><span class="se">\</span>
--DHS<span class="w"> </span>/work2/project/regenet/results/dnaseseq/homo_sapiens/hg19/gm12878/ENCFF664VNB.bam,/work2/project/regenet/results/dnaseseq/homo_sapiens/hg19/gm12878/ENCFF790WER.bam<span class="w"> </span><span class="se">\</span>
--expression_table<span class="w"> </span>/work2/project/regenet/workspace/thoellinger/ABC-Enhancer-Gene-Prediction/BENGI_GM12878/reference/expression/GM12878.ids.ENCFF587ALK_ENCFF766CPS.mean.TPM.txt<span class="w">  </span><span class="se">\</span>
--chrom_sizes<span class="w"> </span>/work2/project/regenet/workspace/thoellinger/ABC-Enhancer-Gene-Prediction/BENGI_GM12878/reference/chr_sizes<span class="w"> </span><span class="se">\</span>
--ubiquitously_expressed_genes<span class="w"> </span>/work2/project/regenet/workspace/thoellinger/ABC-Enhancer-Gene-Prediction/BENGI_GM12878/reference/UbiquitouslyExpressedGenes_ids.txt<span class="w"> </span><span class="se">\</span>
--cellType<span class="w"> </span>GM12878<span class="w"> </span><span class="se">\</span>
--outdir<span class="w"> </span>/work2/project/regenet/workspace/thoellinger/ABC-Enhancer-Gene-Prediction/BENGI_GM12878/ABC_output/Neighborhoods/
</code></pre></div>
<p>WARNING: depending on the length of the inputs, this script may be very RAM-demanding. So we recommend to allocate 32 GB RAM in general to avoid "out of memory" errors.</p>
<div class="highlight"><pre><span></span><code>sbatch<span class="w"> </span>--mem<span class="o">=</span>16G<span class="w"> </span>--cpus-per-task<span class="o">=</span><span class="m">1</span><span class="w"> </span>-J<span class="w"> </span>step2<span class="w"> </span>--mail-user<span class="o">=</span>tristan.hoellinger@inserm.fr<span class="w"> </span>--mail-type<span class="o">=</span>END,FAIL<span class="w"> </span>--workdir<span class="o">=</span><span class="nv">$PWD</span><span class="w"> </span>--export<span class="o">=</span>ALL<span class="w"> </span>-p<span class="w"> </span>workq<span class="w"> </span>hi_slurm/step2.sh
</code></pre></div>
<p>We should find a way to parallelize <code>run.neighborhoods.py</code> as it took almost 3 hours, with 99,9% cpu efficiency. </p>
<h3 id="step-3-computing-the-abc-score">Step 3: computing the ABC score<a class="headerlink" href="#step-3-computing-the-abc-score" title="Permanent link">&para;</a></h3>
<blockquote>
<p>If experimentally derived contact data is not available, one can run the ABC model using the powerlaw estimate only. In this case the <code>--HiCdir</code> argument should be excluded from <code>predict.py</code> and the <code>--score_column powerlaw.Score</code> argument should be included in <code>predict.py</code>. In this case the <code>ABC.Score</code> column of the predictions file will be set to <code>NaN</code>. The <code>powerlaw.Score</code> column of the output prediction files will be the relevant Score column to use.
</p>
</blockquote>
<p>Write the following in <code>BENGI_GM12878/hi_slurm/step3.sh</code>.</p>
<div class="highlight"><pre><span></span><code><span class="ch">#!/bin/sh</span>
python<span class="w"> </span>../src/predict.py<span class="w"> </span><span class="se">\</span>
--enhancers<span class="w"> </span>/work2/project/regenet/workspace/thoellinger/ABC-Enhancer-Gene-Prediction/BENGI_GM12878/ABC_output/Neighborhoods/EnhancerList.txt<span class="w"> </span><span class="se">\</span>
--genes<span class="w"> </span>/work2/project/regenet/workspace/thoellinger/ABC-Enhancer-Gene-Prediction/BENGI_GM12878/ABC_output/Neighborhoods/GeneList.txt<span class="w"> </span><span class="se">\</span>
--score_column<span class="w"> </span>powerlaw.Score<span class="w"> </span><span class="se">\</span>
--hic_resolution<span class="w"> </span><span class="m">5000</span><span class="w"> </span><span class="se">\</span>
--scale_hic_using_powerlaw<span class="w"> </span><span class="se">\</span>
--threshold<span class="w"> </span>.02<span class="w"> </span><span class="se">\</span>
--cellType<span class="w"> </span>GM12878<span class="w"> </span><span class="se">\</span>
--outdir<span class="w"> </span>/work2/project/regenet/workspace/thoellinger/ABC-Enhancer-Gene-Prediction/BENGI_GM12878/ABC_output/Predictions/<span class="w"> </span><span class="se">\</span>
--make_all_putative
</code></pre></div>
<div class="highlight"><pre><span></span><code>sbatch<span class="w"> </span>--mem<span class="o">=</span>32G<span class="w"> </span>--cpus-per-task<span class="o">=</span><span class="m">1</span><span class="w"> </span>-J<span class="w"> </span>step3<span class="w"> </span>--mail-user<span class="o">=</span>tristan.hoellinger@inserm.fr<span class="w"> </span>--mail-type<span class="o">=</span>END,FAIL<span class="w"> </span>--workdir<span class="o">=</span><span class="nv">$PWD</span><span class="w"> </span>--export<span class="o">=</span>ALL<span class="w"> </span>-p<span class="w"> </span>workq<span class="w"> </span>hi_slurm/step3.sh
</code></pre></div>
<h2 id="results">Results<a class="headerlink" href="#results" title="Permanent link">&para;</a></h2>
<h3 id="summary-statistics">Summary statistics<a class="headerlink" href="#summary-statistics" title="Permanent link">&para;</a></h3>
<p>We create a <code>sumstats</code> sub-directory in <code>ABC_output/Predictions</code> then compute summary statistics using <code>bedpe.sumstats.sh</code> script.</p>
<div class="highlight"><pre><span></span><code>module<span class="w"> </span>load<span class="w"> </span>compiler/gcc-7.2.0<span class="w"> </span>system/R-4.0.2_gcc-7.2.0
</code></pre></div>
<p>R packages <code>optparse</code> and <code>reshape2</code> are required. Moreover, when executing <code>bedpe.sumstats.sh</code>, one must have to <code>chmod u+x</code> some dependencies if there are <code>Permission denied</code> errors.</p>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span><span class="w"> </span>ABC_output/Predictions/
gzip<span class="w"> </span>-c<span class="w"> </span>EnhancerPredictions.bedpe<span class="w"> </span>&gt;<span class="w"> </span>EnhancerPredictions.bedpe.gz
<span class="nb">cd</span><span class="w"> </span>sumstats/
sh<span class="w"> </span>../../../../../scripts/bedpe.sumstats.sh<span class="w"> </span>../EnhancerPredictions.bedpe.gz<span class="w"> </span><span class="s2">&quot;0.02-1&quot;</span><span class="w"> </span><span class="s2">&quot;500-500&quot;</span>
</code></pre></div>
<p>Or replace <code>0.02-1</code> with <code>&lt;choose a min&gt;-&lt;choose a max&gt;</code> where <code>min</code> and <code>max</code> can be found with</p>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span><span class="s1">&#39;BEGIN{FS=&quot;\t&quot;; max=0;} {if($8&gt;max){max=$8}} END{print(max);}&#39;</span><span class="w"> </span>../EnhancerPredictions.bedpe

awk<span class="w"> </span><span class="s1">&#39;BEGIN{FS=&quot;\t&quot;; min=1;} {if($8&lt;min){min=$8}} END{print(min);}&#39;</span><span class="w"> </span>../EnhancerPredictions.bedpe
</code></pre></div>
<p><img alt="" src="results1_BENGI_GM12878/Distance.png" /></p>
<p><img alt="" src="results1_BENGI_GM12878/Score.png" /></p>
<p><img alt="" src="results1_BENGI_GM12878/refelt.scorequantile.nbconn.nbtimes.png" /></p>
<h2 id="evaluation-of-abc-predictions-for-gm12878-on-bengi-datasets">Evaluation of ABC predictions for GM12878 on BENGI datasets<a class="headerlink" href="#evaluation-of-abc-predictions-for-gm12878-on-bengi-datasets" title="Permanent link">&para;</a></h2>
<h3 id="intersection-between-abc-prediction-bengi">Intersection between ABC prediction / BENGI<a class="headerlink" href="#intersection-between-abc-prediction-bengi" title="Permanent link">&para;</a></h3>
<h4 id="standard-abc">Standard ABC<a class="headerlink" href="#standard-abc" title="Permanent link">&para;</a></h4>
<p><img alt="Table: intersection between ABC prediction / BENGI" src="results1_BENGI_GM12878/intersection_ABC_standard.png" /></p>
<p>Note that we obtain these values with the notebook, for which we use other histone mark H3K27ac ChIP-seq (same accessions but downloaded from ENCODE instead of GEO -&gt; processed another way). They are very slightly different from what we would have found here.</p>
<h4 id="standard-abc-with-more-peaks">Standard ABC with more peaks<a class="headerlink" href="#standard-abc-with-more-peaks" title="Permanent link">&para;</a></h4>
<p>Here is what we find being more lenient when calling peaks (p=0.2) and taking up to 500,000 peaks (again, using ENCODE accessions for histone marks)</p>
<p><img alt="Table: intersection between ABC prediction / BENGI" src="results1_BENGI_GM12878/ABC_p_e-2_max_500K_peaks.png" /></p>
<h4 id="abc-starting-from-ccre-delss-as-candidate-enhancers">ABC starting from ccRE-dELSs as candidate enhancers<a class="headerlink" href="#abc-starting-from-ccre-delss-as-candidate-enhancers" title="Permanent link">&para;</a></h4>
<p>Starting from distal (&gt;= 2kb) ENCODE ccREs with Enhancer-Like Signature as candidate regions (instead of performing ABC step 1), we obtain, as expected, 100% intersection with BENGI datasets for GM12878. This enables us to compute true Precision - Recall curves and AUPR.</p>
<p><img alt="Table: intersection between ABC prediction / BENGI" src="results1_BENGI_GM12878/intersection_starting_from_ccRE_dELS.png" /></p>
<h3 id="precision-recall-curve">Precision / recall curve<a class="headerlink" href="#precision-recall-curve" title="Permanent link">&para;</a></h3>
<h4 id="abc-starting-from-ccre-delss-as-candidate-enhancers_1">ABC starting from ccRE-dELSs as candidate enhancers<a class="headerlink" href="#abc-starting-from-ccre-delss-as-candidate-enhancers_1" title="Permanent link">&para;</a></h4>
<p>Note that even though these curves are computed over the full BENGI datasets, they still grossly overestimate the Precision, as false positives that fall outside of the BENGI datasets are not taken into account.</p>
<p><img alt="Image: Precision-Recall curves" src="results1_BENGI_GM12878/aupr_abc_starting_from_ENCODE_ccRE_dELSs.png" /></p>
<p>Even with such an overestimate of the Precision of ABC predictions, they are only slightly better than random, and largely outperformed by a baseline distance method.</p>
<h4 id="distance-method">Distance method<a class="headerlink" href="#distance-method" title="Permanent link">&para;</a></h4>
<p><img alt="Image: Precision-Recall curves" src="results1_BENGI_GM12878/aupr_distance.png" /></p>
<p>For information purpose, we also computed pseudo Precision - Recall curves for ABC predictions that made only partial intersect with BENGI - that is, the ones from standard ABC method. Note that the interpretation is limited.</p>
<h4 id="standard-abc_1">Standard ABC<a class="headerlink" href="#standard-abc_1" title="Permanent link">&para;</a></h4>
<p><img alt="Image: Precision-Recall curves" src="results1_BENGI_GM12878/AUPR_abc_standard.png" /></p>
<h4 id="standard-abc-with-more-peaks_1">Standard ABC with more peaks<a class="headerlink" href="#standard-abc-with-more-peaks_1" title="Permanent link">&para;</a></h4>
<p><img alt="Image: Precision-Recall curves" src="results1_BENGI_GM12878/aupr_abc_up_to_500K_peaks.png" /></p>





                
              </article>
            </div>
          
          
        </div>
        
          <a href="#" class="md-top md-icon" data-md-component="top" hidden>
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg>
            Back to top
          </a>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      2020 - 2022 Tristan Hoellinger
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
        <div class="md-social">
  
    
    
    
    
      
      
    
    <a href="https://twitter.com/TristanHoellin1" target="_blank" rel="noopener" title="twitter.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"/></svg>
    </a>
  
    
    
    
    
      
      
    
    <a href="https://fr.linkedin.com/in/tristan-hoellinger" target="_blank" rel="noopener" title="fr.linkedin.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"/></svg>
    </a>
  
    
    
    
    
    <a href="mailto:sarah.djebali@inserm.fr" target="_blank" rel="noopener" title="" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M498.1 5.6c10.1 7 15.4 19.1 13.5 31.2l-64 416c-1.5 9.7-7.4 18.2-16 23s-18.9 5.4-28 1.6L284 427.7l-68.5 74.1c-8.9 9.7-22.9 12.9-35.2 8.1S160 493.2 160 480v-83.6c0-4 1.5-7.8 4.2-10.7l167.6-182.9c5.8-6.3 5.6-16-.4-22s-15.7-6.4-22-.7L106 360.8l-88.3-44.2C7.1 311.3.3 300.7 0 288.9s5.9-22.8 16.1-28.7l448-256c10.7-6.1 23.9-5.5 34 1.4z"/></svg>
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    <script id="__config" type="application/json">{"base": "../..", "features": ["navigation.instant", "navigation.tabs", "navigation.sections", "navigation.top"], "search": "../../assets/javascripts/workers/search.db81ec45.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../../assets/javascripts/bundle.6df46069.min.js"></script>
      
    
  </body>
</html>