
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
        <meta name="author" content="Tristan Hoellinger">
      
      
        <link rel="canonical" href="https://eg.github.io/data_and_results/haemochromatosis/ABC/custom_go.html">
      
      
      
      <link rel="icon" href="../../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.4.2, mkdocs-material-9.0.11">
    
    
      
        <title>New custom script to compute GO enrichment: `GO_enrichment_FDR.R` - Genome-wide identification of E/G interactions</title>
      
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/main.0d440cfe.min.css">
      
        
        <link rel="stylesheet" href="../../../assets/stylesheets/palette.2505c338.min.css">
      
      

    
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../../stylesheets/extra.css">
    
    <script>__md_scope=new URL("../../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="black" data-md-color-accent="">
  
    
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#new-custom-script-to-compute-go-enrichment-go_enrichment_fdrr" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../../.." title="Genome-wide identification of E/G interactions" class="md-header__button md-logo" aria-label="Genome-wide identification of E/G interactions" data-md-component="logo">
      
  <img src="../../../logo/logo-inserm-blc.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Genome-wide identification of E/G interactions
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              New custom script to compute GO enrichment: `GO_enrichment_FDR.R`
            
          </span>
        </div>
      </div>
    </div>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/hoellin/eg" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
  </div>
  <div class="md-source__repository">
    hoellin/eg
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../../guidebooks/introduction.html" class="md-tabs__link">
        Guidebooks
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../../haemochromatosis/introduction.html" class="md-tabs__link">
        Haemochromatosis
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../../notes_BENGI/introduction.html" class="md-tabs__link">
        BENGI
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../../notes_ABC/introduction.html" class="md-tabs__link">
        ABC
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../../index.html" class="md-tabs__link">
        About
      </a>
    </li>
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../../.." title="Genome-wide identification of E/G interactions" class="md-nav__button md-logo" aria-label="Genome-wide identification of E/G interactions" data-md-component="logo">
      
  <img src="../../../logo/logo-inserm-blc.png" alt="logo">

    </a>
    Genome-wide identification of E/G interactions
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/hoellin/eg" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
  </div>
  <div class="md-source__repository">
    hoellin/eg
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1" >
      
      
      
        <label class="md-nav__link" for="__nav_1" id="__nav_1_label" tabindex="0">
          Guidebooks
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_1_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_1">
          <span class="md-nav__icon md-icon"></span>
          Guidebooks
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../guidebooks/introduction.html" class="md-nav__link">
        Introduction
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../guidebooks/script_bedpe_summary/script_bedpe_summary.html" class="md-nav__link">
        Compute summary stats of a bedpe
      </a>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_3" >
      
      
      
        <label class="md-nav__link" for="__nav_1_3" id="__nav_1_3_label" tabindex="0">
          Basics
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_1_3">
          <span class="md-nav__icon md-icon"></span>
          Basics
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../guidebooks/shell.html" class="md-nav__link">
        Shell
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../guidebooks/conda_environments.html" class="md-nav__link">
        Conda basics
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../guidebooks/git.html" class="md-nav__link">
        Git
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../guidebooks/genotoul.html" class="md-nav__link">
        Genotoul cluster
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../guidebooks/notebooks_ssh.html" class="md-nav__link">
        How to use Jupyter through ssh tunneling?
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_4" >
      
      
      
        <label class="md-nav__link" for="__nav_1_4" id="__nav_1_4_label" tabindex="0">
          Misc
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_4_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_1_4">
          <span class="md-nav__icon md-icon"></span>
          Misc
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../guidebooks/compute_correlations.html" class="md-nav__link">
        Compute correlations
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" >
      
      
      
        <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
          Haemochromatosis
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_2">
          <span class="md-nav__icon md-icon"></span>
          Haemochromatosis
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../haemochromatosis/introduction.html" class="md-nav__link">
        Introduction
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../haemochromatosis/data_format.html" class="md-nav__link">
        Data format
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../haemochromatosis/ABC/custom_go.html" class="md-nav__link">
        Custom script GO
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../haemochromatosis/main_results.html" class="md-nav__link">
        Main results
      </a>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2_5" >
      
      
      
        <label class="md-nav__link" for="__nav_2_5" id="__nav_2_5_label" tabindex="0">
          CHiC data
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_2_5_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_2_5">
          <span class="md-nav__icon md-icon"></span>
          CHiC data
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../haemochromatosis/CHiC/preprocess_data.html" class="md-nav__link">
        Preprocessing
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../haemochromatosis/CHiC/preliminary_analysis_chic_v2.html" class="md-nav__link">
        Networks
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../haemochromatosis/CHiC/go_enrichment.html" class="md-nav__link">
        GO enrichment
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2_6" >
      
      
      
        <label class="md-nav__link" for="__nav_2_6" id="__nav_2_6_label" tabindex="0">
          ABC data
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_2_6_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_2_6">
          <span class="md-nav__icon md-icon"></span>
          ABC data
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../haemochromatosis/ABC/data_processing.html" class="md-nav__link">
        Preprocessing
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../haemochromatosis/ABC/preliminary_analysis_v9.html" class="md-nav__link">
        Networks
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../haemochromatosis/ABC/summary_statistics_eg_pairs.html" class="md-nav__link">
        Summary statistics enhancers
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../haemochromatosis/ABC/go_enrichment.html" class="md-nav__link">
        GO enrichment
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../haemochromatosis/ABC/gorilla.html" class="md-nav__link">
        GOrilla
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../haemochromatosis/ABC/compute_avg_nb_of_connections_per_gene.html" class="md-nav__link">
        Avg nb connections per gene
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../haemochromatosis/ABC/gwas.html" class="md-nav__link">
        GWAS (13 initial genes)
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../haemochromatosis/ABC/gwas_10.html" class="md-nav__link">
        GWAS (10 initial genes)
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" >
      
      
      
        <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
          BENGI
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3">
          <span class="md-nav__icon md-icon"></span>
          BENGI
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_BENGI/introduction.html" class="md-nav__link">
        Introduction
      </a>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_2" >
      
      
      
        <label class="md-nav__link" for="__nav_3_2" id="__nav_3_2_label" tabindex="0">
          Notebooks
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_2_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_2">
          <span class="md-nav__icon md-icon"></span>
          Notebooks
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_BENGI/BENGI_to_custom_bedpe.html" class="md-nav__link">
        Convert BENGI datasets to custom bedpe
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_3" >
      
      
      
        <label class="md-nav__link" for="__nav_3_3" id="__nav_3_3_label" tabindex="0">
          Summary statistics
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_3">
          <span class="md-nav__icon md-icon"></span>
          Summary statistics
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_BENGI/summary_statistics/summary_statistics.html" class="md-nav__link">
        Summary statistics on BENGI GM12878 benchmarks
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_BENGI/summary_statistics_BENGI/summary_BENGI_bedpe.html" class="md-nav__link">
        (R code) Compute summary statistics over BENGI
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_4" >
      
      
      
        <label class="md-nav__link" for="__nav_3_4" id="__nav_3_4_label" tabindex="0">
          Evaluation of a few methods over BENGI GM12878
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_4_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_4">
          <span class="md-nav__icon md-icon"></span>
          Evaluation of a few methods over BENGI GM12878
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_4_1" >
      
      
      
        <label class="md-nav__link" for="__nav_3_4_1" id="__nav_3_4_1_label" tabindex="0">
          Distance method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_4_1_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_4_1">
          <span class="md-nav__icon md-icon"></span>
          Distance method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_BENGI/distance_method/distance_evaluation_with_code.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_BENGI/distance_method/summary_distance_evaluation.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_4_2" >
      
      
      
        <label class="md-nav__link" for="__nav_3_4_2" id="__nav_3_4_2_label" tabindex="0">
          DNase-expression correlation method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_4_2_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_4_2">
          <span class="md-nav__icon md-icon"></span>
          DNase-expression correlation method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_BENGI/dnase_expression_correlation/correlation_method_with_code.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_BENGI/dnase_expression_correlation/summary_correlation_method.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_4_3" >
      
      
      
        <label class="md-nav__link" for="__nav_3_4_3" id="__nav_3_4_3_label" tabindex="0">
          Average rank method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_4_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_4_3">
          <span class="md-nav__icon md-icon"></span>
          Average rank method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_BENGI/avg_rank_method/avg_rank_method_with_code.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_BENGI/avg_rank_method/summary_avg_rank_method.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_5" >
      
      
      
        <label class="md-nav__link" for="__nav_3_5" id="__nav_3_5_label" tabindex="0">
          Evaluation of a few methods over Fulco et al 2019 CRISPRi-FlowFISH
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_5_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_5">
          <span class="md-nav__icon md-icon"></span>
          Evaluation of a few methods over Fulco et al 2019 CRISPRi-FlowFISH
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_5_1" >
      
      
      
        <label class="md-nav__link" for="__nav_3_5_1" id="__nav_3_5_1_label" tabindex="0">
          Distance method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_5_1_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_5_1">
          <span class="md-nav__icon md-icon"></span>
          Distance method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_BENGI/CRISPRi_FlowFISH/distance_method/distance_over_fulco_et_al_crispri.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_BENGI/CRISPRi_FlowFISH/distance_method/summary_distance_over_fulco_et_al_crispri.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_5_2" >
      
      
      
        <label class="md-nav__link" for="__nav_3_5_2" id="__nav_3_5_2_label" tabindex="0">
          DNase-expression correlation method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_5_2_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_5_2">
          <span class="md-nav__icon md-icon"></span>
          DNase-expression correlation method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_BENGI/CRISPRi_FlowFISH/dnase_expression_correlation/correlation_method_with_code.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_BENGI/CRISPRi_FlowFISH/dnase_expression_correlation/summary_correlation_method.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_5_3" >
      
      
      
        <label class="md-nav__link" for="__nav_3_5_3" id="__nav_3_5_3_label" tabindex="0">
          Average rank method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_5_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_5_3">
          <span class="md-nav__icon md-icon"></span>
          Average rank method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_BENGI/CRISPRi_FlowFISH/avg_rank_method/avg_rank_method_with_code.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_BENGI/CRISPRi_FlowFISH/avg_rank_method/summary_avg_rank.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" >
      
      
      
        <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
          ABC
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_4">
          <span class="md-nav__icon md-icon"></span>
          ABC
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_ABC/introduction.html" class="md-nav__link">
        Introduction
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_ABC/BENGI/ABC_BENGI_results.html" class="md-nav__link">
        ABC over BENGI for GM12878, main results
      </a>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_3" >
      
      
      
        <label class="md-nav__link" for="__nav_4_3" id="__nav_4_3_label" tabindex="0">
          CRISPRi-FlowFISH
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_4_3">
          <span class="md-nav__icon md-icon"></span>
          CRISPRi-FlowFISH
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_ABC/K562/ABC_K562_CRISPRi_FlowFISH.html" class="md-nav__link">
        Original ABC predictions over K562
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_ABC/K562/recap_investigations_ABC_over_CRiFF.html" class="md-nav__link">
        Summary ABC over CRiFF
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_4" >
      
      
      
        <label class="md-nav__link" for="__nav_4_4" id="__nav_4_4_label" tabindex="0">
          Notebooks
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_4_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_4_4">
          <span class="md-nav__icon md-icon"></span>
          Notebooks
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_ABC/K562/april_K562_56_genes/april_K562_56_genes.html" class="md-nav__link">
        ABC over K562 cCRiFF
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_ABC/K562/april_K562_59_genes/april_K562_59_genes.html" class="md-nav__link">
        ABC over K562 CRiFF
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_ABC/K562/april_K562_candidates_not_in_whitelist_56_genes/april_K562_candidates_NOT_in_whitelist.html" class="md-nav__link">
        ABC over K562 cCRiFF, no whitelist
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_ABC/generic_notebooks/turnkey_notebook_to_run_ABC_with_example_over_GM12878.html" class="md-nav__link">
        Generic notebook to run ABC method & Example over GM12878
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_ABC/BENGI/notebook_ABC_over_BENGI_GM12878.html" class="md-nav__link">
        ABC over GM12878 with same accessions as for BENGI
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_ABC/BENGI/notebook_ABC_over_BENGI_GM12878_from_ccRE_ELSs.html" class="md-nav__link">
        ABC over GM12878 with same accessions as for BENGI, starting with ccRE-dELS
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_5" >
      
      
      
        <label class="md-nav__link" for="__nav_4_5" id="__nav_4_5_label" tabindex="0">
          Full codes
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_5_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_4_5">
          <span class="md-nav__icon md-icon"></span>
          Full codes
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_ABC/test_abc_pipeline.html" class="md-nav__link">
        Test of ABC pipeline over chr22
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_ABC/BENGI/ABC_over_BENGI_GM12878.html" class="md-nav__link">
        ABC over GM12878 with same accessions as for BENGI
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_ABC/differences_example_chr22.html" class="md-nav__link">
        Meld of results over chr22 vs authors' results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" >
      
      
      
        <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
          About
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_5">
          <span class="md-nav__icon md-icon"></span>
          About
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../index.html" class="md-nav__link">
        Getting started
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../about_me.html" class="md-nav__link">
        About me
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#dependencies" class="md-nav__link">
    Dependencies
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#script" class="md-nav__link">
    Script
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#example-of-use" class="md-nav__link">
    Example of use
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  

  
  



<h1 id="new-custom-script-to-compute-go-enrichment-go_enrichment_fdrr">New custom script to compute GO enrichment: <code>GO_enrichment_FDR.R</code><a class="headerlink" href="#new-custom-script-to-compute-go-enrichment-go_enrichment_fdrr" title="Permanent link">&para;</a></h1>
<p>We adapt <code>GO_enrichment.R</code> so that the multiple testing correction is less stringent: we want to control the FDR, rather than the FWER with the Šidák procedure.</p>
<p>We name our script <code>GO_enrichment_FDR.R</code>.</p>
<p>We should look at the <code>enrichGO</code> function from the <code>clusterProfiler</code> library: https://rdrr.io/bioc/clusterProfiler/man/enrichGO.html</p>
<p>https://www.rdocumentation.org/packages/clusterProfiler/versions/3.0.4/topics/enrichGO</p>
<p>https://yulab-smu.top/biomedical-knowledge-mining-book/clusterprofiler-go.html</p>
<h2 id="dependencies">Dependencies<a class="headerlink" href="#dependencies" title="Permanent link">&para;</a></h2>
<p>We use R 4.0. We need the <code>clusterProfiler</code> package.</p>
<blockquote>
<p>The following does not work, so all this quotation is just to keep a record in case we encounter similar problems:</p>
<div class="highlight"><pre><span></span><code><span class="n">BiocManager</span><span class="o">::</span><span class="nf">install</span><span class="p">(</span><span class="s">&quot;clusterProfiler&quot;</span><span class="p">)</span>
</code></pre></div>
<blockquote>
<div class="highlight"><pre><span></span><code><span class="n">Installation</span><span class="w"> </span><span class="n">paths</span><span class="w"> </span><span class="n">not</span><span class="w"> </span><span class="n">writeable</span><span class="p">,</span><span class="w"> </span><span class="n">unable</span><span class="w"> </span><span class="n">to</span><span class="w"> </span><span class="n">update</span><span class="w"> </span><span class="n">packages</span>
<span class="w">  </span><span class="n">path</span><span class="o">:</span><span class="w"> </span><span class="o">/</span><span class="n">usr</span><span class="o">/</span><span class="n">lib</span><span class="o">/</span><span class="n">R</span><span class="o">/</span><span class="n">library</span>
<span class="w">  </span><span class="n">packages</span><span class="o">:</span>
<span class="w">    </span><span class="n">boot</span><span class="p">,</span><span class="w"> </span><span class="n">class</span><span class="p">,</span><span class="w"> </span><span class="n">cluster</span><span class="p">,</span><span class="w"> </span><span class="n">foreign</span><span class="p">,</span><span class="w"> </span><span class="n">KernSmooth</span><span class="p">,</span><span class="w"> </span><span class="n">lattice</span><span class="p">,</span><span class="w"> </span><span class="n">MASS</span><span class="p">,</span><span class="w"> </span><span class="n">Matrix</span><span class="p">,</span><span class="w"> </span><span class="n">mgcv</span><span class="p">,</span>
<span class="w">    </span><span class="n">nlme</span><span class="p">,</span><span class="w"> </span><span class="n">nnet</span><span class="p">,</span><span class="w"> </span><span class="n">spatial</span><span class="p">,</span><span class="w"> </span><span class="n">survival</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code><span class="nf">.libPaths</span><span class="p">()</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code><span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;/home/thoellinger/R/x86_64-pc-linux-gnu-library/4.0&quot;</span>
<span class="p">[</span><span class="m">2</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;/usr/local/lib/R/site-library&quot;</span><span class="w">                      </span>
<span class="p">[</span><span class="m">3</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;/usr/lib/R/site-library&quot;</span><span class="w">                            </span>
<span class="p">[</span><span class="m">4</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;/usr/lib/R/library&quot;</span>
</code></pre></div>
</blockquote>
<div class="highlight"><pre><span></span><code>BiocManager::install<span class="o">(</span>c<span class="o">(</span><span class="s2">&quot;boot&quot;</span>,<span class="w"> </span><span class="s2">&quot;class&quot;</span>,<span class="w"> </span><span class="s2">&quot;cluster&quot;</span>,<span class="w"> </span><span class="s2">&quot;foreign&quot;</span>,<span class="w"> </span><span class="s2">&quot;KernSmooth&quot;</span>,<span class="w"> </span><span class="s2">&quot;lattice&quot;</span>,<span class="w"> </span><span class="s2">&quot;MASS&quot;</span>,<span class="w"> </span><span class="s2">&quot;Matrix&quot;</span>,<span class="w"> </span><span class="s2">&quot;mgcv&quot;</span>,<span class="w"> </span><span class="s2">&quot;nlme&quot;</span>,<span class="w"> </span><span class="s2">&quot;nnet&quot;</span>,<span class="w"> </span><span class="s2">&quot;spatial&quot;</span>,<span class="w"> </span><span class="s2">&quot;survival&quot;</span><span class="o">)</span>,<span class="w"> </span><span class="nv">lib</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">&quot;/home/thoellinger/R/x86_64-pc-linux-gnu-library/4.0&quot;</span><span class="o">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code><span class="n">BiocManager</span><span class="o">::</span><span class="nf">install</span><span class="p">(</span><span class="s">&quot;clusterProfiler&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">lib</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;/home/thoellinger/R/x86_64-pc-linux-gnu-library/4.0&quot;</span><span class="p">)</span>
</code></pre></div>
<blockquote>
<div class="highlight"><pre><span></span><code><span class="n">Error</span><span class="w"> </span><span class="n">in</span><span class="w"> </span><span class="nf">get</span><span class="p">(</span><span class="n">name</span><span class="p">,</span><span class="w"> </span><span class="n">envir</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">asNamespace</span><span class="p">(</span><span class="n">pkg</span><span class="p">),</span><span class="w"> </span><span class="n">inherits</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span><span class="w"> </span><span class="o">:</span><span class="w"> </span>
<span class="w">  </span><span class="n">object</span><span class="w"> </span><span class="s">&#39;get_fun_from_pkg&#39;</span><span class="w"> </span><span class="n">not</span><span class="w"> </span><span class="n">found</span>
<span class="n">Error</span><span class="o">:</span><span class="w"> </span><span class="n">unable</span><span class="w"> </span><span class="n">to</span><span class="w"> </span><span class="n">load</span><span class="w"> </span><span class="n">R</span><span class="w"> </span><span class="n">code</span><span class="w"> </span><span class="n">in</span><span class="w"> </span><span class="n">package</span><span class="w"> </span>‘<span class="n">clusterProfiler</span>’
<span class="n">Execution</span><span class="w"> </span><span class="n">halted</span>
<span class="n">ERROR</span><span class="o">:</span><span class="w"> </span><span class="n">lazy</span><span class="w"> </span><span class="n">loading</span><span class="w"> </span><span class="n">failed</span><span class="w"> </span><span class="n">for</span><span class="w"> </span><span class="n">package</span><span class="w"> </span>‘<span class="n">clusterProfiler</span>’
<span class="o">*</span><span class="w"> </span><span class="n">removing</span><span class="w"> </span>‘<span class="o">/</span><span class="n">home</span><span class="o">/</span><span class="n">thoellinger</span><span class="o">/</span><span class="n">R</span><span class="o">/</span><span class="n">x86_64</span><span class="o">-</span><span class="n">pc</span><span class="o">-</span><span class="n">linux</span><span class="o">-</span><span class="n">gnu</span><span class="o">-</span><span class="n">library</span><span class="o">/</span><span class="m">4.0</span><span class="o">/</span><span class="n">clusterProfiler</span>’

<span class="n">The</span><span class="w"> </span><span class="n">downloaded</span><span class="w"> </span><span class="n">source</span><span class="w"> </span><span class="n">packages</span><span class="w"> </span><span class="n">are</span><span class="w"> </span><span class="n">in</span>
<span class="w">    </span>‘<span class="o">/</span><span class="n">tmp</span><span class="o">/</span><span class="n">Rtmp3Xmv24</span><span class="o">/</span><span class="n">downloaded_packages</span>’
<span class="n">Warning</span><span class="w"> </span><span class="n">message</span><span class="o">:</span>
<span class="n">In</span><span class="w"> </span><span class="nf">.inet_warning</span><span class="p">(</span><span class="n">msg</span><span class="p">)</span><span class="w"> </span><span class="o">:</span>
<span class="w">  </span><span class="n">installation</span><span class="w"> </span><span class="n">of</span><span class="w"> </span><span class="n">package</span><span class="w"> </span>‘<span class="n">clusterProfiler</span>’<span class="w"> </span><span class="n">had</span><span class="w"> </span><span class="n">non</span><span class="o">-</span><span class="n">zero</span><span class="w"> </span><span class="n">exit</span><span class="w"> </span><span class="n">status</span>
</code></pre></div>
</blockquote>
<p>Still does not work. It appears that the problem is the dependency in the <code>rvcheck</code> library, that contained the <code>get_fun_from_pkg</code> up to version 1.8 but does not contain it anymore ; and the authors of <code>clusterProfiler</code> have not updated their dependencies: https://www.biostars.org/p/9491480/</p>
</blockquote>
<p>We need to install an older version of <code>rvcheck</code>, namely, the version 1.8. See https://cran.microsoft.com/snapshot/2020-04-20/web/packages/rvcheck/index.html. We download it and install it from source. On the IRSD computer:</p>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span><span class="w"> </span>Documents/shared/sources
wget<span class="w"> </span>https://cran.microsoft.com/snapshot/2020-04-20/src/contrib/rvcheck_0.1.8.tar.gz<span class="w"> </span>.
</code></pre></div>
<div class="highlight"><pre><span></span><code><span class="nf">install.packages</span><span class="p">(</span><span class="s">&quot;/home/thoellinger/Documents/shared/sources/rvcheck_0.1.8.tar.gz&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">repos</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NULL</span><span class="p">,</span><span class="w"> </span><span class="n">type</span><span class="o">=</span><span class="s">&quot;source&quot;</span><span class="p">)</span>
</code></pre></div>
<p>So now:</p>
<div class="highlight"><pre><span></span><code><span class="n">BiocManager</span><span class="o">::</span><span class="nf">install</span><span class="p">(</span><span class="s">&quot;clusterProfiler&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">lib</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;/home/thoellinger/R/x86_64-pc-linux-gnu-library/4.0&quot;</span><span class="p">)</span>
</code></pre></div>
<p>should work <strong>BUT</strong> one have to select "no" when the prompt asks whether to update the <code>rvcheck</code> package (even if it asks whether to update or not a lot of packages including <code>rvcheck</code>)!</p>
<p>Note that one might also need to install <code>org.Hs.eg.db</code> separately.</p>
<div class="highlight"><pre><span></span><code><span class="n">BiocManager</span><span class="o">::</span><span class="nf">install</span><span class="p">(</span><span class="s">&quot;org.Hs.eg.db&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">lib</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;/home/thoellinger/R/x86_64-pc-linux-gnu-library/4.0&quot;</span><span class="p">)</span>
</code></pre></div>
<p>Again, it is <strong>necessary</strong> to select "no" when the prompt asks whether to update the <code>rvcheck</code> package (even if it asks whether to update or not a lot of packages including <code>rvcheck</code>)!</p>
<p>For graphics, we need a few more packages:</p>
<div class="highlight"><pre><span></span><code><span class="n">BiocManager</span><span class="o">::</span><span class="nf">install</span><span class="p">(</span><span class="s">&quot;ggupset&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">lib</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;/home/thoellinger/R/x86_64-pc-linux-gnu-library/4.0&quot;</span><span class="p">)</span>
<span class="n">BiocManager</span><span class="o">::</span><span class="nf">install</span><span class="p">(</span><span class="s">&quot;ggnewscale&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">lib</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;/home/thoellinger/R/x86_64-pc-linux-gnu-library/4.0&quot;</span><span class="p">)</span>
</code></pre></div>
<p>Again, it is <strong>necessary</strong> to select "no" when the prompt asks whether to update the <code>rvcheck</code> package.</p>
<p>We followed this guide for visual outputs: https://yulab-smu.top/biomedical-knowledge-mining-book/enrichplot.html</p>
<h2 id="script">Script<a class="headerlink" href="#script" title="Permanent link">&para;</a></h2>
<p>The main part of the script will use the <code>groupGO</code> function that basically does everything:</p>
<div class="highlight"><pre><span></span><code><span class="n">res</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">groupGO</span><span class="p">(</span><span class="n">gene</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">U</span><span class="p">,</span><span class="w"> </span><span class="c1"># vector of entrez gene id</span>
<span class="w">              </span><span class="n">OrgDb</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">org.Hs.eg.db</span><span class="p">,</span><span class="w"> </span><span class="c1"># human organism</span>
<span class="w">              </span><span class="n">ont</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;BP&quot;</span><span class="p">,</span><span class="w"> </span><span class="c1"># ontology, &quot;BP&quot;, &quot;MF&quot; or &quot;CC&quot;</span>
<span class="w">              </span><span class="n">universe</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">universe</span><span class="p">,</span><span class="w"> </span><span class="c1"># background genes (entrez)</span>
<span class="w">              </span><span class="n">pvalueCutoff</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.05</span><span class="p">,</span>
<span class="w">              </span><span class="n">pAdjustMethod</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;BH&quot;</span><span class="p">,</span><span class="w"> </span><span class="c1"># one of &quot;holm&quot;, &quot;hochberg&quot;, &quot;hommel&quot;, &quot;bonferroni&quot;,</span>
<span class="w">                                    </span><span class="c1"># &quot;BH&quot;, &quot;BY&quot;, &quot;fdr&quot;, &quot;none&quot;</span>
<span class="w">              </span><span class="n">readable</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span><span class="w"> </span><span class="c1"># whether mapping gene ID to gene Name</span>
</code></pre></div>
<p>The <code>OrgDb</code> parameter is an object specifying the organism of interest. For human, one must use <code>OrgDb = org.Hs.eg.db</code>. Note that <code>OrgDb</code> object are available in Bioconductor for 20 organisms: http://bioconductor.org/packages/release/BiocViews.html#___OrgDb</p>
<p>One can find the list of types of sequence id available (for instance "ENTREZID", "ENSEMBL", "REFSEQ") using <code>columns(org.Hs.eg.db)</code>:</p>
<blockquote>
<div class="highlight"><pre><span></span><code><span class="o">&gt;</span><span class="w"> </span><span class="nf">columns</span><span class="p">(</span><span class="n">org.Hs.eg.db</span><span class="p">)</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;ACCNUM&quot;</span><span class="w">       </span><span class="s">&quot;ALIAS&quot;</span><span class="w">        </span><span class="s">&quot;ENSEMBL&quot;</span><span class="w">      </span><span class="s">&quot;ENSEMBLPROT&quot;</span><span class="w">  </span><span class="s">&quot;ENSEMBLTRANS&quot;</span><span class="w"> </span><span class="s">&quot;ENTREZID&quot;</span><span class="w">     </span><span class="s">&quot;ENZYME&quot;</span><span class="w">       </span><span class="s">&quot;EVIDENCE&quot;</span><span class="w">     </span><span class="s">&quot;EVIDENCEALL&quot;</span><span class="w">  </span><span class="s">&quot;GENENAME&quot;</span><span class="w">     </span><span class="s">&quot;GO&quot;</span><span class="w">           </span><span class="s">&quot;GOALL&quot;</span><span class="w">       </span>
<span class="p">[</span><span class="m">13</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;IPI&quot;</span><span class="w">          </span><span class="s">&quot;MAP&quot;</span><span class="w">          </span><span class="s">&quot;OMIM&quot;</span><span class="w">         </span><span class="s">&quot;ONTOLOGY&quot;</span><span class="w">     </span><span class="s">&quot;ONTOLOGYALL&quot;</span><span class="w">  </span><span class="s">&quot;PATH&quot;</span><span class="w">         </span><span class="s">&quot;PFAM&quot;</span><span class="w">         </span><span class="s">&quot;PMID&quot;</span><span class="w">         </span><span class="s">&quot;PROSITE&quot;</span><span class="w">      </span><span class="s">&quot;REFSEQ&quot;</span><span class="w">       </span><span class="s">&quot;SYMBOL&quot;</span><span class="w">       </span><span class="s">&quot;UCSCKG&quot;</span><span class="w">      </span>
<span class="p">[</span><span class="m">25</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;UNIGENE&quot;</span><span class="w">      </span><span class="s">&quot;UNIPROT&quot;</span>
</code></pre></div>
</blockquote>
<p>Here is the full content of <code>GO_enrichment_FDR.R</code>:</p>
<blockquote>
<div class="highlight"><pre><span></span><code><span class="c1">##################</span>
<span class="c1"># OPTION PARSING #</span>
<span class="c1">##################</span>

<span class="nf">suppressPackageStartupMessages</span><span class="p">(</span><span class="nf">library</span><span class="p">(</span><span class="s">&quot;optparse&quot;</span><span class="p">))</span>

<span class="n">option_list</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span>
<span class="w">    </span><span class="nf">make_option</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;-k&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;--key&quot;</span><span class="p">),</span><span class="w"> </span><span class="n">default</span><span class="o">=</span><span class="s">&quot;ENTREZID&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">help</span><span class="o">=</span><span class="s">&quot;type of gene identifiers used in other arguments (one of \&quot;ENTREZID\&quot;, \&quot;ENSEMBL\&quot;, \&quot;REFSEQ\&quot;, \&quot;GENENAME\&quot;, \&quot;SYMBOL\&quot;) [default=%default]&quot;</span><span class="p">),</span>
<span class="w">    </span><span class="nf">make_option</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;-u&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;--universe&quot;</span><span class="p">),</span><span class="w"> </span><span class="n">default</span><span class="o">=</span><span class="s">&quot;None&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">help</span><span class="o">=</span><span class="s">&quot;list of background gene identifiers (of same type as --key), WITHOUT header.</span>
<span class="s">                 Use \&quot;None\&quot; for default background (given by the clusterProfiler package) [default=%default]&quot;</span><span class="p">),</span>
<span class="w">    </span><span class="nf">make_option</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;-G&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;--genes&quot;</span><span class="p">),</span><span class="w"> </span><span class="n">default</span><span class="o">=</span><span class="s">&quot;stdin&quot;</span><span class="p">,</span>
<span class="w">                 </span><span class="n">help</span><span class="o">=</span><span class="s">&quot;list of foreground gene identifiers (of same type as --key), WITHOUT header [default=%default]&quot;</span><span class="p">),</span>
<span class="w">    </span><span class="nf">make_option</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;-c&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;--ontology&quot;</span><span class="p">),</span><span class="w"> </span><span class="n">help</span><span class="o">=</span><span class="s">&quot;choose the GO category &lt; BP | MF | CC &gt; [default=%default]&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">default</span><span class="o">=</span><span class="s">&quot;BP&quot;</span><span class="p">),</span>
<span class="w">    </span><span class="nf">make_option</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;-p&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;--pval&quot;</span><span class="p">),</span><span class="w"> </span><span class="n">help</span><span class="o">=</span><span class="s">&quot;p-value threshold. All GO terms found to be enriched up to this threshold will be saved as output. [default=%default]&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">default</span><span class="o">=</span><span class="m">0.1</span><span class="p">),</span>
<span class="w">    </span><span class="nf">make_option</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;-q&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;--qval&quot;</span><span class="p">),</span><span class="w"> </span><span class="n">help</span><span class="o">=</span><span class="s">&quot;q-value threshold. All GO terms found to be enriched up to this threshold will be saved as output. [default=%default]&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">default</span><span class="o">=</span><span class="m">0.2</span><span class="p">),</span>
<span class="w">    </span><span class="nf">make_option</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;-a&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;--padjMethod&quot;</span><span class="p">),</span><span class="w"> </span><span class="n">help</span><span class="o">=</span><span class="s">&quot;p-value adjustement method threshold. One of \&quot;holm\&quot;, \&quot;hochberg\&quot;, \&quot;hommel\&quot;, \&quot;bonferroni\&quot;, \&quot;BH\&quot;, \&quot;BY\&quot;, \&quot;fdr\&quot;, \&quot;none\&quot; [default=%default]&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">default</span><span class="o">=</span><span class="s">&quot;fdr&quot;</span><span class="p">),</span>
<span class="w">    </span><span class="nf">make_option</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;-f&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;--fdr&quot;</span><span class="p">),</span><span class="w"> </span><span class="n">help</span><span class="o">=</span><span class="s">&quot;FDR (or adjusted p-value in general) threshold. All GO terms found to be enriched up to this threshold will be saved as output in a separate table. [default=%default]&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">default</span><span class="o">=</span><span class="m">0.1</span><span class="p">),</span>
<span class="w">    </span><span class="nf">make_option</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;-v&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;--verbose&quot;</span><span class="p">),</span><span class="w"> </span><span class="n">default</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">help</span><span class="o">=</span><span class="s">&quot;Verbosity (TRUE/FALSE) [default=%default]&quot;</span><span class="p">),</span>
<span class="w">    </span><span class="nf">make_option</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;-o&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;--output&quot;</span><span class="p">),</span><span class="w"> </span><span class="n">default</span><span class="o">=</span><span class="s">&quot;out&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">help</span><span class="o">=</span><span class="s">&quot;additional tags for otuput [default=%default]&quot;</span><span class="p">),</span>
<span class="w">    </span><span class="nf">make_option</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;-d&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;--output_dir&quot;</span><span class="p">),</span><span class="w"> </span><span class="n">default</span><span class="o">=</span><span class="s">&quot;./GO_output/&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">help</span><span class="o">=</span><span class="s">&quot;directory for the output [default=%default]&quot;</span><span class="p">)</span>
<span class="p">)</span>

<span class="n">parser</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">OptionParser</span><span class="p">(</span><span class="n">usage</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;%prog [options] file&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">option_list</span><span class="o">=</span><span class="n">option_list</span><span class="p">)</span>
<span class="n">arguments</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">parse_args</span><span class="p">(</span><span class="n">parser</span><span class="p">,</span><span class="w"> </span><span class="n">positional_arguments</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span>
<span class="n">opt</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">arguments</span><span class="o">$</span><span class="n">options</span>
<span class="n">verbose</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">opt</span><span class="o">$</span><span class="n">verbose</span>

<span class="c1">#################################</span>
<span class="c1"># Load libraries &amp; Import data #</span>
<span class="c1">#################################</span>

<span class="nf">if</span><span class="p">(</span><span class="n">verbose</span><span class="p">){</span>
<span class="w">    </span><span class="nf">library</span><span class="p">(</span><span class="s">&quot;clusterProfiler&quot;</span><span class="p">)</span>
<span class="w">    </span><span class="nf">library</span><span class="p">(</span><span class="s">&quot;org.Hs.eg.db&quot;</span><span class="p">)</span>
<span class="w">    </span><span class="nf">library</span><span class="p">(</span><span class="n">enrichplot</span><span class="p">)</span>
<span class="w">    </span><span class="nf">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span>
<span class="p">}</span><span class="w"> </span><span class="n">else</span><span class="p">{</span>
<span class="w">    </span><span class="nf">suppressPackageStartupMessages</span><span class="p">(</span><span class="nf">library</span><span class="p">(</span><span class="s">&quot;clusterProfiler&quot;</span><span class="p">))</span>
<span class="w">    </span><span class="nf">suppressPackageStartupMessages</span><span class="p">(</span><span class="nf">library</span><span class="p">(</span><span class="s">&quot;org.Hs.eg.db&quot;</span><span class="p">))</span>
<span class="w">    </span><span class="nf">suppressPackageStartupMessages</span><span class="p">(</span><span class="nf">library</span><span class="p">(</span><span class="n">enrichplot</span><span class="p">))</span>
<span class="w">    </span><span class="nf">suppressPackageStartupMessages</span><span class="p">(</span><span class="nf">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">))</span>
<span class="p">}</span>

<span class="nf">if </span><span class="p">(</span><span class="n">verbose</span><span class="p">)</span><span class="w"> </span><span class="nf">print</span><span class="p">(</span><span class="s">&quot;Loading input data...&quot;</span><span class="p">)</span>
<span class="n">key</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">opt</span><span class="o">$</span><span class="n">key</span>
<span class="nf">if</span><span class="p">(</span><span class="n">opt</span><span class="o">$</span><span class="n">universe</span><span class="o">==</span><span class="s">&quot;None&quot;</span><span class="p">){</span>
<span class="w">    </span><span class="n">universe</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NA</span>
<span class="w">    </span><span class="nf">print</span><span class="p">(</span><span class="s">&quot;Warning: using defaut universe automatically provided by the clusterProfiler package&quot;</span><span class="p">)</span>
<span class="p">}</span><span class="w"> </span><span class="n">else</span><span class="p">{</span>
<span class="w">    </span><span class="n">U</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">read.table</span><span class="p">(</span><span class="n">opt</span><span class="o">$</span><span class="n">universe</span><span class="p">,</span><span class="w"> </span><span class="n">h</span><span class="o">=</span><span class="bp">F</span><span class="p">,</span><span class="w"> </span><span class="n">col.names</span><span class="o">=</span><span class="s">&#39;hs&#39;</span><span class="p">)</span>
<span class="w">    </span><span class="n">universe</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">as.character</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="n">U</span><span class="o">$</span><span class="n">hs</span><span class="p">))</span>
<span class="p">}</span>

<span class="n">G</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">read.table</span><span class="p">(</span><span class="n">opt</span><span class="o">$</span><span class="n">genes</span><span class="p">,</span><span class="w"> </span><span class="n">h</span><span class="o">=</span><span class="bp">F</span><span class="p">,</span><span class="w"> </span><span class="n">col.names</span><span class="o">=</span><span class="s">&#39;hs&#39;</span><span class="p">)</span>
<span class="n">genes</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">unique</span><span class="p">(</span><span class="n">G</span><span class="o">$</span><span class="n">hs</span><span class="p">)</span>
<span class="n">ontology</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">opt</span><span class="o">$</span><span class="n">ontology</span>
<span class="n">pval</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">opt</span><span class="o">$</span><span class="n">pval</span>
<span class="n">qval</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">opt</span><span class="o">$</span><span class="n">qval</span>
<span class="n">pAdjMethod</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">opt</span><span class="o">$</span><span class="n">padjMethod</span>
<span class="n">fdr_threshold</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">opt</span><span class="o">$</span><span class="n">fdr</span>

<span class="nf">if </span><span class="p">(</span><span class="n">verbose</span><span class="p">)</span><span class="w"> </span><span class="nf">print</span><span class="p">(</span><span class="s">&quot;Done.&quot;</span><span class="p">)</span>

<span class="c1">######################</span>
<span class="c1"># Compute enrichment #</span>
<span class="c1">######################</span>

<span class="n">to_readable</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span>
<span class="nf">if </span><span class="p">(</span><span class="n">key</span><span class="o">==</span><span class="s">&quot;SYMBOL&quot;</span><span class="p">)</span><span class="w"> </span><span class="n">to_readable</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span>

<span class="nf">if </span><span class="p">(</span><span class="n">verbose</span><span class="p">)</span><span class="w"> </span><span class="nf">print</span><span class="p">(</span><span class="s">&quot;Computing GO enrichment...&quot;</span><span class="p">)</span>

<span class="n">res</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">enrichGO</span><span class="p">(</span><span class="n">gene</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">genes</span><span class="p">,</span>
<span class="w">                  </span><span class="n">OrgDb</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">org.Hs.eg.db</span><span class="p">,</span>
<span class="w">                  </span><span class="n">keyType</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">key</span><span class="p">,</span>
<span class="w">                  </span><span class="n">ont</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">ontology</span><span class="p">,</span>
<span class="w">                  </span><span class="n">universe</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">universe</span><span class="p">,</span>
<span class="w">                  </span><span class="n">pvalueCutoff</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">pval</span><span class="p">,</span>
<span class="w">                  </span><span class="n">qvalueCutoff</span><span class="w">  </span><span class="o">=</span><span class="w"> </span><span class="n">qval</span><span class="p">,</span>
<span class="w">                  </span><span class="n">pAdjustMethod</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">pAdjMethod</span><span class="p">,</span>
<span class="w">                  </span><span class="n">readable</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">to_readable</span><span class="p">)</span>

<span class="nf">if </span><span class="p">(</span><span class="n">verbose</span><span class="p">)</span><span class="w"> </span><span class="nf">print</span><span class="p">(</span><span class="s">&quot;Done.&quot;</span><span class="p">)</span>

<span class="n">universe_found</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">res</span><span class="o">@</span><span class="n">universe</span>

<span class="c1">###################</span>
<span class="c1"># Compute results #</span>
<span class="c1">###################</span>

<span class="nf">if</span><span class="p">(</span><span class="n">opt</span><span class="o">$</span><span class="n">universe</span><span class="o">==</span><span class="s">&quot;None&quot;</span><span class="p">){</span>
<span class="w">    </span><span class="nf">sprintf</span><span class="p">(</span><span class="s">&quot;%s (default) background genes&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">length</span><span class="p">(</span><span class="n">res</span><span class="o">@</span><span class="n">universe</span><span class="p">))</span>
<span class="p">}</span><span class="w"> </span><span class="n">else</span><span class="p">{</span>
<span class="w">    </span><span class="nf">sprintf</span><span class="p">(</span><span class="s">&quot;%s provided background genes; %s with a corresponding %s id&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">nrow</span><span class="p">(</span><span class="n">U</span><span class="p">),</span><span class="w"> </span><span class="nf">length</span><span class="p">(</span><span class="n">universe</span><span class="p">),</span><span class="w"> </span><span class="n">key</span><span class="p">)</span>
<span class="p">}</span>
<span class="n">n_genes_found</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">strsplit</span><span class="p">(</span><span class="n">res</span><span class="o">@</span><span class="n">result</span><span class="o">$</span><span class="n">GeneRatio</span><span class="p">[[</span><span class="m">1</span><span class="p">]],</span><span class="s">&#39;/&#39;</span><span class="p">)[[</span><span class="m">1</span><span class="p">]][</span><span class="m">2</span><span class="p">]</span>
<span class="nf">sprintf</span><span class="p">(</span><span class="s">&quot;%s provided genes; %s found by `enrichGO`&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">nrow</span><span class="p">(</span><span class="n">G</span><span class="p">),</span><span class="w"> </span><span class="n">n_genes_found</span><span class="p">)</span>
<span class="nf">sprintf</span><span class="p">(</span><span class="s">&quot;Computed GO enrichment (whether significant or not) for %s distinct GO terms&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">nrow</span><span class="p">(</span><span class="n">res</span><span class="o">@</span><span class="n">result</span><span class="p">))</span>
<span class="nf">sprintf</span><span class="p">(</span><span class="s">&quot;Of those %s GO terms, %s have a %s-adjusted p-val &lt; %s&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">nrow</span><span class="p">(</span><span class="n">res</span><span class="o">@</span><span class="n">result</span><span class="p">),</span><span class="w"> </span><span class="nf">nrow</span><span class="p">(</span><span class="n">res</span><span class="o">@</span><span class="n">result</span><span class="p">[</span><span class="n">res</span><span class="o">@</span><span class="n">result</span><span class="o">$</span><span class="n">p.adjust</span><span class="o">&lt;=</span><span class="n">fdr_threshold</span><span class="p">,]),</span><span class="w"> </span><span class="n">pAdjMethod</span><span class="p">,</span><span class="w"> </span><span class="n">fdr_threshold</span><span class="p">)</span>

<span class="c1">################</span>
<span class="c1"># Write output #</span>
<span class="c1">################</span>

<span class="nf">dir.create</span><span class="p">(</span><span class="n">opt</span><span class="o">$</span><span class="n">output_dir</span><span class="p">,</span><span class="w"> </span><span class="n">showWarnings</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> </span><span class="n">recursive</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span>
<span class="n">output</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">sprintf</span><span class="p">(</span><span class="s">&quot;%s/%s.%s.%s%s&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">opt</span><span class="o">$</span><span class="n">output_dir</span><span class="p">,</span><span class="w"> </span><span class="n">opt</span><span class="o">$</span><span class="n">output</span><span class="p">,</span><span class="w"> </span><span class="n">opt</span><span class="o">$</span><span class="n">ontology</span><span class="p">,</span><span class="w"> </span><span class="n">opt</span><span class="o">$</span><span class="n">fdr</span><span class="p">,</span><span class="w"> </span><span class="n">opt</span><span class="o">$</span><span class="n">padjMethod</span><span class="p">)</span>

<span class="nf">if </span><span class="p">(</span><span class="n">verbose</span><span class="p">)</span><span class="w"> </span><span class="nf">print</span><span class="p">(</span><span class="s">&quot;Writing outputs tables...&quot;</span><span class="p">)</span>

<span class="c1"># Full table</span>
<span class="nf">write.table</span><span class="p">(</span><span class="n">res</span><span class="o">@</span><span class="n">result</span><span class="p">,</span><span class="w"> </span><span class="n">file</span><span class="o">=</span><span class="nf">sprintf</span><span class="p">(</span><span class="s">&quot;%s.tsv&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">output</span><span class="p">),</span><span class="w"> </span><span class="n">quote</span><span class="o">=</span><span class="bp">F</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">row.names</span><span class="o">=</span><span class="bp">F</span><span class="p">)</span>

<span class="c1"># Most significant GO terms</span>
<span class="nf">write.table</span><span class="p">(</span><span class="n">res</span><span class="o">@</span><span class="n">result</span><span class="p">[</span><span class="n">res</span><span class="o">@</span><span class="n">result</span><span class="o">$</span><span class="n">p.adjust</span><span class="o">&lt;=</span><span class="n">fdr_threshold</span><span class="p">,],</span><span class="w"> </span><span class="n">file</span><span class="o">=</span><span class="nf">sprintf</span><span class="p">(</span><span class="s">&quot;%s.significant.tsv&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">output</span><span class="p">),</span><span class="w"> </span><span class="n">quote</span><span class="o">=</span><span class="bp">F</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">row.names</span><span class="o">=</span><span class="bp">F</span><span class="p">)</span>

<span class="nf">if </span><span class="p">(</span><span class="n">verbose</span><span class="p">)</span><span class="w"> </span><span class="nf">print</span><span class="p">(</span><span class="s">&quot;Done. Writing output images...&quot;</span><span class="p">)</span>

<span class="nf">png</span><span class="p">(</span><span class="n">file</span><span class="o">=</span><span class="nf">sprintf</span><span class="p">(</span><span class="s">&quot;%s.upset.png&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">output</span><span class="p">),</span>
<span class="w">       </span><span class="n">width</span><span class="o">=</span><span class="m">600</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="o">=</span><span class="m">350</span><span class="p">)</span>
<span class="nf">upsetplot</span><span class="p">(</span><span class="n">res</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="nf">ggtitle</span><span class="p">(</span><span class="s">&quot;UpSet Plot. Number of genes per significant GO term&quot;</span><span class="p">)</span>
<span class="nf">graphics.off</span><span class="p">()</span>

<span class="nf">png</span><span class="p">(</span><span class="n">file</span><span class="o">=</span><span class="nf">sprintf</span><span class="p">(</span><span class="s">&quot;%s.dotplot.png&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">output</span><span class="p">),</span>
<span class="w">       </span><span class="n">width</span><span class="o">=</span><span class="m">600</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="o">=</span><span class="m">350</span><span class="p">)</span>
<span class="nf">dotplot</span><span class="p">(</span><span class="n">res</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="nf">ggtitle</span><span class="p">(</span><span class="nf">sprintf</span><span class="p">(</span><span class="s">&quot;p.adjust based on %s correction&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">pAdjMethod</span><span class="p">))</span>
<span class="nf">graphics.off</span><span class="p">()</span>

<span class="nf">png</span><span class="p">(</span><span class="n">file</span><span class="o">=</span><span class="nf">sprintf</span><span class="p">(</span><span class="s">&quot;%s.gene-concept.png&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">output</span><span class="p">),</span>
<span class="w">       </span><span class="n">width</span><span class="o">=</span><span class="m">600</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="o">=</span><span class="m">350</span><span class="p">)</span>
<span class="nf">cnetplot</span><span class="p">(</span><span class="n">res</span><span class="p">,</span><span class="w"> </span><span class="n">categorySize</span><span class="o">=</span><span class="s">&quot;pvalue&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">foldChange</span><span class="o">=</span><span class="n">genes</span><span class="p">,</span><span class="w"> </span><span class="n">colorEdge</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="nf">ggtitle</span><span class="p">(</span><span class="nf">sprintf</span><span class="p">(</span><span class="s">&quot;Gene-%s Network, circular&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">ontology</span><span class="p">))</span>
<span class="nf">graphics.off</span><span class="p">()</span>

<span class="nf">png</span><span class="p">(</span><span class="n">file</span><span class="o">=</span><span class="nf">sprintf</span><span class="p">(</span><span class="s">&quot;%s.gene-concept.circular.png&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">output</span><span class="p">),</span>
<span class="w">       </span><span class="n">width</span><span class="o">=</span><span class="m">600</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="o">=</span><span class="m">350</span><span class="p">)</span>
<span class="nf">cnetplot</span><span class="p">(</span><span class="n">res</span><span class="p">,</span><span class="w"> </span><span class="n">categorySize</span><span class="o">=</span><span class="s">&quot;pvalue&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">foldChange</span><span class="o">=</span><span class="n">genes</span><span class="p">,</span><span class="w"> </span><span class="n">circular</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">colorEdge</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="nf">ggtitle</span><span class="p">(</span><span class="nf">sprintf</span><span class="p">(</span><span class="s">&quot;Gene-%s Network&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">ontology</span><span class="p">))</span>
<span class="nf">graphics.off</span><span class="p">()</span>

<span class="nf">png</span><span class="p">(</span><span class="n">file</span><span class="o">=</span><span class="nf">sprintf</span><span class="p">(</span><span class="s">&quot;%s.heatplot.png&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">output</span><span class="p">),</span>
<span class="w">       </span><span class="n">width</span><span class="o">=</span><span class="m">600</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="o">=</span><span class="m">350</span><span class="p">)</span>
<span class="nf">heatplot</span><span class="p">(</span><span class="n">res</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="nf">ggtitle</span><span class="p">(</span><span class="s">&quot;Heatplot: significant GO terms against contributing genes&quot;</span><span class="p">)</span>
<span class="nf">graphics.off</span><span class="p">()</span>

<span class="n">pbar</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="kc">pi</span><span class="o">=</span><span class="nf">seq</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="n">length.out</span><span class="o">=</span><span class="m">11</span><span class="p">)</span>
<span class="w">    </span>
<span class="w">    </span><span class="nf">mutate</span><span class="p">(</span><span class="n">x</span><span class="p">,</span><span class="w"> </span><span class="n">pp</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">cut</span><span class="p">(</span><span class="n">p.adjust</span><span class="p">,</span><span class="w"> </span><span class="kc">pi</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">       </span><span class="nf">group_by</span><span class="p">(</span><span class="n">pp</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">       </span><span class="nf">summarise</span><span class="p">(</span><span class="n">cnt</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">n</span><span class="p">())</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">       </span><span class="nf">ggplot</span><span class="p">(</span><span class="nf">aes</span><span class="p">(</span><span class="n">pp</span><span class="p">,</span><span class="w"> </span><span class="n">cnt</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="nf">geom_col</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w"> </span>
<span class="w">       </span><span class="nf">theme_minimal</span><span class="p">()</span><span class="w"> </span><span class="o">+</span>
<span class="w">       </span><span class="nf">xlab</span><span class="p">(</span><span class="s">&quot;p value intervals&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">       </span><span class="nf">ylab</span><span class="p">(</span><span class="s">&quot;Frequency&quot;</span><span class="p">)</span>
<span class="p">}</span><span class="w">   </span>

<span class="nf">if </span><span class="p">(</span><span class="n">verbose</span><span class="p">)</span><span class="w"> </span><span class="nf">print</span><span class="p">(</span><span class="s">&quot;Writing last output image (this one might take some time)...&quot;</span><span class="p">)</span>

<span class="n">random_genes</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">sample</span><span class="p">(</span><span class="n">universe_found</span><span class="p">,</span><span class="w"> </span><span class="nf">as.integer</span><span class="p">(</span><span class="n">n_genes_found</span><span class="p">))</span>
<span class="n">enrich_ref</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">enrichGO</span><span class="p">(</span><span class="n">gene</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">random_genes</span><span class="p">,</span>
<span class="w">                         </span><span class="n">OrgDb</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">org.Hs.eg.db</span><span class="p">,</span>
<span class="w">                         </span><span class="n">keyType</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">key</span><span class="p">,</span>
<span class="w">                         </span><span class="n">ont</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">ontology</span><span class="p">,</span>
<span class="w">                         </span><span class="n">universe</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">universe</span><span class="p">,</span>
<span class="w">                         </span><span class="n">pvalueCutoff</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">pval</span><span class="p">,</span>
<span class="w">                         </span><span class="n">qvalueCutoff</span><span class="w">  </span><span class="o">=</span><span class="w"> </span><span class="n">qval</span><span class="p">,</span>
<span class="w">                         </span><span class="n">pAdjustMethod</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">pAdjMethod</span><span class="p">,</span>
<span class="w">                         </span><span class="n">readable</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">to_readable</span><span class="p">)</span>

<span class="n">p1</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">pbar</span><span class="p">(</span><span class="n">res</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="nf">ggtitle</span><span class="p">(</span><span class="s">&quot;p value distribution&quot;</span><span class="p">)</span>
<span class="n">p2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">pbar</span><span class="p">(</span><span class="n">enrich_ref</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="nf">ggtitle</span><span class="p">(</span><span class="s">&quot;p value distribution (one single random sample of genes)&quot;</span><span class="p">)</span>
<span class="nf">png</span><span class="p">(</span><span class="n">file</span><span class="o">=</span><span class="nf">sprintf</span><span class="p">(</span><span class="s">&quot;%s.pvalues.png&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">output</span><span class="p">),</span>
<span class="w">       </span><span class="n">width</span><span class="o">=</span><span class="m">600</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="o">=</span><span class="m">350</span><span class="p">)</span>
<span class="n">cowplot</span><span class="o">::</span><span class="nf">plot_grid</span><span class="p">(</span><span class="n">p1</span><span class="p">,</span><span class="w"> </span><span class="n">p2</span><span class="p">,</span><span class="w"> </span><span class="n">ncol</span><span class="o">=</span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="n">labels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">LETTERS</span><span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">2</span><span class="p">])</span>
<span class="nf">graphics.off</span><span class="p">()</span>

<span class="nf">if </span><span class="p">(</span><span class="n">verbose</span><span class="p">)</span><span class="w"> </span><span class="nf">print</span><span class="p">(</span><span class="s">&quot;Done.&quot;</span><span class="p">)</span>

<span class="nf">q</span><span class="p">(</span><span class="n">save</span><span class="o">=</span><span class="s">&#39;no&#39;</span><span class="p">)</span>
</code></pre></div>
</blockquote>
<h2 id="example-of-use">Example of use<a class="headerlink" href="#example-of-use" title="Permanent link">&para;</a></h2>
<div class="highlight"><pre><span></span><code>module<span class="w"> </span>load<span class="w"> </span>system/R-4.0.4_gcc-9.3.0
</code></pre></div>
<blockquote>
<div class="highlight"><pre><span></span><code><span class="o">&gt;</span><span class="w"> </span><span class="nf">.libPaths</span><span class="p">()</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;/home/thoellinger/R/x86_64-pc-linux-gnu-library/4.0&quot;</span>
<span class="p">[</span><span class="m">2</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;/tools/R/R-4.0.4_gcc-9.3.0/lib64/R/library&quot;</span>
</code></pre></div>
</blockquote>
<div class="highlight"><pre><span></span><code><span class="nf">install.packages</span><span class="p">(</span><span class="s">&quot;/work2/project/regenet/workspace/thoellinger/shared/sources/rvcheck_0.1.8.tar.gz&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">repos</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NULL</span><span class="p">,</span><span class="w"> </span><span class="n">type</span><span class="o">=</span><span class="s">&quot;source&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">lib</span><span class="o">=</span><span class="s">&quot;/home/thoellinger/R/x86_64-pc-linux-gnu-library/4.0&quot;</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code><span class="n">BiocManager</span><span class="o">::</span><span class="nf">install</span><span class="p">(</span><span class="s">&quot;clusterProfiler&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">lib</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;/home/thoellinger/R/x86_64-pc-linux-gnu-library/4.0&quot;</span><span class="p">)</span>
</code></pre></div>
<p>It is <strong>necessary</strong> to select "no" when the prompt asks whether to update the <code>rvcheck</code> package (even if it asks whether to update or not a lot of packages including <code>rvcheck</code>)!</p>
<blockquote>
<p>We might also have to install <code>org.Hs.eg.db</code> separately:</p>
<div class="highlight"><pre><span></span><code><span class="n">BiocManager</span><span class="o">::</span><span class="nf">install</span><span class="p">(</span><span class="s">&quot;org.Hs.eg.db&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">lib</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;/home/thoellinger/R/x86_64-pc-linux-gnu-library/4.0&quot;</span><span class="p">)</span>
</code></pre></div>
<p>Again, it is <strong>necessary</strong> to select "no" when the prompt asks whether to update the <code>rvcheck</code> package (even if it asks whether to update or not a lot of packages including <code>rvcheck</code>)!</p>
</blockquote>
<p>Finally:</p>
<div class="highlight"><pre><span></span><code>mkdir<span class="w"> </span>-p<span class="w"> </span>GO_FDR/all_genes/symbol
</code></pre></div>
<div class="highlight"><pre><span></span><code>Rscript<span class="w"> </span>GO_enrichment_FDR.R<span class="w"> </span>-k<span class="w"> </span><span class="s2">&quot;SYMBOL&quot;</span><span class="w"> </span>-G<span class="w"> </span>results/new_genes_v7.list<span class="w"> </span>-f<span class="w"> </span><span class="m">0</span>.05<span class="w"> </span>-c<span class="w"> </span><span class="s2">&quot;BP&quot;</span><span class="w"> </span>-a<span class="w"> </span><span class="s2">&quot;BH&quot;</span><span class="w"> </span>-o<span class="w"> </span><span class="s2">&quot;default_universe&quot;</span><span class="w"> </span>-d<span class="w"> </span><span class="s2">&quot;GO_FDR/all_genes/symbol&quot;</span>
</code></pre></div>
<blockquote>
<p><div class="highlight"><pre><span></span><code><span class="kc">...</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Loading input data...&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Warning: using defaut universe automatically provided by the clusterProfiler package&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done.&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Computing GO enrichment...&quot;</span>
<span class="n">`universe`</span><span class="w"> </span><span class="n">is</span><span class="w"> </span><span class="n">not</span><span class="w"> </span><span class="n">in</span><span class="w"> </span><span class="n">character</span><span class="w"> </span><span class="n">and</span><span class="w"> </span><span class="n">will</span><span class="w"> </span><span class="n">be</span><span class="w"> </span><span class="n">ignored...</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done.&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;18866 (default) background genes&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;422 provided genes; 310 found by `enrichGO`&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Computed GO enrichment (whether significant or not) for 3344 distinct GO terms&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Of those 3344 GO terms, 29 have a BH-adjusted p-val &lt; 0.05&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Writing outputs tables...&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done. Writing output images...&quot;</span>
<span class="kc">...</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done.&quot;</span>
</code></pre></div>
</p>
</blockquote>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/all_genes/symbol/default_universe.BP.0.1BH.dotplot.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/all_genes/symbol/default_universe.BP.0.1BH.gene-concept.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/all_genes/symbol/default_universe.BP.0.1BH.gene-concept.circular.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/all_genes/symbol/default_universe.BP.0.1BH.heatplot.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/all_genes/symbol/default_universe.BP.0.1BH.upset.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/all_genes/symbol/default_universe.BP.0.1BH.pvalues.png" /></p>





                
              </article>
            </div>
          
          
        </div>
        
          <a href="#" class="md-top md-icon" data-md-component="top" hidden>
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg>
            Back to top
          </a>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      2020 - 2022 Tristan Hoellinger
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
        <div class="md-social">
  
    
    
    
    
      
      
    
    <a href="https://twitter.com/TristanHoellin1" target="_blank" rel="noopener" title="twitter.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"/></svg>
    </a>
  
    
    
    
    
      
      
    
    <a href="https://fr.linkedin.com/in/tristan-hoellinger" target="_blank" rel="noopener" title="fr.linkedin.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"/></svg>
    </a>
  
    
    
    
    
    <a href="mailto:sarah.djebali@inserm.fr" target="_blank" rel="noopener" title="" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M498.1 5.6c10.1 7 15.4 19.1 13.5 31.2l-64 416c-1.5 9.7-7.4 18.2-16 23s-18.9 5.4-28 1.6L284 427.7l-68.5 74.1c-8.9 9.7-22.9 12.9-35.2 8.1S160 493.2 160 480v-83.6c0-4 1.5-7.8 4.2-10.7l167.6-182.9c5.8-6.3 5.6-16-.4-22s-15.7-6.4-22-.7L106 360.8l-88.3-44.2C7.1 311.3.3 300.7 0 288.9s5.9-22.8 16.1-28.7l448-256c10.7-6.1 23.9-5.5 34 1.4z"/></svg>
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    <script id="__config" type="application/json">{"base": "../../..", "features": ["navigation.instant", "navigation.tabs", "navigation.sections", "navigation.top"], "search": "../../../assets/javascripts/workers/search.db81ec45.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../../../assets/javascripts/bundle.6df46069.min.js"></script>
      
    
  </body>
</html>