
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
        <meta name="author" content="Tristan Hoellinger">
      
      
        <link rel="canonical" href="https://eg.github.io/data_and_results/haemochromatosis/CHiC/preprocess_data.html">
      
      
      
      <link rel="icon" href="../../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.4.2, mkdocs-material-9.0.11">
    
    
      
        <title>Explore and prepare data for further analysis of E-G network with genes involved in hemochromatosis, with E-G pairs based on CHiC data - Genome-wide identification of E/G interactions</title>
      
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/main.0d440cfe.min.css">
      
        
        <link rel="stylesheet" href="../../../assets/stylesheets/palette.2505c338.min.css">
      
      

    
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../../stylesheets/extra.css">
    
    <script>__md_scope=new URL("../../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="black" data-md-color-accent="">
  
    
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#explore-and-prepare-data-for-further-analysis-of-e-g-network-with-genes-involved-in-hemochromatosis-with-e-g-pairs-based-on-chic-data" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../../.." title="Genome-wide identification of E/G interactions" class="md-header__button md-logo" aria-label="Genome-wide identification of E/G interactions" data-md-component="logo">
      
  <img src="../../../logo/logo-inserm-blc.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Genome-wide identification of E/G interactions
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Explore and prepare data for further analysis of E-G network with genes involved in hemochromatosis, with E-G pairs based on CHiC data
            
          </span>
        </div>
      </div>
    </div>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/hoellin/eg" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
  </div>
  <div class="md-source__repository">
    hoellin/eg
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../../guidebooks/introduction.html" class="md-tabs__link">
        Guidebooks
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../../haemochromatosis/introduction.html" class="md-tabs__link">
        Haemochromatosis
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../../notes_BENGI/introduction.html" class="md-tabs__link">
        BENGI
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../../notes_ABC/introduction.html" class="md-tabs__link">
        ABC
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../../index.html" class="md-tabs__link">
        About
      </a>
    </li>
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../../.." title="Genome-wide identification of E/G interactions" class="md-nav__button md-logo" aria-label="Genome-wide identification of E/G interactions" data-md-component="logo">
      
  <img src="../../../logo/logo-inserm-blc.png" alt="logo">

    </a>
    Genome-wide identification of E/G interactions
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/hoellin/eg" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
  </div>
  <div class="md-source__repository">
    hoellin/eg
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1" >
      
      
      
        <label class="md-nav__link" for="__nav_1" id="__nav_1_label" tabindex="0">
          Guidebooks
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_1_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_1">
          <span class="md-nav__icon md-icon"></span>
          Guidebooks
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../guidebooks/introduction.html" class="md-nav__link">
        Introduction
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../guidebooks/script_bedpe_summary/script_bedpe_summary.html" class="md-nav__link">
        Compute summary stats of a bedpe
      </a>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_3" >
      
      
      
        <label class="md-nav__link" for="__nav_1_3" id="__nav_1_3_label" tabindex="0">
          Basics
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_1_3">
          <span class="md-nav__icon md-icon"></span>
          Basics
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../guidebooks/shell.html" class="md-nav__link">
        Shell
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../guidebooks/conda_environments.html" class="md-nav__link">
        Conda basics
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../guidebooks/git.html" class="md-nav__link">
        Git
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../guidebooks/genotoul.html" class="md-nav__link">
        Genotoul cluster
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../guidebooks/notebooks_ssh.html" class="md-nav__link">
        How to use Jupyter through ssh tunneling?
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_4" >
      
      
      
        <label class="md-nav__link" for="__nav_1_4" id="__nav_1_4_label" tabindex="0">
          Misc
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_4_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_1_4">
          <span class="md-nav__icon md-icon"></span>
          Misc
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../guidebooks/compute_correlations.html" class="md-nav__link">
        Compute correlations
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" >
      
      
      
        <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
          Haemochromatosis
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_2">
          <span class="md-nav__icon md-icon"></span>
          Haemochromatosis
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../haemochromatosis/introduction.html" class="md-nav__link">
        Introduction
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../haemochromatosis/data_format.html" class="md-nav__link">
        Data format
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../haemochromatosis/ABC/custom_go.html" class="md-nav__link">
        Custom script GO
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../haemochromatosis/main_results.html" class="md-nav__link">
        Main results
      </a>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2_5" >
      
      
      
        <label class="md-nav__link" for="__nav_2_5" id="__nav_2_5_label" tabindex="0">
          CHiC data
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_2_5_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_2_5">
          <span class="md-nav__icon md-icon"></span>
          CHiC data
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../haemochromatosis/CHiC/preprocess_data.html" class="md-nav__link">
        Preprocessing
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../haemochromatosis/CHiC/preliminary_analysis_chic_v2.html" class="md-nav__link">
        Networks
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../haemochromatosis/CHiC/go_enrichment.html" class="md-nav__link">
        GO enrichment
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2_6" >
      
      
      
        <label class="md-nav__link" for="__nav_2_6" id="__nav_2_6_label" tabindex="0">
          ABC data
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_2_6_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_2_6">
          <span class="md-nav__icon md-icon"></span>
          ABC data
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../haemochromatosis/ABC/data_processing.html" class="md-nav__link">
        Preprocessing
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../haemochromatosis/ABC/preliminary_analysis_v9.html" class="md-nav__link">
        Networks
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../haemochromatosis/ABC/summary_statistics_eg_pairs.html" class="md-nav__link">
        Summary statistics enhancers
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../haemochromatosis/ABC/go_enrichment.html" class="md-nav__link">
        GO enrichment
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../haemochromatosis/ABC/gorilla.html" class="md-nav__link">
        GOrilla
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../haemochromatosis/ABC/compute_avg_nb_of_connections_per_gene.html" class="md-nav__link">
        Avg nb connections per gene
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../haemochromatosis/ABC/gwas.html" class="md-nav__link">
        GWAS (13 initial genes)
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../haemochromatosis/ABC/gwas_10.html" class="md-nav__link">
        GWAS (10 initial genes)
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" >
      
      
      
        <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
          BENGI
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3">
          <span class="md-nav__icon md-icon"></span>
          BENGI
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_BENGI/introduction.html" class="md-nav__link">
        Introduction
      </a>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_2" >
      
      
      
        <label class="md-nav__link" for="__nav_3_2" id="__nav_3_2_label" tabindex="0">
          Notebooks
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_2_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_2">
          <span class="md-nav__icon md-icon"></span>
          Notebooks
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_BENGI/BENGI_to_custom_bedpe.html" class="md-nav__link">
        Convert BENGI datasets to custom bedpe
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_3" >
      
      
      
        <label class="md-nav__link" for="__nav_3_3" id="__nav_3_3_label" tabindex="0">
          Summary statistics
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_3">
          <span class="md-nav__icon md-icon"></span>
          Summary statistics
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_BENGI/summary_statistics/summary_statistics.html" class="md-nav__link">
        Summary statistics on BENGI GM12878 benchmarks
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_BENGI/summary_statistics_BENGI/summary_BENGI_bedpe.html" class="md-nav__link">
        (R code) Compute summary statistics over BENGI
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_4" >
      
      
      
        <label class="md-nav__link" for="__nav_3_4" id="__nav_3_4_label" tabindex="0">
          Evaluation of a few methods over BENGI GM12878
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_4_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_4">
          <span class="md-nav__icon md-icon"></span>
          Evaluation of a few methods over BENGI GM12878
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_4_1" >
      
      
      
        <label class="md-nav__link" for="__nav_3_4_1" id="__nav_3_4_1_label" tabindex="0">
          Distance method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_4_1_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_4_1">
          <span class="md-nav__icon md-icon"></span>
          Distance method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_BENGI/distance_method/distance_evaluation_with_code.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_BENGI/distance_method/summary_distance_evaluation.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_4_2" >
      
      
      
        <label class="md-nav__link" for="__nav_3_4_2" id="__nav_3_4_2_label" tabindex="0">
          DNase-expression correlation method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_4_2_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_4_2">
          <span class="md-nav__icon md-icon"></span>
          DNase-expression correlation method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_BENGI/dnase_expression_correlation/correlation_method_with_code.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_BENGI/dnase_expression_correlation/summary_correlation_method.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_4_3" >
      
      
      
        <label class="md-nav__link" for="__nav_3_4_3" id="__nav_3_4_3_label" tabindex="0">
          Average rank method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_4_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_4_3">
          <span class="md-nav__icon md-icon"></span>
          Average rank method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_BENGI/avg_rank_method/avg_rank_method_with_code.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_BENGI/avg_rank_method/summary_avg_rank_method.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_5" >
      
      
      
        <label class="md-nav__link" for="__nav_3_5" id="__nav_3_5_label" tabindex="0">
          Evaluation of a few methods over Fulco et al 2019 CRISPRi-FlowFISH
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_5_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_5">
          <span class="md-nav__icon md-icon"></span>
          Evaluation of a few methods over Fulco et al 2019 CRISPRi-FlowFISH
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_5_1" >
      
      
      
        <label class="md-nav__link" for="__nav_3_5_1" id="__nav_3_5_1_label" tabindex="0">
          Distance method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_5_1_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_5_1">
          <span class="md-nav__icon md-icon"></span>
          Distance method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_BENGI/CRISPRi_FlowFISH/distance_method/distance_over_fulco_et_al_crispri.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_BENGI/CRISPRi_FlowFISH/distance_method/summary_distance_over_fulco_et_al_crispri.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_5_2" >
      
      
      
        <label class="md-nav__link" for="__nav_3_5_2" id="__nav_3_5_2_label" tabindex="0">
          DNase-expression correlation method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_5_2_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_5_2">
          <span class="md-nav__icon md-icon"></span>
          DNase-expression correlation method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_BENGI/CRISPRi_FlowFISH/dnase_expression_correlation/correlation_method_with_code.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_BENGI/CRISPRi_FlowFISH/dnase_expression_correlation/summary_correlation_method.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_5_3" >
      
      
      
        <label class="md-nav__link" for="__nav_3_5_3" id="__nav_3_5_3_label" tabindex="0">
          Average rank method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_5_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_5_3">
          <span class="md-nav__icon md-icon"></span>
          Average rank method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_BENGI/CRISPRi_FlowFISH/avg_rank_method/avg_rank_method_with_code.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_BENGI/CRISPRi_FlowFISH/avg_rank_method/summary_avg_rank.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" >
      
      
      
        <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
          ABC
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_4">
          <span class="md-nav__icon md-icon"></span>
          ABC
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_ABC/introduction.html" class="md-nav__link">
        Introduction
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_ABC/BENGI/ABC_BENGI_results.html" class="md-nav__link">
        ABC over BENGI for GM12878, main results
      </a>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_3" >
      
      
      
        <label class="md-nav__link" for="__nav_4_3" id="__nav_4_3_label" tabindex="0">
          CRISPRi-FlowFISH
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_4_3">
          <span class="md-nav__icon md-icon"></span>
          CRISPRi-FlowFISH
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_ABC/K562/ABC_K562_CRISPRi_FlowFISH.html" class="md-nav__link">
        Original ABC predictions over K562
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_ABC/K562/recap_investigations_ABC_over_CRiFF.html" class="md-nav__link">
        Summary ABC over CRiFF
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_4" >
      
      
      
        <label class="md-nav__link" for="__nav_4_4" id="__nav_4_4_label" tabindex="0">
          Notebooks
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_4_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_4_4">
          <span class="md-nav__icon md-icon"></span>
          Notebooks
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_ABC/K562/april_K562_56_genes/april_K562_56_genes.html" class="md-nav__link">
        ABC over K562 cCRiFF
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_ABC/K562/april_K562_59_genes/april_K562_59_genes.html" class="md-nav__link">
        ABC over K562 CRiFF
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_ABC/K562/april_K562_candidates_not_in_whitelist_56_genes/april_K562_candidates_NOT_in_whitelist.html" class="md-nav__link">
        ABC over K562 cCRiFF, no whitelist
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_ABC/generic_notebooks/turnkey_notebook_to_run_ABC_with_example_over_GM12878.html" class="md-nav__link">
        Generic notebook to run ABC method & Example over GM12878
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_ABC/BENGI/notebook_ABC_over_BENGI_GM12878.html" class="md-nav__link">
        ABC over GM12878 with same accessions as for BENGI
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_ABC/BENGI/notebook_ABC_over_BENGI_GM12878_from_ccRE_ELSs.html" class="md-nav__link">
        ABC over GM12878 with same accessions as for BENGI, starting with ccRE-dELS
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_5" >
      
      
      
        <label class="md-nav__link" for="__nav_4_5" id="__nav_4_5_label" tabindex="0">
          Full codes
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_5_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_4_5">
          <span class="md-nav__icon md-icon"></span>
          Full codes
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_ABC/test_abc_pipeline.html" class="md-nav__link">
        Test of ABC pipeline over chr22
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_ABC/BENGI/ABC_over_BENGI_GM12878.html" class="md-nav__link">
        ABC over GM12878 with same accessions as for BENGI
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../notes_ABC/differences_example_chr22.html" class="md-nav__link">
        Meld of results over chr22 vs authors' results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" >
      
      
      
        <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
          About
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_5">
          <span class="md-nav__icon md-icon"></span>
          About
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../index.html" class="md-nav__link">
        Getting started
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../about_me.html" class="md-nav__link">
        About me
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#requirements" class="md-nav__link">
    Requirements
  </a>
  
    <nav class="md-nav" aria-label="Requirements">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#data-availability" class="md-nav__link">
    Data availability
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#bedtools" class="md-nav__link">
    bedtools
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#summary-statistics-with-r" class="md-nav__link">
    Summary statistics with R
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#extracting-data-of-interest" class="md-nav__link">
    Extracting data of interest
  </a>
  
    <nav class="md-nav" aria-label="Extracting data of interest">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#exploration" class="md-nav__link">
    Exploration
  </a>
  
    <nav class="md-nav" aria-label="Exploration">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#are-there-self-promoters-if-so-we-have-to-remove-them-for-comparison-purpose-with-abc-data" class="md-nav__link">
    Are there self-promoters ? If so, we have to remove them for comparison purpose with ABC data
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#main-filters" class="md-nav__link">
    Main filters
  </a>
  
    <nav class="md-nav" aria-label="Main filters">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#remove-entries-for-which-no-gene-tss-500-bp-overlaps-the-promoter" class="md-nav__link">
    Remove entries for which no gene TSS+-500 bp overlaps the promoter
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#deal-with-pp-type-of-relation" class="md-nav__link">
    Deal with pp type of relation
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#duplicates-entries-for-which-the-e-prom-pair-is-associated-to-multiple-e-g-pairs-and-modify-file-structure" class="md-nav__link">
    Duplicates entries for which the E-Prom pair is associated to multiple E-G pairs, and modify file structure
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#extract-enhancers" class="md-nav__link">
    Extract enhancers
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#overlap-between-enhancers-and-ccre-els" class="md-nav__link">
    Overlap between enhancers and ccRE-ELS
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#overlap-between-enhancers-and-all-ccre" class="md-nav__link">
    Overlap between enhancers and all ccRE
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#intersect-enhancers-with-nasser2021-enhancers" class="md-nav__link">
    Intersect enhancers with Nasser2021 enhancers
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#keep-only-putative-enhancers-that-overlap-at-least-one-ccre-els" class="md-nav__link">
    Keep only putative enhancers that overlap at least one ccRE-ELS
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  

  
  



<h1 id="explore-and-prepare-data-for-further-analysis-of-e-g-network-with-genes-involved-in-hemochromatosis-with-e-g-pairs-based-on-chic-data">Explore and prepare data for further analysis of E-G network with genes involved in hemochromatosis, with E-G pairs based on CHiC data<a class="headerlink" href="#explore-and-prepare-data-for-further-analysis-of-e-g-network-with-genes-involved-in-hemochromatosis-with-e-g-pairs-based-on-chic-data" title="Permanent link">&para;</a></h1>
<h2 id="requirements">Requirements<a class="headerlink" href="#requirements" title="Permanent link">&para;</a></h2>
<h3 id="data-availability">Data availability<a class="headerlink" href="#data-availability" title="Permanent link">&para;</a></h3>
<p>All data can be found here:</p>
<p><code>/work2/project/regenet/results/phic/homo_sapiens/hg19/jung.ren.2019/LI11/score2/pall/LI11.pall.score2.gninfo.bedpe</code></p>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span><span class="w"> </span>/work2/project/regenet/results/phic/homo_sapiens/hg19/jung.ren.2019/LI11/score2/pall/
mkdir<span class="w"> </span>networks<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span><span class="nb">cd</span><span class="w"> </span>networks
ln<span class="w"> </span>-s<span class="w"> </span>../LI11.pall.score2.gninfo.bedpe<span class="w"> </span>raw_data_liver.full.bedpe
</code></pre></div>
<h3 id="bedtools"><code>bedtools</code><a class="headerlink" href="#bedtools" title="Permanent link">&para;</a></h3>
<p>One can load the appropriate modules with:</p>
<div class="highlight"><pre><span></span><code>conda<span class="w"> </span>activate<span class="w"> </span>base<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>module<span class="w"> </span>load<span class="w"> </span>bioinfo/bedtools-2.27.1<span class="w">  </span>
</code></pre></div>
<h3 id="summary-statistics-with-r"><code>Summary statistics with R</code><a class="headerlink" href="#summary-statistics-with-r" title="Permanent link">&para;</a></h3>
<p>We need to load a module containing R to be able to launch the <code>compute_summary_stats_enhancer_list.R</code> script:</p>
<div class="highlight"><pre><span></span><code>module<span class="w"> </span>load<span class="w"> </span>system/R-4.1.1_gcc-9.3.0
</code></pre></div>
<div class="highlight"><pre><span></span><code>cp<span class="w"> </span>/work2/project/regenet/results/multi/abc.model/Nasser2021/compute_summary_stats_enhancer_list.R<span class="w"> </span>.
</code></pre></div>
<p>We can execute the R script with:</p>
<div class="highlight"><pre><span></span><code>Rscript<span class="w"> </span>compute_summary_stats_enhancer_list.R<span class="w"> </span>&lt;relative_path_to_enhancers_list&gt;
</code></pre></div>
<h2 id="extracting-data-of-interest">Extracting data of interest<a class="headerlink" href="#extracting-data-of-interest" title="Permanent link">&para;</a></h2>
<p>Note that the <code>LI11</code> repository already corresponds to putative E-G pairs in <code>liver</code> cells. The list of all available cell types can be found here; <code>/work2/project/regenet/results/phic/homo_sapiens/hg19/jung.ren.2019/metadata.tsv</code>.</p>
<p>There are 12 columns:</p>
<blockquote>
<div class="highlight"><pre><span></span><code>head<span class="w"> </span>-n<span class="w"> </span><span class="m">1</span><span class="w"> </span>raw_data_liver.full.bedpe
</code></pre></div>
<div class="highlight"><pre><span></span><code>chr1<span class="w">  </span><span class="m">943049</span><span class="w">  </span><span class="m">965801</span><span class="w">  </span>chr1<span class="w">    </span><span class="m">1183838</span><span class="w"> </span><span class="m">1209657</span><span class="w"> </span>chr1:943049:965801,chr1:1183838:1209657<span class="w"> </span><span class="m">2</span>.76719343781528..<span class="w">  </span>pp<span class="w">  </span><span class="m">11837</span><span class="w">   </span><span class="m">22511</span><span class="w">   </span><span class="m">7</span><span class="w">   </span><span class="m">242322</span><span class="w">  </span><span class="m">1</span>.9015<span class="w">  </span><span class="m">3</span>.6813<span class="w">  </span><span class="m">1</span>.1317<span class="w">  </span><span class="m">2</span>.1959<span class="w">  </span><span class="m">2</span><span class="w">   </span>chr1:948802:949920:+:ENSG00000187608.5:ISG15,chr1:955502:991496:+:ENSG00000188157.9:AGRN,<span class="w">   </span><span class="m">1</span><span class="w">   </span>chr1:1189288:1209265:-:ENSG00000160087.16:UBE2J2,<span class="w">   </span>elt2B<span class="w">   </span><span class="m">218037</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code><span class="m">1</span><span class="w"> </span>chr1<span class="w"> </span><span class="c1"># promoter of the gene</span>
<span class="m">2</span><span class="w"> </span>start1<span class="w"> </span><span class="c1"># 0-based</span>
<span class="m">3</span><span class="w"> </span>end1<span class="w"> </span><span class="c1"># 1-based</span>
<span class="m">4</span><span class="w"> </span>chr2<span class="w"> </span><span class="c1"># elt2 (putative enhancer)</span>
<span class="m">5</span><span class="w"> </span>start2<span class="w"> </span><span class="c1"># 0-based</span>
<span class="m">6</span><span class="w"> </span>end2<span class="w"> </span><span class="c1"># 1-based</span>
<span class="m">7</span><span class="w"> </span>short<span class="w"> </span>name<span class="w"> </span>&lt;chr1:start1:end1,<span class="w"> </span>chr2:start2:end2&gt;<span class="w"> </span><span class="c1"># &lt;prom_coord, elt2_coord&gt;</span>
<span class="m">8</span><span class="w"> </span>strand1<span class="w"> </span><span class="c1"># promoter</span>
<span class="m">9</span><span class="w"> </span>strand2<span class="w"> </span><span class="c1"># elt2</span>
<span class="m">10</span><span class="w"> </span>score<span class="w"> </span><span class="o">(</span>does<span class="w"> </span>not<span class="w"> </span>matter<span class="w"> </span>because<span class="w"> </span>all<span class="w"> </span>p-values<span class="w"> </span>are<span class="w"> </span>small<span class="w"> </span>enough<span class="w"> </span><span class="k">in</span><span class="w"> </span>those<span class="w"> </span>files<span class="o">)</span>
<span class="m">11</span><span class="w"> </span><span class="nb">type</span><span class="w"> </span>of<span class="w"> </span>relation<span class="w"> </span><span class="o">(</span>pp<span class="w"> </span>or<span class="w"> </span>po<span class="o">)</span><span class="w">  </span><span class="c1"># I think that</span>
<span class="w">                              </span><span class="c1"># pp stands for promoter-promoter</span>
<span class="w">                              </span><span class="c1"># po stands for promoter-other</span>
<span class="m">12</span>-19<span class="w"> </span>does<span class="w"> </span>not<span class="w"> </span>matter<span class="w"> </span><span class="c1"># cf mail &quot;sens des champs des *gninfo.bedpe.gz&quot; for details</span>
<span class="m">20</span><span class="w"> </span>number<span class="w"> </span>of<span class="w"> </span>genes<span class="w"> </span>whose<span class="w"> </span>tss+-500bp<span class="w"> </span>overlaps<span class="w"> </span>the<span class="w"> </span>prom<span class="w"> </span>parts
<span class="m">21</span><span class="w"> </span>list<span class="w"> </span>of<span class="w"> </span>such<span class="w"> </span>genes<span class="w"> </span><span class="k">in</span><span class="w"> </span>the<span class="w"> </span>form<span class="w"> </span>&lt;chr1:start1:end1:strand1:gene_id:gene_symbol,chr2:...,...&gt;
<span class="m">22</span><span class="w"> </span>number<span class="w"> </span>of<span class="w"> </span>genes<span class="w"> </span>whose<span class="w"> </span>tss+-500bp<span class="w"> </span>overlaps<span class="w"> </span>the<span class="w"> </span>elt2<span class="w"> </span>part
<span class="m">23</span><span class="w"> </span>the<span class="w"> </span>list<span class="w"> </span>of<span class="w"> </span>such<span class="w"> </span>genes<span class="w"> </span><span class="k">in</span><span class="w"> </span>the<span class="w"> </span>same<span class="w"> </span>form<span class="w"> </span>as<span class="w"> </span><span class="k">for</span><span class="w"> </span>prom<span class="w"> </span><span class="o">(</span><span class="m">21</span><span class="o">)</span>
<span class="m">24</span><span class="w"> </span><span class="nb">type</span><span class="w"> </span>of<span class="w"> </span>elt2<span class="w"> </span><span class="c1"># two possible values:</span>
<span class="w">              </span><span class="c1"># elt2A if the element 2 is never a promoter anywhere else in the file</span>
<span class="w">              </span><span class="c1"># elt2B otherwise, that is if the element 2 also appears as a promoter somewhere else in the file</span>
<span class="m">25</span><span class="w"> </span>interfragment<span class="w"> </span>distance<span class="w"> </span><span class="o">(</span>from<span class="w"> </span>end<span class="w"> </span>to<span class="w"> </span>beg<span class="o">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;{types[$24]++} END{for(u in types){print u, types[u]}}&#39;</span><span class="w"> </span>raw_data_liver.full.bedpe<span class="w"> </span><span class="p">|</span>sort<span class="w"> </span>-nrk2,2
</code></pre></div>
<div class="highlight"><pre><span></span><code>elt2A<span class="w"> </span><span class="m">33553</span>
elt2B<span class="w"> </span><span class="m">4706</span>
</code></pre></div>
</blockquote>
<h3 id="exploration">Exploration<a class="headerlink" href="#exploration" title="Permanent link">&para;</a></h3>
<h4 id="are-there-self-promoters-if-so-we-have-to-remove-them-for-comparison-purpose-with-abc-data">Are there self-promoters ? If so, we have to remove them for comparison purpose with ABC data<a class="headerlink" href="#are-there-self-promoters-if-so-we-have-to-remove-them-for-comparison-purpose-with-abc-data" title="Permanent link">&para;</a></h4>
<blockquote>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;$24~/(^elt2B$)/ {if($1&quot;:&quot;$2&quot;:&quot;$3==$4&quot;:&quot;$5&quot;:&quot;$6){print $0}}&#39;</span><span class="w"> </span>raw_data_liver.bedpe<span class="w"> </span><span class="p">|</span>wc<span class="w"> </span>-l
</code></pre></div>
<div class="highlight"><pre><span></span><code>0
</code></pre></div>
</blockquote>
<p>=&gt; No. OK.</p>
<h3 id="main-filters">Main filters<a class="headerlink" href="#main-filters" title="Permanent link">&para;</a></h3>
<h4 id="remove-entries-for-which-no-gene-tss-500-bp-overlaps-the-promoter">Remove entries for which no gene TSS+-500 bp overlaps the promoter<a class="headerlink" href="#remove-entries-for-which-no-gene-tss-500-bp-overlaps-the-promoter" title="Permanent link">&para;</a></h4>
<p>This is the case for 46 entires:</p>
<blockquote>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;{nb[$20]++} END{for(u in nb){print u, nb[u]}}&#39;</span><span class="w"> </span>raw_data_liver.full.bedpe<span class="w"> </span><span class="p">|</span>sort<span class="w"> </span>-nrk2,2
</code></pre></div>
<div class="highlight"><pre><span></span><code><span class="m">1</span><span class="w"> </span><span class="m">30065</span>
<span class="m">2</span><span class="w"> </span><span class="m">6672</span>
<span class="m">3</span><span class="w"> </span><span class="m">1114</span>
<span class="m">4</span><span class="w"> </span><span class="m">286</span>
<span class="m">0</span><span class="w"> </span><span class="m">46</span>
<span class="m">5</span><span class="w"> </span><span class="m">39</span>
<span class="m">6</span><span class="w"> </span><span class="m">30</span>
<span class="m">7</span><span class="w"> </span><span class="m">7</span>
</code></pre></div>
</blockquote>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;$20&gt;0&#39;</span><span class="w"> </span>raw_data_liver.full.bedpe<span class="w"> </span>&gt;<span class="w"> </span>temp.bedpe
mv<span class="w"> </span>temp.bedpe<span class="w"> </span>raw_data_liver.full.bedpe
</code></pre></div>
<blockquote>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;{nb[$20]++} END{for(u in nb){print u, nb[u]}}&#39;</span><span class="w"> </span>raw_data_liver.full.bedpe<span class="w"> </span><span class="p">|</span>sort<span class="w"> </span>-nrk2,2
</code></pre></div>
<div class="highlight"><pre><span></span><code><span class="m">1</span><span class="w"> </span><span class="m">30065</span>
<span class="m">2</span><span class="w"> </span><span class="m">6672</span>
<span class="m">3</span><span class="w"> </span><span class="m">1114</span>
<span class="m">4</span><span class="w"> </span><span class="m">286</span>
<span class="m">5</span><span class="w"> </span><span class="m">39</span>
<span class="m">6</span><span class="w"> </span><span class="m">30</span>
<span class="m">7</span><span class="w"> </span><span class="m">7</span>
</code></pre></div>
</blockquote>
<h4 id="deal-with-pp-type-of-relation">Deal with <code>pp</code> type of relation<a class="headerlink" href="#deal-with-pp-type-of-relation" title="Permanent link">&para;</a></h4>
<p>Well there should be no problem keeping them:</p>
<blockquote>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;{type[$11]++} END{for(u in type){print u, type[u]}}&#39;</span><span class="w"> </span>raw_data_liver.full.bedpe<span class="w"> </span><span class="p">|</span>sort<span class="w"> </span>-nrk2,2
</code></pre></div>
<div class="highlight"><pre><span></span><code>po<span class="w"> </span><span class="m">33506</span>
pp<span class="w"> </span><span class="m">4707</span>
</code></pre></div>
</blockquote>
<h4 id="duplicates-entries-for-which-the-e-prom-pair-is-associated-to-multiple-e-g-pairs-and-modify-file-structure">Duplicates entries for which the E-Prom pair is associated to multiple E-G pairs, and modify file structure<a class="headerlink" href="#duplicates-entries-for-which-the-e-prom-pair-is-associated-to-multiple-e-g-pairs-and-modify-file-structure" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;BEGIN{OFS=&quot;\t&quot;} {split($21,genes,&quot;,&quot;); i=0; n=length(genes)-1; while(i&lt;n){i=i+1; gene=genes[i]; split(gene,parts,&quot;:&quot;); entry[$4&quot;\t&quot;$5&quot;\t&quot;$6&quot;\t&quot;parts[1]&quot;\t&quot;parts[2]&quot;\t&quot;parts[3]&quot;\t&quot;parts[6]&quot;\t&quot;parts[5]]=$4&quot;\t&quot;$5&quot;\t&quot;$6&quot;\t&quot;parts[1]&quot;\t&quot;parts[2]&quot;\t&quot;parts[3]&quot;\t&quot;$4&quot;:&quot;$5&quot;-&quot;$6&quot;::&quot;parts[5]&quot;::&quot;parts[6]&quot;\t&quot;$8&quot;\t&quot;$10&quot;\t&quot;$9&quot;\tliver\t&quot;parts[6]&quot;\t&quot;$25&quot;\t&quot;parts[5]}} END{for(u in entry){print entry[u]}}&#39;</span><span class="w"> </span>raw_data_liver.full.bedpe<span class="w"> </span>&gt;<span class="w"> </span>liver.all_putative_enhancers.bedpe
</code></pre></div>
<p>OK.</p>
<p><strong>Warning</strong>:</p>
<ul>
<li>in <code>raw_data_liver.full.bedpe</code>, element 1 is the promoter and element 2 is the putative enhancer</li>
<li>in <code>liver.all_putative_enhancers.bedpe</code>, element 1 is the putative enhancer and element 2 is the gene</li>
</ul>
<p>Also, let us note that there are some duplicates, ie in a few cases (279), multiple entries in the original raw file, correspond to the same "id" (coordinates of the putative enhancer + coordinates, id and name of the gene) in the new file:</p>
<blockquote>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;BEGIN{OFS=&quot;\t&quot;} {split($21,genes,&quot;,&quot;); i=0; n=length(genes)-1; while(i&lt;n){i=i+1; gene=genes[i]; split(gene,parts,&quot;:&quot;); entry[$4&quot;:&quot;$5&quot;:&quot;$6&quot;::&quot;parts[1]&quot;:&quot;parts[2]&quot;:&quot;parts[3]&quot;::&quot;parts[6]&quot;::&quot;parts[5]]++}} END{for(u in entry){if(entry[u]&gt;1){print u, entry[u]}}}&#39;</span><span class="w"> </span>raw_data_liver.full.bedpe<span class="w"> </span><span class="p">|</span>sort<span class="w"> </span>-nrk2,2<span class="w"> </span><span class="p">|</span>wc<span class="w"> </span>-l
</code></pre></div>
<p>279</p>
</blockquote>
<p>More precisely, there are 2 new entries that both match 3 old entries, and 277 new entries matching 2 old entries each.</p>
<p>The reason must be (I did not verify though, but this is the only explanation I see) that for some given putative enhancer, let us say for E, the are multiple promoters in the original file, let's say P1 and P2, each one overlapping the same TSS+- 500 bp of a gene G.</p>
<p>Well, for now we don't care.</p>
<div class="highlight"><pre><span></span><code>cp<span class="w"> </span>/work2/project/regenet/results/multi/abc.model/Nasser2021/chr_sizes<span class="w"> </span>chr_sizes
</code></pre></div>
<div class="highlight"><pre><span></span><code>bedtools<span class="w"> </span>sort<span class="w"> </span>-faidx<span class="w"> </span>chr_sizes<span class="w"> </span>-i<span class="w"> </span>liver.all_putative_enhancers.bedpe<span class="w"> </span>&gt;<span class="w"> </span>liver.all_putative_enhancers.sorted.bedpe
</code></pre></div>
<h4 id="extract-enhancers">Extract enhancers<a class="headerlink" href="#extract-enhancers" title="Permanent link">&para;</a></h4>
<p>Now we extract the list of all enhancers in our liver biosample:</p>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span><span class="s1">&#39;BEGIN{FS=&quot;\t&quot;; OFS=&quot;\t&quot;} {print $1, $2, $3}&#39;</span><span class="w"> </span>liver.all_putative_enhancers.sorted.bedpe<span class="w"> </span><span class="p">|</span>uniq<span class="w"> </span>&gt;<span class="w"> </span>list_all_enhancers.bed
</code></pre></div>
<p>OK. Contains 31,749 enhancers.</p>
<p>And we define the list of merged enhancers, such that none are overlapping:</p>
<div class="highlight"><pre><span></span><code>bedtools<span class="w"> </span>merge<span class="w"> </span>-i<span class="w"> </span>list_all_enhancers.bed<span class="w"> </span>&gt;<span class="w"> </span>list_all_enhancers.merged_liver.bed
</code></pre></div>
<p>It did not change anything, so there was no overlap, which is expected as we are focused on only 1 biosample.</p>
<div class="highlight"><pre><span></span><code>rm<span class="w"> </span>list_all_enhancers.merged_liver.bed
</code></pre></div>
<p>Here are the summary statistics on those enhancers:</p>
<div class="highlight"><pre><span></span><code><span class="w">   </span><span class="n">Min.</span><span class="w"> </span><span class="m">1</span><span class="n">st</span><span class="w"> </span><span class="n">Qu.</span><span class="w">  </span><span class="n">Median</span><span class="w">    </span><span class="n">Mean</span><span class="w"> </span><span class="m">3</span><span class="n">rd</span><span class="w"> </span><span class="n">Qu.</span><span class="w">    </span><span class="n">Max.</span><span class="w"> </span>
<span class="w">   </span><span class="m">1111</span><span class="w">    </span><span class="m">4247</span><span class="w">    </span><span class="m">5596</span><span class="w">    </span><span class="m">7007</span><span class="w">    </span><span class="m">7756</span><span class="w"> </span><span class="m">1553733</span>
</code></pre></div>
<p><strong>Note that they are pretty large!</strong></p>
<h4 id="overlap-between-enhancers-and-ccre-els">Overlap between enhancers and ccRE-ELS<a class="headerlink" href="#overlap-between-enhancers-and-ccre-els" title="Permanent link">&para;</a></h4>
<p>We take the ccRE from <code>/work2/project/regenet/results/multi/bengi/Annotations/hg19-cCREs.bed</code>.</p>
<div class="highlight"><pre><span></span><code>ln<span class="w"> </span>-s<span class="w"> </span>/work2/project/regenet/results/multi/bengi/Annotations/hg19-cCREs.bed<span class="w"> </span>.
</code></pre></div>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;$6~/(^Enhancer-like$)/&#39;</span><span class="w"> </span>hg19-cCREs.bed<span class="w"> </span>&gt;<span class="w"> </span>ccRE-ELS.bed
</code></pre></div>
<p>989,712 ccRE-ELS</p>
<div class="highlight"><pre><span></span><code><span class="w">   </span><span class="n">Min.</span><span class="w"> </span><span class="m">1</span><span class="n">st</span><span class="w"> </span><span class="n">Qu.</span><span class="w">  </span><span class="n">Median</span><span class="w">    </span><span class="n">Mean</span><span class="w"> </span><span class="m">3</span><span class="n">rd</span><span class="w"> </span><span class="n">Qu.</span><span class="w">    </span><span class="n">Max.</span><span class="w"> </span>
<span class="w">   </span><span class="m">50.0</span><span class="w">   </span><span class="m">247.0</span><span class="w">   </span><span class="m">352.0</span><span class="w">   </span><span class="m">423.4</span><span class="w">   </span><span class="m">519.0</span><span class="w"> </span><span class="m">16633.0</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>bedtools<span class="w"> </span>intersect<span class="w"> </span>-c<span class="w"> </span>-a<span class="w"> </span>list_all_enhancers.bed<span class="w"> </span>-b<span class="w"> </span>ccRE-ELS.bed<span class="w"> </span><span class="p">|</span>awk<span class="w"> </span><span class="s1">&#39;BEGIN{FS=&quot;\t&quot;; OFS=&quot;\t&quot;} {nbs[$4]++} END{for(u in nbs){print u, nbs[u]}}&#39;</span><span class="w"> </span><span class="p">|</span>sort<span class="w"> </span>-nrk2,2<span class="w">   </span>
</code></pre></div>
<blockquote>
<div class="highlight"><pre><span></span><code><span class="m">0</span><span class="w"> </span><span class="m">6149</span>
<span class="m">1</span><span class="w"> </span><span class="m">4355</span>
<span class="m">2</span><span class="w"> </span><span class="m">4238</span>
<span class="m">3</span><span class="w"> </span><span class="m">3869</span>
<span class="m">4</span><span class="w"> </span><span class="m">3302</span>
<span class="m">5</span><span class="w"> </span><span class="m">2650</span>
<span class="m">6</span><span class="w"> </span><span class="m">1975</span>
<span class="m">7</span><span class="w"> </span><span class="m">1402</span>
<span class="m">8</span><span class="w"> </span><span class="m">1040</span>
<span class="m">9</span><span class="w"> </span><span class="m">744</span>
<span class="m">10</span><span class="w">    </span><span class="m">568</span>
<span class="m">11</span><span class="w">    </span><span class="m">363</span>
<span class="m">12</span><span class="w">    </span><span class="m">247</span>
<span class="m">13</span><span class="w">    </span><span class="m">176</span>
<span class="m">14</span><span class="w">    </span><span class="m">148</span>
<span class="m">15</span><span class="w">    </span><span class="m">105</span>
<span class="m">16</span><span class="w">    </span><span class="m">90</span>
<span class="m">17</span><span class="w">    </span><span class="m">73</span>
<span class="m">18</span><span class="w">    </span><span class="m">48</span>
<span class="m">19</span><span class="w">    </span><span class="m">34</span>
<span class="m">20</span><span class="w">    </span><span class="m">31</span>
<span class="m">22</span><span class="w">    </span><span class="m">19</span>
<span class="m">23</span><span class="w">    </span><span class="m">18</span>
<span class="m">21</span><span class="w">    </span><span class="m">18</span>
<span class="m">25</span><span class="w">    </span><span class="m">13</span>
<span class="m">24</span><span class="w">    </span><span class="m">13</span>
<span class="m">26</span><span class="w">    </span><span class="m">9</span>
<span class="m">34</span><span class="w">    </span><span class="m">7</span>
<span class="m">30</span><span class="w">    </span><span class="m">7</span>
<span class="m">28</span><span class="w">    </span><span class="m">7</span>
<span class="m">27</span><span class="w">    </span><span class="m">5</span>
<span class="m">32</span><span class="w">    </span><span class="m">4</span>
<span class="m">29</span><span class="w">    </span><span class="m">4</span>
<span class="m">33</span><span class="w">    </span><span class="m">3</span>
<span class="m">49</span><span class="w">    </span><span class="m">2</span>
<span class="m">39</span><span class="w">    </span><span class="m">2</span>
<span class="m">37</span><span class="w">    </span><span class="m">2</span>
<span class="m">36</span><span class="w">    </span><span class="m">2</span>
<span class="m">31</span><span class="w">    </span><span class="m">2</span>
<span class="m">51</span><span class="w">    </span><span class="m">1</span>
<span class="m">42</span><span class="w">    </span><span class="m">1</span>
<span class="m">41</span><span class="w">    </span><span class="m">1</span>
<span class="m">40</span><span class="w">    </span><span class="m">1</span>
<span class="m">35</span><span class="w">    </span><span class="m">1</span>
</code></pre></div>
</blockquote>
<p>6,149 (19%) of putative enhancers, do not match any ccRE-ELS. Only 4,355 (14%) of putative enhancers match exactly one ccRE-ELS. </p>
<h4 id="overlap-between-enhancers-and-all-ccre">Overlap between enhancers and all ccRE<a class="headerlink" href="#overlap-between-enhancers-and-all-ccre" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code>bedtools<span class="w"> </span>intersect<span class="w"> </span>-c<span class="w"> </span>-a<span class="w"> </span>list_all_enhancers.bed<span class="w"> </span>-b<span class="w"> </span>hg19-cCREs.bed<span class="w"> </span><span class="p">|</span>awk<span class="w"> </span><span class="s1">&#39;BEGIN{FS=&quot;\t&quot;; OFS=&quot;\t&quot;} {nbs[$4]++} END{for(u in nbs){print u, nbs[u]}}&#39;</span><span class="w"> </span><span class="p">|</span>sort<span class="w"> </span>-nrk2,2
</code></pre></div>
<blockquote>
<div class="highlight"><pre><span></span><code><span class="m">3</span><span class="w"> </span><span class="m">3962</span>
<span class="m">2</span><span class="w"> </span><span class="m">3805</span>
<span class="m">4</span><span class="w"> </span><span class="m">3629</span>
<span class="m">1</span><span class="w"> </span><span class="m">3524</span>
<span class="m">0</span><span class="w"> </span><span class="m">3184</span>
<span class="m">5</span><span class="w"> </span><span class="m">3010</span>
<span class="m">6</span><span class="w"> </span><span class="m">2443</span>
<span class="m">7</span><span class="w"> </span><span class="m">1757</span>
<span class="m">8</span><span class="w"> </span><span class="m">1397</span>
<span class="m">9</span><span class="w"> </span><span class="m">998</span>
<span class="m">10</span><span class="w">    </span><span class="m">785</span>
<span class="m">11</span><span class="w">    </span><span class="m">589</span>
<span class="m">12</span><span class="w">    </span><span class="m">435</span>
<span class="m">13</span><span class="w">    </span><span class="m">351</span>
<span class="m">14</span><span class="w">    </span><span class="m">292</span>
<span class="m">15</span><span class="w">    </span><span class="m">237</span>
<span class="m">16</span><span class="w">    </span><span class="m">198</span>
<span class="m">17</span><span class="w">    </span><span class="m">162</span>
<span class="m">19</span><span class="w">    </span><span class="m">131</span>
<span class="m">18</span><span class="w">    </span><span class="m">120</span>
<span class="m">20</span><span class="w">    </span><span class="m">93</span>
<span class="m">21</span><span class="w">    </span><span class="m">76</span>
<span class="m">22</span><span class="w">    </span><span class="m">75</span>
<span class="m">24</span><span class="w">    </span><span class="m">63</span>
<span class="m">23</span><span class="w">    </span><span class="m">56</span>
<span class="m">26</span><span class="w">    </span><span class="m">36</span>
<span class="m">28</span><span class="w">    </span><span class="m">34</span>
<span class="m">25</span><span class="w">    </span><span class="m">34</span>
<span class="m">29</span><span class="w">    </span><span class="m">32</span>
<span class="m">27</span><span class="w">    </span><span class="m">25</span>
<span class="m">30</span><span class="w">    </span><span class="m">24</span>
<span class="m">33</span><span class="w">    </span><span class="m">22</span>
<span class="m">31</span><span class="w">    </span><span class="m">22</span>
<span class="m">32</span><span class="w">    </span><span class="m">16</span>
<span class="m">36</span><span class="w">    </span><span class="m">13</span>
<span class="m">35</span><span class="w">    </span><span class="m">13</span>
<span class="m">38</span><span class="w">    </span><span class="m">12</span>
<span class="m">37</span><span class="w">    </span><span class="m">12</span>
<span class="m">34</span><span class="w">    </span><span class="m">12</span>
<span class="m">41</span><span class="w">    </span><span class="m">11</span>
<span class="m">40</span><span class="w">    </span><span class="m">6</span>
<span class="m">39</span><span class="w">    </span><span class="m">5</span>
<span class="m">53</span><span class="w">    </span><span class="m">4</span>
<span class="m">52</span><span class="w">    </span><span class="m">4</span>
<span class="m">51</span><span class="w">    </span><span class="m">4</span>
<span class="m">45</span><span class="w">    </span><span class="m">4</span>
<span class="m">42</span><span class="w">    </span><span class="m">4</span>
<span class="m">50</span><span class="w">    </span><span class="m">3</span>
<span class="m">49</span><span class="w">    </span><span class="m">3</span>
<span class="m">44</span><span class="w">    </span><span class="m">3</span>
<span class="m">43</span><span class="w">    </span><span class="m">3</span>
<span class="m">48</span><span class="w">    </span><span class="m">2</span>
<span class="m">46</span><span class="w">    </span><span class="m">2</span>
<span class="m">79</span><span class="w">    </span><span class="m">1</span>
<span class="m">76</span><span class="w">    </span><span class="m">1</span>
<span class="m">74</span><span class="w">    </span><span class="m">1</span>
<span class="m">69</span><span class="w">    </span><span class="m">1</span>
<span class="m">64</span><span class="w">    </span><span class="m">1</span>
<span class="m">61</span><span class="w">    </span><span class="m">1</span>
<span class="m">60</span><span class="w">    </span><span class="m">1</span>
<span class="m">58</span><span class="w">    </span><span class="m">1</span>
<span class="m">57</span><span class="w">    </span><span class="m">1</span>
<span class="m">55</span><span class="w">    </span><span class="m">1</span>
<span class="m">54</span><span class="w">    </span><span class="m">1</span>
<span class="m">47</span><span class="w">    </span><span class="m">1</span>
</code></pre></div>
</blockquote>
<p>3,184 (10%) of all putative enhancers, do not match any ccRE.</p>
<h4 id="intersect-enhancers-with-nasser2021-enhancers">Intersect enhancers with Nasser2021 enhancers<a class="headerlink" href="#intersect-enhancers-with-nasser2021-enhancers" title="Permanent link">&para;</a></h4>
<p>TODO. Not essential.</p>
<h4 id="keep-only-putative-enhancers-that-overlap-at-least-one-ccre-els">Keep only putative enhancers that overlap at least one ccRE-ELS<a class="headerlink" href="#keep-only-putative-enhancers-that-overlap-at-least-one-ccre-els" title="Permanent link">&para;</a></h4>
<p>At the end of the day we chose to cast out putative enhancers that do not overlap any ccRE-ELS, in order to remove element-Gene pairs in which the element is highly suspected not to be an enhancer. Note that we have not done so for Nasser2021 enhancers (instead, for Nasser2021 enhancers, we will create a confidence label on the resulting predictions according to whether the enhancer overlaps a ccRE-ELS or not), but for the ones here it is necessary given the important size of elements: if they do not overlap any ccRE-ELS despite their large size, they are very unlikely to be actual enhancers.</p>
<p>Doing so, we lost 19% of our enhancers, resulting in a list of 31,7496,149 = 25,600 enhancers:</p>
<div class="highlight"><pre><span></span><code>bedtools<span class="w"> </span>intersect<span class="w"> </span>-wa<span class="w"> </span>-a<span class="w"> </span>list_all_enhancers.bed<span class="w"> </span>-b<span class="w"> </span>ccRE-ELS.bed<span class="w"> </span><span class="p">|</span>uniq<span class="w"> </span>&gt;<span class="w"> </span>list_all_enhancers.overlapping_ccRE-ELS.bed
</code></pre></div>
<blockquote>
<div class="highlight"><pre><span></span><code>wc<span class="w"> </span>-l<span class="w"> </span>list_all_enhancers.overlapping_ccRE-ELS.bed
</code></pre></div>
<p>25,600</p>
</blockquote>
<p>Then, over 48,038 initial putative E-G pairs in liver, there are 39,252 remaining pairs for which the enhancer overlaps at least one ccRE-ELS.</p>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;BEGIN{OFS=&quot;\t&quot;} {if(NR==FNR){found[$1&quot;:&quot;$2&quot;-&quot;$3]++; next}; if(found[$1&quot;:&quot;$2&quot;-&quot;$3]){print $0}}&#39;</span><span class="w"> </span>list_all_enhancers.overlapping_ccRE-ELS.bed<span class="w"> </span>liver.all_putative_enhancers.sorted.bedpe<span class="w"> </span>&gt;<span class="w"> </span>liver.all_putative_enhancers.overlapping_ccRE-ELS.sorted.bedpe
</code></pre></div>
<blockquote>
<div class="highlight"><pre><span></span><code>wc<span class="w"> </span>-l<span class="w"> </span>liver.all_putative_enhancers.overlapping_ccRE-ELS.sorted.bedpe
</code></pre></div>
<p>39,252</p>
</blockquote>
<p>The file <code>liver.all_putative_enhancers.overlapping_ccRE-ELS.sorted.bedpe</code> is the one we will use from now.</p>





                
              </article>
            </div>
          
          
        </div>
        
          <a href="#" class="md-top md-icon" data-md-component="top" hidden>
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg>
            Back to top
          </a>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      2020 - 2022 Tristan Hoellinger
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
        <div class="md-social">
  
    
    
    
    
      
      
    
    <a href="https://twitter.com/TristanHoellin1" target="_blank" rel="noopener" title="twitter.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"/></svg>
    </a>
  
    
    
    
    
      
      
    
    <a href="https://fr.linkedin.com/in/tristan-hoellinger" target="_blank" rel="noopener" title="fr.linkedin.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"/></svg>
    </a>
  
    
    
    
    
    <a href="mailto:sarah.djebali@inserm.fr" target="_blank" rel="noopener" title="" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M498.1 5.6c10.1 7 15.4 19.1 13.5 31.2l-64 416c-1.5 9.7-7.4 18.2-16 23s-18.9 5.4-28 1.6L284 427.7l-68.5 74.1c-8.9 9.7-22.9 12.9-35.2 8.1S160 493.2 160 480v-83.6c0-4 1.5-7.8 4.2-10.7l167.6-182.9c5.8-6.3 5.6-16-.4-22s-15.7-6.4-22-.7L106 360.8l-88.3-44.2C7.1 311.3.3 300.7 0 288.9s5.9-22.8 16.1-28.7l448-256c10.7-6.1 23.9-5.5 34 1.4z"/></svg>
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    <script id="__config" type="application/json">{"base": "../../..", "features": ["navigation.instant", "navigation.tabs", "navigation.sections", "navigation.top"], "search": "../../../assets/javascripts/workers/search.db81ec45.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../../../assets/javascripts/bundle.6df46069.min.js"></script>
      
    
  </body>
</html>