
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
        <meta name="author" content="Tristan Hoellinger">
      
      
        <link rel="canonical" href="https://eg.github.io/notes_ABC/generic_notebooks/turnkey_notebook_to_run_ABC_with_example_over_GM12878.html">
      
      
        <link rel="prev" href="../K562/april_K562_candidates_not_in_whitelist_56_genes/april_K562_candidates_NOT_in_whitelist.html">
      
      
        <link rel="next" href="../BENGI/notebook_ABC_over_BENGI_GM12878.html">
      
      <link rel="icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.4.2, mkdocs-material-9.0.11">
    
    
      
        <title>Generic notebook to run ABC method & Example over GM12878 - Genome-wide identification of E/G interactions</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.0d440cfe.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.2505c338.min.css">
      
      

    
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../stylesheets/extra.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="black" data-md-color-accent="">
  
    
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#turnkey-executable-notebook-to-run-abc-model-example-over-gm12878" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Genome-wide identification of E/G interactions" class="md-header__button md-logo" aria-label="Genome-wide identification of E/G interactions" data-md-component="logo">
      
  <img src="../../logo/logo-inserm-blc.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Genome-wide identification of E/G interactions
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Generic notebook to run ABC method & Example over GM12878
            
          </span>
        </div>
      </div>
    </div>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/hoellin/eg" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
  </div>
  <div class="md-source__repository">
    hoellin/eg
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../guidebooks/introduction.html" class="md-tabs__link">
        Guidebooks
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../haemochromatosis/introduction.html" class="md-tabs__link">
        Haemochromatosis
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../notes_BENGI/introduction.html" class="md-tabs__link">
        BENGI
      </a>
    </li>
  

      
        
  
  
    
  


  
  
  
    <li class="md-tabs__item">
      <a href="../introduction.html" class="md-tabs__link md-tabs__link--active">
        ABC
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../index.html" class="md-tabs__link">
        About
      </a>
    </li>
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Genome-wide identification of E/G interactions" class="md-nav__button md-logo" aria-label="Genome-wide identification of E/G interactions" data-md-component="logo">
      
  <img src="../../logo/logo-inserm-blc.png" alt="logo">

    </a>
    Genome-wide identification of E/G interactions
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/hoellin/eg" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
  </div>
  <div class="md-source__repository">
    hoellin/eg
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1" >
      
      
      
        <label class="md-nav__link" for="__nav_1" id="__nav_1_label" tabindex="0">
          Guidebooks
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_1_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_1">
          <span class="md-nav__icon md-icon"></span>
          Guidebooks
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../guidebooks/introduction.html" class="md-nav__link">
        Introduction
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../guidebooks/script_bedpe_summary/script_bedpe_summary.html" class="md-nav__link">
        Compute summary stats of a bedpe
      </a>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_3" >
      
      
      
        <label class="md-nav__link" for="__nav_1_3" id="__nav_1_3_label" tabindex="0">
          Basics
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_1_3">
          <span class="md-nav__icon md-icon"></span>
          Basics
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../guidebooks/shell.html" class="md-nav__link">
        Shell
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../guidebooks/conda_environments.html" class="md-nav__link">
        Conda basics
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../guidebooks/git.html" class="md-nav__link">
        Git
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../guidebooks/genotoul.html" class="md-nav__link">
        Genotoul cluster
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../guidebooks/notebooks_ssh.html" class="md-nav__link">
        How to use Jupyter through ssh tunneling?
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_4" >
      
      
      
        <label class="md-nav__link" for="__nav_1_4" id="__nav_1_4_label" tabindex="0">
          Misc
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_4_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_1_4">
          <span class="md-nav__icon md-icon"></span>
          Misc
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../guidebooks/compute_correlations.html" class="md-nav__link">
        Compute correlations
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" >
      
      
      
        <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
          Haemochromatosis
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_2">
          <span class="md-nav__icon md-icon"></span>
          Haemochromatosis
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../haemochromatosis/introduction.html" class="md-nav__link">
        Introduction
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../haemochromatosis/data_format.html" class="md-nav__link">
        Data format
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../haemochromatosis/ABC/custom_go.html" class="md-nav__link">
        Custom script GO
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../haemochromatosis/main_results.html" class="md-nav__link">
        Main results
      </a>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2_5" >
      
      
      
        <label class="md-nav__link" for="__nav_2_5" id="__nav_2_5_label" tabindex="0">
          CHiC data
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_2_5_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_2_5">
          <span class="md-nav__icon md-icon"></span>
          CHiC data
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../haemochromatosis/CHiC/preprocess_data.html" class="md-nav__link">
        Preprocessing
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../haemochromatosis/CHiC/preliminary_analysis_chic_v2.html" class="md-nav__link">
        Networks
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../haemochromatosis/CHiC/go_enrichment.html" class="md-nav__link">
        GO enrichment
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2_6" >
      
      
      
        <label class="md-nav__link" for="__nav_2_6" id="__nav_2_6_label" tabindex="0">
          ABC data
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_2_6_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_2_6">
          <span class="md-nav__icon md-icon"></span>
          ABC data
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../haemochromatosis/ABC/data_processing.html" class="md-nav__link">
        Preprocessing
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../haemochromatosis/ABC/preliminary_analysis_v9.html" class="md-nav__link">
        Networks
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../haemochromatosis/ABC/summary_statistics_eg_pairs.html" class="md-nav__link">
        Summary statistics enhancers
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../haemochromatosis/ABC/go_enrichment.html" class="md-nav__link">
        GO enrichment
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../haemochromatosis/ABC/gorilla.html" class="md-nav__link">
        GOrilla
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../haemochromatosis/ABC/compute_avg_nb_of_connections_per_gene.html" class="md-nav__link">
        Avg nb connections per gene
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../haemochromatosis/ABC/gwas.html" class="md-nav__link">
        GWAS (13 initial genes)
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../haemochromatosis/ABC/gwas_10.html" class="md-nav__link">
        GWAS (10 initial genes)
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" >
      
      
      
        <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
          BENGI
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3">
          <span class="md-nav__icon md-icon"></span>
          BENGI
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/introduction.html" class="md-nav__link">
        Introduction
      </a>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_2" >
      
      
      
        <label class="md-nav__link" for="__nav_3_2" id="__nav_3_2_label" tabindex="0">
          Notebooks
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_2_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_2">
          <span class="md-nav__icon md-icon"></span>
          Notebooks
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/BENGI_to_custom_bedpe.html" class="md-nav__link">
        Convert BENGI datasets to custom bedpe
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_3" >
      
      
      
        <label class="md-nav__link" for="__nav_3_3" id="__nav_3_3_label" tabindex="0">
          Summary statistics
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_3">
          <span class="md-nav__icon md-icon"></span>
          Summary statistics
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/summary_statistics/summary_statistics.html" class="md-nav__link">
        Summary statistics on BENGI GM12878 benchmarks
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/summary_statistics_BENGI/summary_BENGI_bedpe.html" class="md-nav__link">
        (R code) Compute summary statistics over BENGI
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_4" >
      
      
      
        <label class="md-nav__link" for="__nav_3_4" id="__nav_3_4_label" tabindex="0">
          Evaluation of a few methods over BENGI GM12878
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_4_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_4">
          <span class="md-nav__icon md-icon"></span>
          Evaluation of a few methods over BENGI GM12878
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_4_1" >
      
      
      
        <label class="md-nav__link" for="__nav_3_4_1" id="__nav_3_4_1_label" tabindex="0">
          Distance method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_4_1_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_4_1">
          <span class="md-nav__icon md-icon"></span>
          Distance method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/distance_method/distance_evaluation_with_code.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/distance_method/summary_distance_evaluation.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_4_2" >
      
      
      
        <label class="md-nav__link" for="__nav_3_4_2" id="__nav_3_4_2_label" tabindex="0">
          DNase-expression correlation method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_4_2_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_4_2">
          <span class="md-nav__icon md-icon"></span>
          DNase-expression correlation method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/dnase_expression_correlation/correlation_method_with_code.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/dnase_expression_correlation/summary_correlation_method.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_4_3" >
      
      
      
        <label class="md-nav__link" for="__nav_3_4_3" id="__nav_3_4_3_label" tabindex="0">
          Average rank method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_4_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_4_3">
          <span class="md-nav__icon md-icon"></span>
          Average rank method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/avg_rank_method/avg_rank_method_with_code.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/avg_rank_method/summary_avg_rank_method.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_5" >
      
      
      
        <label class="md-nav__link" for="__nav_3_5" id="__nav_3_5_label" tabindex="0">
          Evaluation of a few methods over Fulco et al 2019 CRISPRi-FlowFISH
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_5_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_5">
          <span class="md-nav__icon md-icon"></span>
          Evaluation of a few methods over Fulco et al 2019 CRISPRi-FlowFISH
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_5_1" >
      
      
      
        <label class="md-nav__link" for="__nav_3_5_1" id="__nav_3_5_1_label" tabindex="0">
          Distance method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_5_1_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_5_1">
          <span class="md-nav__icon md-icon"></span>
          Distance method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/CRISPRi_FlowFISH/distance_method/distance_over_fulco_et_al_crispri.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/CRISPRi_FlowFISH/distance_method/summary_distance_over_fulco_et_al_crispri.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_5_2" >
      
      
      
        <label class="md-nav__link" for="__nav_3_5_2" id="__nav_3_5_2_label" tabindex="0">
          DNase-expression correlation method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_5_2_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_5_2">
          <span class="md-nav__icon md-icon"></span>
          DNase-expression correlation method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/CRISPRi_FlowFISH/dnase_expression_correlation/correlation_method_with_code.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/CRISPRi_FlowFISH/dnase_expression_correlation/summary_correlation_method.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_5_3" >
      
      
      
        <label class="md-nav__link" for="__nav_3_5_3" id="__nav_3_5_3_label" tabindex="0">
          Average rank method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_5_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_5_3">
          <span class="md-nav__icon md-icon"></span>
          Average rank method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/CRISPRi_FlowFISH/avg_rank_method/avg_rank_method_with_code.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/CRISPRi_FlowFISH/avg_rank_method/summary_avg_rank.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" checked>
      
      
      
        <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
          ABC
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="true">
        <label class="md-nav__title" for="__nav_4">
          <span class="md-nav__icon md-icon"></span>
          ABC
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../introduction.html" class="md-nav__link">
        Introduction
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../BENGI/ABC_BENGI_results.html" class="md-nav__link">
        ABC over BENGI for GM12878, main results
      </a>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_3" >
      
      
      
        <label class="md-nav__link" for="__nav_4_3" id="__nav_4_3_label" tabindex="0">
          CRISPRi-FlowFISH
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_4_3">
          <span class="md-nav__icon md-icon"></span>
          CRISPRi-FlowFISH
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../K562/ABC_K562_CRISPRi_FlowFISH.html" class="md-nav__link">
        Original ABC predictions over K562
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../K562/recap_investigations_ABC_over_CRiFF.html" class="md-nav__link">
        Summary ABC over CRiFF
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
    
  
  
    
      
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_4" checked>
      
      
      
        <label class="md-nav__link" for="__nav_4_4" id="__nav_4_4_label" tabindex="0">
          Notebooks
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_4_label" aria-expanded="true">
        <label class="md-nav__title" for="__nav_4_4">
          <span class="md-nav__icon md-icon"></span>
          Notebooks
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../K562/april_K562_56_genes/april_K562_56_genes.html" class="md-nav__link">
        ABC over K562 cCRiFF
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../K562/april_K562_59_genes/april_K562_59_genes.html" class="md-nav__link">
        ABC over K562 CRiFF
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../K562/april_K562_candidates_not_in_whitelist_56_genes/april_K562_candidates_NOT_in_whitelist.html" class="md-nav__link">
        ABC over K562 cCRiFF, no whitelist
      </a>
    </li>
  

            
          
            
              
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          Generic notebook to run ABC method & Example over GM12878
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="turnkey_notebook_to_run_ABC_with_example_over_GM12878.html" class="md-nav__link md-nav__link--active">
        Generic notebook to run ABC method & Example over GM12878
      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#how-to-use-this-notebook" class="md-nav__link">
    How to use this notebook?
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#import-required-librairies" class="md-nav__link">
    Import required librairies
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#set-working-directory" class="md-nav__link">
    Set working directory
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#data-acquisition" class="md-nav__link">
    Data acquisition
  </a>
  
    <nav class="md-nav" aria-label="Data acquisition">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#chromatin-accessibility-dnase-seq" class="md-nav__link">
    Chromatin accessibility (DNase-seq)
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#blacklist" class="md-nav__link">
    Blacklist
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#histone-mark-h3k27ac-chip-seq" class="md-nav__link">
    Histone mark H3K27ac ChIP-seq
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gene-expression-polya-rna-seq" class="md-nav__link">
    Gene expression (polyA+ RNA-seq)
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ubiquitously-expressed-genes" class="md-nav__link">
    Ubiquitously expressed genes
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#genome-file" class="md-nav__link">
    Genome file
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ccres" class="md-nav__link">
    ccREs
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#data-reprocessing" class="md-nav__link">
    Data reprocessing
  </a>
  
    <nav class="md-nav" aria-label="Data reprocessing">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#create-expression-table" class="md-nav__link">
    Create expression table
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#filter-gene-annotation" class="md-nav__link">
    Filter gene annotation
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#running-the-abc-model" class="md-nav__link">
    Running the ABC model
  </a>
  
    <nav class="md-nav" aria-label="Running the ABC model">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#step-1-define-candidate-elements" class="md-nav__link">
    Step 1: define candidate elements
  </a>
  
    <nav class="md-nav" aria-label="Step 1: define candidate elements">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#call-peaks-with-macs2" class="md-nav__link">
    Call peaks with macs2
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#use-abc-makecandidateregionspy-to-define-candidate-regions" class="md-nav__link">
    Use ABC makeCandidateRegions.py to define candidate regions
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-2-quantifying-enhancer-activity" class="md-nav__link">
    Step 2: quantifying enhancer activity
  </a>
  
    <nav class="md-nav" aria-label="Step 2: quantifying enhancer activity">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#use-abc-runneighborhoodspy" class="md-nav__link">
    Use ABC run.neighborhoods.py
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-3-computing-the-abc-score" class="md-nav__link">
    Step 3: computing the ABC score
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#evaluation-of-abc-predictions-on-bengi-datasets" class="md-nav__link">
    Evaluation of ABC predictions on BENGI datasets
  </a>
  
    <nav class="md-nav" aria-label="Evaluation of ABC predictions on BENGI datasets">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#find-ccre-delss" class="md-nav__link">
    Find ccRE-dELSs
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#data-processing" class="md-nav__link">
    Data processing
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#intersection-between-abc-predictions-bengi" class="md-nav__link">
    Intersection between ABC predictions / BENGI
  </a>
  
    <nav class="md-nav" aria-label="Intersection between ABC predictions / BENGI">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#import-files" class="md-nav__link">
    Import files
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#export-files" class="md-nav__link">
    Export files
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../BENGI/notebook_ABC_over_BENGI_GM12878.html" class="md-nav__link">
        ABC over GM12878 with same accessions as for BENGI
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../BENGI/notebook_ABC_over_BENGI_GM12878_from_ccRE_ELSs.html" class="md-nav__link">
        ABC over GM12878 with same accessions as for BENGI, starting with ccRE-dELS
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_5" >
      
      
      
        <label class="md-nav__link" for="__nav_4_5" id="__nav_4_5_label" tabindex="0">
          Full codes
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_5_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_4_5">
          <span class="md-nav__icon md-icon"></span>
          Full codes
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../test_abc_pipeline.html" class="md-nav__link">
        Test of ABC pipeline over chr22
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../BENGI/ABC_over_BENGI_GM12878.html" class="md-nav__link">
        ABC over GM12878 with same accessions as for BENGI
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../differences_example_chr22.html" class="md-nav__link">
        Meld of results over chr22 vs authors' results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" >
      
      
      
        <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
          About
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_5">
          <span class="md-nav__icon md-icon"></span>
          About
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../index.html" class="md-nav__link">
        Getting started
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../about_me.html" class="md-nav__link">
        About me
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#how-to-use-this-notebook" class="md-nav__link">
    How to use this notebook?
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#import-required-librairies" class="md-nav__link">
    Import required librairies
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#set-working-directory" class="md-nav__link">
    Set working directory
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#data-acquisition" class="md-nav__link">
    Data acquisition
  </a>
  
    <nav class="md-nav" aria-label="Data acquisition">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#chromatin-accessibility-dnase-seq" class="md-nav__link">
    Chromatin accessibility (DNase-seq)
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#blacklist" class="md-nav__link">
    Blacklist
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#histone-mark-h3k27ac-chip-seq" class="md-nav__link">
    Histone mark H3K27ac ChIP-seq
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gene-expression-polya-rna-seq" class="md-nav__link">
    Gene expression (polyA+ RNA-seq)
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ubiquitously-expressed-genes" class="md-nav__link">
    Ubiquitously expressed genes
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#genome-file" class="md-nav__link">
    Genome file
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ccres" class="md-nav__link">
    ccREs
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#data-reprocessing" class="md-nav__link">
    Data reprocessing
  </a>
  
    <nav class="md-nav" aria-label="Data reprocessing">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#create-expression-table" class="md-nav__link">
    Create expression table
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#filter-gene-annotation" class="md-nav__link">
    Filter gene annotation
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#running-the-abc-model" class="md-nav__link">
    Running the ABC model
  </a>
  
    <nav class="md-nav" aria-label="Running the ABC model">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#step-1-define-candidate-elements" class="md-nav__link">
    Step 1: define candidate elements
  </a>
  
    <nav class="md-nav" aria-label="Step 1: define candidate elements">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#call-peaks-with-macs2" class="md-nav__link">
    Call peaks with macs2
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#use-abc-makecandidateregionspy-to-define-candidate-regions" class="md-nav__link">
    Use ABC makeCandidateRegions.py to define candidate regions
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-2-quantifying-enhancer-activity" class="md-nav__link">
    Step 2: quantifying enhancer activity
  </a>
  
    <nav class="md-nav" aria-label="Step 2: quantifying enhancer activity">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#use-abc-runneighborhoodspy" class="md-nav__link">
    Use ABC run.neighborhoods.py
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-3-computing-the-abc-score" class="md-nav__link">
    Step 3: computing the ABC score
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#evaluation-of-abc-predictions-on-bengi-datasets" class="md-nav__link">
    Evaluation of ABC predictions on BENGI datasets
  </a>
  
    <nav class="md-nav" aria-label="Evaluation of ABC predictions on BENGI datasets">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#find-ccre-delss" class="md-nav__link">
    Find ccRE-dELSs
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#data-processing" class="md-nav__link">
    Data processing
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#intersection-between-abc-predictions-bengi" class="md-nav__link">
    Intersection between ABC predictions / BENGI
  </a>
  
    <nav class="md-nav" aria-label="Intersection between ABC predictions / BENGI">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#import-files" class="md-nav__link">
    Import files
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#export-files" class="md-nav__link">
    Export files
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  

  
  



<h1 id="turnkey-executable-notebook-to-run-abc-model-example-over-gm12878">Turnkey executable notebook to run ABC model. Example over GM12878.<a class="headerlink" href="#turnkey-executable-notebook-to-run-abc-model-example-over-gm12878" title="Permanent link">&para;</a></h1>
<h2 id="how-to-use-this-notebook">How to use this notebook?<a class="headerlink" href="#how-to-use-this-notebook" title="Permanent link">&para;</a></h2>
<p>First, make a copy of <a href="../../../scripts/notebooks/ipynb/ABC/ABC_generic.ipynb">this notebook</a> on your computer / cluster.</p>
<p>Then, to use this notebook, one should only have to carefully modify the content of the <a href="#set-working-directory">Set working directory</a> section, then to execute the notebook cell by cell, in the correct order. After execution of each cell, remember to check for errors before executing the next one.</p>
<p>One can use the following to switch notebook theme:</p>
<div class="highlight"><pre><span></span><code><span class="c1"># night theme</span>
<span class="c1">#!jt -t monokai -f fira -fs 10 -nf ptsans -nfs 11 -N -kl -cursw 2 -cursc r -cellw 95% -T</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code><span class="c1"># standard theme</span>
<span class="c1">#!jt -r</span>
</code></pre></div>
<h2 id="import-required-librairies">Import required librairies<a class="headerlink" href="#import-required-librairies" title="Permanent link">&para;</a></h2>
<div class="highlight"><pre><span></span><code><span class="kn">import</span> <span class="nn">os.path</span> <span class="c1">#Or see https://stackoverflow.com/a/82852 for an object-oriented approach</span>
<span class="kn">from</span> <span class="nn">IPython.core.magic</span> <span class="kn">import</span> <span class="n">register_line_cell_magic</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code><span class="nd">@register_line_cell_magic</span>
<span class="k">def</span> <span class="nf">writetemplate</span><span class="p">(</span><span class="n">line</span><span class="p">,</span> <span class="n">cell</span><span class="p">):</span>
    <span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="n">line</span><span class="p">,</span> <span class="s1">&#39;w&#39;</span><span class="p">)</span> <span class="k">as</span> <span class="n">f</span><span class="p">:</span>
        <span class="n">f</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="n">cell</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="o">**</span><span class="nb">globals</span><span class="p">()))</span>
</code></pre></div>
<h2 id="set-working-directory">Set working directory<a class="headerlink" href="#set-working-directory" title="Permanent link">&para;</a></h2>
<div class="highlight"><pre><span></span><code><span class="n">mail_user</span> <span class="o">=</span> <span class="s2">&quot;tristan.hoellinger@inserm.fr&quot;</span> <span class="c1"># slurm notifications</span>
<span class="n">version</span> <span class="o">=</span> <span class="s2">&quot;hg19&quot;</span> <span class="c1"># used only in the current cell to compute some paths </span>
<span class="n">cell_type</span> <span class="o">=</span> <span class="s2">&quot;gm12878&quot;</span> <span class="c1"># used all along the notebook</span>
<span class="n">specie</span> <span class="o">=</span> <span class="s2">&quot;homo_sapiens/&quot;</span><span class="o">+</span><span class="n">version</span><span class="o">+</span><span class="s2">&quot;/&quot;</span> <span class="c1"># used only in the current cell to compute some paths </span>

<span class="c1"># Where to store run-specific references, scripts, intermediate files, predictions, etc</span>
<span class="n">work_dir</span> <span class="o">=</span> <span class="s2">&quot;/work2/project/regenet/workspace/thoellinger/ABC-Enhancer-Gene-Prediction/500K_peaks_BENGI_GM12878/&quot;</span>

<span class="n">scripts</span> <span class="o">=</span> <span class="n">work_dir</span><span class="o">+</span><span class="s2">&quot;hi_slurm/&quot;</span> <span class="c1"># where to store scripts specific to this run</span>
<span class="n">results_dir</span> <span class="o">=</span> <span class="s2">&quot;/work2/project/regenet/results/&quot;</span> <span class="c1"># used to compute paths where data for this cell type is</span>
                                                <span class="c1"># stored / will be downloaded</span>
<span class="n">dnase_dir</span> <span class="o">=</span> <span class="n">results_dir</span><span class="o">+</span><span class="s2">&quot;dnaseseq/&quot;</span><span class="o">+</span><span class="n">specie</span><span class="o">+</span><span class="n">cell_type</span><span class="o">+</span><span class="s1">&#39;/&#39;</span>
<span class="n">chipseq_dir</span> <span class="o">=</span> <span class="n">results_dir</span><span class="o">+</span><span class="s2">&quot;chipseq/h3k27ac/&quot;</span><span class="o">+</span><span class="n">specie</span><span class="o">+</span><span class="n">cell_type</span><span class="o">+</span><span class="s1">&#39;/&#39;</span>
<span class="n">expression_dir</span> <span class="o">=</span> <span class="n">results_dir</span><span class="o">+</span><span class="s2">&quot;rnaseq/&quot;</span><span class="o">+</span><span class="n">specie</span><span class="o">+</span><span class="n">cell_type</span><span class="o">+</span><span class="s1">&#39;/&#39;</span>
<span class="n">blacklist_dir</span> <span class="o">=</span> <span class="s2">&quot;/work2/project/regenet/data/species/&quot;</span><span class="o">+</span><span class="n">specie</span> <span class="c1"># where the ENCODE blacklist is (going to be) stored</span>

<span class="n">blacklist</span> <span class="o">=</span> <span class="n">blacklist_dir</span><span class="o">+</span><span class="n">version</span><span class="o">+</span><span class="s2">&quot;-blacklist.v2.bed.gz&quot;</span> <span class="c1"># need not exist yet</span>

<span class="c1"># Gene annotation in the gtf format and gene name / gene id table (1st column id, 2nd column name)</span>
<span class="n">gene_annotation</span> <span class="o">=</span> <span class="s2">&quot;/work2/project/fragencode/data/species/homo_sapiens/hg19.gencv19/homo_sapiens.gtf&quot;</span>
<span class="n">gnid_gname</span> <span class="o">=</span> <span class="s2">&quot;/work2/project/fragencode/data/species/homo_sapiens/hg19.gencv19/homo_sapiens.gnid.gnname.tsv&quot;</span>

<span class="c1"># Accessions for the current run. Just put accession numbers here, no need to download these accessions &quot;by hand&quot;.</span>
<span class="n">dnase_rep1</span> <span class="o">=</span> <span class="s2">&quot;ENCFF664VNB&quot;</span> <span class="c1"># exp ENCSR000COQ</span>
<span class="n">dnase_rep2</span> <span class="o">=</span> <span class="s2">&quot;ENCFF790WER&quot;</span> <span class="c1"># exp ENCSR000COQ</span>
<span class="n">h3k27_rep1</span> <span class="o">=</span> <span class="s2">&quot;ENCFF197QHX&quot;</span> <span class="c1"># exp ENCSR000AKC</span>
<span class="n">h3k27_rep2</span> <span class="o">=</span> <span class="s2">&quot;ENCFF882PRP&quot;</span> <span class="c1"># exp ENCSR000AKC</span>
<span class="n">rnaseq_rep1</span> <span class="o">=</span> <span class="s2">&quot;ENCFF587ALK&quot;</span> <span class="c1"># exp ENCSR000COQ</span>
<span class="n">rnaseq_rep2</span> <span class="o">=</span> <span class="s2">&quot;ENCFF766CPS&quot;</span> <span class="c1"># exp ENCSR000COQ</span>

<span class="c1"># Where to download the blacklist. Will not be used unless blacklist not found yet.</span>
<span class="n">blacklist_link</span> <span class="o">=</span> <span class="s2">&quot;https://github.com/Boyle-Lab/Blacklist/raw/master/lists/hg19-blacklist.v2.bed.gz&quot;</span>

<span class="n">dnase_file_rep1</span> <span class="o">=</span> <span class="n">dnase_dir</span><span class="o">+</span><span class="n">dnase_rep1</span><span class="o">+</span><span class="s2">&quot;.bam&quot;</span> <span class="c1"># do not change</span>
<span class="n">dnase_file_rep2</span> <span class="o">=</span> <span class="n">dnase_dir</span><span class="o">+</span><span class="n">dnase_rep2</span><span class="o">+</span><span class="s2">&quot;.bam&quot;</span>  <span class="c1"># do not change</span>
<span class="n">h3k27_file_rep1</span> <span class="o">=</span> <span class="n">chipseq_dir</span><span class="o">+</span><span class="n">h3k27_rep1</span><span class="o">+</span><span class="s2">&quot;.bam&quot;</span>  <span class="c1"># do not change</span>
<span class="n">h3k27_file_rep2</span> <span class="o">=</span> <span class="n">chipseq_dir</span><span class="o">+</span><span class="n">h3k27_rep2</span><span class="o">+</span><span class="s2">&quot;.bam&quot;</span>  <span class="c1"># do not change</span>
<span class="n">rnaseq_file_rep1</span> <span class="o">=</span> <span class="n">expression_dir</span><span class="o">+</span><span class="n">rnaseq_rep1</span><span class="o">+</span><span class="s2">&quot;.tsv&quot;</span>  <span class="c1"># do not change</span>
<span class="n">rnaseq_file_rep2</span> <span class="o">=</span> <span class="n">expression_dir</span><span class="o">+</span><span class="n">rnaseq_rep2</span><span class="o">+</span><span class="s2">&quot;.tsv&quot;</span>  <span class="c1"># do not change</span>

<span class="n">reference_dir</span> <span class="o">=</span> <span class="n">work_dir</span><span class="o">+</span><span class="s2">&quot;reference/&quot;</span> <span class="c1"># do not change</span>
<span class="n">annotations_dir</span> <span class="o">=</span> <span class="n">reference_dir</span><span class="o">+</span><span class="s2">&quot;gene_annotation/&quot;</span> <span class="c1"># do not change</span>
<span class="n">peaks</span> <span class="o">=</span> <span class="n">work_dir</span><span class="o">+</span><span class="s2">&quot;ABC_output/Peaks/&quot;</span> <span class="c1"># do not change</span>
<span class="n">neighborhoods</span> <span class="o">=</span> <span class="n">work_dir</span><span class="o">+</span><span class="s2">&quot;ABC_output/Neighborhoods/&quot;</span> <span class="c1"># do not change</span>
<span class="n">predictions</span> <span class="o">=</span> <span class="n">work_dir</span><span class="o">+</span><span class="s2">&quot;ABC_output/Predictions/&quot;</span> <span class="c1"># do not change</span>

<span class="n">light_annotation</span> <span class="o">=</span> <span class="n">annotations_dir</span><span class="o">+</span><span class="s2">&quot;gene_ids.bed&quot;</span>

<span class="c1"># Ubiquitously expressed genes (gene names). If provided, `ubiquitously_expressed_genes` will be</span>
<span class="c1"># automatically generated with gene ids instead of gene names. If not (left empty), </span>
<span class="c1"># `ubiquitously_expressed_genes` must be provided.</span>
<span class="n">ubiquitous_gene_names</span> <span class="o">=</span> <span class="n">work_dir</span><span class="o">+</span><span class="s2">&quot;../reference/UbiquitouslyExpressedGenesHG19.txt&quot;</span>
<span class="c1"># Ubiquitously expressed genes (gene ids). Please provide either `ubiquitously_expressed_genes` or</span>
<span class="c1"># `ubiquitous_gene_names`</span>
<span class="n">ubiquitously_expressed_genes</span> <span class="o">=</span> <span class="n">reference_dir</span><span class="o">+</span><span class="s2">&quot;UbiquitouslyExpressedGenes_ids.txt&quot;</span> <span class="c1"># need not exist</span>

<span class="n">compute_mean_expression</span> <span class="o">=</span> <span class="n">work_dir</span><span class="o">+</span><span class="s2">&quot;../compute_mean_expression.awk&quot;</span>
<span class="c1"># compute_mean_expression.awk computes an expression table (&lt;gene id&gt; &lt;expression&gt;) taking the mean</span>
<span class="c1"># TPM expression found in the 2 input polyA+ RNA-seq tsv.</span>

<span class="n">make_candidate_regions</span> <span class="o">=</span> <span class="n">work_dir</span><span class="o">+</span><span class="s2">&quot;../src/makeCandidateRegions.py&quot;</span>
<span class="n">run_neighborhoods</span> <span class="o">=</span> <span class="n">work_dir</span><span class="o">+</span><span class="s2">&quot;../src/run.neighborhoods.py&quot;</span>
<span class="n">predict</span> <span class="o">=</span> <span class="n">work_dir</span><span class="o">+</span><span class="s2">&quot;../src/predict.py&quot;</span>

<span class="n">genome_file</span> <span class="o">=</span> <span class="n">reference_dir</span><span class="o">+</span><span class="s2">&quot;chr_sizes&quot;</span> <span class="c1"># shall not exist yet</span>

<span class="n">gene_expression_table</span> <span class="o">=</span> <span class="n">reference_dir</span><span class="o">+</span><span class="s2">&quot;expression/&quot;</span><span class="o">+</span><span class="n">cell_type</span><span class="o">+</span><span class="s2">&quot;.&quot;</span><span class="o">+</span><span class="n">rnaseq_rep1</span><span class="o">+</span><span class="s2">&quot;_&quot;</span><span class="o">+</span><span class="n">rnaseq_rep2</span><span class="o">+</span><span class="s2">&quot;.mean.TPM.txt&quot;</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code><span class="k">if</span> <span class="ow">not</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">isdir</span><span class="p">(</span><span class="n">reference_dir</span><span class="p">):</span>
    <span class="err">!</span><span class="n">mkdir</span> <span class="err">$</span><span class="n">reference_dir</span>
<span class="k">if</span> <span class="ow">not</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">isdir</span><span class="p">(</span><span class="n">scripts</span><span class="p">):</span>
    <span class="err">!</span><span class="n">mkdir</span> <span class="err">$</span><span class="n">scripts</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code><span class="c1"># One may only change what is between quotes here</span>
<span class="n">slurm_1_1</span> <span class="o">=</span> <span class="n">scripts</span><span class="o">+</span><span class="s2">&quot;step1.1.sh&quot;</span>
<span class="n">slurm_1_2</span> <span class="o">=</span> <span class="n">scripts</span><span class="o">+</span><span class="s2">&quot;step1.2.sh&quot;</span>
<span class="n">slurm_1_3</span> <span class="o">=</span> <span class="n">scripts</span><span class="o">+</span><span class="s2">&quot;step1.3.sh&quot;</span>
<span class="n">slurm_2</span> <span class="o">=</span> <span class="n">scripts</span><span class="o">+</span><span class="s2">&quot;step2.sh&quot;</span>
<span class="n">slurm_3</span> <span class="o">=</span> <span class="n">scripts</span><span class="o">+</span><span class="s2">&quot;step3.sh&quot;</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code><span class="n">ccREs_dir</span> <span class="o">=</span> <span class="n">results_dir</span><span class="o">+</span><span class="s2">&quot;ccREs/&quot;</span><span class="o">+</span><span class="n">specie</span><span class="o">+</span><span class="s2">&quot;V1/&quot;</span><span class="o">+</span><span class="n">cell_type</span><span class="o">+</span><span class="s1">&#39;/&#39;</span>

<span class="n">ccREs_accession</span> <span class="o">=</span> <span class="s2">&quot;ENCFF028SGJ&quot;</span> <span class="c1"># ENCODE ccREs with annotations for the cell type considered</span>
<span class="n">TSS_list</span> <span class="o">=</span> <span class="s2">&quot;/work2/project/regenet/workspace/thoellinger/BENGI/Benchmark/Annotations/GENCODEv19-TSSs.bed&quot;</span>

<span class="n">ccREs</span> <span class="o">=</span> <span class="n">ccREs_dir</span><span class="o">+</span><span class="n">ccREs_accession</span><span class="o">+</span><span class="s2">&quot;.bed&quot;</span> <span class="c1"># need not exist yet</span>
<span class="n">ccREs_ELS</span> <span class="o">=</span> <span class="n">ccREs_dir</span><span class="o">+</span><span class="n">cell_type</span><span class="o">+</span><span class="s2">&quot;_ccRE_ELSs.bed&quot;</span> <span class="c1"># need not exist yet</span>
<span class="n">ccREs_dELS</span> <span class="o">=</span> <span class="n">ccREs_dir</span><span class="o">+</span><span class="n">cell_type</span><span class="o">+</span><span class="s2">&quot;_ccRE_dELSs.bed&quot;</span> <span class="c1"># need not exist yet</span>
</code></pre></div>
<h2 id="data-acquisition">Data acquisition<a class="headerlink" href="#data-acquisition" title="Permanent link">&para;</a></h2>
<h3 id="chromatin-accessibility-dnase-seq">Chromatin accessibility (DNase-seq)<a class="headerlink" href="#chromatin-accessibility-dnase-seq" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code><span class="k">if</span> <span class="ow">not</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">isfile</span><span class="p">(</span><span class="n">dnase_file_rep1</span><span class="p">):</span>
    <span class="err">!</span><span class="n">wget</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">www</span><span class="o">.</span><span class="n">encodeproject</span><span class="o">.</span><span class="n">org</span><span class="o">/</span><span class="n">files</span><span class="o">/</span><span class="err">$</span><span class="n">dnase_rep1</span><span class="o">/@</span><span class="nd">@download</span><span class="o">/</span><span class="err">$</span><span class="n">dnase_rep1</span><span class="err">$</span><span class="s2">&quot;.bam&quot;</span> <span class="o">-</span><span class="n">P</span> <span class="err">$</span><span class="n">dnase_dir</span>

<span class="k">if</span> <span class="ow">not</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">isfile</span><span class="p">(</span><span class="n">dnase_file_rep2</span><span class="p">):</span>
    <span class="err">!</span><span class="n">wget</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">www</span><span class="o">.</span><span class="n">encodeproject</span><span class="o">.</span><span class="n">org</span><span class="o">/</span><span class="n">files</span><span class="o">/</span><span class="err">$</span><span class="n">dnase_rep2</span><span class="o">/@</span><span class="nd">@download</span><span class="o">/</span><span class="err">$</span><span class="n">dnase_rep2</span><span class="err">$</span><span class="s2">&quot;.bam&quot;</span> <span class="o">-</span><span class="n">P</span> <span class="err">$</span><span class="n">dnase_dir</span>
</code></pre></div>
<h3 id="blacklist">Blacklist<a class="headerlink" href="#blacklist" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code><span class="k">if</span> <span class="ow">not</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">isfile</span><span class="p">(</span><span class="n">blacklist</span><span class="p">):</span>
    <span class="err">!</span><span class="n">wget</span> <span class="err">$</span><span class="n">blacklist_link</span> <span class="o">-</span><span class="n">P</span> <span class="err">$</span><span class="n">blacklist_dir</span>
</code></pre></div>
<h3 id="histone-mark-h3k27ac-chip-seq">Histone mark H3K27ac ChIP-seq<a class="headerlink" href="#histone-mark-h3k27ac-chip-seq" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code><span class="k">if</span> <span class="ow">not</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">isfile</span><span class="p">(</span><span class="n">h3k27_file_rep1</span><span class="p">):</span>
    <span class="err">!</span><span class="n">wget</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">www</span><span class="o">.</span><span class="n">encodeproject</span><span class="o">.</span><span class="n">org</span><span class="o">/</span><span class="n">files</span><span class="o">/</span><span class="err">$</span><span class="n">h3k27_rep1</span><span class="o">/@</span><span class="nd">@download</span><span class="o">/</span><span class="err">$</span><span class="n">h3k27_rep1</span><span class="err">$</span><span class="s2">&quot;.bam&quot;</span> <span class="o">-</span><span class="n">P</span> <span class="err">$</span><span class="n">chipseq_dir</span>

<span class="k">if</span> <span class="ow">not</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">isfile</span><span class="p">(</span><span class="n">h3k27_file_rep2</span><span class="p">):</span>
    <span class="err">!</span><span class="n">wget</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">www</span><span class="o">.</span><span class="n">encodeproject</span><span class="o">.</span><span class="n">org</span><span class="o">/</span><span class="n">files</span><span class="o">/</span><span class="err">$</span><span class="n">h3k27_rep2</span><span class="o">/@</span><span class="nd">@download</span><span class="o">/</span><span class="err">$</span><span class="n">h3k27_rep2</span><span class="err">$</span><span class="s2">&quot;.bam&quot;</span> <span class="o">-</span><span class="n">P</span> <span class="err">$</span><span class="n">chipseq_dir</span>
</code></pre></div>
<h3 id="gene-expression-polya-rna-seq">Gene expression (polyA+ RNA-seq)<a class="headerlink" href="#gene-expression-polya-rna-seq" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code><span class="k">if</span> <span class="ow">not</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">isfile</span><span class="p">(</span><span class="n">rnaseq_file_rep1</span><span class="p">):</span>
    <span class="err">!</span><span class="n">wget</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">www</span><span class="o">.</span><span class="n">encodeproject</span><span class="o">.</span><span class="n">org</span><span class="o">/</span><span class="n">files</span><span class="o">/</span><span class="err">$</span><span class="n">rnaseq_rep1</span><span class="o">/@</span><span class="nd">@download</span><span class="o">/</span><span class="err">$</span><span class="n">rnaseq_rep1</span><span class="err">$</span><span class="s2">&quot;.bam&quot;</span> <span class="o">-</span><span class="n">P</span> <span class="err">$</span><span class="n">expression_dir</span>

<span class="k">if</span> <span class="ow">not</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">isfile</span><span class="p">(</span><span class="n">rnaseq_file_rep2</span><span class="p">):</span>
    <span class="err">!</span><span class="n">wget</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">www</span><span class="o">.</span><span class="n">encodeproject</span><span class="o">.</span><span class="n">org</span><span class="o">/</span><span class="n">files</span><span class="o">/</span><span class="err">$</span><span class="n">rnaseq_rep2</span><span class="o">/@</span><span class="nd">@download</span><span class="o">/</span><span class="err">$</span><span class="n">rnaseq_rep2</span><span class="err">$</span><span class="s2">&quot;.bam&quot;</span> <span class="o">-</span><span class="n">P</span> <span class="err">$</span><span class="n">expression_dir</span>
</code></pre></div>
<h3 id="ubiquitously-expressed-genes">Ubiquitously expressed genes<a class="headerlink" href="#ubiquitously-expressed-genes" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code>%%bash<span class="w"> </span>-s<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$gnid_gname</span><span class="s2">&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$ubiquitous_gene_names</span><span class="s2">&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$ubiquitously_expressed_genes</span><span class="s2">&quot;</span>
<span class="k">if</span><span class="w"> </span><span class="o">[[</span><span class="w"> </span>-f<span class="w"> </span><span class="nv">$2</span><span class="w"> </span><span class="o">]]</span><span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span><span class="o">[[</span><span class="w"> </span>!<span class="w"> </span>-f<span class="w"> </span><span class="nv">$3</span><span class="w"> </span><span class="o">]]</span><span class="p">;</span><span class="w"> </span><span class="k">then</span>
<span class="w">    </span>awk<span class="w"> </span><span class="s1">&#39;BEGIN{sep=&quot;\t&quot;} NR==FNR {id[$2]=$1; next} id[$1] {print id[$1]}&#39;</span><span class="w"> </span><span class="nv">$1</span><span class="w"> </span><span class="nv">$2</span><span class="w"> </span>&gt;<span class="w"> </span><span class="nv">$3</span>
<span class="k">fi</span>
</code></pre></div>
<h3 id="genome-file">Genome file<a class="headerlink" href="#genome-file" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code>%%bash<span class="w"> </span>-s<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$work_dir</span><span class="s2">&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$dnase_file_rep1</span><span class="s2">&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$genome_file</span><span class="s2">&quot;</span>
<span class="nv">CONDA_BASE</span><span class="o">=</span><span class="k">$(</span>conda<span class="w"> </span>info<span class="w"> </span>--base<span class="k">)</span><span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span><span class="nb">source</span><span class="w"> </span><span class="nv">$CONDA_BASE</span>/etc/profile.d/conda.sh
conda<span class="w"> </span>activate<span class="w"> </span>base<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>module<span class="w"> </span>load<span class="w"> </span>bioinfo/samtools-1.9
samtools<span class="w"> </span>view<span class="w"> </span>-H<span class="w"> </span><span class="nv">$2</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>grep<span class="w"> </span>SQ<span class="w"> </span><span class="p">|</span><span class="w"> </span>cut<span class="w"> </span>-f<span class="w"> </span><span class="m">2</span>,3<span class="w"> </span><span class="p">|</span>cut<span class="w"> </span>-c<span class="w"> </span><span class="m">4</span>-<span class="w"> </span><span class="p">|</span>awk<span class="w"> </span><span class="s1">&#39;BEGIN{FS=&quot;\t&quot;} {split($2,locus,&quot;:&quot;); if($1 ~ /^(chr)([0-9]{1,2}$)|(M$)|(X$)|(Y$)/){print($1&quot;\t&quot;locus[2])}}&#39;</span><span class="w"> </span>&gt;<span class="w"> </span><span class="nv">$3</span>
awk<span class="w"> </span><span class="s1">&#39;{print $1&quot;\t&quot;0&quot;\t&quot;$2}&#39;</span><span class="w"> </span><span class="nv">$3</span><span class="w"> </span>&gt;<span class="w"> </span><span class="nv">$3</span><span class="s2">&quot;.bed&quot;</span>
</code></pre></div>
<h3 id="ccres">ccREs<a class="headerlink" href="#ccres" title="Permanent link">&para;</a></h3>
<p>We  need to download ccREs with annotations for this cell type, in order to intersect ABC predictions with distal ccREs with Enhancer-like signature, to later intersect them with BENGI datasets. The correct accession can be found in Supplentary Figure 1 Table S1a of Moore et al. 2020.</p>
<div class="highlight"><pre><span></span><code><span class="k">if</span> <span class="ow">not</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">isfile</span><span class="p">(</span><span class="n">ccREs</span><span class="o">+</span><span class="s2">&quot;.gz&quot;</span><span class="p">):</span>
    <span class="err">!</span><span class="n">wget</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">www</span><span class="o">.</span><span class="n">encodeproject</span><span class="o">.</span><span class="n">org</span><span class="o">/</span><span class="n">files</span><span class="o">/</span><span class="err">$</span><span class="n">ccREs_accession</span><span class="o">/@</span><span class="nd">@download</span><span class="o">/</span><span class="err">$</span><span class="n">ccREs_accession</span><span class="err">$</span><span class="s2">&quot;.bed.gz&quot;</span> <span class="o">-</span><span class="n">P</span> <span class="err">$</span><span class="n">ccREs_dir</span>

<span class="k">if</span> <span class="ow">not</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">isfile</span><span class="p">(</span><span class="n">ccREs</span><span class="p">):</span>
    <span class="err">!</span><span class="n">gzip</span> <span class="o">-</span><span class="n">cd</span> <span class="err">$</span><span class="n">ccREs</span><span class="s2">&quot;.gz&quot;</span> <span class="o">&gt;</span> <span class="err">$</span><span class="n">ccREs</span>
</code></pre></div>
<h2 id="data-reprocessing">Data reprocessing<a class="headerlink" href="#data-reprocessing" title="Permanent link">&para;</a></h2>
<h3 id="create-expression-table">Create expression table<a class="headerlink" href="#create-expression-table" title="Permanent link">&para;</a></h3>
<p>We use <code>compute_mean_expression.awk</code> to take as the reference TPM value, the mean value for the 2 replicates.</p>
<div class="highlight"><pre><span></span><code>%%bash<span class="w"> </span>-s<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$compute_mean_expression</span><span class="s2">&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$reference_dir</span><span class="s2">&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$rnaseq_file_rep1</span><span class="s2">&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$rnaseq_file_rep2</span><span class="s2">&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$cell_type</span><span class="s2">&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$rnaseq_rep1</span><span class="s2">&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$rnaseq_rep2</span><span class="s2">&quot;</span>
<span class="k">if</span><span class="w"> </span><span class="o">[[</span><span class="w"> </span>!<span class="w"> </span>-d<span class="w"> </span><span class="nv">$2</span><span class="s2">&quot;expression/&quot;</span><span class="w"> </span><span class="o">]]</span><span class="p">;</span><span class="w"> </span><span class="k">then</span><span class="w"> </span>mkdir<span class="w"> </span><span class="nv">$2</span><span class="s2">&quot;expression/&quot;</span><span class="p">;</span><span class="w"> </span><span class="k">fi</span>
awk<span class="w"> </span>-f<span class="w"> </span><span class="nv">$1</span><span class="w"> </span><span class="nv">$3</span><span class="w"> </span><span class="nv">$4</span><span class="w"> </span>&gt;<span class="w"> </span><span class="nv">$2</span><span class="s2">&quot;expression/&quot;</span><span class="nv">$5</span><span class="s2">&quot;.&quot;</span><span class="nv">$6</span><span class="s2">&quot;_&quot;</span><span class="nv">$7</span><span class="s2">&quot;.mean.TPM.txt&quot;</span>
</code></pre></div>
<h3 id="filter-gene-annotation">Filter gene annotation<a class="headerlink" href="#filter-gene-annotation" title="Permanent link">&para;</a></h3>
<p>We keep genes only, and <code>gene_ids.bed</code> must fit following format:</p>
<blockquote>
<div class="highlight"><pre><span></span><code>chrX<span class="w">  </span><span class="m">99883667</span><span class="w">    </span><span class="m">99894988</span><span class="w">    </span>ENSG00000000003.10<span class="w">  </span><span class="m">0</span><span class="w">   </span>-
chrX<span class="w">  </span><span class="m">99839799</span><span class="w">    </span><span class="m">99854882</span><span class="w">    </span>ENSG00000000005.5<span class="w">   </span><span class="m">0</span><span class="w">   </span>+
</code></pre></div>
</blockquote>
<div class="highlight"><pre><span></span><code>%%bash<span class="w"> </span>-s<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$work_dir</span><span class="s2">&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$genome_file</span><span class="s2">&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$light_annotation</span><span class="s2">&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$gene_annotation</span><span class="s2">&quot;</span>
<span class="k">if</span><span class="w"> </span><span class="o">[[</span><span class="w"> </span>!<span class="w"> </span>-d<span class="w"> </span><span class="nv">$3</span><span class="w"> </span><span class="o">]]</span><span class="p">;</span><span class="w"> </span><span class="k">then</span><span class="w"> </span>mkdir<span class="w"> </span><span class="nv">$3</span><span class="p">;</span><span class="w"> </span><span class="k">fi</span>
<span class="nv">CONDA_BASE</span><span class="o">=</span><span class="k">$(</span>conda<span class="w"> </span>info<span class="w"> </span>--base<span class="k">)</span><span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span><span class="nb">source</span><span class="w"> </span><span class="nv">$CONDA_BASE</span>/etc/profile.d/conda.sh
conda<span class="w"> </span>activate<span class="w"> </span>base<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>module<span class="w"> </span>load<span class="w"> </span>bioinfo/bedtools-2.27.1
awk<span class="w"> </span><span class="s1">&#39;{if(NR==FNR){!genome[$1]++; next}; if(genome[$1]){if($3 ~ /(^gene$)/){print $1&quot;\t&quot;$4&quot;\t&quot;$5&quot;\t&quot;substr($10,2,length($10)-3)&quot;\t&quot;0&quot;\t&quot;$7}}}&#39;</span><span class="w"> </span><span class="nv">$2</span><span class="w"> </span><span class="nv">$4</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>bedtools<span class="w"> </span>sort<span class="w"> </span>-g<span class="w"> </span><span class="nv">$2</span><span class="w"> </span>-i<span class="w"> </span>stdin<span class="w"> </span>&gt;<span class="w"> </span><span class="nv">$3</span>
</code></pre></div>
<h2 id="running-the-abc-model">Running the ABC model<a class="headerlink" href="#running-the-abc-model" title="Permanent link">&para;</a></h2>
<h3 id="step-1-define-candidate-elements">Step 1: define candidate elements<a class="headerlink" href="#step-1-define-candidate-elements" title="Permanent link">&para;</a></h3>
<h4 id="call-peaks-with-macs2">Call peaks with <code>macs2</code><a class="headerlink" href="#call-peaks-with-macs2" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code><span class="o">%%</span><span class="n">writetemplate</span> <span class="err">$</span><span class="n">slurm_1_1</span>
<span class="c1">#!/bin/sh                                                                                                                                                                                                                    </span>
<span class="c1"># dependencies: python2</span>
<span class="n">macs2</span> <span class="n">callpeak</span> \
<span class="o">-</span><span class="n">t</span> <span class="p">{</span><span class="n">dnase_file_rep1</span><span class="p">}</span> <span class="p">{</span><span class="n">dnase_file_rep2</span><span class="p">}</span> \
<span class="o">-</span><span class="n">n</span> <span class="p">{</span><span class="n">dnase_rep1</span><span class="p">}</span><span class="n">_</span><span class="p">{</span><span class="n">dnase_rep2</span><span class="p">}</span><span class="o">.</span><span class="n">macs2</span> \
<span class="o">-</span><span class="n">f</span> <span class="n">BAM</span> \
<span class="o">-</span><span class="n">g</span> <span class="n">hs</span> \
<span class="o">-</span><span class="n">p</span> <span class="mf">.1</span> \
<span class="o">--</span><span class="n">call</span><span class="o">-</span><span class="n">summits</span> \
<span class="o">--</span><span class="n">outdir</span> <span class="p">{</span><span class="n">peaks</span><span class="p">}</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>%%bash<span class="w"> </span>-s<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$work_dir</span><span class="s2">&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$slurm_1_1</span><span class="s2">&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$mail_user</span><span class="s2">&quot;</span>
<span class="nv">CONDA_BASE</span><span class="o">=</span><span class="k">$(</span>conda<span class="w"> </span>info<span class="w"> </span>--base<span class="k">)</span><span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span><span class="nb">source</span><span class="w"> </span><span class="nv">$CONDA_BASE</span>/etc/profile.d/conda.sh
conda<span class="w"> </span>activate<span class="w"> </span>base<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>conda<span class="w"> </span>activate<span class="w"> </span>abcmodel
module<span class="w"> </span>load<span class="w"> </span>bioinfo/samtools-1.9<span class="w"> </span>bioinfo/bedtools-2.27.1<span class="w"> </span>bioinfo/tabix-0.2.5<span class="w"> </span>bioinfo/juicer-1.5.6
conda<span class="w"> </span>activate<span class="w"> </span>py2<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>module<span class="w"> </span>load<span class="w"> </span>system/Python-2.7.2
sbatch<span class="w"> </span>--mem<span class="o">=</span>4G<span class="w"> </span>--cpus-per-task<span class="o">=</span><span class="m">1</span><span class="w"> </span>-J<span class="w"> </span>step1.1<span class="w"> </span>--mail-user<span class="o">=</span><span class="nv">$3</span><span class="w"> </span>--mail-type<span class="o">=</span>END,FAIL<span class="w"> </span>--workdir<span class="o">=</span><span class="nv">$1</span><span class="w"> </span>--export<span class="o">=</span>ALL<span class="w"> </span>-p<span class="w"> </span>workq<span class="w"> </span><span class="nv">$2</span>
conda<span class="w"> </span>deactivate<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>module<span class="w"> </span>unload<span class="w"> </span>system/Python-2.7.2
</code></pre></div>
<div class="highlight"><pre><span></span><code>Submitted batch job 22130631
</code></pre></div>
<h4 id="use-abc-makecandidateregionspy-to-define-candidate-regions">Use ABC <code>makeCandidateRegions.py</code> to define candidate regions<a class="headerlink" href="#use-abc-makecandidateregionspy-to-define-candidate-regions" title="Permanent link">&para;</a></h4>
<p>As there are peaks over many kinds of scaffolds listed in the <code>.narrowPeak</code> and <code>.xls</code> output files, we modify them keep only chromosomes only (otherwise <code>bedtools sort</code> below would not work). <strong>But we have to keep in mind that the <code>.r</code> will still corresponds to DNase peaks over all scaffolds</strong>.</p>
<div class="highlight"><pre><span></span><code>%%bash<span class="w"> </span>-s<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$peaks</span><span class="s2">&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$dnase_rep1</span><span class="s2">&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$dnase_rep2</span><span class="s2">&quot;</span>
awk<span class="w"> </span><span class="s1">&#39;$1 ~ /^(chr)([0-9]{1,2}$)|(M$)|(X$)|(Y$)/ {print $0}&#39;</span><span class="w"> </span><span class="nv">$1$2</span><span class="s2">&quot;_&quot;</span><span class="nv">$3</span><span class="s2">&quot;.macs2_peaks.narrowPeak&quot;</span><span class="w"> </span>&gt;<span class="w"> </span><span class="nv">$1</span>/<span class="nv">$2</span><span class="s2">&quot;_&quot;</span><span class="nv">$3</span><span class="s2">&quot;.chromosomes_only.macs2_peaks.narrowPeak&quot;</span>
awk<span class="w"> </span><span class="s1">&#39;NR &lt;= 28 || $1 ~ /^(chr)([0-9]{1,2}$)|(M$)|(X$)|(Y$)/ {print $0}&#39;</span><span class="w"> </span><span class="nv">$1$2</span><span class="s2">&quot;_&quot;</span><span class="nv">$3</span><span class="s2">&quot;.macs2_peaks.xls&quot;</span><span class="w"> </span>&gt;<span class="w"> </span><span class="nv">$1$2</span><span class="s2">&quot;_&quot;</span><span class="nv">$3</span><span class="s2">&quot;.chromosomes_only.macs2_peaks.xls&quot;</span>
</code></pre></div>
<p>Now we sort the result (note that it seems quite tricky to use <code>srun</code> from a notebook as we first need to export environment variables - maybe there is another way but as for now I could not find better):</p>
<div class="highlight"><pre><span></span><code>%%bash<span class="w"> </span>-s<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$peaks</span><span class="s2">&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$dnase_rep1</span><span class="s2">&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$dnase_rep2</span><span class="s2">&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$genome_file</span><span class="s2">&quot;</span>
<span class="nb">export</span><span class="w"> </span><span class="nv">V1</span><span class="o">=</span><span class="nv">$1</span><span class="w"> </span><span class="nv">V2</span><span class="o">=</span><span class="nv">$2</span><span class="w"> </span><span class="nv">V3</span><span class="o">=</span><span class="nv">$3</span><span class="w"> </span><span class="nv">V4</span><span class="o">=</span><span class="nv">$4</span>
<span class="nv">CONDA_BASE</span><span class="o">=</span><span class="k">$(</span>conda<span class="w"> </span>info<span class="w"> </span>--base<span class="k">)</span><span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span><span class="nb">source</span><span class="w"> </span><span class="nv">$CONDA_BASE</span>/etc/profile.d/conda.sh
conda<span class="w"> </span>activate<span class="w"> </span>base<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>conda<span class="w"> </span>activate<span class="w"> </span>abcmodel
module<span class="w"> </span>load<span class="w"> </span>bioinfo/bedtools-2.27.1
srun<span class="w"> </span>bash
bedtools<span class="w"> </span>sort<span class="w"> </span>-faidx<span class="w"> </span><span class="nv">$V4</span><span class="w"> </span>-i<span class="w"> </span><span class="nv">$V1$V2</span><span class="s2">&quot;_&quot;</span><span class="nv">$V3</span><span class="s2">&quot;.chromosomes_only.macs2_peaks.narrowPeak&quot;</span><span class="w"> </span>&gt;<span class="w"> </span><span class="nv">$V1$V2</span><span class="s2">&quot;_&quot;</span><span class="nv">$V3</span><span class="s2">&quot;.macs2_peaks.narrowPeak.sorted&quot;</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>srun: job 22134742 queued and waiting for resources
srun: job 22134742 has been allocated resources
</code></pre></div>
<p>Now as <code>work_dir ../src/makeCandidateRegions.py</code> does not support multiple <code>bam</code> inputs, we concatenate the bam of the 2 replicates.  Indeed, <code>makeCandidateRegions.py</code> will use the bam input to count DNase reads on each peak of the<code>.macs2_peaks.narrowPeak.sorted</code> output found with <code>macs2</code>.</p>
<p>(WARNING: depending on the size of the input bam, which is much variable, the execution on 1 thread only can take from a few minutes to several hours, so better use at least 8 threads)</p>
<div class="highlight"><pre><span></span><code><span class="n">n_threads</span> <span class="o">=</span> <span class="mi">8</span> <span class="c1">#we recommend using at least 8 cpu</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code><span class="o">%%</span><span class="n">writetemplate</span> <span class="err">$</span><span class="n">slurm_1_2</span>
<span class="c1">#!/bin/sh</span>
<span class="c1"># dependencies: bioinfo/samtools-1.9 bioinfo/bedtools-2.27.1</span>

<span class="k">if</span> <span class="p">[[</span> <span class="err">!</span> <span class="o">-</span><span class="n">f</span> <span class="p">{</span><span class="n">dnase_dir</span><span class="p">}{</span><span class="n">dnase_rep1</span><span class="p">}</span><span class="n">_</span><span class="p">{</span><span class="n">dnase_rep2</span><span class="p">}</span><span class="o">.</span><span class="n">bam</span> <span class="p">]]</span>
<span class="n">then</span>
    <span class="k">if</span> <span class="p">[[</span> <span class="err">!</span> <span class="o">-</span><span class="n">f</span> <span class="p">{</span><span class="n">dnase_dir</span><span class="p">}{</span><span class="n">dnase_rep1</span><span class="p">}</span><span class="n">_</span><span class="p">{</span><span class="n">dnase_rep2</span><span class="p">}</span><span class="o">.</span><span class="n">sam</span> <span class="p">]]</span>
    <span class="n">then</span>
        <span class="n">samtools</span> <span class="n">view</span> <span class="o">-@</span> <span class="p">{</span><span class="n">n_threads</span><span class="p">}</span> <span class="o">-</span><span class="n">h</span> <span class="p">{</span><span class="n">dnase_file_rep1</span><span class="p">}</span> <span class="o">&gt;</span> <span class="p">{</span><span class="n">dnase_dir</span><span class="p">}{</span><span class="n">dnase_rep1</span><span class="p">}</span><span class="n">_</span><span class="p">{</span><span class="n">dnase_rep2</span><span class="p">}</span><span class="o">.</span><span class="n">sam</span>
        <span class="n">samtools</span> <span class="n">view</span> <span class="o">-@</span> <span class="p">{</span><span class="n">n_threads</span><span class="p">}</span> <span class="p">{</span><span class="n">dnase_file_rep2</span><span class="p">}</span> <span class="o">&gt;&gt;</span> <span class="p">{</span><span class="n">dnase_dir</span><span class="p">}{</span><span class="n">dnase_rep1</span><span class="p">}</span><span class="n">_</span><span class="p">{</span><span class="n">dnase_rep2</span><span class="p">}</span><span class="o">.</span><span class="n">sam</span>
        <span class="n">samtools</span> <span class="n">view</span> <span class="o">-@</span> <span class="p">{</span><span class="n">n_threads</span><span class="p">}</span> <span class="o">-</span><span class="n">S</span> <span class="o">-</span><span class="n">b</span> <span class="p">{</span><span class="n">dnase_dir</span><span class="p">}{</span><span class="n">dnase_rep1</span><span class="p">}</span><span class="n">_</span><span class="p">{</span><span class="n">dnase_rep2</span><span class="p">}</span><span class="o">.</span><span class="n">sam</span> <span class="o">&gt;</span> <span class="p">{</span><span class="n">dnase_dir</span><span class="p">}{</span><span class="n">dnase_rep1</span><span class="p">}</span><span class="n">_</span><span class="p">{</span><span class="n">dnase_rep2</span><span class="p">}</span><span class="o">.</span><span class="n">bam</span>
        <span class="n">rm</span> <span class="p">{</span><span class="n">dnase_dir</span><span class="p">}{</span><span class="n">dnase_rep1</span><span class="p">}</span><span class="n">_</span><span class="p">{</span><span class="n">dnase_rep2</span><span class="p">}</span><span class="o">.</span><span class="n">sam</span>
    <span class="k">else</span>
        <span class="n">samtools</span> <span class="n">view</span> <span class="o">-@</span> <span class="p">{</span><span class="n">n_threads</span><span class="p">}</span> <span class="o">-</span><span class="n">S</span> <span class="o">-</span><span class="n">b</span> <span class="p">{</span><span class="n">dnase_dir</span><span class="p">}{</span><span class="n">dnase_rep1</span><span class="p">}</span><span class="n">_</span><span class="p">{</span><span class="n">dnase_rep2</span><span class="p">}</span><span class="o">.</span><span class="n">sam</span> <span class="o">&gt;</span> <span class="p">{</span><span class="n">dnase_dir</span><span class="p">}{</span><span class="n">dnase_rep1</span><span class="p">}</span><span class="n">_</span><span class="p">{</span><span class="n">dnase_rep2</span><span class="p">}</span><span class="o">.</span><span class="n">bam</span>
    <span class="n">fi</span>
<span class="n">fi</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>%%bash<span class="w"> </span>-s<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$work_dir</span><span class="s2">&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$slurm_1_2</span><span class="s2">&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$mail_user</span><span class="s2">&quot;</span>
<span class="nv">CONDA_BASE</span><span class="o">=</span><span class="k">$(</span>conda<span class="w"> </span>info<span class="w"> </span>--base<span class="k">)</span><span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span><span class="nb">source</span><span class="w"> </span><span class="nv">$CONDA_BASE</span>/etc/profile.d/conda.sh
conda<span class="w"> </span>activate<span class="w"> </span>base<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>conda<span class="w"> </span>activate<span class="w"> </span>abcmodel
module<span class="w"> </span>load<span class="w"> </span>bioinfo/samtools-1.9
sbatch<span class="w"> </span>--mem<span class="o">=</span>2G<span class="w"> </span>--cpus-per-task<span class="o">=</span><span class="m">8</span><span class="w"> </span>-J<span class="w"> </span>step1.2<span class="w"> </span>--mail-user<span class="o">=</span><span class="nv">$3</span><span class="w"> </span>--mail-type<span class="o">=</span>END,FAIL<span class="w"> </span>--workdir<span class="o">=</span><span class="nv">$1</span><span class="w"> </span>--export<span class="o">=</span>ALL<span class="w"> </span>-p<span class="w"> </span>workq<span class="w"> </span><span class="nv">$2</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>Submitted batch job 22138998
</code></pre></div>
<p>Now we can launch <code>makeCandidateRegions</code> on slurm:</p>
<div class="highlight"><pre><span></span><code><span class="o">%%</span><span class="n">writetemplate</span> <span class="err">$</span><span class="n">slurm_1_3</span>
<span class="c1">#!/bin/sh</span>
<span class="c1"># dependencies: python3 bioinfo/samtools-1.9 bioinfo/bedtools-2.27.1 bioinfo/tabix-0.2.5 bioinfo/juicer-1.5.6</span>
<span class="n">python</span> <span class="p">{</span><span class="n">make_candidate_regions</span><span class="p">}</span> \
<span class="o">--</span><span class="n">narrowPeak</span> <span class="p">{</span><span class="n">peaks</span><span class="p">}{</span><span class="n">dnase_rep1</span><span class="p">}</span><span class="n">_</span><span class="p">{</span><span class="n">dnase_rep2</span><span class="p">}</span><span class="o">.</span><span class="n">macs2_peaks</span><span class="o">.</span><span class="n">narrowPeak</span><span class="o">.</span><span class="n">sorted</span> \
<span class="o">--</span><span class="n">bam</span> <span class="p">{</span><span class="n">dnase_dir</span><span class="p">}{</span><span class="n">dnase_rep1</span><span class="p">}</span><span class="n">_</span><span class="p">{</span><span class="n">dnase_rep2</span><span class="p">}</span><span class="o">.</span><span class="n">bam</span> \
<span class="o">--</span><span class="n">outDir</span> <span class="p">{</span><span class="n">peaks</span><span class="p">}</span> \
<span class="o">--</span><span class="n">chrom_sizes</span> <span class="p">{</span><span class="n">genome_file</span><span class="p">}</span> \
<span class="o">--</span><span class="n">regions_blacklist</span> <span class="p">{</span><span class="n">blacklist</span><span class="p">}</span> \
<span class="o">--</span><span class="n">peakExtendFromSummit</span> <span class="mi">250</span> \
<span class="o">--</span><span class="n">nStrongestPeaks</span> <span class="mi">500000</span>
<span class="c1">#Expected output: params.txt, foo1_foo2.macs2_peaks.narrowPeak.sorted.candidateRegions.bed, foo1_foo2.macs2_peaks.narrowPeak.sorted.foo1_foo2.bam.Counts.bed</span>
</code></pre></div>
<p>WARNING: depending on the length of the inputs, this script may be very RAM-demanding. So we recommend to allocate at least 16 GB RAM to avoid "out of memory" errors.</p>
<div class="highlight"><pre><span></span><code>%%bash<span class="w"> </span>-s<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$work_dir</span><span class="s2">&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$slurm_1_3</span><span class="s2">&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$mail_user</span><span class="s2">&quot;</span>
<span class="nv">CONDA_BASE</span><span class="o">=</span><span class="k">$(</span>conda<span class="w"> </span>info<span class="w"> </span>--base<span class="k">)</span><span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span><span class="nb">source</span><span class="w"> </span><span class="nv">$CONDA_BASE</span>/etc/profile.d/conda.sh
conda<span class="w"> </span>activate<span class="w"> </span>base<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>conda<span class="w"> </span>activate<span class="w"> </span>abcmodel
module<span class="w"> </span>load<span class="w"> </span>bioinfo/samtools-1.9<span class="w"> </span>bioinfo/bedtools-2.27.1<span class="w"> </span>bioinfo/tabix-0.2.5<span class="w"> </span>bioinfo/juicer-1.5.6
sbatch<span class="w"> </span>--mem<span class="o">=</span>16G<span class="w"> </span>--cpus-per-task<span class="o">=</span><span class="m">1</span><span class="w"> </span>-J<span class="w"> </span>step1.3<span class="w"> </span>--mail-user<span class="o">=</span><span class="nv">$3</span><span class="w"> </span>--mail-type<span class="o">=</span>END,FAIL<span class="w"> </span>--workdir<span class="o">=</span><span class="nv">$1</span><span class="w"> </span>--export<span class="o">=</span>ALL<span class="w"> </span>-p<span class="w"> </span>workq<span class="w"> </span><span class="nv">$2</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>Submitted batch job 22139094
</code></pre></div>
<h3 id="step-2-quantifying-enhancer-activity">Step 2: quantifying enhancer activity<a class="headerlink" href="#step-2-quantifying-enhancer-activity" title="Permanent link">&para;</a></h3>
<h4 id="use-abc-runneighborhoodspy">Use ABC <code>run.neighborhoods.py</code><a class="headerlink" href="#use-abc-runneighborhoodspy" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code><span class="o">%%</span><span class="n">writetemplate</span> <span class="err">$</span><span class="n">slurm_2</span>
<span class="c1">#!/bin/sh</span>
<span class="n">python</span> <span class="p">{</span><span class="n">run_neighborhoods</span><span class="p">}</span> \
<span class="o">--</span><span class="n">candidate_enhancer_regions</span> <span class="p">{</span><span class="n">peaks</span><span class="p">}{</span><span class="n">dnase_rep1</span><span class="p">}</span><span class="n">_</span><span class="p">{</span><span class="n">dnase_rep2</span><span class="p">}</span><span class="o">.</span><span class="n">macs2_peaks</span><span class="o">.</span><span class="n">narrowPeak</span><span class="o">.</span><span class="n">sorted</span><span class="o">.</span><span class="n">candidateRegions</span><span class="o">.</span><span class="n">bed</span> \
<span class="o">--</span><span class="n">genes</span> <span class="p">{</span><span class="n">annotations_dir</span><span class="p">}</span><span class="n">gene_ids</span><span class="o">.</span><span class="n">bed</span> \
<span class="o">--</span><span class="n">H3K27ac</span> <span class="p">{</span><span class="n">h3k27_file_rep1</span><span class="p">},{</span><span class="n">h3k27_file_rep1</span><span class="p">}</span> \
<span class="o">--</span><span class="n">DHS</span> <span class="p">{</span><span class="n">dnase_file_rep1</span><span class="p">},{</span><span class="n">dnase_file_rep2</span><span class="p">}</span> \
<span class="o">--</span><span class="n">expression_table</span> <span class="p">{</span><span class="n">gene_expression_table</span><span class="p">}</span>  \
<span class="o">--</span><span class="n">chrom_sizes</span> <span class="p">{</span><span class="n">genome_file</span><span class="p">}</span> \
<span class="o">--</span><span class="n">ubiquitously_expressed_genes</span> <span class="p">{</span><span class="n">ubiquitously_expressed_genes</span><span class="p">}</span> \
<span class="o">--</span><span class="n">cellType</span> <span class="p">{</span><span class="n">cell_type</span><span class="p">}</span> \
<span class="o">--</span><span class="n">outdir</span> <span class="p">{</span><span class="n">neighborhoods</span><span class="p">}</span>
</code></pre></div>
<p>WARNING: depending on the length of the inputs, this script may be very RAM-demanding. So we recommend to allocate at least 32 GB RAM to avoid "out of memory" errors.</p>
<div class="highlight"><pre><span></span><code>%%bash<span class="w"> </span>-s<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$work_dir</span><span class="s2">&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$slurm_2</span><span class="s2">&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$mail_user</span><span class="s2">&quot;</span>
<span class="nv">CONDA_BASE</span><span class="o">=</span><span class="k">$(</span>conda<span class="w"> </span>info<span class="w"> </span>--base<span class="k">)</span><span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span><span class="nb">source</span><span class="w"> </span><span class="nv">$CONDA_BASE</span>/etc/profile.d/conda.sh
conda<span class="w"> </span>activate<span class="w"> </span>base<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>conda<span class="w"> </span>activate<span class="w"> </span>abcmodel
module<span class="w"> </span>load<span class="w"> </span>bioinfo/samtools-1.9<span class="w"> </span>bioinfo/bedtools-2.27.1<span class="w"> </span>bioinfo/tabix-0.2.5<span class="w"> </span>bioinfo/juicer-1.5.6
sbatch<span class="w"> </span>--mem<span class="o">=</span>32G<span class="w"> </span>--cpus-per-task<span class="o">=</span><span class="m">1</span><span class="w"> </span>-J<span class="w"> </span>step2<span class="w"> </span>--mail-user<span class="o">=</span><span class="nv">$3</span><span class="w"> </span>--mail-type<span class="o">=</span>END,FAIL<span class="w"> </span>--workdir<span class="o">=</span><span class="nv">$1</span><span class="w"> </span>--export<span class="o">=</span>ALL<span class="w"> </span>-p<span class="w"> </span>workq<span class="w"> </span><span class="nv">$2</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>Submitted batch job 22142275
</code></pre></div>
<h3 id="step-3-computing-the-abc-score">Step 3: computing the ABC score<a class="headerlink" href="#step-3-computing-the-abc-score" title="Permanent link">&para;</a></h3>
<p>If experimentally derived contact data is not available, one can run the ABC model using the powerlaw estimate only. In this case the <code>--HiCdir</code> argument should be excluded from <code>predict.py</code> and the <code>--score_column powerlaw.Score</code> argument should be included in <code>predict.py</code>. In this case the <code>ABC.Score</code> column of the predictions file will be set to <code>NaN</code>. The <code>powerlaw.Score</code> column of the output prediction files will be the relevant Score column to use.</p>
<div class="highlight"><pre><span></span><code><span class="o">%%</span><span class="n">writetemplate</span> <span class="err">$</span><span class="n">slurm_3</span>
<span class="c1">#!/bin/sh</span>
<span class="n">python</span> <span class="p">{</span><span class="n">predict</span><span class="p">}</span> \
<span class="o">--</span><span class="n">enhancers</span> <span class="p">{</span><span class="n">neighborhoods</span><span class="p">}</span><span class="n">EnhancerList</span><span class="o">.</span><span class="n">txt</span> \
<span class="o">--</span><span class="n">genes</span> <span class="p">{</span><span class="n">neighborhoods</span><span class="p">}</span><span class="n">GeneList</span><span class="o">.</span><span class="n">txt</span> \
<span class="o">--</span><span class="n">score_column</span> <span class="n">powerlaw</span><span class="o">.</span><span class="n">Score</span> \
<span class="o">--</span><span class="n">hic_resolution</span> <span class="mi">5000</span> \
<span class="o">--</span><span class="n">scale_hic_using_powerlaw</span> \
<span class="o">--</span><span class="n">threshold</span> <span class="mf">.02</span> \
<span class="o">--</span><span class="n">cellType</span> <span class="p">{</span><span class="n">cell_type</span><span class="p">}</span> \
<span class="o">--</span><span class="n">outdir</span> <span class="p">{</span><span class="n">predictions</span><span class="p">}</span> \
<span class="o">--</span><span class="n">make_all_putative</span>
</code></pre></div>
<p>Again, that step is very RAM-demanding so we recommend to allocate at least 32G of RAM.</p>
<div class="highlight"><pre><span></span><code>%%bash<span class="w"> </span>-s<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$work_dir</span><span class="s2">&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$slurm_3</span><span class="s2">&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$mail_user</span><span class="s2">&quot;</span>
<span class="nv">CONDA_BASE</span><span class="o">=</span><span class="k">$(</span>conda<span class="w"> </span>info<span class="w"> </span>--base<span class="k">)</span><span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span><span class="nb">source</span><span class="w"> </span><span class="nv">$CONDA_BASE</span>/etc/profile.d/conda.sh
conda<span class="w"> </span>activate<span class="w"> </span>base<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>conda<span class="w"> </span>activate<span class="w"> </span>abcmodel
module<span class="w"> </span>load<span class="w"> </span>bioinfo/samtools-1.9<span class="w"> </span>bioinfo/bedtools-2.27.1<span class="w"> </span>bioinfo/tabix-0.2.5<span class="w"> </span>bioinfo/juicer-1.5.6
sbatch<span class="w"> </span>--mem<span class="o">=</span>64G<span class="w"> </span>--cpus-per-task<span class="o">=</span><span class="m">1</span><span class="w"> </span>-J<span class="w"> </span>step3<span class="w"> </span>--mail-user<span class="o">=</span><span class="nv">$3</span><span class="w"> </span>--mail-type<span class="o">=</span>END,FAIL<span class="w"> </span>--workdir<span class="o">=</span><span class="nv">$1</span><span class="w"> </span>--export<span class="o">=</span>ALL<span class="w"> </span>-p<span class="w"> </span>workq<span class="w"> </span><span class="nv">$2</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>Submitted batch job 22149081
</code></pre></div>
<h2 id="evaluation-of-abc-predictions-on-bengi-datasets">Evaluation of ABC predictions on BENGI datasets<a class="headerlink" href="#evaluation-of-abc-predictions-on-bengi-datasets" title="Permanent link">&para;</a></h2>
<h3 id="find-ccre-delss">Find ccRE-dELSs<a class="headerlink" href="#find-ccre-delss" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code>%%bash<span class="w"> </span>-s<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$ccREs</span><span class="s2">&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$ccREs_ELS</span><span class="s2">&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$genome_file</span><span class="s2">&quot;</span>
<span class="k">if</span><span class="w"> </span><span class="o">[[</span><span class="w"> </span>!<span class="w"> </span>-f<span class="w"> </span><span class="nv">$2</span><span class="w"> </span><span class="o">||</span><span class="w"> </span>!<span class="w"> </span>-f<span class="w"> </span><span class="nv">$2</span><span class="s2">&quot;.sorted&quot;</span><span class="w"> </span><span class="o">]]</span>
<span class="k">then</span>
<span class="w">    </span><span class="nb">export</span><span class="w"> </span><span class="nv">V1</span><span class="o">=</span><span class="nv">$1</span><span class="w"> </span><span class="nv">V2</span><span class="o">=</span><span class="nv">$2</span><span class="w"> </span><span class="nv">V3</span><span class="o">=</span><span class="nv">$3</span>
<span class="w">    </span>awk<span class="w"> </span>-v<span class="w"> </span><span class="nv">OFS</span><span class="o">=</span><span class="s1">&#39;\t&#39;</span><span class="w"> </span><span class="s1">&#39;$9 ~ /(^255,205,0)/ {print $1, $2, $3, &quot;.&quot;, $4, &quot;Enhancer-like&quot;}&#39;</span><span class="w"> </span><span class="nv">$V1</span><span class="w"> </span>&gt;<span class="w"> </span><span class="nv">$V2</span>
<span class="w">    </span>srun<span class="w"> </span>bash
<span class="w">    </span>bedtools<span class="w"> </span>sort<span class="w"> </span>-faidx<span class="w"> </span><span class="nv">$V3</span><span class="w"> </span>-i<span class="w"> </span><span class="nv">$V2</span><span class="w"> </span>&gt;<span class="w"> </span><span class="nv">$V2</span><span class="s2">&quot;.sorted&quot;</span>
<span class="k">fi</span>
</code></pre></div>
<p>Then using <code>bedtools closest</code> we keep only distal ccRE-ELS (ie greater than 2 kb from an annotated TSS, GENCODE v19).</p>
<div class="highlight"><pre><span></span><code>%%bash<span class="w"> </span>-s<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$genome_file</span><span class="s2">&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$ccREs_ELS</span><span class="s2">&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$TSS_list</span><span class="s2">&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$ccREs_dELS</span><span class="s2">&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$genome_file</span><span class="s2">&quot;</span>
<span class="k">if</span><span class="w"> </span><span class="o">[[</span><span class="w"> </span>!<span class="w"> </span>-f<span class="w"> </span><span class="nv">$4</span><span class="w"> </span><span class="o">]]</span>
<span class="k">then</span>
<span class="w">    </span><span class="nb">export</span><span class="w"> </span><span class="nv">V1</span><span class="o">=</span><span class="nv">$1</span><span class="w"> </span><span class="nv">V2</span><span class="o">=</span><span class="nv">$2</span><span class="w"> </span><span class="nv">V3</span><span class="o">=</span><span class="nv">$3</span><span class="w"> </span><span class="nv">V4</span><span class="o">=</span><span class="nv">$4</span><span class="w"> </span><span class="nv">V5</span><span class="o">=</span><span class="nv">$5</span>
<span class="w">    </span><span class="nv">CONDA_BASE</span><span class="o">=</span><span class="k">$(</span>conda<span class="w"> </span>info<span class="w"> </span>--base<span class="k">)</span><span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span><span class="nb">source</span><span class="w"> </span><span class="nv">$CONDA_BASE</span>/etc/profile.d/conda.sh
<span class="w">    </span>conda<span class="w"> </span>activate<span class="w"> </span>base<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>conda<span class="w"> </span>activate<span class="w"> </span>abcmodel
<span class="w">    </span>module<span class="w"> </span>load<span class="w"> </span>bioinfo/bedtools-2.27.1
<span class="w">    </span>srun<span class="w"> </span>bash
<span class="w">    </span>awk<span class="w"> </span><span class="s1">&#39;{if(NR==FNR){chr[$1]++; next}; if(chr[$1]){print $0}}&#39;</span><span class="w"> </span><span class="nv">$V5</span><span class="w"> </span><span class="nv">$V3</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>bedtools<span class="w"> </span>sort<span class="w"> </span>-faidx<span class="w"> </span><span class="nv">$V5</span><span class="w"> </span>-i<span class="w"> </span>stdin<span class="w"> </span><span class="p">|</span><span class="w"> </span>bedtools<span class="w"> </span>closest<span class="w"> </span>-d<span class="w"> </span>-t<span class="w"> </span><span class="s2">&quot;first&quot;</span><span class="w"> </span>-g<span class="w"> </span><span class="nv">$V1</span><span class="w"> </span>-a<span class="w"> </span><span class="nv">$V2</span><span class="s2">&quot;.sorted&quot;</span><span class="w"> </span>-b<span class="w"> </span>stdin<span class="w"> </span><span class="p">|</span><span class="w"> </span>awk<span class="w"> </span>-v<span class="w"> </span><span class="nv">OFS</span><span class="o">=</span><span class="s1">&#39;\t&#39;</span><span class="w"> </span><span class="s1">&#39;$14 &gt;= 2000 {print $1, $2, $3, $4, $5, &quot;distal-ELS&quot;}&#39;</span><span class="w"> </span>&gt;<span class="w"> </span><span class="nv">$V4</span>
<span class="k">fi</span>
</code></pre></div>
<h3 id="data-processing">Data processing<a class="headerlink" href="#data-processing" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code><span class="n">predictions_non_expressed</span> <span class="o">=</span> <span class="n">predictions</span><span class="o">+</span><span class="s2">&quot;EnhancerPredictionsAllPutativeNonExpressedGenes.txt&quot;</span>
<span class="n">predictions_expressed</span> <span class="o">=</span> <span class="n">predictions</span><span class="o">+</span><span class="s2">&quot;EnhancerPredictionsAllPutative.txt&quot;</span>
<span class="n">all_predictions</span> <span class="o">=</span> <span class="n">predictions</span><span class="o">+</span><span class="s2">&quot;AllPredictions.bedpe&quot;</span> <span class="c1"># does not exist yet</span>
<span class="n">all_predictions_sorted</span> <span class="o">=</span> <span class="n">all_predictions</span><span class="o">+</span><span class="s2">&quot;.sorted&quot;</span>
<span class="n">all_predictions_ccre_els</span> <span class="o">=</span> <span class="n">all_predictions_sorted</span><span class="o">+</span><span class="s2">&quot;.ccRE_intersect.txt&quot;</span>
<span class="n">all_predictions_ccre_els_bedpe</span> <span class="o">=</span> <span class="n">all_predictions_sorted</span><span class="o">+</span><span class="s2">&quot;.ccRE_intersect.bedpe&quot;</span>

<span class="n">bengi</span> <span class="o">=</span> <span class="s2">&quot;/work2/project/regenet/workspace/thoellinger/BENGI/Benchmark/All-Pairs.Natural-Ratio/&quot;</span>
<span class="n">nb_benchmarks</span> <span class="o">=</span> <span class="mi">6</span>
<span class="n">file_names</span> <span class="o">=</span> <span class="nb">list</span><span class="p">()</span>
<span class="c1"># benchmark file names without extensions</span>
<span class="n">file_names</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;GM12878.CHiC-Benchmark.v3&quot;</span><span class="p">)</span>
<span class="n">file_names</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;GM12878.CTCF-ChIAPET-Benchmark.v3&quot;</span><span class="p">)</span>
<span class="n">file_names</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;GM12878.GEUVADIS-Benchmark.v3&quot;</span><span class="p">)</span>
<span class="n">file_names</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;GM12878.GTEx-Benchmark.v3&quot;</span><span class="p">)</span>
<span class="n">file_names</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;GM12878.HiC-Benchmark.v3&quot;</span><span class="p">)</span>
<span class="n">file_names</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;GM12878.RNAPII-ChIAPET-Benchmark.v3&quot;</span><span class="p">)</span>

<span class="c1"># short custom names for benchmarks, same order as above</span>
<span class="n">names</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;CHiC&quot;</span><span class="p">,</span> <span class="s2">&quot;CTCF&quot;</span><span class="p">,</span> <span class="s2">&quot;GEUVADIS&quot;</span><span class="p">,</span> <span class="s2">&quot;GTEx&quot;</span><span class="p">,</span> <span class="s2">&quot;HiC&quot;</span><span class="p">,</span> <span class="s2">&quot;RNAPII&quot;</span><span class="p">]</span>

<span class="n">files_dict</span> <span class="o">=</span> <span class="p">{}</span>
<span class="k">for</span> <span class="n">k</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="n">nb_benchmarks</span><span class="p">):</span>
    <span class="n">files_dict</span><span class="p">[</span><span class="n">names</span><span class="p">[</span><span class="n">k</span><span class="p">]]</span> <span class="o">=</span> <span class="n">bengi</span><span class="o">+</span><span class="n">file_names</span><span class="p">[</span><span class="n">k</span><span class="p">]</span><span class="o">+</span><span class="s2">&quot;.new.bedpe&quot;</span>
</code></pre></div>
<p>Unzip ABC predictions.</p>
<div class="highlight"><pre><span></span><code><span class="k">if</span> <span class="ow">not</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">isfile</span><span class="p">(</span><span class="n">predictions_non_expressed</span><span class="p">):</span>
    <span class="err">!</span><span class="n">gzip</span> <span class="o">-</span><span class="n">cd</span> <span class="err">$</span><span class="n">predictions_non_expressed</span><span class="s2">&quot;.gz&quot;</span> <span class="o">&gt;</span> <span class="err">$</span><span class="n">predictions_non_expressed</span>
<span class="k">if</span> <span class="ow">not</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">isfile</span><span class="p">(</span><span class="n">predictions_expressed</span><span class="p">):</span>
    <span class="err">!</span><span class="n">gzip</span> <span class="o">-</span><span class="n">cd</span> <span class="err">$</span><span class="n">predictions_expressed</span><span class="s2">&quot;.gz&quot;</span> <span class="o">&gt;</span> <span class="err">$</span><span class="n">predictions_expressed</span>    
</code></pre></div>
<p>Merge ABC predictions for expressed and non expressed genes, keep only columns of interest and sort the result.</p>
<div class="highlight"><pre><span></span><code>%%bash<span class="w"> </span>-s<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$all_predictions</span><span class="s2">&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$predictions_expressed</span><span class="s2">&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$predictions_non_expressed</span><span class="s2">&quot;</span>
<span class="k">if</span><span class="w"> </span><span class="o">[[</span><span class="w"> </span>!<span class="w"> </span>-f<span class="w"> </span><span class="nv">$1</span><span class="w"> </span><span class="o">]]</span>
<span class="k">then</span>
<span class="w">    </span>awk<span class="w"> </span><span class="s1">&#39;{print $1&quot;\t&quot;$2&quot;\t&quot;$3&quot;\t&quot;$1&quot;\t&quot;$10&quot;\t&quot;$10&quot;\t&quot;$9&quot;\t&quot;$20&quot;\t.\t.\t&quot;$13&quot;\t&quot;$11}&#39;</span><span class="w"> </span><span class="nv">$2</span><span class="w"> </span>&gt;<span class="w"> </span><span class="nv">$1</span><span class="p">;</span>
<span class="w">    </span>tail<span class="w"> </span>-n+2<span class="w"> </span><span class="nv">$3</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>awk<span class="w"> </span><span class="s1">&#39;{print $1&quot;\t&quot;$2&quot;\t&quot;$3&quot;\t&quot;$1&quot;\t&quot;$10&quot;\t&quot;$10&quot;\t&quot;$9&quot;\t&quot;$20&quot;\t.\t.\t&quot;$13&quot;\t&quot;$11}&#39;</span><span class="w"> </span>&gt;&gt;<span class="w"> </span><span class="nv">$1</span><span class="p">;</span>
<span class="k">fi</span>
</code></pre></div>
<p>The following is very RAM-demanding, we recommend using at least 64GB of ram to avoid "out of memory" errors.</p>
<div class="highlight"><pre><span></span><code>%%bash<span class="w"> </span>-s<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$all_predictions</span><span class="s2">&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$all_predictions_sorted</span><span class="s2">&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$genome_file</span><span class="s2">&quot;</span>
<span class="k">if</span><span class="w"> </span><span class="o">[[</span><span class="w"> </span>!<span class="w"> </span>-f<span class="w"> </span><span class="nv">$V2</span><span class="w"> </span><span class="o">]]</span>
<span class="k">then</span>
<span class="w">    </span><span class="nb">export</span><span class="w"> </span><span class="nv">V1</span><span class="o">=</span><span class="nv">$1</span><span class="w"> </span><span class="nv">V2</span><span class="o">=</span><span class="nv">$2</span><span class="w"> </span><span class="nv">V3</span><span class="o">=</span><span class="nv">$3</span>
<span class="w">    </span><span class="nv">CONDA_BASE</span><span class="o">=</span><span class="k">$(</span>conda<span class="w"> </span>info<span class="w"> </span>--base<span class="k">)</span><span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span><span class="nb">source</span><span class="w"> </span><span class="nv">$CONDA_BASE</span>/etc/profile.d/conda.sh
<span class="w">    </span>conda<span class="w"> </span>activate<span class="w"> </span>base<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>conda<span class="w"> </span>activate<span class="w"> </span>abcmodel
<span class="w">    </span>module<span class="w"> </span>load<span class="w"> </span>bioinfo/bedtools-2.27.1
<span class="w">    </span>srun<span class="w"> </span>--mem<span class="o">=</span>64G<span class="w"> </span>bash
<span class="w">    </span>tail<span class="w"> </span>-n+2<span class="w"> </span><span class="nv">$V1</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>bedtools<span class="w"> </span>sort<span class="w"> </span>-faidx<span class="w"> </span><span class="nv">$V3</span><span class="w"> </span>-i<span class="w"> </span>stdin<span class="w">  </span>&gt;<span class="w"> </span><span class="nv">$V2</span>
<span class="k">fi</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>srun: job 22250468 queued and waiting for resources
srun: job 22250468 has been allocated resources
</code></pre></div>
<p>Intersect ABC candidate regions with ccRE-ELS.</p>
<div class="highlight"><pre><span></span><code>%%bash<span class="w"> </span>-s<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$ccREs_dELS</span><span class="s2">&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$all_predictions_sorted</span><span class="s2">&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$all_predictions_ccre_els</span><span class="s2">&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$genome_file</span><span class="s2">&quot;</span>
<span class="k">if</span><span class="w"> </span><span class="o">[[</span><span class="w"> </span>!<span class="w"> </span>-f<span class="w"> </span><span class="nv">$3</span><span class="w"> </span><span class="o">]]</span>
<span class="k">then</span>
<span class="w">    </span><span class="nb">export</span><span class="w"> </span><span class="nv">V1</span><span class="o">=</span><span class="nv">$1</span><span class="w"> </span><span class="nv">V2</span><span class="o">=</span><span class="nv">$2</span><span class="w"> </span><span class="nv">V3</span><span class="o">=</span><span class="nv">$3</span><span class="w"> </span><span class="nv">V4</span><span class="o">=</span><span class="nv">$4</span>
<span class="w">    </span><span class="nv">CONDA_BASE</span><span class="o">=</span><span class="k">$(</span>conda<span class="w"> </span>info<span class="w"> </span>--base<span class="k">)</span><span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span><span class="nb">source</span><span class="w"> </span><span class="nv">$CONDA_BASE</span>/etc/profile.d/conda.sh
<span class="w">    </span>conda<span class="w"> </span>activate<span class="w"> </span>base<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>conda<span class="w"> </span>activate<span class="w"> </span>abcmodel
<span class="w">    </span>module<span class="w"> </span>load<span class="w"> </span>bioinfo/bedtools-2.27.1
<span class="w">    </span>srun<span class="w"> </span>--mem<span class="o">=</span>32G<span class="w"> </span>bash
<span class="w">    </span>bedtools<span class="w"> </span>intersect<span class="w"> </span>-sorted<span class="w"> </span>-wo<span class="w"> </span>-g<span class="w"> </span><span class="nv">$V4</span><span class="w"> </span>-a<span class="w"> </span><span class="nv">$V1</span><span class="w"> </span>-b<span class="w"> </span><span class="nv">$V2</span><span class="w"> </span>&gt;<span class="w"> </span><span class="nv">$V3</span>
<span class="k">fi</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>srun: job 22250614 queued and waiting for resources
srun: job 22250614 has been allocated resources
</code></pre></div>
<p>When multiple ABC candidate enhancers for a given gene overlap the same ccRE-ELS, keep only the one that maximize the ABC score (the "powerlaw.Score" if ABC ran without 3D data).</p>
<div class="highlight"><pre><span></span><code>%%bash<span class="w"> </span>-s<span class="w"> </span><span class="s2">&quot;</span><span class="nv">$light_annotation</span><span class="s2">&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$all_predictions_ccre_els</span><span class="s2">&quot;</span><span class="w"> </span><span class="s2">&quot;</span><span class="nv">$all_predictions_ccre_els_bedpe</span><span class="s2">&quot;</span>
<span class="k">if</span><span class="w"> </span><span class="o">[[</span><span class="w"> </span>!<span class="w"> </span>-f<span class="w"> </span><span class="nv">$3</span><span class="w"> </span><span class="o">]]</span>
<span class="k">then</span>
<span class="w">    </span><span class="nb">export</span><span class="w"> </span><span class="nv">V1</span><span class="o">=</span><span class="nv">$1</span><span class="w"> </span><span class="nv">V2</span><span class="o">=</span><span class="nv">$2</span><span class="w"> </span><span class="nv">V3</span><span class="o">=</span><span class="nv">$3</span>
<span class="w">    </span><span class="nv">CONDA_BASE</span><span class="o">=</span><span class="k">$(</span>conda<span class="w"> </span>info<span class="w"> </span>--base<span class="k">)</span><span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span><span class="nb">source</span><span class="w"> </span><span class="nv">$CONDA_BASE</span>/etc/profile.d/conda.sh
<span class="w">    </span>conda<span class="w"> </span>activate<span class="w"> </span>base<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>conda<span class="w"> </span>activate<span class="w"> </span>abcmodel
<span class="w">    </span>module<span class="w"> </span>load<span class="w"> </span>bioinfo/bedtools-2.27.1
<span class="w">    </span>srun<span class="w"> </span>--mem<span class="o">=</span>32G<span class="w"> </span>bash
<span class="w">    </span>awk<span class="w"> </span><span class="s1">&#39;{ \</span>
<span class="s1">        if(NR==FNR){ \</span>
<span class="s1">            start[$4]=$2; \</span>
<span class="s1">            end[$4]=$3; \</span>
<span class="s1">            next; \</span>
<span class="s1">        }; \</span>
<span class="s1">        if(!pair[$5$13]){ \</span>
<span class="s1">            pair[$5$13]=$1&quot;\t&quot;start[$13]&quot;\t&quot;end[$13]&quot;\t&quot;$1&quot;\t&quot;$2&quot;\t&quot;$3&quot;\t&quot;$13&quot;:&quot;$5&quot;\t&quot;$14&quot;\t&quot;$15&quot;\t&quot;$16&quot;\t&quot;$2&quot;\t&quot;$19&quot;\t&quot;$17&quot;\t&quot;$18; \</span>
<span class="s1">            next}; \</span>
<span class="s1">        split(pair[$5$13],current_pair,&quot;\t&quot;); \</span>
<span class="s1">        if($14&gt;current_pair[$14]){ \</span>
<span class="s1">            pair[$5$13]=$1&quot;\t&quot;start[$13]&quot;\t&quot;end[$13]&quot;\t&quot;$1&quot;\t&quot;$2&quot;\t&quot;$3&quot;\t&quot;$13&quot;:&quot;$5&quot;\t&quot;$14&quot;\t&quot;$15&quot;\t&quot;$16&quot;\t&quot;$2&quot;\t&quot;$19&quot;\t&quot;$17&quot;\t&quot;$18; \</span>
<span class="s1">        }} \</span>
<span class="s1">    END{for(u in pair){ \</span>
<span class="s1">        print pair[u] \</span>
<span class="s1">        }}&#39;</span><span class="w"> </span><span class="nv">$V1</span><span class="w"> </span><span class="nv">$V2</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>bedtools<span class="w"> </span>sort<span class="w"> </span>-i<span class="w"> </span>stdin<span class="w"> </span>&gt;<span class="w"> </span><span class="nv">$V3</span>
<span class="k">fi</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>srun: job 22250736 queued and waiting for resources
srun: job 22250736 has been allocated resources
</code></pre></div>
<h3 id="intersection-between-abc-predictions-bengi">Intersection between ABC predictions / BENGI<a class="headerlink" href="#intersection-between-abc-predictions-bengi" title="Permanent link">&para;</a></h3>
<h4 id="import-files">Import files<a class="headerlink" href="#import-files" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code><span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code><span class="n">col_names</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;chr1&quot;</span><span class="p">,</span> <span class="s2">&quot;start1&quot;</span><span class="p">,</span> <span class="s2">&quot;end1&quot;</span><span class="p">,</span> <span class="s2">&quot;chr2&quot;</span><span class="p">,</span> <span class="s2">&quot;start2&quot;</span><span class="p">,</span> <span class="s2">&quot;end2&quot;</span><span class="p">,</span> <span class="s2">&quot;name&quot;</span><span class="p">,</span> <span class="s2">&quot;is.positive&quot;</span><span class="p">,</span> <span class="s2">&quot;strand1&quot;</span><span class="p">,</span> <span class="s2">&quot;strand2&quot;</span><span class="p">,</span> <span class="s2">&quot;TSSs&quot;</span><span class="p">]</span> <span class="c1"># column names in bedpe benchmarks</span>
<span class="n">col_names_abc</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;chr1&quot;</span><span class="p">,</span> <span class="s2">&quot;start1&quot;</span><span class="p">,</span> <span class="s2">&quot;end1&quot;</span><span class="p">,</span> <span class="s2">&quot;chr2&quot;</span><span class="p">,</span> <span class="s2">&quot;start2&quot;</span><span class="p">,</span> <span class="s2">&quot;end2&quot;</span><span class="p">,</span> <span class="s2">&quot;name&quot;</span><span class="p">,</span> <span class="s2">&quot;score&quot;</span><span class="p">,</span> <span class="s2">&quot;strand1&quot;</span><span class="p">,</span> <span class="s2">&quot;strand2&quot;</span><span class="p">,</span> <span class="s2">&quot;TSS&quot;</span><span class="p">,</span> <span class="s2">&quot;overlap&quot;</span><span class="p">,</span> <span class="s2">&quot;is.expressed&quot;</span><span class="p">,</span> <span class="s2">&quot;expression&quot;</span><span class="p">]</span> <span class="c1"># column names in abc bedpe</span>
<span class="n">benchmarks</span> <span class="o">=</span> <span class="p">{}</span>
<span class="k">for</span> <span class="n">k</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="n">nb_benchmarks</span><span class="p">):</span>
    <span class="n">benchmarks</span><span class="p">[</span><span class="n">names</span><span class="p">[</span><span class="n">k</span><span class="p">]]</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">files_dict</span><span class="p">[</span><span class="n">names</span><span class="p">[</span><span class="n">k</span><span class="p">]],</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">usecols</span><span class="o">=</span><span class="p">[</span><span class="mi">0</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="mi">4</span><span class="p">,</span> <span class="mi">5</span><span class="p">,</span> <span class="mi">6</span><span class="p">,</span> <span class="mi">7</span><span class="p">,</span> <span class="mi">10</span><span class="p">],</span> <span class="n">names</span><span class="o">=</span><span class="n">col_names</span><span class="p">,</span> <span class="n">dtype</span><span class="o">=</span><span class="s1">&#39;str&#39;</span><span class="p">,</span> <span class="n">engine</span><span class="o">=</span><span class="s1">&#39;c&#39;</span><span class="p">)</span>

<span class="n">abc</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">all_predictions_ccre_els_bedpe</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">usecols</span><span class="o">=</span><span class="p">[</span><span class="mi">0</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="mi">4</span><span class="p">,</span> <span class="mi">5</span><span class="p">,</span> <span class="mi">6</span><span class="p">,</span> <span class="mi">7</span><span class="p">,</span> <span class="mi">10</span><span class="p">,</span> <span class="mi">11</span><span class="p">],</span> <span class="n">names</span><span class="o">=</span><span class="n">col_names_abc</span><span class="p">,</span> <span class="n">dtype</span><span class="o">=</span><span class="s1">&#39;str&#39;</span><span class="p">,</span> <span class="n">engine</span><span class="o">=</span><span class="s1">&#39;c&#39;</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code><span class="n">intersection</span> <span class="o">=</span> <span class="p">{}</span>
<span class="k">for</span> <span class="n">k</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="n">nb_benchmarks</span><span class="p">):</span>
    <span class="n">intersection</span><span class="p">[</span><span class="n">names</span><span class="p">[</span><span class="n">k</span><span class="p">]]</span> <span class="o">=</span> <span class="n">benchmarks</span><span class="p">[</span><span class="n">names</span><span class="p">[</span><span class="n">k</span><span class="p">]]</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span><span class="n">abc</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:,[</span><span class="mi">5</span><span class="p">,</span><span class="mi">6</span><span class="p">,</span><span class="mi">8</span><span class="p">]],</span> <span class="n">how</span><span class="o">=</span><span class="s1">&#39;inner&#39;</span><span class="p">,</span> <span class="n">left_on</span><span class="o">=</span><span class="n">col_names</span><span class="p">[</span><span class="mi">6</span><span class="p">],</span> <span class="n">right_on</span><span class="o">=</span><span class="n">col_names_abc</span><span class="p">[</span><span class="mi">6</span><span class="p">])</span>

<span class="n">col_summary</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;ABC.length&#39;</span><span class="p">,</span> <span class="s1">&#39;BENGI.length&#39;</span><span class="p">,</span> <span class="s1">&#39;intersection.length&#39;</span><span class="p">,</span> <span class="s1">&#39;intersection.length (%)&#39;</span><span class="p">,</span> <span class="s1">&#39;BENGI.positives&#39;</span><span class="p">,</span> <span class="s1">&#39;intersection.positives&#39;</span><span class="p">,</span> <span class="s1">&#39;intersection.positives (%)&#39;</span><span class="p">]</span>
<span class="n">summary</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="n">col_summary</span><span class="p">)</span>
<span class="k">for</span> <span class="n">k</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="n">nb_benchmarks</span><span class="p">):</span>
    <span class="n">len_abc</span> <span class="o">=</span> <span class="nb">len</span><span class="p">(</span><span class="n">abc</span><span class="p">)</span>
    <span class="n">len_bengi</span> <span class="o">=</span> <span class="nb">len</span><span class="p">(</span><span class="n">benchmarks</span><span class="p">[</span><span class="n">names</span><span class="p">[</span><span class="n">k</span><span class="p">]])</span>
    <span class="n">len_inter</span> <span class="o">=</span> <span class="nb">len</span><span class="p">(</span><span class="n">intersection</span><span class="p">[</span><span class="n">names</span><span class="p">[</span><span class="n">k</span><span class="p">]])</span>
    <span class="n">ratio</span> <span class="o">=</span> <span class="n">len_inter</span><span class="o">/</span><span class="n">len_bengi</span><span class="o">*</span><span class="mi">100</span>
    <span class="n">bengi_positives</span> <span class="o">=</span> <span class="nb">sum</span><span class="p">(</span><span class="n">benchmarks</span><span class="p">[</span><span class="n">names</span><span class="p">[</span><span class="n">k</span><span class="p">]][</span><span class="n">col_names</span><span class="p">[</span><span class="mi">7</span><span class="p">]]</span> <span class="o">==</span> <span class="s2">&quot;1&quot;</span><span class="p">)</span>
    <span class="n">inter_positives</span> <span class="o">=</span> <span class="nb">sum</span><span class="p">(</span><span class="n">intersection</span><span class="p">[</span><span class="n">names</span><span class="p">[</span><span class="n">k</span><span class="p">]][</span><span class="n">col_names</span><span class="p">[</span><span class="mi">7</span><span class="p">]]</span> <span class="o">==</span> <span class="s2">&quot;1&quot;</span><span class="p">)</span>
    <span class="n">ratio_positives</span> <span class="o">=</span> <span class="n">inter_positives</span><span class="o">/</span><span class="n">bengi_positives</span><span class="o">*</span><span class="mi">100</span>
    <span class="n">summary</span> <span class="o">=</span> <span class="n">summary</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">([[</span><span class="n">len_abc</span><span class="p">,</span> <span class="n">len_bengi</span><span class="p">,</span> <span class="n">len_inter</span><span class="p">,</span> <span class="n">ratio</span><span class="p">,</span> <span class="n">bengi_positives</span><span class="p">,</span> <span class="n">inter_positives</span><span class="p">,</span> <span class="n">ratio_positives</span><span class="p">]],</span> <span class="n">columns</span><span class="o">=</span><span class="n">col_summary</span><span class="p">),</span> <span class="n">ignore_index</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">summary</span><span class="o">.</span><span class="n">index</span> <span class="o">=</span> <span class="n">names</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code><span class="n">summary</span>
</code></pre></div>
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>ABC.length</th>
      <th>BENGI.length</th>
      <th>intersection.length</th>
      <th>intersection.length (%)</th>
      <th>BENGI.positives</th>
      <th>intersection.positives</th>
      <th>intersection.positives (%)</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>CHiC</th>
      <td>1828627</td>
      <td>375728</td>
      <td>118550</td>
      <td>31.552080</td>
      <td>88245</td>
      <td>28194</td>
      <td>31.949686</td>
    </tr>
    <tr>
      <th>CTCF</th>
      <td>1828627</td>
      <td>105016</td>
      <td>32037</td>
      <td>30.506780</td>
      <td>7591</td>
      <td>2222</td>
      <td>29.271506</td>
    </tr>
    <tr>
      <th>GEUVADIS</th>
      <td>1828627</td>
      <td>50999</td>
      <td>17452</td>
      <td>34.220279</td>
      <td>2073</td>
      <td>779</td>
      <td>37.578389</td>
    </tr>
    <tr>
      <th>GTEx</th>
      <td>1828627</td>
      <td>38200</td>
      <td>14224</td>
      <td>37.235602</td>
      <td>1301</td>
      <td>546</td>
      <td>41.967717</td>
    </tr>
    <tr>
      <th>HiC</th>
      <td>1828627</td>
      <td>153739</td>
      <td>50071</td>
      <td>32.568834</td>
      <td>3404</td>
      <td>1123</td>
      <td>32.990599</td>
    </tr>
    <tr>
      <th>RNAPII</th>
      <td>1828627</td>
      <td>157235</td>
      <td>52821</td>
      <td>33.593666</td>
      <td>23699</td>
      <td>8300</td>
      <td>35.022575</td>
    </tr>
  </tbody>
</table>
</div>

<h4 id="export-files">Export files<a class="headerlink" href="#export-files" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code><span class="k">if</span> <span class="ow">not</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">isdir</span><span class="p">(</span><span class="n">predictions</span><span class="o">+</span><span class="s2">&quot;BENGI_intersect/&quot;</span><span class="p">):</span>
    <span class="err">!</span><span class="n">mkdir</span> <span class="err">$</span><span class="n">predictions</span><span class="s2">&quot;BENGI_intersect/&quot;</span>
<span class="k">for</span> <span class="n">k</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="n">nb_benchmarks</span><span class="p">):</span>
    <span class="n">intersection</span><span class="p">[</span><span class="n">names</span><span class="p">[</span><span class="n">k</span><span class="p">]]</span><span class="o">.</span><span class="n">drop</span><span class="p">(</span><span class="n">intersection</span><span class="p">[</span><span class="n">names</span><span class="p">[</span><span class="mi">1</span><span class="p">]]</span><span class="o">.</span><span class="n">columns</span><span class="p">[</span><span class="o">-</span><span class="mi">1</span><span class="p">],</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="n">predictions</span><span class="o">+</span><span class="s2">&quot;BENGI_intersect/AllPredictions.intersect.&quot;</span><span class="o">+</span><span class="n">file_names</span><span class="p">[</span><span class="n">k</span><span class="p">]</span><span class="o">+</span><span class="s2">&quot;.bedpe&quot;</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">,</span><span class="n">header</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span><span class="n">index</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
</code></pre></div>





                
              </article>
            </div>
          
          
        </div>
        
          <a href="#" class="md-top md-icon" data-md-component="top" hidden>
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg>
            Back to top
          </a>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      2020 - 2022 Tristan Hoellinger
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
        <div class="md-social">
  
    
    
    
    
      
      
    
    <a href="https://twitter.com/TristanHoellin1" target="_blank" rel="noopener" title="twitter.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"/></svg>
    </a>
  
    
    
    
    
      
      
    
    <a href="https://fr.linkedin.com/in/tristan-hoellinger" target="_blank" rel="noopener" title="fr.linkedin.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"/></svg>
    </a>
  
    
    
    
    
    <a href="mailto:sarah.djebali@inserm.fr" target="_blank" rel="noopener" title="" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M498.1 5.6c10.1 7 15.4 19.1 13.5 31.2l-64 416c-1.5 9.7-7.4 18.2-16 23s-18.9 5.4-28 1.6L284 427.7l-68.5 74.1c-8.9 9.7-22.9 12.9-35.2 8.1S160 493.2 160 480v-83.6c0-4 1.5-7.8 4.2-10.7l167.6-182.9c5.8-6.3 5.6-16-.4-22s-15.7-6.4-22-.7L106 360.8l-88.3-44.2C7.1 311.3.3 300.7 0 288.9s5.9-22.8 16.1-28.7l448-256c10.7-6.1 23.9-5.5 34 1.4z"/></svg>
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    <script id="__config" type="application/json">{"base": "../..", "features": ["navigation.instant", "navigation.tabs", "navigation.sections", "navigation.top"], "search": "../../assets/javascripts/workers/search.db81ec45.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../../assets/javascripts/bundle.6df46069.min.js"></script>
      
    
  </body>
</html>