
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
        <meta name="author" content="Tristan Hoellinger">
      
      
        <link rel="canonical" href="https://eg.github.io/haemochromatosis/CHiC/go_enrichment.html">
      
      
        <link rel="prev" href="preliminary_analysis_chic_v2.html">
      
      
        <link rel="next" href="../ABC/data_processing.html">
      
      <link rel="icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.4.2, mkdocs-material-9.0.11">
    
    
      
        <title>GO enrichment - Genome-wide identification of E/G interactions</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.0d440cfe.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.2505c338.min.css">
      
      

    
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../stylesheets/extra.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="black" data-md-color-accent="">
  
    
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#go-enrichment-of-all-genes-obtained" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Genome-wide identification of E/G interactions" class="md-header__button md-logo" aria-label="Genome-wide identification of E/G interactions" data-md-component="logo">
      
  <img src="../../logo/logo-inserm-blc.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Genome-wide identification of E/G interactions
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              GO enrichment
            
          </span>
        </div>
      </div>
    </div>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/hoellin/eg" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
  </div>
  <div class="md-source__repository">
    hoellin/eg
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../guidebooks/introduction.html" class="md-tabs__link">
        Guidebooks
      </a>
    </li>
  

      
        
  
  
    
  


  
  
  
    <li class="md-tabs__item">
      <a href="../introduction.html" class="md-tabs__link md-tabs__link--active">
        Haemochromatosis
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../notes_BENGI/introduction.html" class="md-tabs__link">
        BENGI
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../notes_ABC/introduction.html" class="md-tabs__link">
        ABC
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../index.html" class="md-tabs__link">
        About
      </a>
    </li>
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Genome-wide identification of E/G interactions" class="md-nav__button md-logo" aria-label="Genome-wide identification of E/G interactions" data-md-component="logo">
      
  <img src="../../logo/logo-inserm-blc.png" alt="logo">

    </a>
    Genome-wide identification of E/G interactions
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/hoellin/eg" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
  </div>
  <div class="md-source__repository">
    hoellin/eg
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1" >
      
      
      
        <label class="md-nav__link" for="__nav_1" id="__nav_1_label" tabindex="0">
          Guidebooks
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_1_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_1">
          <span class="md-nav__icon md-icon"></span>
          Guidebooks
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../guidebooks/introduction.html" class="md-nav__link">
        Introduction
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../guidebooks/script_bedpe_summary/script_bedpe_summary.html" class="md-nav__link">
        Compute summary stats of a bedpe
      </a>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_3" >
      
      
      
        <label class="md-nav__link" for="__nav_1_3" id="__nav_1_3_label" tabindex="0">
          Basics
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_1_3">
          <span class="md-nav__icon md-icon"></span>
          Basics
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../guidebooks/shell.html" class="md-nav__link">
        Shell
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../guidebooks/conda_environments.html" class="md-nav__link">
        Conda basics
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../guidebooks/git.html" class="md-nav__link">
        Git
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../guidebooks/genotoul.html" class="md-nav__link">
        Genotoul cluster
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../guidebooks/notebooks_ssh.html" class="md-nav__link">
        How to use Jupyter through ssh tunneling?
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_4" >
      
      
      
        <label class="md-nav__link" for="__nav_1_4" id="__nav_1_4_label" tabindex="0">
          Misc
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_4_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_1_4">
          <span class="md-nav__icon md-icon"></span>
          Misc
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../guidebooks/compute_correlations.html" class="md-nav__link">
        Compute correlations
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" checked>
      
      
      
        <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
          Haemochromatosis
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="true">
        <label class="md-nav__title" for="__nav_2">
          <span class="md-nav__icon md-icon"></span>
          Haemochromatosis
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../introduction.html" class="md-nav__link">
        Introduction
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../data_format.html" class="md-nav__link">
        Data format
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../ABC/custom_go.html" class="md-nav__link">
        Custom script GO
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../main_results.html" class="md-nav__link">
        Main results
      </a>
    </li>
  

            
          
            
              
  
  
    
  
  
    
      
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2_5" checked>
      
      
      
        <label class="md-nav__link" for="__nav_2_5" id="__nav_2_5_label" tabindex="0">
          CHiC data
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_2_5_label" aria-expanded="true">
        <label class="md-nav__title" for="__nav_2_5">
          <span class="md-nav__icon md-icon"></span>
          CHiC data
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="preprocess_data.html" class="md-nav__link">
        Preprocessing
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="preliminary_analysis_chic_v2.html" class="md-nav__link">
        Networks
      </a>
    </li>
  

            
          
            
              
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          GO enrichment
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="go_enrichment.html" class="md-nav__link md-nav__link--active">
        GO enrichment
      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#dependencies" class="md-nav__link">
    Dependencies
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#main-remarks" class="md-nav__link">
    Main remarks
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#collecting-the-list-of-genes-of-interest" class="md-nav__link">
    Collecting the list of genes of interest
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#go-enrichment" class="md-nav__link">
    GO enrichment
  </a>
  
    <nav class="md-nav" aria-label="GO enrichment">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#all-54-genes-original-inferred" class="md-nav__link">
    All 54 genes (original + inferred)
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#all-42-genes-inferred-only-wo-original-genes" class="md-nav__link">
    All 42 genes (inferred only ; w/o original genes)
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#all-genes-co-expressed-in-liver" class="md-nav__link">
    All genes co-expressed in liver
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#all-genes-expressed-in-intestine" class="md-nav__link">
    All genes expressed in intestine
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#each-original-gene-separately" class="md-nav__link">
    Each original gene separately
  </a>
  
    <nav class="md-nav" aria-label="Each original gene separately">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#inferred-from-ciapin1" class="md-nav__link">
    Inferred from CIAPIN1
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#inferred-from-ciapin1-wo-ciapin1" class="md-nav__link">
    Inferred from CIAPIN1 w/o CIAPIN1
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#inferred-from-hfe2" class="md-nav__link">
    Inferred from HFE2
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#inferred-from-hfe2-wo-hfe2" class="md-nav__link">
    Inferred from HFE2 w/o HFE2
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#inferred-from-neo1" class="md-nav__link">
    Inferred from NEO1
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#inferred-from-neo1-wo-neo1" class="md-nav__link">
    Inferred from NEO1 w/o NEO1
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#inferred-from-slc11a2" class="md-nav__link">
    Inferred from SLC11A2
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#inferred-from-slc11a2-wo-slc11a2" class="md-nav__link">
    Inferred from SLC11A2 w/o SLC11A2
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#inferred-from-slc39a14" class="md-nav__link">
    Inferred from SLC39A14
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#inferred-from-slc39a14-wo-slc39a14" class="md-nav__link">
    Inferred from SLC39A14 w/o SLC39A14
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#inferred-from-tfr2" class="md-nav__link">
    Inferred from TFR2
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#inferred-from-tfr2-wo-tfr2" class="md-nav__link">
    Inferred from TFR2 w/o TFR2
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#inferred-from-tmprss6" class="md-nav__link">
    Inferred from TMPRSS6
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#inferred-from-tmprss6-wo-tmprss6" class="md-nav__link">
    Inferred from TMPRSS6 w/o TMPRSS6
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2_6" >
      
      
      
        <label class="md-nav__link" for="__nav_2_6" id="__nav_2_6_label" tabindex="0">
          ABC data
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_2_6_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_2_6">
          <span class="md-nav__icon md-icon"></span>
          ABC data
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../ABC/data_processing.html" class="md-nav__link">
        Preprocessing
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../ABC/preliminary_analysis_v9.html" class="md-nav__link">
        Networks
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../ABC/summary_statistics_eg_pairs.html" class="md-nav__link">
        Summary statistics enhancers
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../ABC/go_enrichment.html" class="md-nav__link">
        GO enrichment
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../ABC/gorilla.html" class="md-nav__link">
        GOrilla
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../ABC/compute_avg_nb_of_connections_per_gene.html" class="md-nav__link">
        Avg nb connections per gene
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../ABC/gwas.html" class="md-nav__link">
        GWAS (13 initial genes)
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../ABC/gwas_10.html" class="md-nav__link">
        GWAS (10 initial genes)
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" >
      
      
      
        <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
          BENGI
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3">
          <span class="md-nav__icon md-icon"></span>
          BENGI
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/introduction.html" class="md-nav__link">
        Introduction
      </a>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_2" >
      
      
      
        <label class="md-nav__link" for="__nav_3_2" id="__nav_3_2_label" tabindex="0">
          Notebooks
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_2_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_2">
          <span class="md-nav__icon md-icon"></span>
          Notebooks
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/BENGI_to_custom_bedpe.html" class="md-nav__link">
        Convert BENGI datasets to custom bedpe
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_3" >
      
      
      
        <label class="md-nav__link" for="__nav_3_3" id="__nav_3_3_label" tabindex="0">
          Summary statistics
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_3">
          <span class="md-nav__icon md-icon"></span>
          Summary statistics
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/summary_statistics/summary_statistics.html" class="md-nav__link">
        Summary statistics on BENGI GM12878 benchmarks
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/summary_statistics_BENGI/summary_BENGI_bedpe.html" class="md-nav__link">
        (R code) Compute summary statistics over BENGI
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_4" >
      
      
      
        <label class="md-nav__link" for="__nav_3_4" id="__nav_3_4_label" tabindex="0">
          Evaluation of a few methods over BENGI GM12878
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_4_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_4">
          <span class="md-nav__icon md-icon"></span>
          Evaluation of a few methods over BENGI GM12878
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_4_1" >
      
      
      
        <label class="md-nav__link" for="__nav_3_4_1" id="__nav_3_4_1_label" tabindex="0">
          Distance method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_4_1_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_4_1">
          <span class="md-nav__icon md-icon"></span>
          Distance method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/distance_method/distance_evaluation_with_code.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/distance_method/summary_distance_evaluation.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_4_2" >
      
      
      
        <label class="md-nav__link" for="__nav_3_4_2" id="__nav_3_4_2_label" tabindex="0">
          DNase-expression correlation method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_4_2_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_4_2">
          <span class="md-nav__icon md-icon"></span>
          DNase-expression correlation method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/dnase_expression_correlation/correlation_method_with_code.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/dnase_expression_correlation/summary_correlation_method.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_4_3" >
      
      
      
        <label class="md-nav__link" for="__nav_3_4_3" id="__nav_3_4_3_label" tabindex="0">
          Average rank method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_4_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_4_3">
          <span class="md-nav__icon md-icon"></span>
          Average rank method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/avg_rank_method/avg_rank_method_with_code.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/avg_rank_method/summary_avg_rank_method.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_5" >
      
      
      
        <label class="md-nav__link" for="__nav_3_5" id="__nav_3_5_label" tabindex="0">
          Evaluation of a few methods over Fulco et al 2019 CRISPRi-FlowFISH
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_5_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_5">
          <span class="md-nav__icon md-icon"></span>
          Evaluation of a few methods over Fulco et al 2019 CRISPRi-FlowFISH
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_5_1" >
      
      
      
        <label class="md-nav__link" for="__nav_3_5_1" id="__nav_3_5_1_label" tabindex="0">
          Distance method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_5_1_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_5_1">
          <span class="md-nav__icon md-icon"></span>
          Distance method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/CRISPRi_FlowFISH/distance_method/distance_over_fulco_et_al_crispri.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/CRISPRi_FlowFISH/distance_method/summary_distance_over_fulco_et_al_crispri.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_5_2" >
      
      
      
        <label class="md-nav__link" for="__nav_3_5_2" id="__nav_3_5_2_label" tabindex="0">
          DNase-expression correlation method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_5_2_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_5_2">
          <span class="md-nav__icon md-icon"></span>
          DNase-expression correlation method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/CRISPRi_FlowFISH/dnase_expression_correlation/correlation_method_with_code.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/CRISPRi_FlowFISH/dnase_expression_correlation/summary_correlation_method.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_5_3" >
      
      
      
        <label class="md-nav__link" for="__nav_3_5_3" id="__nav_3_5_3_label" tabindex="0">
          Average rank method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_5_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_5_3">
          <span class="md-nav__icon md-icon"></span>
          Average rank method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/CRISPRi_FlowFISH/avg_rank_method/avg_rank_method_with_code.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/CRISPRi_FlowFISH/avg_rank_method/summary_avg_rank.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" >
      
      
      
        <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
          ABC
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_4">
          <span class="md-nav__icon md-icon"></span>
          ABC
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/introduction.html" class="md-nav__link">
        Introduction
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/BENGI/ABC_BENGI_results.html" class="md-nav__link">
        ABC over BENGI for GM12878, main results
      </a>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_3" >
      
      
      
        <label class="md-nav__link" for="__nav_4_3" id="__nav_4_3_label" tabindex="0">
          CRISPRi-FlowFISH
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_4_3">
          <span class="md-nav__icon md-icon"></span>
          CRISPRi-FlowFISH
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/K562/ABC_K562_CRISPRi_FlowFISH.html" class="md-nav__link">
        Original ABC predictions over K562
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/K562/recap_investigations_ABC_over_CRiFF.html" class="md-nav__link">
        Summary ABC over CRiFF
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_4" >
      
      
      
        <label class="md-nav__link" for="__nav_4_4" id="__nav_4_4_label" tabindex="0">
          Notebooks
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_4_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_4_4">
          <span class="md-nav__icon md-icon"></span>
          Notebooks
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/K562/april_K562_56_genes/april_K562_56_genes.html" class="md-nav__link">
        ABC over K562 cCRiFF
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/K562/april_K562_59_genes/april_K562_59_genes.html" class="md-nav__link">
        ABC over K562 CRiFF
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/K562/april_K562_candidates_not_in_whitelist_56_genes/april_K562_candidates_NOT_in_whitelist.html" class="md-nav__link">
        ABC over K562 cCRiFF, no whitelist
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/generic_notebooks/turnkey_notebook_to_run_ABC_with_example_over_GM12878.html" class="md-nav__link">
        Generic notebook to run ABC method & Example over GM12878
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/BENGI/notebook_ABC_over_BENGI_GM12878.html" class="md-nav__link">
        ABC over GM12878 with same accessions as for BENGI
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/BENGI/notebook_ABC_over_BENGI_GM12878_from_ccRE_ELSs.html" class="md-nav__link">
        ABC over GM12878 with same accessions as for BENGI, starting with ccRE-dELS
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_5" >
      
      
      
        <label class="md-nav__link" for="__nav_4_5" id="__nav_4_5_label" tabindex="0">
          Full codes
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_5_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_4_5">
          <span class="md-nav__icon md-icon"></span>
          Full codes
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/test_abc_pipeline.html" class="md-nav__link">
        Test of ABC pipeline over chr22
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/BENGI/ABC_over_BENGI_GM12878.html" class="md-nav__link">
        ABC over GM12878 with same accessions as for BENGI
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/differences_example_chr22.html" class="md-nav__link">
        Meld of results over chr22 vs authors' results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" >
      
      
      
        <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
          About
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_5">
          <span class="md-nav__icon md-icon"></span>
          About
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../index.html" class="md-nav__link">
        Getting started
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../about_me.html" class="md-nav__link">
        About me
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#dependencies" class="md-nav__link">
    Dependencies
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#main-remarks" class="md-nav__link">
    Main remarks
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#collecting-the-list-of-genes-of-interest" class="md-nav__link">
    Collecting the list of genes of interest
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#go-enrichment" class="md-nav__link">
    GO enrichment
  </a>
  
    <nav class="md-nav" aria-label="GO enrichment">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#all-54-genes-original-inferred" class="md-nav__link">
    All 54 genes (original + inferred)
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#all-42-genes-inferred-only-wo-original-genes" class="md-nav__link">
    All 42 genes (inferred only ; w/o original genes)
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#all-genes-co-expressed-in-liver" class="md-nav__link">
    All genes co-expressed in liver
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#all-genes-expressed-in-intestine" class="md-nav__link">
    All genes expressed in intestine
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#each-original-gene-separately" class="md-nav__link">
    Each original gene separately
  </a>
  
    <nav class="md-nav" aria-label="Each original gene separately">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#inferred-from-ciapin1" class="md-nav__link">
    Inferred from CIAPIN1
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#inferred-from-ciapin1-wo-ciapin1" class="md-nav__link">
    Inferred from CIAPIN1 w/o CIAPIN1
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#inferred-from-hfe2" class="md-nav__link">
    Inferred from HFE2
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#inferred-from-hfe2-wo-hfe2" class="md-nav__link">
    Inferred from HFE2 w/o HFE2
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#inferred-from-neo1" class="md-nav__link">
    Inferred from NEO1
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#inferred-from-neo1-wo-neo1" class="md-nav__link">
    Inferred from NEO1 w/o NEO1
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#inferred-from-slc11a2" class="md-nav__link">
    Inferred from SLC11A2
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#inferred-from-slc11a2-wo-slc11a2" class="md-nav__link">
    Inferred from SLC11A2 w/o SLC11A2
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#inferred-from-slc39a14" class="md-nav__link">
    Inferred from SLC39A14
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#inferred-from-slc39a14-wo-slc39a14" class="md-nav__link">
    Inferred from SLC39A14 w/o SLC39A14
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#inferred-from-tfr2" class="md-nav__link">
    Inferred from TFR2
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#inferred-from-tfr2-wo-tfr2" class="md-nav__link">
    Inferred from TFR2 w/o TFR2
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#inferred-from-tmprss6" class="md-nav__link">
    Inferred from TMPRSS6
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#inferred-from-tmprss6-wo-tmprss6" class="md-nav__link">
    Inferred from TMPRSS6 w/o TMPRSS6
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  

  
  



<h1 id="go-enrichment-of-all-genes-obtained">GO enrichment of all genes obtained<a class="headerlink" href="#go-enrichment-of-all-genes-obtained" title="Permanent link">&para;</a></h1>
<p>We use <code>GO_enrichment_FDR.R</code> which allows us to choose between multiple procedures for multiple testing correction. Here we are interested in controlling the FDR ; we can use either <code>pAdjustMethod = "BH"</code> (Benjamini-Hochberg, default parameter) or <code>pAdjustMethod = "fdr"</code>.</p>
<p>All results (list of genes and detailed GO enrichment) can be found here:</p>
<div class="highlight"><pre><span></span><code>/work2/project/regenet/workspace/thoellinger/shared/2022/promoter_capture_hic/
</code></pre></div>
<h2 id="dependencies">Dependencies<a class="headerlink" href="#dependencies" title="Permanent link">&para;</a></h2>
<p>We use R 4.0. We need the <code>clusterProfiler</code> package. See <a href="../ABC/custom_go.html"><code>custom_go.md</code></a> for installation details.</p>
<h2 id="main-remarks">Main remarks<a class="headerlink" href="#main-remarks" title="Permanent link">&para;</a></h2>
<ul>
<li>From time to time, one may get the following error while using <code>GO_enrichment_FDR.R</code>:</li>
</ul>
<div class="highlight"><pre><span></span><code><span class="n">Error</span><span class="w"> </span><span class="n">in</span><span class="w"> </span><span class="nf">`[.data.frame`</span><span class="p">(</span><span class="n">d</span><span class="p">,</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">)</span><span class="w"> </span><span class="o">:</span><span class="w"> </span><span class="n">undefined</span><span class="w"> </span><span class="n">columns</span><span class="w"> </span><span class="n">selected</span>
</code></pre></div>
<p>This is expected, and should be considered as a warning only (I did not take the time to handle the case with an exception). This happens when no significant GO terms are found ; in such case, no graphical outputs are computed.</p>
<h2 id="collecting-the-list-of-genes-of-interest">Collecting the list of genes of interest<a class="headerlink" href="#collecting-the-list-of-genes-of-interest" title="Permanent link">&para;</a></h2>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span><span class="w"> </span>/work2/project/regenet/workspace/thoellinger/shared/2022/promoter_capture_hic/
</code></pre></div>
<p>We obtained this list using our R markdown doing network analysis of E-G pairs starting from genes directly involved in hemochromatosis or involved in the regulation of iron metabolism.</p>
<p>We saved this list of 54 (12 known + 42 inferred) genes as <code>new_genes_v1.list</code>. For each one of the 13 initial genes, we also saved separately the corresponding inferred genes. I a few cases, no genes were inferred starting from one of the 13 genes (+ for one of the 13, we did not have CHiC data at all):</p>
<blockquote>
<div class="highlight"><pre><span></span><code>└──<span class="w"> </span>results
<span class="w"> </span>...
├──<span class="w"> </span>13_initial_genes.list<span class="w"> </span><span class="c1"># 13</span>
├──<span class="w"> </span>new_genes_v1.list<span class="w"> </span><span class="c1"># 54</span>
└──<span class="w"> </span>separate
<span class="w">    </span>├──<span class="w"> </span>BMP6.list<span class="w"> </span><span class="m">1</span>
<span class="w">    </span>├──<span class="w"> </span>CIAPIN1.list<span class="w"> </span><span class="c1"># 8</span>
<span class="w">    </span>├──<span class="w"> </span>CYBRD1.list<span class="w"> </span><span class="m">1</span>
<span class="w">    </span>├──<span class="w"> </span>HFE2.list<span class="w"> </span><span class="c1"># 5</span>
<span class="w">    </span>├──<span class="w"> </span>HFE.list<span class="w"> </span><span class="m">1</span>
<span class="w">    </span>├──<span class="w"> </span>NEO1.list<span class="w"> </span><span class="c1"># 2</span>
<span class="w">    </span>├──<span class="w"> </span>SLC11A2.list<span class="w"> </span><span class="c1"># 8</span>
<span class="w">    </span>├──<span class="w"> </span>SLC39A14.list<span class="w"> </span><span class="c1"># 4</span>
<span class="w">    </span>├──<span class="w"> </span>SLC40A1.list<span class="w"> </span><span class="m">1</span>
<span class="w">    </span>├──<span class="w"> </span>TFR2.list<span class="w"> </span><span class="c1"># 15</span>
<span class="w">    </span>├──<span class="w"> </span>TFRC.list<span class="w"> </span><span class="m">1</span>
<span class="w">    </span>└──<span class="w"> </span>TMPRSS6.list<span class="w"> </span><span class="c1"># 7</span>
</code></pre></div>
</blockquote>
<h2 id="go-enrichment">GO enrichment<a class="headerlink" href="#go-enrichment" title="Permanent link">&para;</a></h2>
<div class="highlight"><pre><span></span><code>module<span class="w"> </span>load<span class="w"> </span>system/R-4.0.4_gcc-9.3.0
</code></pre></div>
<h3 id="all-54-genes-original-inferred">All 54 genes (original + inferred)<a class="headerlink" href="#all-54-genes-original-inferred" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code>mkdir<span class="w"> </span>-p<span class="w"> </span>GO_FDR/all_genes/symbol
</code></pre></div>
<div class="highlight"><pre><span></span><code>Rscript<span class="w"> </span>GO_enrichment_FDR.R<span class="w"> </span>-k<span class="w"> </span><span class="s2">&quot;SYMBOL&quot;</span><span class="w"> </span>-G<span class="w"> </span>results/new_genes_v1.list<span class="w"> </span>-f<span class="w"> </span><span class="m">0</span>.05<span class="w"> </span>-c<span class="w"> </span><span class="s2">&quot;BP&quot;</span><span class="w"> </span>-a<span class="w"> </span><span class="s2">&quot;BH&quot;</span><span class="w"> </span>-o<span class="w"> </span><span class="s2">&quot;default_universe&quot;</span><span class="w"> </span>-d<span class="w"> </span><span class="s2">&quot;GO_FDR/all_genes/symbol&quot;</span>
</code></pre></div>
<blockquote>
<div class="highlight"><pre><span></span><code><span class="kc">...</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Loading input data...&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Warning: using defaut universe automatically provided by the clusterProfiler package&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done.&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Computing GO enrichment...&quot;</span>
<span class="n">`universe`</span><span class="w"> </span><span class="n">is</span><span class="w"> </span><span class="n">not</span><span class="w"> </span><span class="n">in</span><span class="w"> </span><span class="n">character</span><span class="w"> </span><span class="n">and</span><span class="w"> </span><span class="n">will</span><span class="w"> </span><span class="n">be</span><span class="w"> </span><span class="n">ignored...</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done.&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;18866 (default) background genes&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;54 provided genes; 49 found by `enrichGO`&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Computed GO enrichment (whether significant or not) for 1086 distinct GO terms&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Of those 1086 GO terms, 18 have a BH-adjusted p-val &lt; 0.05&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Writing outputs tables...&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done. Writing output images...&quot;</span>
<span class="kc">...</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done.&quot;</span>
</code></pre></div>
</blockquote>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/all_genes/symbol/default_universe.BP.0.05BH.dotplot.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/all_genes/symbol/default_universe.BP.0.05BH.gene-concept.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/all_genes/symbol/default_universe.BP.0.05BH.gene-concept.circular.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/all_genes/symbol/default_universe.BP.0.05BH.heatplot.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/all_genes/symbol/default_universe.BP.0.05BH.upset.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/all_genes/symbol/default_universe.BP.0.05BH.pvalues.png" /></p>
<h3 id="all-42-genes-inferred-only-wo-original-genes">All 42 genes (inferred only ; w/o original genes)<a class="headerlink" href="#all-42-genes-inferred-only-wo-original-genes" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code>mkdir<span class="w"> </span>-p<span class="w"> </span>GO_FDR/all_genes_wo_orig/symbol
</code></pre></div>
<div class="highlight"><pre><span></span><code>Rscript<span class="w"> </span>GO_enrichment_FDR.R<span class="w"> </span>-k<span class="w"> </span><span class="s2">&quot;SYMBOL&quot;</span><span class="w"> </span>-G<span class="w"> </span>results/new_genes_v1_without_original.list<span class="w"> </span>-f<span class="w"> </span><span class="m">0</span>.05<span class="w"> </span>-c<span class="w"> </span><span class="s2">&quot;BP&quot;</span><span class="w"> </span>-a<span class="w"> </span><span class="s2">&quot;BH&quot;</span><span class="w"> </span>-o<span class="w"> </span><span class="s2">&quot;default_universe&quot;</span><span class="w"> </span>-d<span class="w"> </span><span class="s2">&quot;GO_FDR/all_genes_wo_orig/symbol&quot;</span>
</code></pre></div>
<blockquote>
<div class="highlight"><pre><span></span><code><span class="kc">...</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Loading input data...&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Warning: using defaut universe automatically provided by the clusterProfiler package&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done.&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Computing GO enrichment...&quot;</span>
<span class="n">`universe`</span><span class="w"> </span><span class="n">is</span><span class="w"> </span><span class="n">not</span><span class="w"> </span><span class="n">in</span><span class="w"> </span><span class="n">character</span><span class="w"> </span><span class="n">and</span><span class="w"> </span><span class="n">will</span><span class="w"> </span><span class="n">be</span><span class="w"> </span><span class="n">ignored...</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done.&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;18866 (default) background genes&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;42 provided genes; 38 found by `enrichGO`&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Computed GO enrichment (whether significant or not) for 740 distinct GO terms&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Of those 740 GO terms, 0 have a BH-adjusted p-val &lt; 0.05&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Writing outputs tables...&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done. Writing output images...&quot;</span>
<span class="kc">...</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done.&quot;</span>
</code></pre></div>
</blockquote>
<h3 id="all-genes-co-expressed-in-liver">All genes co-expressed in liver<a class="headerlink" href="#all-genes-co-expressed-in-liver" title="Permanent link">&para;</a></h3>
<p>Those are: HFE + TFR2 + HJV (HFE2) + HAMP + TMPRSS6</p>
<p>Note: we have no CHiC data for HAMP, and there are no inferred genes from HFE. </p>
<div class="highlight"><pre><span></span><code>mkdir<span class="w"> </span>-p<span class="w"> </span>GO_FDR/genes_inferred_from_that_expressed_in_liver/symbol
</code></pre></div>
<div class="highlight"><pre><span></span><code>Rscript<span class="w"> </span>GO_enrichment_FDR.R<span class="w"> </span>-k<span class="w"> </span><span class="s2">&quot;SYMBOL&quot;</span><span class="w"> </span>-G<span class="w"> </span>results/genes_inferred_from_that_expressed_in_liver.list<span class="w"> </span>-f<span class="w"> </span><span class="m">0</span>.05<span class="w"> </span>-c<span class="w"> </span><span class="s2">&quot;BP&quot;</span><span class="w"> </span>-a<span class="w"> </span><span class="s2">&quot;BH&quot;</span><span class="w"> </span>-o<span class="w"> </span><span class="s2">&quot;default_universe&quot;</span><span class="w"> </span>-d<span class="w"> </span><span class="s2">&quot;GO_FDR/genes_inferred_from_that_expressed_in_liver/symbol&quot;</span>
</code></pre></div>
<blockquote>
<div class="highlight"><pre><span></span><code><span class="kc">...</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Loading input data...&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Warning: using defaut universe automatically provided by the clusterProfiler package&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done.&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Computing GO enrichment...&quot;</span>
<span class="n">`universe`</span><span class="w"> </span><span class="n">is</span><span class="w"> </span><span class="n">not</span><span class="w"> </span><span class="n">in</span><span class="w"> </span><span class="n">character</span><span class="w"> </span><span class="n">and</span><span class="w"> </span><span class="n">will</span><span class="w"> </span><span class="n">be</span><span class="w"> </span><span class="n">ignored...</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done.&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;18866 (default) background genes&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;24 provided genes; 21 found by `enrichGO`&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Computed GO enrichment (whether significant or not) for 421 distinct GO terms&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Of those 421 GO terms, 4 have a BH-adjusted p-val &lt; 0.05&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Writing outputs tables...&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done. Writing output images...&quot;</span>
<span class="kc">...</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done.&quot;</span>
</code></pre></div>
</blockquote>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/genes_inferred_from_that_expressed_in_liver/symbol/default_universe.BP.0.05BH.dotplot.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/genes_inferred_from_that_expressed_in_liver/symbol/default_universe.BP.0.05BH.gene-concept.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/genes_inferred_from_that_expressed_in_liver/symbol/default_universe.BP.0.05BH.gene-concept.circular.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/genes_inferred_from_that_expressed_in_liver/symbol/default_universe.BP.0.05BH.heatplot.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/genes_inferred_from_that_expressed_in_liver/symbol/default_universe.BP.0.05BH.upset.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/genes_inferred_from_that_expressed_in_liver/symbol/default_universe.BP.0.05BH.pvalues.png" /></p>
<h3 id="all-genes-expressed-in-intestine">All genes expressed in intestine<a class="headerlink" href="#all-genes-expressed-in-intestine" title="Permanent link">&para;</a></h3>
<p>Those are: DCYTB (CYBRD1) / DMT1 (SLC11A2) / SLC40A1</p>
<p>Note: there are no inferred genes from CYBRD1 nor SLC40A1 =&gt; so the GO enrichment is basically that of SLC11A2.</p>
<h3 id="each-original-gene-separately">Each original gene separately<a class="headerlink" href="#each-original-gene-separately" title="Permanent link">&para;</a></h3>
<h4 id="inferred-from-ciapin1">Inferred from CIAPIN1<a class="headerlink" href="#inferred-from-ciapin1" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code>mkdir<span class="w"> </span>-p<span class="w"> </span>GO_FDR/separate/CIAPIN1/symbol
</code></pre></div>
<div class="highlight"><pre><span></span><code>Rscript<span class="w"> </span>GO_enrichment_FDR.R<span class="w"> </span>-k<span class="w"> </span><span class="s2">&quot;SYMBOL&quot;</span><span class="w"> </span>-G<span class="w"> </span>results/separate/CIAPIN1.list<span class="w"> </span>-f<span class="w"> </span><span class="m">0</span>.05<span class="w"> </span>-c<span class="w"> </span><span class="s2">&quot;BP&quot;</span><span class="w"> </span>-a<span class="w"> </span><span class="s2">&quot;BH&quot;</span><span class="w"> </span>-o<span class="w"> </span><span class="s2">&quot;default_universe&quot;</span><span class="w"> </span>-d<span class="w"> </span><span class="s2">&quot;GO_FDR/separate/CIAPIN1/symbol&quot;</span>
</code></pre></div>
<blockquote>
<div class="highlight"><pre><span></span><code><span class="kc">...</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Loading input data...&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Warning: using defaut universe automatically provided by the clusterProfiler package&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done.&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Computing GO enrichment...&quot;</span>
<span class="n">`universe`</span><span class="w"> </span><span class="n">is</span><span class="w"> </span><span class="n">not</span><span class="w"> </span><span class="n">in</span><span class="w"> </span><span class="n">character</span><span class="w"> </span><span class="n">and</span><span class="w"> </span><span class="n">will</span><span class="w"> </span><span class="n">be</span><span class="w"> </span><span class="n">ignored...</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done.&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;18866 (default) background genes&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;8 provided genes; 7 found by `enrichGO`&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Computed GO enrichment (whether significant or not) for 313 distinct GO terms&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Of those 313 GO terms, 64 have a BH-adjusted p-val &lt; 0.05&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Writing outputs tables...&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done. Writing output images...&quot;</span>
<span class="kc">...</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done.&quot;</span>
</code></pre></div>
</blockquote>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/CIAPIN1/symbol/default_universe.BP.0.05BH.dotplot.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/CIAPIN1/symbol/default_universe.BP.0.05BH.gene-concept.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/CIAPIN1/symbol/default_universe.BP.0.05BH.gene-concept.circular.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/CIAPIN1/symbol/default_universe.BP.0.05BH.heatplot.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/CIAPIN1/symbol/default_universe.BP.0.05BH.upset.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/CIAPIN1/symbol/default_universe.BP.0.05BH.pvalues.png" /></p>
<h4 id="inferred-from-ciapin1-wo-ciapin1">Inferred from CIAPIN1 w/o CIAPIN1<a class="headerlink" href="#inferred-from-ciapin1-wo-ciapin1" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code>mkdir<span class="w"> </span>-p<span class="w"> </span>GO_FDR/separate/CIAPIN1_wo_orig/symbol
</code></pre></div>
<div class="highlight"><pre><span></span><code>Rscript<span class="w"> </span>GO_enrichment_FDR.R<span class="w"> </span>-k<span class="w"> </span><span class="s2">&quot;SYMBOL&quot;</span><span class="w"> </span>-G<span class="w"> </span>results/separate/CIAPIN1_wo_original.list<span class="w"> </span>-f<span class="w"> </span><span class="m">0</span>.05<span class="w"> </span>-c<span class="w"> </span><span class="s2">&quot;BP&quot;</span><span class="w"> </span>-a<span class="w"> </span><span class="s2">&quot;BH&quot;</span><span class="w"> </span>-o<span class="w"> </span><span class="s2">&quot;default_universe&quot;</span><span class="w"> </span>-d<span class="w"> </span><span class="s2">&quot;GO_FDR/separate/CIAPIN1_wo_orig/symbol&quot;</span>
</code></pre></div>
<blockquote>
<div class="highlight"><pre><span></span><code><span class="kc">...</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Loading input data...&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Warning: using defaut universe automatically provided by the clusterProfiler package&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done.&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Computing GO enrichment...&quot;</span>
<span class="n">`universe`</span><span class="w"> </span><span class="n">is</span><span class="w"> </span><span class="n">not</span><span class="w"> </span><span class="n">in</span><span class="w"> </span><span class="n">character</span><span class="w"> </span><span class="n">and</span><span class="w"> </span><span class="n">will</span><span class="w"> </span><span class="n">be</span><span class="w"> </span><span class="n">ignored...</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done.&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;18866 (default) background genes&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;7 provided genes; 6 found by `enrichGO`&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Computed GO enrichment (whether significant or not) for 310 distinct GO terms&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Of those 310 GO terms, 73 have a BH-adjusted p-val &lt; 0.05&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Writing outputs tables...&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done. Writing output images...&quot;</span>
<span class="kc">...</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done.&quot;</span>
</code></pre></div>
</blockquote>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/CIAPIN1_wo_orig/symbol/default_universe.BP.0.05BH.dotplot.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/CIAPIN1_wo_orig/symbol/default_universe.BP.0.05BH.gene-concept.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/CIAPIN1_wo_orig/symbol/default_universe.BP.0.05BH.gene-concept.circular.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/CIAPIN1_wo_orig/symbol/default_universe.BP.0.05BH.heatplot.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/CIAPIN1_wo_orig/symbol/default_universe.BP.0.05BH.upset.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/CIAPIN1_wo_orig/symbol/default_universe.BP.0.05BH.pvalues.png" /></p>
<h4 id="inferred-from-hfe2">Inferred from HFE2<a class="headerlink" href="#inferred-from-hfe2" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code>mkdir<span class="w"> </span>-p<span class="w"> </span>GO_FDR/separate/HFE2/symbol
</code></pre></div>
<div class="highlight"><pre><span></span><code>Rscript<span class="w"> </span>GO_enrichment_FDR.R<span class="w"> </span>-k<span class="w"> </span><span class="s2">&quot;SYMBOL&quot;</span><span class="w"> </span>-G<span class="w"> </span>results/separate/HFE2.list<span class="w"> </span>-f<span class="w"> </span><span class="m">0</span>.05<span class="w"> </span>-c<span class="w"> </span><span class="s2">&quot;BP&quot;</span><span class="w"> </span>-a<span class="w"> </span><span class="s2">&quot;BH&quot;</span><span class="w"> </span>-o<span class="w"> </span><span class="s2">&quot;default_universe&quot;</span><span class="w"> </span>-d<span class="w"> </span><span class="s2">&quot;GO_FDR/separate/HFE2/symbol&quot;</span>
</code></pre></div>
<blockquote>
<div class="highlight"><pre><span></span><code><span class="kc">...</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Loading input data...&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Warning: using defaut universe automatically provided by the clusterProfiler package&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done.&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Computing GO enrichment...&quot;</span>
<span class="n">`universe`</span><span class="w"> </span><span class="n">is</span><span class="w"> </span><span class="n">not</span><span class="w"> </span><span class="n">in</span><span class="w"> </span><span class="n">character</span><span class="w"> </span><span class="n">and</span><span class="w"> </span><span class="n">will</span><span class="w"> </span><span class="n">be</span><span class="w"> </span><span class="n">ignored...</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done.&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;18866 (default) background genes&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;5 provided genes; 3 found by `enrichGO`&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Computed GO enrichment (whether significant or not) for 17 distinct GO terms&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Of those 17 GO terms, 15 have a BH-adjusted p-val &lt; 0.05&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Writing outputs tables...&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done. Writing output images...&quot;</span>
<span class="kc">...</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done.&quot;</span>
</code></pre></div>
</blockquote>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/HFE2/symbol/default_universe.BP.0.05BH.dotplot.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/HFE2/symbol/default_universe.BP.0.05BH.gene-concept.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/HFE2/symbol/default_universe.BP.0.05BH.gene-concept.circular.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/HFE2/symbol/default_universe.BP.0.05BH.heatplot.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/HFE2/symbol/default_universe.BP.0.05BH.upset.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/HFE2/symbol/default_universe.BP.0.05BH.pvalues.png" /></p>
<h4 id="inferred-from-hfe2-wo-hfe2">Inferred from HFE2 w/o HFE2<a class="headerlink" href="#inferred-from-hfe2-wo-hfe2" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code>mkdir<span class="w"> </span>-p<span class="w"> </span>GO_FDR/separate/HFE2_wo_orig/symbol
</code></pre></div>
<div class="highlight"><pre><span></span><code>Rscript<span class="w"> </span>GO_enrichment_FDR.R<span class="w"> </span>-k<span class="w"> </span><span class="s2">&quot;SYMBOL&quot;</span><span class="w"> </span>-G<span class="w"> </span>results/separate/HFE2_wo_original.list<span class="w"> </span>-f<span class="w"> </span><span class="m">0</span>.05<span class="w"> </span>-c<span class="w"> </span><span class="s2">&quot;BP&quot;</span><span class="w"> </span>-a<span class="w"> </span><span class="s2">&quot;BH&quot;</span><span class="w"> </span>-o<span class="w"> </span><span class="s2">&quot;default_universe&quot;</span><span class="w"> </span>-d<span class="w"> </span><span class="s2">&quot;GO_FDR/separate/HFE2_wo_orig/symbol&quot;</span>
</code></pre></div>
<blockquote>
<div class="highlight"><pre><span></span><code><span class="kc">...</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Loading input data...&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Warning: using defaut universe automatically provided by the clusterProfiler package&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done.&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Computing GO enrichment...&quot;</span>
<span class="n">`universe`</span><span class="w"> </span><span class="n">is</span><span class="w"> </span><span class="n">not</span><span class="w"> </span><span class="n">in</span><span class="w"> </span><span class="n">character</span><span class="w"> </span><span class="n">and</span><span class="w"> </span><span class="n">will</span><span class="w"> </span><span class="n">be</span><span class="w"> </span><span class="n">ignored...</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done.&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;18866 (default) background genes&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;4 provided genes; 3 found by `enrichGO`&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Computed GO enrichment (whether significant or not) for 17 distinct GO terms&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Of those 17 GO terms, 15 have a BH-adjusted p-val &lt; 0.05&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Writing outputs tables...&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done. Writing output images...&quot;</span>
<span class="kc">...</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done.&quot;</span>
</code></pre></div>
</blockquote>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/HFE2_wo_orig/symbol/default_universe.BP.0.05BH.dotplot.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/HFE2_wo_orig/symbol/default_universe.BP.0.05BH.gene-concept.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/HFE2_wo_orig/symbol/default_universe.BP.0.05BH.gene-concept.circular.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/HFE2_wo_orig/symbol/default_universe.BP.0.05BH.heatplot.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/HFE2_wo_orig/symbol/default_universe.BP.0.05BH.upset.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/HFE2_wo_orig/symbol/default_universe.BP.0.05BH.pvalues.png" /></p>
<h4 id="inferred-from-neo1">Inferred from NEO1<a class="headerlink" href="#inferred-from-neo1" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code>mkdir<span class="w"> </span>-p<span class="w"> </span>GO_FDR/separate/NEO1/symbol
</code></pre></div>
<div class="highlight"><pre><span></span><code>Rscript<span class="w"> </span>GO_enrichment_FDR.R<span class="w"> </span>-k<span class="w"> </span><span class="s2">&quot;SYMBOL&quot;</span><span class="w"> </span>-G<span class="w"> </span>results/separate/NEO1.list<span class="w"> </span>-c<span class="w"> </span><span class="s2">&quot;BP&quot;</span><span class="w"> </span>-a<span class="w"> </span><span class="s2">&quot;BH&quot;</span><span class="w"> </span>-o<span class="w"> </span><span class="s2">&quot;default_universe&quot;</span><span class="w"> </span>-d<span class="w"> </span><span class="s2">&quot;GO_FDR/separate/NEO1/symbol&quot;</span>
</code></pre></div>
<blockquote>
<div class="highlight"><pre><span></span><code><span class="kc">...</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Loading input data...&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Warning: using defaut universe automatically provided by the clusterProfiler package&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done.&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Computing GO enrichment...&quot;</span>
<span class="n">`universe`</span><span class="w"> </span><span class="n">is</span><span class="w"> </span><span class="n">not</span><span class="w"> </span><span class="n">in</span><span class="w"> </span><span class="n">character</span><span class="w"> </span><span class="n">and</span><span class="w"> </span><span class="n">will</span><span class="w"> </span><span class="n">be</span><span class="w"> </span><span class="n">ignored...</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done.&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;18866 (default) background genes&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;2 provided genes; 2 found by `enrichGO`&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Computed GO enrichment (whether significant or not) for 45 distinct GO terms&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Of those 45 GO terms, 45 have a BH-adjusted p-val &lt; 0.1&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Writing outputs tables...&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done. Writing output images...&quot;</span>
<span class="kc">...</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done.&quot;</span>
</code></pre></div>
</blockquote>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/NEO1/symbol/default_universe.BP.0.1BH.dotplot.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/NEO1/symbol/default_universe.BP.0.1BH.gene-concept.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/NEO1/symbol/default_universe.BP.0.1BH.gene-concept.circular.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/NEO1/symbol/default_universe.BP.0.1BH.heatplot.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/NEO1/symbol/default_universe.BP.0.1BH.upset.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/NEO1/symbol/default_universe.BP.0.1BH.pvalues.png" /></p>
<h4 id="inferred-from-neo1-wo-neo1">Inferred from NEO1 w/o NEO1<a class="headerlink" href="#inferred-from-neo1-wo-neo1" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code>mkdir<span class="w"> </span>-p<span class="w"> </span>GO_FDR/separate/NEO1_wo_orig/symbol
</code></pre></div>
<div class="highlight"><pre><span></span><code>Rscript<span class="w"> </span>GO_enrichment_FDR.R<span class="w"> </span>-k<span class="w"> </span><span class="s2">&quot;SYMBOL&quot;</span><span class="w"> </span>-G<span class="w"> </span>results/separate/NEO1_wo_original.list<span class="w"> </span>-c<span class="w"> </span><span class="s2">&quot;BP&quot;</span><span class="w"> </span>-a<span class="w"> </span><span class="s2">&quot;BH&quot;</span><span class="w"> </span>-o<span class="w"> </span><span class="s2">&quot;default_universe&quot;</span><span class="w"> </span>-d<span class="w"> </span><span class="s2">&quot;GO_FDR/separate/NEO1_wo_orig/symbol&quot;</span>
</code></pre></div>
<blockquote>
<div class="highlight"><pre><span></span><code><span class="kc">...</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Loading input data...&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Warning: using defaut universe automatically provided by the clusterProfiler package&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done.&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Computing GO enrichment...&quot;</span>
<span class="n">`universe`</span><span class="w"> </span><span class="n">is</span><span class="w"> </span><span class="n">not</span><span class="w"> </span><span class="n">in</span><span class="w"> </span><span class="n">character</span><span class="w"> </span><span class="n">and</span><span class="w"> </span><span class="n">will</span><span class="w"> </span><span class="n">be</span><span class="w"> </span><span class="n">ignored...</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done.&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;18866 (default) background genes&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;1 provided genes; 1 found by `enrichGO`&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Computed GO enrichment (whether significant or not) for 31 distinct GO terms&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Of those 31 GO terms, 31 have a BH-adjusted p-val &lt; 0.1&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Writing outputs tables...&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done. Writing output images...&quot;</span>
<span class="n">wrong</span><span class="w"> </span><span class="n">orderBy</span><span class="w"> </span><span class="n">parameter</span><span class="p">;</span><span class="w"> </span><span class="n">set</span><span class="w"> </span><span class="n">to</span><span class="w"> </span><span class="n">default</span><span class="w"> </span><span class="n">`orderBy = &quot;x&quot;`</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Writing last output image (this one might take some time)...&quot;</span>
<span class="n">`universe`</span><span class="w"> </span><span class="n">is</span><span class="w"> </span><span class="n">not</span><span class="w"> </span><span class="n">in</span><span class="w"> </span><span class="n">character</span><span class="w"> </span><span class="n">and</span><span class="w"> </span><span class="n">will</span><span class="w"> </span><span class="n">be</span><span class="w"> </span><span class="n">ignored...</span>
<span class="n">No</span><span class="w"> </span><span class="n">gene</span><span class="w"> </span><span class="n">set</span><span class="w"> </span><span class="n">have</span><span class="w"> </span><span class="n">size</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">10</span><span class="w"> </span><span class="kc">...</span>
<span class="o">--&gt;</span><span class="w"> </span><span class="n">return</span><span class="w"> </span><span class="n">NULL...</span>
<span class="n">Warning</span><span class="w"> </span><span class="n">message</span><span class="o">:</span>
<span class="n">The</span><span class="w"> </span><span class="n">`add`</span><span class="w"> </span><span class="n">argument</span><span class="w"> </span><span class="n">of</span><span class="w"> </span><span class="n">`group_by()`</span><span class="w"> </span><span class="n">is</span><span class="w"> </span><span class="n">deprecated</span><span class="w"> </span><span class="n">as</span><span class="w"> </span><span class="n">of</span><span class="w"> </span><span class="n">dplyr</span><span class="w"> </span><span class="m">1.0</span><span class="n">.</span><span class="m">0</span><span class="n">.</span>
<span class="n">Please</span><span class="w"> </span><span class="n">use</span><span class="w"> </span><span class="n">the</span><span class="w"> </span><span class="n">`.add`</span><span class="w"> </span><span class="n">argument</span><span class="w"> </span><span class="n">instead.</span>
<span class="n">This</span><span class="w"> </span><span class="n">warning</span><span class="w"> </span><span class="n">is</span><span class="w"> </span><span class="n">displayed</span><span class="w"> </span><span class="n">once</span><span class="w"> </span><span class="n">every</span><span class="w"> </span><span class="m">8</span><span class="w"> </span><span class="n">hours.</span>
<span class="n">Call</span><span class="w"> </span><span class="n">`lifecycle::last_warnings()`</span><span class="w"> </span><span class="n">to</span><span class="w"> </span><span class="n">see</span><span class="w"> </span><span class="n">where</span><span class="w"> </span><span class="n">this</span><span class="w"> </span><span class="n">warning</span><span class="w"> </span><span class="n">was</span><span class="w"> </span><span class="n">generated.</span><span class="w"> </span>
<span class="n">Error</span><span class="w"> </span><span class="n">in</span><span class="w"> </span><span class="nf">UseMethod</span><span class="p">(</span><span class="s">&quot;mutate&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">:</span><span class="w"> </span>
<span class="w">  </span><span class="n">no</span><span class="w"> </span><span class="n">applicable</span><span class="w"> </span><span class="n">method</span><span class="w"> </span><span class="n">for</span><span class="w"> </span><span class="s">&#39;mutate&#39;</span><span class="w"> </span><span class="n">applied</span><span class="w"> </span><span class="n">to</span><span class="w"> </span><span class="n">an</span><span class="w"> </span><span class="n">object</span><span class="w"> </span><span class="n">of</span><span class="w"> </span><span class="n">class</span><span class="w"> </span><span class="s">&quot;NULL&quot;</span>
<span class="n">Calls</span><span class="o">:</span><span class="w"> </span><span class="n">pbar</span><span class="w"> </span><span class="o">-&gt;</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="o">-&gt;</span><span class="w"> </span><span class="n">ggplot</span><span class="w"> </span><span class="o">-&gt;</span><span class="w"> </span><span class="n">summarise</span><span class="w"> </span><span class="o">-&gt;</span><span class="w"> </span><span class="n">group_by</span><span class="w"> </span><span class="o">-&gt;</span><span class="w"> </span><span class="n">mutate</span>
<span class="n">Execution</span><span class="w"> </span><span class="n">halted</span>
</code></pre></div>
</blockquote>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/NEO1_wo_orig/symbol/default_universe.BP.0.1BH.dotplot.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/NEO1_wo_orig/symbol/default_universe.BP.0.1BH.gene-concept.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/NEO1_wo_orig/symbol/default_universe.BP.0.1BH.gene-concept.circular.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/NEO1_wo_orig/symbol/default_universe.BP.0.1BH.heatplot.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/NEO1_wo_orig/symbol/default_universe.BP.0.1BH.upset.png" /></p>
<h4 id="inferred-from-slc11a2">Inferred from SLC11A2<a class="headerlink" href="#inferred-from-slc11a2" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code>mkdir<span class="w"> </span>-p<span class="w"> </span>GO_FDR/separate/SLC11A2/symbol
</code></pre></div>
<div class="highlight"><pre><span></span><code>Rscript<span class="w"> </span>GO_enrichment_FDR.R<span class="w"> </span>-k<span class="w"> </span><span class="s2">&quot;SYMBOL&quot;</span><span class="w"> </span>-G<span class="w"> </span>results/separate/SLC11A2.list<span class="w"> </span>-c<span class="w"> </span><span class="s2">&quot;BP&quot;</span><span class="w"> </span>-a<span class="w"> </span><span class="s2">&quot;BH&quot;</span><span class="w"> </span>-o<span class="w"> </span><span class="s2">&quot;default_universe&quot;</span><span class="w"> </span>-d<span class="w"> </span><span class="s2">&quot;GO_FDR/separate/SLC11A2/symbol&quot;</span>
</code></pre></div>
<blockquote>
<div class="highlight"><pre><span></span><code><span class="kc">...</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Loading input data...&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Warning: using defaut universe automatically provided by the clusterProfiler package&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done.&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Computing GO enrichment...&quot;</span>
<span class="n">`universe`</span><span class="w"> </span><span class="n">is</span><span class="w"> </span><span class="n">not</span><span class="w"> </span><span class="n">in</span><span class="w"> </span><span class="n">character</span><span class="w"> </span><span class="n">and</span><span class="w"> </span><span class="n">will</span><span class="w"> </span><span class="n">be</span><span class="w"> </span><span class="n">ignored...</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done.&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;18866 (default) background genes&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;8 provided genes; 8 found by `enrichGO`&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Computed GO enrichment (whether significant or not) for 104 distinct GO terms&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Of those 104 GO terms, 18 have a BH-adjusted p-val &lt; 0.1&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Writing outputs tables...&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done. Writing output images...&quot;</span>
<span class="kc">...</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done.&quot;</span>
</code></pre></div>
</blockquote>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/SLC11A2/symbol/default_universe.BP.0.1BH.dotplot.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/SLC11A2/symbol/default_universe.BP.0.1BH.gene-concept.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/SLC11A2/symbol/default_universe.BP.0.1BH.gene-concept.circular.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/SLC11A2/symbol/default_universe.BP.0.1BH.heatplot.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/SLC11A2/symbol/default_universe.BP.0.1BH.upset.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/SLC11A2/symbol/default_universe.BP.0.1BH.pvalues.png" /></p>
<h4 id="inferred-from-slc11a2-wo-slc11a2">Inferred from SLC11A2 w/o SLC11A2<a class="headerlink" href="#inferred-from-slc11a2-wo-slc11a2" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code>mkdir<span class="w"> </span>-p<span class="w"> </span>GO_FDR/separate/SLC11A2_wo_orig/symbol
</code></pre></div>
<div class="highlight"><pre><span></span><code>Rscript<span class="w"> </span>GO_enrichment_FDR.R<span class="w"> </span>-k<span class="w"> </span><span class="s2">&quot;SYMBOL&quot;</span><span class="w"> </span>-G<span class="w"> </span>results/separate/SLC11A2_wo_original.list<span class="w"> </span>-c<span class="w"> </span><span class="s2">&quot;BP&quot;</span><span class="w"> </span>-a<span class="w"> </span><span class="s2">&quot;BH&quot;</span><span class="w"> </span>-o<span class="w"> </span><span class="s2">&quot;default_universe&quot;</span><span class="w"> </span>-d<span class="w"> </span><span class="s2">&quot;GO_FDR/separate/SLC11A2_wo_orig/symbol&quot;</span>
</code></pre></div>
<blockquote>
<div class="highlight"><pre><span></span><code><span class="kc">...</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Loading input data...&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Warning: using defaut universe automatically provided by the clusterProfiler package&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done.&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Computing GO enrichment...&quot;</span>
<span class="n">`universe`</span><span class="w"> </span><span class="n">is</span><span class="w"> </span><span class="n">not</span><span class="w"> </span><span class="n">in</span><span class="w"> </span><span class="n">character</span><span class="w"> </span><span class="n">and</span><span class="w"> </span><span class="n">will</span><span class="w"> </span><span class="n">be</span><span class="w"> </span><span class="n">ignored...</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done.&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;18866 (default) background genes&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;7 provided genes; 7 found by `enrichGO`&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Computed GO enrichment (whether significant or not) for 68 distinct GO terms&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Of those 68 GO terms, 0 have a BH-adjusted p-val &lt; 0.1&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Writing outputs tables...&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done. Writing output images...&quot;</span>
<span class="n">Error</span><span class="w"> </span><span class="n">in</span><span class="w"> </span><span class="nf">`[.data.frame`</span><span class="p">(</span><span class="n">d</span><span class="p">,</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">)</span><span class="w"> </span><span class="o">:</span><span class="w"> </span><span class="n">undefined</span><span class="w"> </span><span class="n">columns</span><span class="w"> </span><span class="n">selected</span>
<span class="n">Calls</span><span class="o">:</span><span class="w"> </span><span class="n">upsetplot</span><span class="w"> </span><span class="kc">...</span><span class="w"> </span><span class="n">eval_tidy</span><span class="w"> </span><span class="o">-&gt;</span><span class="w"> </span><span class="n">split</span><span class="w"> </span><span class="o">-&gt;</span><span class="w"> </span><span class="n">split.default</span><span class="w"> </span><span class="o">-&gt;</span><span class="w"> </span><span class="p">[</span><span class="w"> </span><span class="o">-&gt;</span><span class="w"> </span><span class="p">[</span><span class="n">.data.frame</span>
<span class="n">Execution</span><span class="w"> </span><span class="n">halted</span>
</code></pre></div>
</blockquote>
<h4 id="inferred-from-slc39a14">Inferred from SLC39A14<a class="headerlink" href="#inferred-from-slc39a14" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code>mkdir<span class="w"> </span>-p<span class="w"> </span>GO_FDR/separate/SLC39A14/symbol
</code></pre></div>
<div class="highlight"><pre><span></span><code>Rscript<span class="w"> </span>GO_enrichment_FDR.R<span class="w"> </span>-k<span class="w"> </span><span class="s2">&quot;SYMBOL&quot;</span><span class="w"> </span>-G<span class="w"> </span>results/separate/SLC39A14.list<span class="w"> </span>-c<span class="w"> </span><span class="s2">&quot;BP&quot;</span><span class="w"> </span>-a<span class="w"> </span><span class="s2">&quot;BH&quot;</span><span class="w"> </span>-o<span class="w"> </span><span class="s2">&quot;default_universe&quot;</span><span class="w"> </span>-d<span class="w"> </span><span class="s2">&quot;GO_FDR/separate/SLC39A14/symbol&quot;</span>
</code></pre></div>
<blockquote>
<div class="highlight"><pre><span></span><code><span class="kc">...</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Loading input data...&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Warning: using defaut universe automatically provided by the clusterProfiler package&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done.&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Computing GO enrichment...&quot;</span>
<span class="n">`universe`</span><span class="w"> </span><span class="n">is</span><span class="w"> </span><span class="n">not</span><span class="w"> </span><span class="n">in</span><span class="w"> </span><span class="n">character</span><span class="w"> </span><span class="n">and</span><span class="w"> </span><span class="n">will</span><span class="w"> </span><span class="n">be</span><span class="w"> </span><span class="n">ignored...</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done.&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;18866 (default) background genes&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;4 provided genes; 4 found by `enrichGO`&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Computed GO enrichment (whether significant or not) for 70 distinct GO terms&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Of those 70 GO terms, 69 have a BH-adjusted p-val &lt; 0.1&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Writing outputs tables...&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done. Writing output images...&quot;</span>
<span class="kc">...</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done.&quot;</span>
</code></pre></div>
</blockquote>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/SLC39A14/symbol/default_universe.BP.0.1BH.dotplot.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/SLC39A14/symbol/default_universe.BP.0.1BH.gene-concept.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/SLC39A14/symbol/default_universe.BP.0.1BH.gene-concept.circular.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/SLC39A14/symbol/default_universe.BP.0.1BH.heatplot.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/SLC39A14/symbol/default_universe.BP.0.1BH.upset.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/SLC39A14/symbol/default_universe.BP.0.1BH.pvalues.png" /></p>
<h4 id="inferred-from-slc39a14-wo-slc39a14">Inferred from SLC39A14 w/o SLC39A14<a class="headerlink" href="#inferred-from-slc39a14-wo-slc39a14" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code>mkdir<span class="w"> </span>-p<span class="w"> </span>GO_FDR/separate/SLC39A14_wo_orig/symbol
</code></pre></div>
<div class="highlight"><pre><span></span><code>Rscript<span class="w"> </span>GO_enrichment_FDR.R<span class="w"> </span>-k<span class="w"> </span><span class="s2">&quot;SYMBOL&quot;</span><span class="w"> </span>-G<span class="w"> </span>results/separate/SLC39A14_wo_original.list<span class="w"> </span>-c<span class="w"> </span><span class="s2">&quot;BP&quot;</span><span class="w"> </span>-a<span class="w"> </span><span class="s2">&quot;BH&quot;</span><span class="w"> </span>-o<span class="w"> </span><span class="s2">&quot;default_universe&quot;</span><span class="w"> </span>-d<span class="w"> </span><span class="s2">&quot;GO_FDR/separate/SLC39A14_wo_orig/symbol&quot;</span>
</code></pre></div>
<blockquote>
<div class="highlight"><pre><span></span><code><span class="kc">...</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Loading input data...&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Warning: using defaut universe automatically provided by the clusterProfiler package&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done.&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Computing GO enrichment...&quot;</span>
<span class="n">`universe`</span><span class="w"> </span><span class="n">is</span><span class="w"> </span><span class="n">not</span><span class="w"> </span><span class="n">in</span><span class="w"> </span><span class="n">character</span><span class="w"> </span><span class="n">and</span><span class="w"> </span><span class="n">will</span><span class="w"> </span><span class="n">be</span><span class="w"> </span><span class="n">ignored...</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done.&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;18866 (default) background genes&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;3 provided genes; 3 found by `enrichGO`&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Computed GO enrichment (whether significant or not) for 19 distinct GO terms&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Of those 19 GO terms, 19 have a BH-adjusted p-val &lt; 0.1&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Writing outputs tables...&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done. Writing output images...&quot;</span>
<span class="kc">...</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done.&quot;</span>
</code></pre></div>
</blockquote>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/SLC39A14_wo_orig/symbol/default_universe.BP.0.1BH.dotplot.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/SLC39A14_wo_orig/symbol/default_universe.BP.0.1BH.gene-concept.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/SLC39A14_wo_orig/symbol/default_universe.BP.0.1BH.gene-concept.circular.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/SLC39A14_wo_orig/symbol/default_universe.BP.0.1BH.heatplot.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/SLC39A14_wo_orig/symbol/default_universe.BP.0.1BH.upset.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/SLC39A14_wo_orig/symbol/default_universe.BP.0.1BH.pvalues.png" /></p>
<h4 id="inferred-from-tfr2">Inferred from TFR2<a class="headerlink" href="#inferred-from-tfr2" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code>mkdir<span class="w"> </span>-p<span class="w"> </span>GO_FDR/separate/TFR2/symbol
</code></pre></div>
<div class="highlight"><pre><span></span><code>Rscript<span class="w"> </span>GO_enrichment_FDR.R<span class="w"> </span>-k<span class="w"> </span><span class="s2">&quot;SYMBOL&quot;</span><span class="w"> </span>-G<span class="w"> </span>results/separate/TFR2.list<span class="w"> </span>-c<span class="w"> </span><span class="s2">&quot;BP&quot;</span><span class="w"> </span>-a<span class="w"> </span><span class="s2">&quot;BH&quot;</span><span class="w"> </span>-o<span class="w"> </span><span class="s2">&quot;default_universe&quot;</span><span class="w"> </span>-d<span class="w"> </span><span class="s2">&quot;GO_FDR/separate/TFR2/symbol&quot;</span>
</code></pre></div>
<blockquote>
<div class="highlight"><pre><span></span><code><span class="kc">...</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Loading input data...&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Warning: using defaut universe automatically provided by the clusterProfiler package&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done.&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Computing GO enrichment...&quot;</span>
<span class="n">`universe`</span><span class="w"> </span><span class="n">is</span><span class="w"> </span><span class="n">not</span><span class="w"> </span><span class="n">in</span><span class="w"> </span><span class="n">character</span><span class="w"> </span><span class="n">and</span><span class="w"> </span><span class="n">will</span><span class="w"> </span><span class="n">be</span><span class="w"> </span><span class="n">ignored...</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done.&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;18866 (default) background genes&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;15 provided genes; 14 found by `enrichGO`&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Computed GO enrichment (whether significant or not) for 305 distinct GO terms&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Of those 305 GO terms, 0 have a BH-adjusted p-val &lt; 0.1&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Writing outputs tables...&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done. Writing output images...&quot;</span>
<span class="n">Error</span><span class="w"> </span><span class="n">in</span><span class="w"> </span><span class="nf">`[.data.frame`</span><span class="p">(</span><span class="n">d</span><span class="p">,</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">)</span><span class="w"> </span><span class="o">:</span><span class="w"> </span><span class="n">undefined</span><span class="w"> </span><span class="n">columns</span><span class="w"> </span><span class="n">selected</span>
<span class="n">Calls</span><span class="o">:</span><span class="w"> </span><span class="n">upsetplot</span><span class="w"> </span><span class="kc">...</span><span class="w"> </span><span class="n">eval_tidy</span><span class="w"> </span><span class="o">-&gt;</span><span class="w"> </span><span class="n">split</span><span class="w"> </span><span class="o">-&gt;</span><span class="w"> </span><span class="n">split.default</span><span class="w"> </span><span class="o">-&gt;</span><span class="w"> </span><span class="p">[</span><span class="w"> </span><span class="o">-&gt;</span><span class="w"> </span><span class="p">[</span><span class="n">.data.frame</span>
<span class="n">Execution</span><span class="w"> </span><span class="n">halted</span>
</code></pre></div>
</blockquote>
<p>No significant enrichment found in any GO term.</p>
<h4 id="inferred-from-tfr2-wo-tfr2">Inferred from TFR2 w/o TFR2<a class="headerlink" href="#inferred-from-tfr2-wo-tfr2" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code>mkdir<span class="w"> </span>-p<span class="w"> </span>GO_FDR/separate/TFR2_wo_orig/symbol
</code></pre></div>
<div class="highlight"><pre><span></span><code>Rscript<span class="w"> </span>GO_enrichment_FDR.R<span class="w"> </span>-k<span class="w"> </span><span class="s2">&quot;SYMBOL&quot;</span><span class="w"> </span>-G<span class="w"> </span>results/separate/TFR2_wo_original.list<span class="w"> </span>-c<span class="w"> </span><span class="s2">&quot;BP&quot;</span><span class="w"> </span>-a<span class="w"> </span><span class="s2">&quot;BH&quot;</span><span class="w"> </span>-o<span class="w"> </span><span class="s2">&quot;default_universe&quot;</span><span class="w"> </span>-d<span class="w"> </span><span class="s2">&quot;GO_FDR/separate/TFR2_wo_orig/symbol&quot;</span>
</code></pre></div>
<blockquote>
<div class="highlight"><pre><span></span><code><span class="kc">...</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Loading input data...&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Warning: using defaut universe automatically provided by the clusterProfiler package&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done.&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Computing GO enrichment...&quot;</span>
<span class="n">`universe`</span><span class="w"> </span><span class="n">is</span><span class="w"> </span><span class="n">not</span><span class="w"> </span><span class="n">in</span><span class="w"> </span><span class="n">character</span><span class="w"> </span><span class="n">and</span><span class="w"> </span><span class="n">will</span><span class="w"> </span><span class="n">be</span><span class="w"> </span><span class="n">ignored...</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done.&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;18866 (default) background genes&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;14 provided genes; 13 found by `enrichGO`&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Computed GO enrichment (whether significant or not) for 276 distinct GO terms&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Of those 276 GO terms, 0 have a BH-adjusted p-val &lt; 0.1&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Writing outputs tables...&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done. Writing output images...&quot;</span>
<span class="n">Error</span><span class="w"> </span><span class="n">in</span><span class="w"> </span><span class="nf">`[.data.frame`</span><span class="p">(</span><span class="n">d</span><span class="p">,</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">)</span><span class="w"> </span><span class="o">:</span><span class="w"> </span><span class="n">undefined</span><span class="w"> </span><span class="n">columns</span><span class="w"> </span><span class="n">selected</span>
<span class="n">Calls</span><span class="o">:</span><span class="w"> </span><span class="n">upsetplot</span><span class="w"> </span><span class="kc">...</span><span class="w"> </span><span class="n">eval_tidy</span><span class="w"> </span><span class="o">-&gt;</span><span class="w"> </span><span class="n">split</span><span class="w"> </span><span class="o">-&gt;</span><span class="w"> </span><span class="n">split.default</span><span class="w"> </span><span class="o">-&gt;</span><span class="w"> </span><span class="p">[</span><span class="w"> </span><span class="o">-&gt;</span><span class="w"> </span><span class="p">[</span><span class="n">.data.frame</span>
<span class="n">Execution</span><span class="w"> </span><span class="n">halted</span>
</code></pre></div>
</blockquote>
<p>No significant enrichment found in any GO term.</p>
<h4 id="inferred-from-tmprss6">Inferred from TMPRSS6<a class="headerlink" href="#inferred-from-tmprss6" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code>mkdir<span class="w"> </span>-p<span class="w"> </span>GO_FDR/separate/TMPRSS6/symbol
</code></pre></div>
<div class="highlight"><pre><span></span><code>Rscript<span class="w"> </span>GO_enrichment_FDR.R<span class="w"> </span>-k<span class="w"> </span><span class="s2">&quot;SYMBOL&quot;</span><span class="w"> </span>-G<span class="w"> </span>results/separate/TMPRSS6.list<span class="w"> </span>-f<span class="w"> </span><span class="m">0</span>.05<span class="w"> </span>-c<span class="w"> </span><span class="s2">&quot;BP&quot;</span><span class="w"> </span>-a<span class="w"> </span><span class="s2">&quot;BH&quot;</span><span class="w"> </span>-o<span class="w"> </span><span class="s2">&quot;default_universe&quot;</span><span class="w"> </span>-d<span class="w"> </span><span class="s2">&quot;GO_FDR/separate/TMPRSS6/symbol&quot;</span>
</code></pre></div>
<blockquote>
<div class="highlight"><pre><span></span><code><span class="kc">...</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Loading input data...&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Warning: using defaut universe automatically provided by the clusterProfiler package&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done.&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Computing GO enrichment...&quot;</span>
<span class="n">`universe`</span><span class="w"> </span><span class="n">is</span><span class="w"> </span><span class="n">not</span><span class="w"> </span><span class="n">in</span><span class="w"> </span><span class="n">character</span><span class="w"> </span><span class="n">and</span><span class="w"> </span><span class="n">will</span><span class="w"> </span><span class="n">be</span><span class="w"> </span><span class="n">ignored...</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done.&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;18866 (default) background genes&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;7 provided genes; 6 found by `enrichGO`&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Computed GO enrichment (whether significant or not) for 185 distinct GO terms&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Of those 185 GO terms, 46 have a BH-adjusted p-val &lt; 0.05&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Writing outputs tables...&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done. Writing output images...&quot;</span>
<span class="kc">...</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done.&quot;</span>
</code></pre></div>
</blockquote>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/TMPRSS6/symbol/default_universe.BP.0.05BH.dotplot.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/TMPRSS6/symbol/default_universe.BP.0.05BH.gene-concept.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/TMPRSS6/symbol/default_universe.BP.0.05BH.gene-concept.circular.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/TMPRSS6/symbol/default_universe.BP.0.05BH.heatplot.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/TMPRSS6/symbol/default_universe.BP.0.05BH.upset.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/TMPRSS6/symbol/default_universe.BP.0.05BH.pvalues.png" /></p>
<h4 id="inferred-from-tmprss6-wo-tmprss6">Inferred from TMPRSS6 w/o TMPRSS6<a class="headerlink" href="#inferred-from-tmprss6-wo-tmprss6" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code>mkdir<span class="w"> </span>-p<span class="w"> </span>GO_FDR/separate/TMPRSS6_wo_orig/symbol
</code></pre></div>
<div class="highlight"><pre><span></span><code>Rscript<span class="w"> </span>GO_enrichment_FDR.R<span class="w"> </span>-k<span class="w"> </span><span class="s2">&quot;SYMBOL&quot;</span><span class="w"> </span>-G<span class="w"> </span>results/separate/TMPRSS6_wo_original.list<span class="w"> </span>-f<span class="w"> </span><span class="m">0</span>.05<span class="w"> </span>-c<span class="w"> </span><span class="s2">&quot;BP&quot;</span><span class="w"> </span>-a<span class="w"> </span><span class="s2">&quot;BH&quot;</span><span class="w"> </span>-o<span class="w"> </span><span class="s2">&quot;default_universe&quot;</span><span class="w"> </span>-d<span class="w"> </span><span class="s2">&quot;GO_FDR/separate/TMPRSS6_wo_orig/symbol&quot;</span>
</code></pre></div>
<blockquote>
<div class="highlight"><pre><span></span><code><span class="kc">...</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Loading input data...&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Warning: using defaut universe automatically provided by the clusterProfiler package&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done.&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Computing GO enrichment...&quot;</span>
<span class="n">`universe`</span><span class="w"> </span><span class="n">is</span><span class="w"> </span><span class="n">not</span><span class="w"> </span><span class="n">in</span><span class="w"> </span><span class="n">character</span><span class="w"> </span><span class="n">and</span><span class="w"> </span><span class="n">will</span><span class="w"> </span><span class="n">be</span><span class="w"> </span><span class="n">ignored...</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done.&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;18866 (default) background genes&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;6 provided genes; 5 found by `enrichGO`&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Computed GO enrichment (whether significant or not) for 165 distinct GO terms&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Of those 165 GO terms, 69 have a BH-adjusted p-val &lt; 0.05&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Writing outputs tables...&quot;</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done. Writing output images...&quot;</span>
<span class="kc">...</span>
<span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s">&quot;Done.&quot;</span>
</code></pre></div>
</blockquote>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/TMPRSS6_wo_orig/symbol/default_universe.BP.0.05BH.dotplot.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/TMPRSS6_wo_orig/symbol/default_universe.BP.0.05BH.gene-concept.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/TMPRSS6_wo_orig/symbol/default_universe.BP.0.05BH.gene-concept.circular.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/TMPRSS6_wo_orig/symbol/default_universe.BP.0.05BH.heatplot.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/TMPRSS6_wo_orig/symbol/default_universe.BP.0.05BH.upset.png" /></p>
<p><img alt="" src="images_go_enrichment_md/GO_FDR/separate/TMPRSS6_wo_orig/symbol/default_universe.BP.0.05BH.pvalues.png" /></p>





                
              </article>
            </div>
          
          
        </div>
        
          <a href="#" class="md-top md-icon" data-md-component="top" hidden>
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg>
            Back to top
          </a>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      2020 - 2022 Tristan Hoellinger
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
        <div class="md-social">
  
    
    
    
    
      
      
    
    <a href="https://twitter.com/TristanHoellin1" target="_blank" rel="noopener" title="twitter.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"/></svg>
    </a>
  
    
    
    
    
      
      
    
    <a href="https://fr.linkedin.com/in/tristan-hoellinger" target="_blank" rel="noopener" title="fr.linkedin.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"/></svg>
    </a>
  
    
    
    
    
    <a href="mailto:sarah.djebali@inserm.fr" target="_blank" rel="noopener" title="" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M498.1 5.6c10.1 7 15.4 19.1 13.5 31.2l-64 416c-1.5 9.7-7.4 18.2-16 23s-18.9 5.4-28 1.6L284 427.7l-68.5 74.1c-8.9 9.7-22.9 12.9-35.2 8.1S160 493.2 160 480v-83.6c0-4 1.5-7.8 4.2-10.7l167.6-182.9c5.8-6.3 5.6-16-.4-22s-15.7-6.4-22-.7L106 360.8l-88.3-44.2C7.1 311.3.3 300.7 0 288.9s5.9-22.8 16.1-28.7l448-256c10.7-6.1 23.9-5.5 34 1.4z"/></svg>
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    <script id="__config" type="application/json">{"base": "../..", "features": ["navigation.instant", "navigation.tabs", "navigation.sections", "navigation.top"], "search": "../../assets/javascripts/workers/search.db81ec45.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../../assets/javascripts/bundle.6df46069.min.js"></script>
      
    
  </body>
</html>