<!doctype html>
<html>
<head>
<meta charset='UTF-8'><meta name='viewport' content='width=device-width initial-scale=1'>
<title>gwas</title></head>
<body><h1>Intersect enhancers suspected to regulate the 13 initial genes (involved in hemochromatosis or iron metabolism regulation), with the NHGRI-EBI GWAS Catalog</h1>
<h2>Data acquisition</h2>
<h3>Download the latest GWAS catalog</h3>
<p>The version information on the GWAS website were the following when I downloaded the catalog:</p>
<blockquote><p>As of 2021-10-22, the GWAS Catalog contains 5419 publications and 316782 associations. GWAS Catalog data is currently mapped to Genome Assembly GRCh38.p13 and dbSNP Build 154.</p>
</blockquote>
<pre><code class='language-shell' lang='shell'>cd /work2/project/regenet/results/gwas/gwas.catalog
</code></pre>
<pre><code class='language-shell' lang='shell'>wget https://www.ebi.ac.uk/gwas/api/search/downloads/alternative -O /work2/project/regenet/results/gwas/gwas.catalog/v1.0.2_r2021-10-02/gwas_catalog_v1.0.2-associations_r2021-10-02.tsv
</code></pre>
<h3>List of enhancers suspected to regulate genes involved in hemochromatosis</h3>
<p>We obtained it while computing networks with the R markdown.</p>
<pre><code class='language-shell' lang='shell'>cd /work2/project/regenet/results/multi/abc.model/Nasser2021/
mv GWAS/ GWAS_10_initial_genes_fall_2021/
mkdir GWAS_13_initial_genes/ &amp;&amp; cd GWAS_13_initial_genes/
</code></pre>
<pre><code class='language-shell' lang='shell'>cp /work2/project/regenet/workspace/thoellinger/shared/2022/networks_hemochromatosis/results/enhancers.list .
</code></pre>
<h3>List of enhancers suspected to regulate the 457 original+new genes</h3>
<pre><code class='language-shell' lang='shell'>cd /work2/project/regenet/results/multi/abc.model/Nasser2021/GWAS_13_initial_genes/
cp /work2/project/regenet/workspace/thoellinger/shared/2022/networks_hemochromatosis/results/enhancers.new_genes.list .
</code></pre>
<h2>Data pre-processing</h2>
<h3>Enhancer lists</h3>
<p>We want to convert them in the bed format so we do the following:</p>
<pre><code class='language-shell' lang='shell'>conda activate base &amp;&amp; module load bioinfo/bedtools-2.27.1
</code></pre>
<pre><code class='language-shell' lang='shell'>awk &#39;BEGIN{FS=&quot;\t&quot;; OFS=&quot;\t&quot;}; {split($1,l1,&quot;:&quot;); split(l1[2],l2,&quot;-&quot;); print l1[1], l2[1], l2[2]}&#39; enhancers.list |bedtools sort -faidx ../chr_sizes -i stdin &gt; enhancers.sorted.bed
</code></pre>
<pre><code class='language-shell' lang='shell'>awk &#39;BEGIN{FS=&quot;\t&quot;; OFS=&quot;\t&quot;}; {split($1,l1,&quot;:&quot;); split(l1[2],l2,&quot;-&quot;); print l1[1], l2[1], l2[2]}&#39; enhancers.new_genes.list |bedtools sort -faidx ../chr_sizes -i stdin &gt; enhancers.new_genes.sorted.bed
</code></pre>
<h3>SNP related to hemochromatosis</h3>
<pre><code class='language-shell' lang='shell'>cd /work2/project/regenet/results/gwas/gwas.catalog/v1.0.2_r2021-10-02/
</code></pre>
<p>We want to extract from the GWAS catalog only the SNP related to hemochromatosis. Let us first investigate the list of all diseases / traits in the GWAS catalog.</p>
<blockquote><pre><code class='language-shell' lang='shell'>awk &#39;BEGIN{FS=&quot;\t&quot;; OFS=&quot;\t&quot;} {traits[$8]++} END{for(u in traits){print u, traits[u]}}&#39; gwas_catalog_v1.0.2-associations_r2021-10-02.tsv |wc -l
</code></pre>
<p>5845</p>
</blockquote>
<blockquote><pre><code class='language-shell' lang='shell'>awk &#39;BEGIN{FS=&quot;\t&quot;; OFS=&quot;\t&quot;} {traits[$8]++} END{for(u in traits){print u, traits[u]}}&#39; gwas_catalog_v1.0.2-associations_r2021-10-02.tsv |sort -nrk2,2 |head -n 30
</code></pre>
<pre><code class='language-shell' lang='shell'>Height	6144
Schizophrenia	2161
Hematocrit	1913
Triglycerides	1487
Neuroticism	1171
Asthma	1168
Weight	1037
Plateletcrit	1014
Hemoglobin	973
Intelligence	581
Chronotype	479
Psoriasis	422
Insomnia	422
Hypertension	308
Hypothyroidism	291
Depression	290
Morningness	274
Tonsillectomy	241
Eczema	214
FEV1	206
Neurociticism	181
Longevity	169
Adventurousness	167
Glaucoma	162
Snoring	158
Cataracts	154
Melanoma	149
Endometriosis	148
Worry	147
Migraine	146
</code></pre>
</blockquote>
<blockquote><pre><code class='language-shell' lang='shell'>awk -F &quot;\t&quot; &#39;$8~/([Hh]emochromatosis)/&#39; gwas_catalog_v1.0.2-associations_r2021-10-02.tsv |wc -l
</code></pre>
<p>2</p>
</blockquote>
<pre><code class='language-shell' lang='shell'>grep emochromatos gwas_catalog_v1.0.2-associations_r2021-10-02.tsv |awk &#39;BEGIN{FS=&quot;\t&quot;; OFS=&quot;\t&quot;} {traits[$8]++} END{for(u in traits){print u, traits[u]}}&#39; |sort -t $&#39;\t&#39; -nrk2,2
</code></pre>
<blockquote><pre><code class='language-shell' lang='shell'>Iron status biomarkers (transferrin levels)	9
Iron status biomarkers (transferrin saturation)	6
Iron status biomarkers (iron levels)	5
Iron status biomarkers (ferritin levels)	5
Hereditary hemochromatosis-related traits (HFE mutation homozygotes)	2
</code></pre>
</blockquote>
<p>Only 2 entries involve hemochromatosis directly, and a few others (25) involve studies about hemochromatosis but not SNP directly related to hemochromatosis. We check that no entries correspond to the other name of hemochromatosis, namely iron overload:</p>
<blockquote><pre><code class='language-shell' lang='shell'>grep &#39;iron.*overload&#39; gwas_catalog_v1.0.2-associations_r2021-10-02.tsv |wc -l
</code></pre>
<p>0</p>
</blockquote>
<p>We conclude that there are too few data to focus on those involving hemochromatosis. We will rather focus on those for which the trait is <code>Iron status biomarkers</code></p>
<h3>SNP related to iron in general</h3>
<blockquote><pre><code class='language-shell' lang='shell'>grep &#39;[iI]ron.*biomarkers&#39; gwas_catalog_v1.0.2-associations_r2021-10-02.tsv |wc -l
</code></pre>
<p>250</p>
</blockquote>
<p>Well we use the following filter to be sure not to miss any trait involving iron, then we will filter out by hand those which do not actually correspond to iron (such as &quot;env<strong>iron</strong>ment&quot;) or that do not interest us:</p>
<blockquote><pre><code class='language-shell' lang='shell'>awk -F &quot;\t&quot; &#39;$8~/([iI]ron)/&#39; gwas_catalog_v1.0.2-associations_r2021-10-02.tsv |wc -l
</code></pre>
<p>378</p>
</blockquote>
<pre><code class='language-shell' lang='shell'>awk -F &quot;\t&quot; &#39;if($8~/([iI]ron)/ {traits[$8]++}; END{for(u in traits){print u, traits[u]}}&#39; gwas_catalog_v1.0.2-associations_r2021-10-02.tsv |sort -t $&#39;\t&#39; -nrk2,2
</code></pre>
<blockquote><pre><code class='language-shell' lang='shell'>Iron status biomarkers (ferritin levels)	55
Iron status biomarkers	54
Iron status biomarkers (total iron binding capacity)	48
Sensitivity to environmental stress and adversity	47
Iron status biomarkers (transferrin saturation)	47
Iron status biomarkers (iron levels)	37
Liver iron content	23
Economic and political preferences (environmentalism)	17
BMI x  environmental factors (excluding physical activity) interaction	11
Iron status biomarkers (transferrin levels)	9
Forced expiratory volume in 1 second (occupational environmental exposures interaction)	7
BMI x environmental factors (including physical activity) interaction	7
Iron deficiency anemia	5
Type 2 diabetes (dietary heme iron intake interaction)	4
Forced expiratory volume in 1 second (environmental tobacco smoke interaction)	3
Pancreas iron content	1
Iron levels	1
Iron deficiency	1
Cleft plate (environmental tobacco smoke interaction)	1
</code></pre>
</blockquote>
<p>So the only ones that could interest us are the following:</p>
<blockquote><pre><code class='language-shell' lang='shell'>Iron status biomarkers (ferritin levels)    55
Iron status biomarkers    54
Iron status biomarkers (total iron binding capacity)    48
Iron status biomarkers (transferrin saturation)    47
Iron status biomarkers (iron levels)    37
Liver iron content    23
Iron status biomarkers (transferrin levels)    9
Iron deficiency anemia    5
Type 2 diabetes (dietary heme iron intake interaction)    4
Pancreas iron content    1
Iron levels    1
Iron deficiency    1
Hereditary hemochromatosis-related traits (HFE mutation homozygotes) 2
</code></pre>
</blockquote>
<p>We remove those involved in diabetes and pancreas, leading to the following list:</p>
<blockquote><pre><code class='language-shell' lang='shell'>Iron status biomarkers (ferritin levels)
Iron status biomarkers
Iron status biomarkers (total iron binding capacity)
Iron status biomarkers (transferrin saturation)
Iron status biomarkers (iron levels)
Liver iron content
Iron status biomarkers (transferrin levels)
Iron deficiency anemia
Iron levels
Iron deficiency
Hereditary hemochromatosis-related traits (HFE mutation homozygotes)
</code></pre>
</blockquote>
<p>So now let&#39;s go back to our working directory and extract only the lines of interest:</p>
<pre><code class='language-shell' lang='shell'>cd /work2/project/regenet/results/multi/abc.model/Nasser2021/GWAS_13_initial_genes
gwas=&quot;/work2/project/regenet/results/gwas/gwas.catalog/v1.0.2_r2021-10-02/gwas_catalog_v1.0.2-associations_r2021-10-02.tsv&quot;
</code></pre>
<pre><code class='language-shell' lang='shell'>awk -F &quot;\t&quot; &#39;$8~/(Iron status biomarkers)|(Liver iron content)|(Iron deficiency anemia)|(Iron levels)|(Iron deficiency)|(Hereditary hemochromatosis-related traits (HFE mutation homozygotes))/&#39; $gwas &gt; associations_full.tsv
</code></pre>
<p><code>associations_full.tsv</code> contains 280 rows. Remark: we see in the following section that only 269 of those lines are valid.</p>
<h2>Perform intersection (will enhancers regulating the 10 initial genes)</h2>
<h3>TSV to bed</h3>
<p>Fields $12 and $13 in <code>associations_full.tsv</code> are <chr id> and <chr pos> of the SNP. The 3 first columns of the bed file shall be the position of the SNP, and in column 4, as feature name, we concatenate all the columns of <code>associations_full.tsv</code>  with a separator like &quot;::&quot;, so that the output is strictly in bed format.</p>
<pre><code class='language-shell' lang='shell'>awk &#39;BEGIN{FS=&quot;\t&quot;; OFS=&quot;\t&quot;} {if($13 &amp;&amp; $14){ORS=&quot;\t&quot;; print &quot;chr&quot;$12, $13, $13+1; ORS=&quot;\n&quot;; gsub(/\t/,&quot;::&quot;,$0); print}}&#39; associations_full.tsv |bedtools sort -faidx ../chr_sizes -i stdin &gt; associations_full.bed
</code></pre>
<h3>Liftover</h3>
<p>The Genome Assembly used in our GWAS catalog is GRCh38.p13 whereas it&#39;s GRCh37/hg19 in the Nasser2021 predictions, so we need to start with a liftover.</p>
<p>In order to do this we need several modules, that should contain the keywords <code>kent</code> and <code>utils</code>. Note that the output of <code>module av</code> is actually directed to stderr so we use <code>2&gt;&amp;1</code> to redirect stderr to stdout, then <code>&gt;/dev/null</code> to redirect stdout to <code>/dev/null</code>.</p>
<pre><code class='language-shell' lang='shell'>(module av) 2&gt;&amp;1 &gt;/dev/null |grep kent
</code></pre>
<blockquote><pre><code class='language-shell' lang='shell'>bioinfo/kentUtils-302.1.0
bioinfo/kentUtils-v370
</code></pre>
</blockquote>
<p>So we can finally load those modules:</p>
<pre><code class='language-shell' lang='shell'>module load bioinfo/kentUtils-302.1.0 bioinfo/kentUtils-v370
</code></pre>
<p>(actually <code>bioinfo/kentUtils-v370</code> should suffice, cf mail from Sarah 07/12/2020, 14:34).</p>
<p>Now we can simply use <code>liftOver</code>:</p>
<blockquote><pre><code class='language-shell' lang='shell'>liftOver
</code></pre>
<pre><code>liftOver - Move annotations from one assembly to another
usage:
liftOver oldFile map.chain newFile unMapped
oldFile and newFile are in bed format by default, but can be in GFF and
maybe eventually others with the appropriate flags below.
The map.chain file has the old genome as the target and the new genome
as the query.
</code></pre>
</blockquote>
<p><code>unlifted.bed</code> file is not an optional argument and will contain all genome positions that cannot be lifted in case it happens for some. Full documentation here: <a href='https://genome.sph.umich.edu/wiki/LiftOver' target='_blank' class='url'>https://genome.sph.umich.edu/wiki/LiftOver</a></p>
<p>One can find the <code>chain</code> file that goes from hg38 to hg19 here: <a href='http://hgdownload.soe.ucsc.edu/goldenPath/hg38/liftOver/' target='_blank' class='url'>http://hgdownload.soe.ucsc.edu/goldenPath/hg38/liftOver/</a>. Let&#39;s download it:</p>
<pre><code class='language-shell' lang='shell'>rsync -avzP rsync://hgdownload.cse.ucsc.edu/goldenPath/hg38/liftOver/hg38ToHg19.over.chain.gz .
</code></pre>
<blockquote><p>Not this chain file was last modified at the date: 2013-12-31 17:24</p>
</blockquote>
<p>So now we can perform the liftover. <code>enhancers.sorted.bed</code> is already mapped over the hg19 assembly, so all we need to do is to convert <code>associations_full.bed</code> to hg19.</p>
<pre><code class='language-shell' lang='shell'>liftOver -bedPlus=1 -tab associations_full.bed hg38ToHg19.over.chain associations_full.hg19_liftover.bed unlifted.bed
</code></pre>
<p>OK.</p>
<p>Note that using <code>meld</code> we found that all rows are different between <code>associations_full.bed</code> and <code>associations_full.hg19_liftover.bed</code>.</p>
<h3>Intersect</h3>
<pre><code class='language-shell' lang='shell'>srun --mem=32G --pty bash
conda activate base &amp;&amp; module load bioinfo/bedtools-2.27.1
</code></pre>
<pre><code class='language-shell' lang='shell'>bedtools intersect -wo -a enhancers.sorted.bed -b associations_full.hg19_liftover.bed &gt; enhancers.overlap_GWAS.bedpe
</code></pre>
<blockquote><pre><code class='language-shell' lang='shell'>wc -l enhancers.overlap_GWAS.bedpe
</code></pre>
<p>4</p>
<pre><code class='language-shell' lang='shell'>cat enhancers.overlap_GWAS.bedpe
</code></pre>
<pre><code class='language-shell' lang='shell'>chr2	190405339	190407623	chr2	190407500	190407501	2021-08-13::34128465::Liu Y::2021-06-15::Elife::www.ncbi.nlm.nih.gov/pubmed/34128465::Genetic architecture of 11 organ traits derived from abdominal MRI using deep learning.::Liver iron content::32,858 European ancestry individuals::NA::2q32.2::2::189542774::SLC40A1::KDM3AP1 - SLC40A1::ENSG00000233996::ENSG00000138449::::54782::17816::rs115380467-?::rs115380467::0::115380467::regulatory_region_variant::1::NR::3E-12::11.522878745280337::(conditional)::0.181::NR mg/g increase::Affymetrix [9390170] (imputed)::N::liver iron measurement::http://www.ebi.ac.uk/efo/EFO_0010056::GCST90016674::Genome-wide genotyping array	1
chr3	195799723	195801054	chr3	195800811	195800812	2021-03-22::33536631::Bell S::2021-02-03::Commun Biol::www.ncbi.nlm.nih.gov/pubmed/33536631::A genome-wide meta-analysis yields 46 new loci associating with biomarkers of iron homeostasis.::Iron status biomarkers (total iron binding capacity)::135,430 European ancestry individuals::NA::3q29::3::196073940::TFRC::TFRC::::::ENSG00000072274::::::rs3817672-C::rs3817672::0::3817672::missense_variant::0::0.44::6E-11::10.221848749616356::::0.031::[0.04-0.022] SD decrease::Affymetrix, Illumina [40000000] (imputed)::N::total iron binding capacity::http://www.ebi.ac.uk/efo/EFO_0006334::GCST011368::Genome-wide genotyping array	1
chr3	195799723	195801054	chr3	195800811	195800812	2021-03-22::33536631::Bell S::2021-02-03::Commun Biol::www.ncbi.nlm.nih.gov/pubmed/33536631::A genome-wide meta-analysis yields 46 new loci associating with biomarkers of iron homeostasis.::Iron status biomarkers (transferrin saturation)::131,471 European ancestry individuals::NA::3q29::3::196073940::TFRC::TFRC::::::ENSG00000072274::::::rs3817672-C::rs3817672::0::3817672::missense_variant::0::0.44::2E-8::7.698970004336019::::0.026::[0.017-0.034] SD increase::Affymetrix, Illumina [40000000] (imputed)::N::transferrin saturation measurement::http://www.ebi.ac.uk/efo/EFO_0006333::GCST011366::Genome-wide genotyping array	1
chr22	37505175	37505880	chr22	37505552	37505553	2015-06-14::25352340::Benyamin B::2014-10-29::Nat Commun::www.ncbi.nlm.nih.gov/pubmed/25352340::Novel loci affecting iron homeostasis and their effects in individuals at risk for hemochromatosis.::Iron status biomarkers (iron levels)::23,986 European ancestry individuals::Up to 24,986 European ancestry individuals::22q12.3::22::37109512::TMPRSS6::TMPRSS6::::::ENSG00000187045::::::rs228916-T::rs228916::0::228916::5_prime_UTR_variant::0::0.875::3E-8::7.522878745280337::::0.086::[0.055-0.117] unit decrease::Affymetrix, Illumina [~ 2100000] (imputed)::N::iron biomarker measurement, serum iron measurement::http://www.ebi.ac.uk/efo/EFO_0004461, http://www.ebi.ac.uk/efo/EFO_0006332::GCST002679::Genome-wide genotyping array	1
</code></pre>
</blockquote>
<p>Well, there are only 4 overlap, involving only 3 unique enhancers. This is expected, as we are searching for overlap between a list of only 137 enhancer over the whole genome, and a list of only 269 SNP over the whole genome.</p>
<h2>Perform intersection (with enhancers regulating the 457 original + newly found genes)</h2>
<pre><code class='language-shell' lang='shell'>srun --mem=32G --pty bash
conda activate base &amp;&amp; module load bioinfo/bedtools-2.27.1
</code></pre>
<pre><code class='language-shell' lang='shell'>bedtools intersect -wo -a enhancers.new_genes.sorted.bed -b associations_full.hg19_liftover.bed &gt; enhancers.new_genes.overlap_GWAS.bedpe
</code></pre>
<p>There is only one more overlap, namely:</p>
<blockquote><pre><code class='language-shell' lang='shell'>chr6	26103318	26105346	chr6	26104632	26104633	2021-08-13::34128465::Liu Y::2021-06-15::Elife::www.ncbi.nlm.nih.gov/pubmed/34128465::Genetic architecture of 11 organ traits derived from abdominal MRI using deep learning.::Liver iron content::32,858 European ancestry individuals::NA::6p22.2::6::26104404::HIST1H4C::H4C3 - H1-6::ENSG00000197061::ENSG00000187475::::67::3008::rs198851-?::rs198851::0::198851::TF_binding_site_variant::1::NR::3E-73::72.52287874528034::(conditional)::0.194::NR mg/g decrease::Affymetrix [9390170] (imputed)::N::liver iron measurement::http://www.ebi.ac.uk/efo/EFO_0010056::GCST90016674::Genome-wide genotyping array	1
</code></pre>
</blockquote>
<h2>Are the genes regulated by the enhancers found to intersect a GWAS SNP, the same as those given in GWAS as associated to the SNP?</h2>
<p>We did this analysis starting from 10 instead of 13 genes, but as it conduces to only 1 more overlap, we do not re-perform it. So here is the analysis: <a href='https://genoweb.toulouse.inra.fr/~thoellinger/fall_2021/gwas.html' target='_blank' class='url'>https://genoweb.toulouse.inra.fr/~thoellinger/fall_2021/gwas.html</a></p>
</body>
</html>