
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
        <meta name="author" content="Tristan Hoellinger">
      
      
        <link rel="canonical" href="https://eg.github.io/haemochromatosis/data_format.html">
      
      
        <link rel="prev" href="introduction.html">
      
      
        <link rel="next" href="ABC/custom_go.html">
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.4.2, mkdocs-material-9.0.11">
    
    
      
        <title>Data format - Genome-wide identification of E/G interactions</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.0d440cfe.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.2505c338.min.css">
      
      

    
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../stylesheets/extra.css">
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="black" data-md-color-accent="">
  
    
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#compute-a-list-of-reliable-putative-new-genes" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="Genome-wide identification of E/G interactions" class="md-header__button md-logo" aria-label="Genome-wide identification of E/G interactions" data-md-component="logo">
      
  <img src="../logo/logo-inserm-blc.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Genome-wide identification of E/G interactions
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Data format
            
          </span>
        </div>
      </div>
    </div>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/hoellin/eg" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
  </div>
  <div class="md-source__repository">
    hoellin/eg
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../guidebooks/introduction.html" class="md-tabs__link">
        Guidebooks
      </a>
    </li>
  

      
        
  
  
    
  


  
  
  
    <li class="md-tabs__item">
      <a href="introduction.html" class="md-tabs__link md-tabs__link--active">
        Haemochromatosis
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../notes_BENGI/introduction.html" class="md-tabs__link">
        BENGI
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../notes_ABC/introduction.html" class="md-tabs__link">
        ABC
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../index.html" class="md-tabs__link">
        About
      </a>
    </li>
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="Genome-wide identification of E/G interactions" class="md-nav__button md-logo" aria-label="Genome-wide identification of E/G interactions" data-md-component="logo">
      
  <img src="../logo/logo-inserm-blc.png" alt="logo">

    </a>
    Genome-wide identification of E/G interactions
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/hoellin/eg" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
  </div>
  <div class="md-source__repository">
    hoellin/eg
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1" >
      
      
      
        <label class="md-nav__link" for="__nav_1" id="__nav_1_label" tabindex="0">
          Guidebooks
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_1_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_1">
          <span class="md-nav__icon md-icon"></span>
          Guidebooks
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../guidebooks/introduction.html" class="md-nav__link">
        Introduction
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../guidebooks/script_bedpe_summary/script_bedpe_summary.html" class="md-nav__link">
        Compute summary stats of a bedpe
      </a>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_3" >
      
      
      
        <label class="md-nav__link" for="__nav_1_3" id="__nav_1_3_label" tabindex="0">
          Basics
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_1_3">
          <span class="md-nav__icon md-icon"></span>
          Basics
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../guidebooks/shell.html" class="md-nav__link">
        Shell
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../guidebooks/conda_environments.html" class="md-nav__link">
        Conda basics
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../guidebooks/git.html" class="md-nav__link">
        Git
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../guidebooks/genotoul.html" class="md-nav__link">
        Genotoul cluster
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../guidebooks/notebooks_ssh.html" class="md-nav__link">
        How to use Jupyter through ssh tunneling?
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_4" >
      
      
      
        <label class="md-nav__link" for="__nav_1_4" id="__nav_1_4_label" tabindex="0">
          Misc
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_4_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_1_4">
          <span class="md-nav__icon md-icon"></span>
          Misc
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../guidebooks/compute_correlations.html" class="md-nav__link">
        Compute correlations
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" checked>
      
      
      
        <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
          Haemochromatosis
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="true">
        <label class="md-nav__title" for="__nav_2">
          <span class="md-nav__icon md-icon"></span>
          Haemochromatosis
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="introduction.html" class="md-nav__link">
        Introduction
      </a>
    </li>
  

            
          
            
              
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          Data format
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="data_format.html" class="md-nav__link md-nav__link--active">
        Data format
      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#context" class="md-nav__link">
    Context
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#results" class="md-nav__link">
    Results
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#remarks" class="md-nav__link">
    Remarks
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#data-format" class="md-nav__link">
    Data format
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#steps-to-obtain-this-file" class="md-nav__link">
    Steps to obtain this file
  </a>
  
    <nav class="md-nav" aria-label="Steps to obtain this file">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#compute-list-of-abc-inferred-genes" class="md-nav__link">
    Compute list of ABC-inferred genes
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#compute-list-of-chic-inferred-genes" class="md-nav__link">
    Compute list of CHiC-inferred genes
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#compute-a-table-giving-for-each-gene-symbol-its-corresponding-ensembl-id-whenever-known" class="md-nav__link">
    Compute a table giving, for each gene symbol, its corresponding ENSEMBL id, whenever known
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#combine-everything" class="md-nav__link">
    Combine everything
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#add-gene-ids-whenever-possible" class="md-nav__link">
    Add gene ids whenever possible
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#add-source-gene-symbol" class="md-nav__link">
    Add source gene symbol
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="ABC/custom_go.html" class="md-nav__link">
        Custom script GO
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="main_results.html" class="md-nav__link">
        Main results
      </a>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2_5" >
      
      
      
        <label class="md-nav__link" for="__nav_2_5" id="__nav_2_5_label" tabindex="0">
          CHiC data
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_2_5_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_2_5">
          <span class="md-nav__icon md-icon"></span>
          CHiC data
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="CHiC/preprocess_data.html" class="md-nav__link">
        Preprocessing
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="CHiC/preliminary_analysis_chic_v2.html" class="md-nav__link">
        Networks
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="CHiC/go_enrichment.html" class="md-nav__link">
        GO enrichment
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2_6" >
      
      
      
        <label class="md-nav__link" for="__nav_2_6" id="__nav_2_6_label" tabindex="0">
          ABC data
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_2_6_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_2_6">
          <span class="md-nav__icon md-icon"></span>
          ABC data
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="ABC/data_processing.html" class="md-nav__link">
        Preprocessing
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="ABC/preliminary_analysis_v9.html" class="md-nav__link">
        Networks
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="ABC/summary_statistics_eg_pairs.html" class="md-nav__link">
        Summary statistics enhancers
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="ABC/go_enrichment.html" class="md-nav__link">
        GO enrichment
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="ABC/gorilla.html" class="md-nav__link">
        GOrilla
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="ABC/compute_avg_nb_of_connections_per_gene.html" class="md-nav__link">
        Avg nb connections per gene
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="ABC/gwas.html" class="md-nav__link">
        GWAS (13 initial genes)
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="ABC/gwas_10.html" class="md-nav__link">
        GWAS (10 initial genes)
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" >
      
      
      
        <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
          BENGI
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3">
          <span class="md-nav__icon md-icon"></span>
          BENGI
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../notes_BENGI/introduction.html" class="md-nav__link">
        Introduction
      </a>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_2" >
      
      
      
        <label class="md-nav__link" for="__nav_3_2" id="__nav_3_2_label" tabindex="0">
          Notebooks
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_2_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_2">
          <span class="md-nav__icon md-icon"></span>
          Notebooks
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../notes_BENGI/BENGI_to_custom_bedpe.html" class="md-nav__link">
        Convert BENGI datasets to custom bedpe
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_3" >
      
      
      
        <label class="md-nav__link" for="__nav_3_3" id="__nav_3_3_label" tabindex="0">
          Summary statistics
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_3">
          <span class="md-nav__icon md-icon"></span>
          Summary statistics
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../notes_BENGI/summary_statistics/summary_statistics.html" class="md-nav__link">
        Summary statistics on BENGI GM12878 benchmarks
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../notes_BENGI/summary_statistics_BENGI/summary_BENGI_bedpe.html" class="md-nav__link">
        (R code) Compute summary statistics over BENGI
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_4" >
      
      
      
        <label class="md-nav__link" for="__nav_3_4" id="__nav_3_4_label" tabindex="0">
          Evaluation of a few methods over BENGI GM12878
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_4_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_4">
          <span class="md-nav__icon md-icon"></span>
          Evaluation of a few methods over BENGI GM12878
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_4_1" >
      
      
      
        <label class="md-nav__link" for="__nav_3_4_1" id="__nav_3_4_1_label" tabindex="0">
          Distance method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_4_1_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_4_1">
          <span class="md-nav__icon md-icon"></span>
          Distance method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../notes_BENGI/distance_method/distance_evaluation_with_code.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../notes_BENGI/distance_method/summary_distance_evaluation.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_4_2" >
      
      
      
        <label class="md-nav__link" for="__nav_3_4_2" id="__nav_3_4_2_label" tabindex="0">
          DNase-expression correlation method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_4_2_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_4_2">
          <span class="md-nav__icon md-icon"></span>
          DNase-expression correlation method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../notes_BENGI/dnase_expression_correlation/correlation_method_with_code.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../notes_BENGI/dnase_expression_correlation/summary_correlation_method.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_4_3" >
      
      
      
        <label class="md-nav__link" for="__nav_3_4_3" id="__nav_3_4_3_label" tabindex="0">
          Average rank method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_4_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_4_3">
          <span class="md-nav__icon md-icon"></span>
          Average rank method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../notes_BENGI/avg_rank_method/avg_rank_method_with_code.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../notes_BENGI/avg_rank_method/summary_avg_rank_method.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_5" >
      
      
      
        <label class="md-nav__link" for="__nav_3_5" id="__nav_3_5_label" tabindex="0">
          Evaluation of a few methods over Fulco et al 2019 CRISPRi-FlowFISH
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_5_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_5">
          <span class="md-nav__icon md-icon"></span>
          Evaluation of a few methods over Fulco et al 2019 CRISPRi-FlowFISH
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_5_1" >
      
      
      
        <label class="md-nav__link" for="__nav_3_5_1" id="__nav_3_5_1_label" tabindex="0">
          Distance method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_5_1_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_5_1">
          <span class="md-nav__icon md-icon"></span>
          Distance method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../notes_BENGI/CRISPRi_FlowFISH/distance_method/distance_over_fulco_et_al_crispri.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../notes_BENGI/CRISPRi_FlowFISH/distance_method/summary_distance_over_fulco_et_al_crispri.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_5_2" >
      
      
      
        <label class="md-nav__link" for="__nav_3_5_2" id="__nav_3_5_2_label" tabindex="0">
          DNase-expression correlation method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_5_2_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_5_2">
          <span class="md-nav__icon md-icon"></span>
          DNase-expression correlation method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../notes_BENGI/CRISPRi_FlowFISH/dnase_expression_correlation/correlation_method_with_code.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../notes_BENGI/CRISPRi_FlowFISH/dnase_expression_correlation/summary_correlation_method.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_5_3" >
      
      
      
        <label class="md-nav__link" for="__nav_3_5_3" id="__nav_3_5_3_label" tabindex="0">
          Average rank method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_5_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_5_3">
          <span class="md-nav__icon md-icon"></span>
          Average rank method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../notes_BENGI/CRISPRi_FlowFISH/avg_rank_method/avg_rank_method_with_code.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../notes_BENGI/CRISPRi_FlowFISH/avg_rank_method/summary_avg_rank.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" >
      
      
      
        <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
          ABC
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_4">
          <span class="md-nav__icon md-icon"></span>
          ABC
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../notes_ABC/introduction.html" class="md-nav__link">
        Introduction
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../notes_ABC/BENGI/ABC_BENGI_results.html" class="md-nav__link">
        ABC over BENGI for GM12878, main results
      </a>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_3" >
      
      
      
        <label class="md-nav__link" for="__nav_4_3" id="__nav_4_3_label" tabindex="0">
          CRISPRi-FlowFISH
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_4_3">
          <span class="md-nav__icon md-icon"></span>
          CRISPRi-FlowFISH
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../notes_ABC/K562/ABC_K562_CRISPRi_FlowFISH.html" class="md-nav__link">
        Original ABC predictions over K562
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../notes_ABC/K562/recap_investigations_ABC_over_CRiFF.html" class="md-nav__link">
        Summary ABC over CRiFF
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_4" >
      
      
      
        <label class="md-nav__link" for="__nav_4_4" id="__nav_4_4_label" tabindex="0">
          Notebooks
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_4_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_4_4">
          <span class="md-nav__icon md-icon"></span>
          Notebooks
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../notes_ABC/K562/april_K562_56_genes/april_K562_56_genes.html" class="md-nav__link">
        ABC over K562 cCRiFF
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../notes_ABC/K562/april_K562_59_genes/april_K562_59_genes.html" class="md-nav__link">
        ABC over K562 CRiFF
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../notes_ABC/K562/april_K562_candidates_not_in_whitelist_56_genes/april_K562_candidates_NOT_in_whitelist.html" class="md-nav__link">
        ABC over K562 cCRiFF, no whitelist
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../notes_ABC/generic_notebooks/turnkey_notebook_to_run_ABC_with_example_over_GM12878.html" class="md-nav__link">
        Generic notebook to run ABC method & Example over GM12878
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../notes_ABC/BENGI/notebook_ABC_over_BENGI_GM12878.html" class="md-nav__link">
        ABC over GM12878 with same accessions as for BENGI
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../notes_ABC/BENGI/notebook_ABC_over_BENGI_GM12878_from_ccRE_ELSs.html" class="md-nav__link">
        ABC over GM12878 with same accessions as for BENGI, starting with ccRE-dELS
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_5" >
      
      
      
        <label class="md-nav__link" for="__nav_4_5" id="__nav_4_5_label" tabindex="0">
          Full codes
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_5_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_4_5">
          <span class="md-nav__icon md-icon"></span>
          Full codes
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../notes_ABC/test_abc_pipeline.html" class="md-nav__link">
        Test of ABC pipeline over chr22
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../notes_ABC/BENGI/ABC_over_BENGI_GM12878.html" class="md-nav__link">
        ABC over GM12878 with same accessions as for BENGI
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../notes_ABC/differences_example_chr22.html" class="md-nav__link">
        Meld of results over chr22 vs authors' results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" >
      
      
      
        <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
          About
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_5">
          <span class="md-nav__icon md-icon"></span>
          About
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../index.html" class="md-nav__link">
        Getting started
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../about_me.html" class="md-nav__link">
        About me
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#context" class="md-nav__link">
    Context
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#results" class="md-nav__link">
    Results
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#remarks" class="md-nav__link">
    Remarks
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#data-format" class="md-nav__link">
    Data format
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#steps-to-obtain-this-file" class="md-nav__link">
    Steps to obtain this file
  </a>
  
    <nav class="md-nav" aria-label="Steps to obtain this file">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#compute-list-of-abc-inferred-genes" class="md-nav__link">
    Compute list of ABC-inferred genes
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#compute-list-of-chic-inferred-genes" class="md-nav__link">
    Compute list of CHiC-inferred genes
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#compute-a-table-giving-for-each-gene-symbol-its-corresponding-ensembl-id-whenever-known" class="md-nav__link">
    Compute a table giving, for each gene symbol, its corresponding ENSEMBL id, whenever known
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#combine-everything" class="md-nav__link">
    Combine everything
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#add-gene-ids-whenever-possible" class="md-nav__link">
    Add gene ids whenever possible
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#add-source-gene-symbol" class="md-nav__link">
    Add source gene symbol
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  

  
  



<h1 id="compute-a-list-of-reliable-putative-new-genes">Compute a list of reliable putative new genes<a class="headerlink" href="#compute-a-list-of-reliable-putative-new-genes" title="Permanent link">&para;</a></h1>
<h2 id="context">Context<a class="headerlink" href="#context" title="Permanent link">&para;</a></h2>
<p>We aim at finding new genes potentially related to haemochromatosis / involved in the severity of haemochromatosis. To that purpose, we started from a list of up to 13 genes, 6 of which being the ones known to be causally involved in haemochromatosis, and the other ones being strongly involved in iron metabolism regulation in liver and intestine.</p>
<p>We worked with E-G pairs obtained with either ABC or CHiC data, and inferred new genes as follows:</p>
<ol>
<li>Find all the enhancers E regulating the list of 13 initial genes (later one we will denote by "I" a gene from this list)</li>
<li>Find all genes G regulated by those very same enhancers E</li>
</ol>
<p>In other words, if G is a gene inferred, it means it is regulated by an enhancer E that is also regulating a initial gene, so we have a relation of the form "I &lt;- E -&gt; G".</p>
<p>We worked with both ABC and CHiC data, and obtained two separate list of inferred genes (of respective lengths 444 and 42). For more details on how those list have been obtained, one can refer to:</p>
<ul>
<li><code>data_and_results/haemochromatosis/networks_hemochromatosis/preliminary_analysis_v9.html</code> for ABC data</li>
<li><code>data_and_results/haemochromatosis/promoter_capture_hic/preliminary_analysis_chic_v2.html</code> for CHiC data</li>
</ul>
<p>Now, we want to merge these list together, and to attribute a confidence label to each gene, according to multiple factors. The exact process is explained below.</p>
<h2 id="results">Results<a class="headerlink" href="#results" title="Permanent link">&para;</a></h2>
<p>At the end of the day, one can directly use the following file:</p>
<p><code>data_and_results/haemochromatosis//merged_inferred_genes_v1.csv</code></p>
<h2 id="remarks">Remarks<a class="headerlink" href="#remarks" title="Permanent link">&para;</a></h2>
<ul>
<li>The field 2 (gene ids) is completed using a table then "by hand" at the very end</li>
<li>The field 4 (source gene symbol) was added by hand at the very end (because I did not planned to add it at first), but could have been handled since the very beginning (as I added it in the input files, namely <code>new_genes_abc_v9_more_info.list</code> and <code>new_genes_chic_v2_more_info.list</code>)</li>
</ul>
<h2 id="data-format">Data format<a class="headerlink" href="#data-format" title="Permanent link">&para;</a></h2>
<p>12 columns:</p>
<ol>
<li>
<p><strong>gene</strong> Gene symbol</p>
</li>
<li>
<p><strong>gene.id</strong> ENSEMBL id (if applicable, "." otherwise)</p>
</li>
<li>
<p><strong>score.adj</strong> The sum of all numerical $5-$9 labels (total "score" of the inferred gene) except that $9 (the number of enhancers) is divided by 4. See details of the following fields to understand why this score is expected to be relevant.</p>
</li>
<li>
<p><strong>source.gene</strong> Gene symbol of the "source" gene (the initial gene I that shares an enhancer E with the current gene G)</p>
</li>
<li>
<p><strong>score.tot</strong> The sum of all numerical $5-$9 labels (total "score" of the inferred gene, other approach). See details of the following fields to understand why this score is expected to be relevant.</p>
</li>
<li>
<p><strong>abc.enhancer.intersect.label</strong> Label between 0 and 2. Let us denote G the current gene, and I &lt;- E -&gt; G the way it has been obtained.</p>
<ul>
<li>2 (the best) if E is a putative enhancer of the ABC data, and intersects &gt;= 1 ccRE-ELS</li>
<li>1 if E is a putative enhancer of the ABC data, and intersects 0 ccRE-ELS but &gt;=1 ccRE</li>
<li>0 otherwise (if the enhancer does not intersect any ccRE <strong>or</strong> if the gene G is not inferred from ABC element-Gene pairs)</li>
</ul>
</li>
<li>
<p><strong>ABC.product.label</strong> Label between 0 and 3. The higher the better. 0 if not inferred from ABC data, 1 if small ABC scores, 2 medium ABC score, 3 high ABC score ; according to the product of the ABC scores of:</p>
<ul>
<li>the initial gene - enhancer pair (I-E)</li>
<li>the enhancer - gene pair (E-G)</li>
</ul>
<p>See the dedicated R markdown for more details ("small", "medium" and "high" have been defined in an arbitrary manner).</p>
</li>
<li>
<p><strong>chic.enhancer.intersect.label</strong> Label between 0 and 2. Let us denote G the current gene, and I &lt;- E -&gt; G the way it has been obtained.</p>
<ul>
<li>2 (the best) if E is a putative enhancer of the CHiC data, and intersects &gt;= 1 ccRE-ELS</li>
<li>1 if E is a putative enhancer of the CHiC data, and  intersects 0 ccRE-ELS but &gt;=1 ccRE</li>
<li>0 otherwise (if the enhancer does not intersect any ccRE <strong>or</strong> if the gene G is not inferred from CHiC element-Gene pairs)</li>
</ul>
<p>Remark: as a matter of fact, there are no "1" in the csv file, because, to define the list of genes inferred with CHiC data, we restricted ourselves to E-G pairs in which the enhancers intersects &gt;= 1 ccRE-ELS. We did so because putative enhancers in CHiC data are pretty big, and it would be surprising if such a large actual enhancer would not intersect any ccRE-ELS</p>
</li>
<li>
<p><strong>contact.product.label</strong> Label between 0 and 3. The higher the better. 0 if not inferred from CHiC data, 1 if small contact probability, 2 medium contact probability, 3 high contact probability ; according to the product of the contact probabilities of:</p>
<ul>
<li>the initial gene - enhancer pair (I-E)</li>
<li>the enhancer - gene pair (E-G)</li>
</ul>
<p>See the dedicated R markdown for more details ("small", "medium" and "high" have been defined in an arbitrary manner).</p>
</li>
<li>
<p><strong>count</strong> Label by the total number of E-G pairs involving the current gene (irrespective of if they are obtained from ABC or CHiC data). Note that this is nothing but the number of enhancers in $12.</p>
</li>
<li>
<p><strong>datatype</strong> Indicates whether the gene comes from ABC data, CHiC data, or both "ABC,CHiC" data.</p>
</li>
<li>
<p><strong>sources</strong> The list of enhancer coordinates of the I-E-G tuple (initial gene, enhancer, inferred gene).</p>
</li>
</ol>
<p>The file is sorted by value of column $3.</p>
<h2 id="steps-to-obtain-this-file">Steps to obtain this file<a class="headerlink" href="#steps-to-obtain-this-file" title="Permanent link">&para;</a></h2>
<h3 id="compute-list-of-abc-inferred-genes">Compute list of ABC-inferred genes<a class="headerlink" href="#compute-list-of-abc-inferred-genes" title="Permanent link">&para;</a></h3>
<p>In order to do so, we started with a dedicated R markdown. Details to be found here: <code>data_and_results/haemochromatosis/networks_hemochromatosis/preliminary_analysis_v9.html</code></p>
<p>It produces an output file looking as follows</p>
<div class="highlight"><pre><span></span><code>&lt;gene_symbol&gt;   &lt;list_of_enhancers_ids&gt; &lt;label_product_ABC_scores&gt;  &lt;nb_of_enhancers&gt;
</code></pre></div>
<p>Now, we want to add relevant information to this list of genes (whether the associated enhancers intersect ccRE, etc). The steps are given below.</p>
<p>First, we compute:</p>
<ul>
<li>
<p>a table which indicates, for every ABC enhancers id in our ABC data, whether the enhancer intersects &gt;= 1 ccRE-ELS</p>
<p>In <code>/work2/project/regenet/results/multi/abc.model/Nasser2021/</code>, we should find the 3 following files:</p>
<ul>
<li><code>list_all_enhancers.merged.bed</code></li>
<li><code>ccRE-ELS.bed</code></li>
<li><code>hg19-cCREs.bed</code></li>
</ul>
<p>We copy them in our current working directory, namely, <code>data_and_results/haemochromatosis/networks_hemochromatosis/data</code>. Now, we run the following commands:</p>
<div class="highlight"><pre><span></span><code>conda<span class="w"> </span>activate<span class="w"> </span>base<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>module<span class="w"> </span>load<span class="w"> </span>bioinfo/bedtools-2.27.1
bedtools<span class="w"> </span>intersect<span class="w"> </span>-c<span class="w"> </span>-a<span class="w"> </span>list_all_enhancers.merged.bed<span class="w"> </span>-b<span class="w"> </span>ccRE-ELS.bed<span class="w"> </span><span class="p">|</span>awk<span class="w"> </span><span class="s1">&#39;BEGIN{FS=&quot;\t&quot;; OFS=&quot;\t&quot;} {nbs[$4]++} END{for(u in nbs){print u, nbs[u]}}&#39;</span><span class="w"> </span><span class="p">|</span>sort<span class="w"> </span>-nrk2,2<span class="w">    </span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>bedtools<span class="w"> </span>intersect<span class="w"> </span>-c<span class="w"> </span>-a<span class="w"> </span>list_all_abc_enhancers.merged.bed<span class="w"> </span>-b<span class="w"> </span>ccRE-ELS.bed<span class="w"> </span><span class="p">|</span>awk<span class="w"> </span><span class="s1">&#39;$4&gt;=1 {print $1&quot;:&quot;$2&quot;-&quot;$3}&#39;</span><span class="w"> </span>&gt;<span class="w"> </span>list_all_abc_enhancers_intersecting_ccRE_ELS.bed
</code></pre></div>
</li>
<li>
<p>a table which indicates, for every ABC enhancers id in our ABC data, whether the enhancer intersects &gt;= 1 ccRE</p>
<div class="highlight"><pre><span></span><code>bedtools<span class="w"> </span>intersect<span class="w"> </span>-c<span class="w"> </span>-a<span class="w"> </span>list_all_abc_enhancers.merged.bed<span class="w"> </span>-b<span class="w"> </span>hg19-cCREs.bed<span class="w"> </span><span class="p">|</span>awk<span class="w"> </span><span class="s1">&#39;$4&gt;=1 {print $1&quot;:&quot;$2&quot;-&quot;$3}&#39;</span><span class="w"> </span>&gt;<span class="w"> </span>list_all_abc_enhancers_intersecting_ccRE.bed
</code></pre></div>
</li>
<li>
<p>starting from those 2 tables, a table which associates to each ABC enhancer id:</p>
<ul>
<li>2 if it intersects &gt;= 1 ccRE-ELS</li>
<li>1 if it intersects 0 ccRE-ELS but &gt;= 1 ccRE</li>
<li>0 otherwise</li>
</ul>
</li>
</ul>
<p>Well, let us compute this table only for enhancers actually found in our data:</p>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;BEGIN{OFS=&quot;\t&quot;} {if(NR==FNR){els[$1]++; next}; if(els[$1]){print $1, 2} else{print $1, $2}}&#39;</span><span class="w"> </span>data/list_all_abc_enhancers_intersecting_ccRE_ELS.bed<span class="w"> </span>&lt;<span class="o">(</span>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;BEGIN{OFS=&quot;\t&quot;} {if(NR==FNR){ccre[$1]++; next}; if(ccre[$1]){print $1, 1} else{print $1, 0}}&#39;</span><span class="w"> </span>data/list_all_abc_enhancers_intersecting_ccRE.bed<span class="w"> </span>networks_hemochromatosis/results/enhancers.new_genes.list<span class="o">)</span><span class="w"> </span>&gt;<span class="w"> </span>networks_hemochromatosis/results/enhancers.new_genes.intersection_label.list
</code></pre></div>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span><span class="w"> </span>/work2/project/regenet/workspace/thoellinger/shared/2022/networks_hemochromatosis/results
awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;{nb[$2]++} END{for(u in nb){print u, nb[u]}}&#39;</span><span class="w"> </span>enhancers.new_genes.intersection_label.list
</code></pre></div>
<blockquote>
<p>0 51</p>
<p>1 501</p>
<p>2 666</p>
</blockquote>
<p>Now, in awk, add a column ($4) to our output files, which contains the list of scores of all enhancers id in column $2. Then, replace this column with the max in the list.</p>
<blockquote>
<p>Note: let us remark that there are only 101 distinct enhancers in our data:</p>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;BEGIN{OFS=&quot;\t&quot;} {split($2,liste,&quot;,&quot;); for(i in liste){e=liste[i]; enhancers[e]++}} END{for(u in enhancers){print u, enhancers[u]}}&#39;</span><span class="w"> </span>&lt;<span class="o">(</span>tail<span class="w"> </span>-n+2<span class="w"> </span>new_genes_abc_v9_more_info.list<span class="o">)</span><span class="w"> </span><span class="p">|</span>wc<span class="w"> </span>-l
</code></pre></div>
<p>101</p>
</blockquote>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;BEGIN{OFS=&quot;\t&quot;} {print $1, $2, &quot;enhancer.intersect&quot;, $3, $4}&#39;</span><span class="w"> </span>&lt;<span class="o">(</span>head<span class="w"> </span>-n<span class="w"> </span><span class="m">1</span><span class="w"> </span>new_genes_abc_v9_more_info.list<span class="o">)</span><span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;BEGIN{OFS=&quot;\t&quot;} {if(NR==FNR){label[$1]=$2; next}; split($2,liste,&quot;,&quot;); labs=&quot;&quot;; for(i in liste){e=liste[i]; labs=labs&quot;&quot;label[e]}; print $1, $2, labs, $3, $4}&#39;</span><span class="w"> </span>enhancers.new_genes.intersection_label.list<span class="w"> </span>&lt;<span class="o">(</span>tail<span class="w"> </span>-n+2<span class="w"> </span>new_genes_abc_v9_more_info.list<span class="o">)</span><span class="w"> </span><span class="p">|</span>head
</code></pre></div>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;BEGIN{OFS=&quot;\t&quot;} {print $1, $2, &quot;enhancer.intersect&quot;, $3, $4}&#39;</span><span class="w"> </span>&lt;<span class="o">(</span>head<span class="w"> </span>-n<span class="w"> </span><span class="m">1</span><span class="w"> </span>new_genes_abc_v9_more_info.list<span class="o">)</span><span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;BEGIN{OFS=&quot;\t&quot;} {if(NR==FNR){label[$1]=$2; next}; split($2,liste,&quot;,&quot;); labs=&quot;&quot;; for(i in liste){e=liste[i]; labs=labs&quot;&quot;label[e]}; print $1, $2, labs, $3, $4}&#39;</span><span class="w"> </span>enhancers.new_genes.intersection_label.list<span class="w"> </span>&lt;<span class="o">(</span>tail<span class="w"> </span>-n+2<span class="w"> </span>new_genes_abc_v9_more_info.list<span class="o">)</span><span class="w"> </span><span class="p">|</span>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;BEGIN{OFS=&quot;\t&quot;} {split($3,liste,&quot;&quot;); max=0; for(i in liste){e=liste[i]; if(e&gt;max){max=e}}; print $1, $2, max, $4, $5}&#39;</span><span class="w"> </span><span class="p">|</span>head
</code></pre></div>
<div class="highlight"><pre><span></span><code><span class="o">(</span>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;BEGIN{OFS=&quot;\t&quot;} {print $1, &quot;gene.id&quot;, $2, &quot;enhancer.intersect&quot;, $3, $4}&#39;</span><span class="w"> </span>&lt;<span class="o">(</span>head<span class="w"> </span>-n<span class="w"> </span><span class="m">1</span><span class="w"> </span>new_genes_abc_v9_more_info.list<span class="o">)</span><span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;BEGIN{OFS=&quot;\t&quot;} {if(NR==FNR){label[$1]=$2; next}; split($2,liste,&quot;,&quot;); labs=&quot;&quot;; for(i in liste){e=liste[i]; labs=labs&quot;&quot;label[e]}; print $1, $2, labs, $3, $4}&#39;</span><span class="w"> </span>enhancers.new_genes.intersection_label.list<span class="w"> </span>&lt;<span class="o">(</span>tail<span class="w"> </span>-n+2<span class="w"> </span>new_genes_abc_v9_more_info.list<span class="o">)</span><span class="w"> </span><span class="p">|</span>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;BEGIN{OFS=&quot;\t&quot;} {split($3,liste,&quot;&quot;); max=0; for(i in liste){e=liste[i]; if(e&gt;max){max=e}}; print $1, &quot;.&quot;, $2, max, $4, $5}&#39;</span><span class="o">)</span><span class="w"> </span>&gt;<span class="w"> </span>abc_eg.details.list
</code></pre></div>
<ul>
<li>Now, I should have an output file looking as follows</li>
</ul>
<div class="highlight"><pre><span></span><code>&lt;gene symbol&gt; &lt;list_of_enhancers_ids&gt; &lt;max_validity_score_of_enhaner&gt; &lt;label_product_ABC_scores&gt;  &lt;nb_of_enhancers&gt;
</code></pre></div>
<blockquote>
<p>So at the end of the day, how many Genes inferred are associated to at least one enhancer overlapping a ccRE-ELS or a ccRE?</p>
<div class="highlight"><pre><span></span><code>tail<span class="w"> </span>-n+2<span class="w"> </span>abc_eg.details.list<span class="w"> </span><span class="p">|</span>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;{nb[$4]++} END{for(u in nb){print u, nb[u]}}&#39;</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code><span class="m">1</span><span class="w"> </span><span class="m">138</span><span class="w"> </span><span class="c1"># at least 1 enhancer overlapping a ccRE</span>
<span class="m">2</span><span class="w"> </span><span class="m">306</span><span class="w"> </span><span class="c1"># at least 1 enhancer overlapping a ccRE-ELS</span>
</code></pre></div>
</blockquote>
<p>It appears that all the 444 inferred genes were obtain through at least 1 enhancer overlapping a ccRE.</p>
<h3 id="compute-list-of-chic-inferred-genes">Compute list of CHiC-inferred genes<a class="headerlink" href="#compute-list-of-chic-inferred-genes" title="Permanent link">&para;</a></h3>
<p>In order to do so, we started with a dedicated R markdown. Details here: http://genoweb.toulouse.inra.fr/~thoellinger/2022/CHiC_ingerred_genes/preliminary_analysis_chic_v2.html</p>
<p>It produces an output file looking as follows:</p>
<div class="highlight"><pre><span></span><code>&lt;gene_symbol&gt;   &lt;gene_id&gt;   &lt;list_of_enhancers_ids&gt; &lt;label_product_CHiC_scores&gt; &lt;nb_of_enhancers&gt;
</code></pre></div>
<p>Now, we want to add relevant information to this list of genes (whether the associated enhancers intersect ccRE, etc). The steps are given below.</p>
<blockquote>
<p>Well, it was not really necessary to check whether the associated enhancers intersect ccRE or ccRE-ELS, because they all do: we restricted ourselves to CHiC data for which they do, since the beginning.</p>
</blockquote>
<ul>
<li>We add a column ($4) to our output files, which contain the list of scores of all enhancers id in column $2. Then, we replace this column with the max in the list.</li>
</ul>
<blockquote>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;BEGIN{OFS=&quot;\t&quot;} {split($3,liste,&quot;,&quot;); for(i in liste){e=liste[i]; enhancers[e]++}} END{for(u in enhancers){print u, enhancers[u]}}&#39;</span><span class="w"> </span>&lt;<span class="o">(</span>tail<span class="w"> </span>-n+2<span class="w"> </span>new_genes_chic_v2_more_info.list<span class="o">)</span><span class="w"> </span><span class="p">|</span>wc<span class="w"> </span>-l
</code></pre></div>
<p>There are only 20 distinct enhancers involved.</p>
<p>They must overlap with ccRE as with already restricted ourselves to data for which they do, but still, let us verify that they overlap ccRE-ELS:</p>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;BEGIN{OFS=&quot;\t&quot;} {if(NR==FNR){enhancers[$1&quot;:&quot;$2&quot;-&quot;$3]++; next}; if(enhancers[$1]){print $0}}&#39;</span><span class="w"> </span>../data/list_all_enhancers.overlapping_ccRE-ELS.bed<span class="w"> </span>&lt;<span class="o">(</span>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;BEGIN{OFS=&quot;\t&quot;} {split($3,liste,&quot;,&quot;); for(i in liste){e=liste[i]; enhancers[e]++}} END{for(u in enhancers){print u, enhancers[u]}}&#39;</span><span class="w"> </span>&lt;<span class="o">(</span>tail<span class="w"> </span>-n+2<span class="w"> </span>new_genes_chic_v2_more_info.list<span class="o">))</span><span class="w"> </span><span class="p">|</span>wc<span class="w"> </span>-l
</code></pre></div>
<p>20</p>
<p>They do.</p>
</blockquote>
<p>All enhancers involved overlap &gt;= 1 ccRE-ELS, so we add a score 2 everywhere, in the appropriate column.</p>
<div class="highlight"><pre><span></span><code><span class="o">(</span>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;BEGIN{OFS=&quot;\t&quot;} {print $1, $2, $3, &quot;enhancer.intersect&quot;, $4, $5}&#39;</span><span class="w"> </span>&lt;<span class="o">(</span>head<span class="w"> </span>-n<span class="w"> </span><span class="m">1</span><span class="w"> </span>new_genes_chic_v2_more_info.list<span class="o">)</span><span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;BEGIN{OFS=&quot;\t&quot;} {print $1, $2, $3, 2, $4, $5}&#39;</span><span class="w"> </span>&lt;<span class="o">(</span>tail<span class="w"> </span>-n+2<span class="w"> </span>new_genes_chic_v2_more_info.list<span class="o">))</span><span class="w"> </span>&gt;<span class="w"> </span>chic_eg.details.list
</code></pre></div>
<ul>
<li>Now, I should have an output file looking as follows</li>
</ul>
<div class="highlight"><pre><span></span><code>&lt;gene symbol&gt; &lt;gene_id&gt;   &lt;list_of_enhancers_ids&gt; &lt;max_validity_score_of_enhaner = 2&gt; &lt;label_product_CHiC_scores&gt; &lt;nb_of_enhancers&gt;
</code></pre></div>
<h3 id="compute-a-table-giving-for-each-gene-symbol-its-corresponding-ensembl-id-whenever-known">Compute a table giving, for each gene symbol, its corresponding ENSEMBL id, whenever known<a class="headerlink" href="#compute-a-table-giving-for-each-gene-symbol-its-corresponding-ensembl-id-whenever-known" title="Permanent link">&para;</a></h3>
<p>TODO.</p>
<h3 id="combine-everything">Combine everything<a class="headerlink" href="#combine-everything" title="Permanent link">&para;</a></h3>
<p>First, transform the file <code>abc_eg.details.list</code>, ie the one looking like:</p>
<blockquote>
<div class="highlight"><pre><span></span><code>&lt;gene symbol&gt; .   &lt;list_of_enhancers_ids&gt; &lt;label_product_ABC_scores&gt;  &lt;max_validity_score_of_enhaner&gt; &lt;nb_of_enhancers&gt;
</code></pre></div>
</blockquote>
<p>so that it looks like:</p>
<blockquote>
<div class="highlight"><pre><span></span><code>&lt;gene symbol&gt; .   &lt;list_of_enhancers_ids&gt; .   &lt;max_validity_score_of_ABC_enhaner&gt; &lt;label_product_ABC_scores&gt;  0   0   &lt;nb_of_enhancers&gt;   ABC
</code></pre></div>
</blockquote>
<p>So, go to <code>data_and_results/haemochromatosis/networks_hemochromatosis/results/</code> and do:</p>
<div class="highlight"><pre><span></span><code><span class="o">(</span><span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;gene\tgene.id\tsources\tabc.enhancer.intersect.label\tABC.product.label\tchic.enhancer.intersect.label\tcontact.product.label\tcount&quot;</span><span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;BEGIN{OFS=&quot;\t&quot;} {print $1, $2, $3, $4, $5, 0, 0, $6, &quot;ABC&quot;}&#39;</span><span class="w"> </span>&lt;<span class="o">(</span>tail<span class="w"> </span>-n+2<span class="w"> </span>abc_eg.details.list<span class="o">))</span><span class="w"> </span>&gt;<span class="w"> </span>abc_part.list
</code></pre></div>
<p>Then, transform the file looking like:</p>
<blockquote>
<div class="highlight"><pre><span></span><code>&lt;gene symbol&gt; &lt;list_of_enhancers_ids&gt; &lt;label_product_CHiC_scores&gt; &lt;max_validity_score_of_enhaner&gt; &lt;nb_of_enhancers&gt;
</code></pre></div>
</blockquote>
<p>so that it looks like:</p>
<blockquote>
<div class="highlight"><pre><span></span><code>&lt;gene symbol&gt; .   &lt;list_of_enhancers_ids&gt; .   0   0   &lt;max_validity_score_of_ChiC_enhancer&gt;   &lt;label_product_CHiC_scores&gt; &lt;nb_of_enhancers&gt;   CHiC
</code></pre></div>
</blockquote>
<p>So, go to <code>/work2/project/regenet/workspace/thoellinger/shared/2022/promoter_capture_hic/results</code> and do:</p>
<div class="highlight"><pre><span></span><code><span class="o">(</span><span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;gene\tgene.id\tsources\tabc.enhancer.intersect.label\tABC.product.label\tchic.enhancer.intersect.label\tcontact.product.label\tcount&quot;</span><span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;BEGIN{OFS=&quot;\t&quot;} {print $1, $2, $3, 0, 0, $4, $5, $6, &quot;CHiC&quot;}&#39;</span><span class="w"> </span>&lt;<span class="o">(</span>tail<span class="w"> </span>-n+2<span class="w"> </span>chic_eg.details.list<span class="o">))</span><span class="w"> </span>&gt;<span class="w"> </span>chic_part.list
</code></pre></div>
<p>Then, concatenate the two files obtained above.</p>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span><span class="w"> </span>/work2/project/regenet/workspace/thoellinger/shared/2022
</code></pre></div>
<div class="highlight"><pre><span></span><code><span class="o">(</span>cat<span class="w"> </span>networks_hemochromatosis/results/abc_part.list<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>tail<span class="w"> </span>-n+2<span class="w"> </span>promoter_capture_hic/results/chic_part.list<span class="o">)</span><span class="w"> </span><span class="p">|</span>wc<span class="w"> </span>-l
</code></pre></div>
<p>Now, we have to merge rows for which the gene symbol (column $1) is the same. There should not be more than 2 occurrences of the same gene symbol because we started with only 2 lists, of unique gene symbols. So here are the steps (the code is given afterwards):</p>
<p>In awk, parse the file a first time and maintain, for each new gene, a table containing all column values. Whenever the current row corresponds to a gene already encountered in the above rows, do the following (note that the file is not sorted at the moment, so there should be first all ABC-inferred genes, then all CHiC-inferred genes, so whenever there is a gene already encountered, it means that the current row corresponds to a CHiC prediction):</p>
<ul>
<li>let us denote G the gene, and <code>info[G]</code> the information saved for this gene when previously encountered (which necessarily correspond to a row of ABC inference). Let $n denote the n-th column of the current row - that is, the information corresponding to the CHiC inference. We have to do the following:</li>
</ul>
<div class="highlight"><pre><span></span><code><span class="kr">split</span><span class="p">(</span><span class="nx">info</span><span class="p">[</span><span class="nx">G</span><span class="p">],</span><span class="nx">parts</span><span class="p">,</span><span class="s2">&quot;\t&quot;</span><span class="p">)</span> <span class="c1"># get the fields corresponding to the previously encountered occurence of the current gene</span>
<span class="nx">parts</span><span class="p">[</span><span class="mi">2</span><span class="p">]</span> <span class="o">=</span> <span class="o">$</span><span class="mi">2</span> <span class="c1"># replace the gene id by the current one (this won&#39;t change anything at the end, but we do so because the gene id is already known in CHiC data whereas it is not in ABC data ; but at the end of the day we will add all missing ids so it really changes nothing)</span>
<span class="nx">parts</span><span class="p">[</span><span class="mi">3</span><span class="p">]</span> <span class="o">=</span> <span class="nx">parts</span><span class="p">[</span><span class="mi">3</span><span class="p">]</span><span class="s2">&quot;,&quot;</span><span class="o">$</span><span class="mi">3</span> <span class="c1"># add the current enhancers to the list</span>
<span class="nx">parts</span><span class="p">[</span><span class="mi">6</span><span class="p">]</span> <span class="o">=</span> <span class="o">$</span><span class="mi">6</span> <span class="c1"># add the CHiC enhancer label (which will always be 2 here because all CHiC enhancers we kept do intersect at least one ccRE-ELS) </span>
<span class="nx">parts</span><span class="p">[</span><span class="mi">7</span><span class="p">]</span> <span class="o">=</span> <span class="o">$</span><span class="mi">7</span> <span class="c1"># add the CHiC contact product label (see definition in &quot;Data format&quot;)</span>
<span class="nx">parts</span><span class="p">[</span><span class="mi">8</span><span class="p">]</span> <span class="o">=</span> <span class="nx">parts</span><span class="p">[</span><span class="mi">8</span><span class="p">]</span><span class="o">+$</span><span class="mi">8</span> <span class="c1"># sum the number of enhancers (ABC + CHiC)</span>
<span class="nx">parts</span><span class="p">[</span><span class="mi">9</span><span class="p">]</span> <span class="o">=</span> <span class="s2">&quot;ABC,CHiC&quot;</span> <span class="c1"># if the gene is already encountered it means that it comes from both ABC and CHiC data</span>
</code></pre></div>
<ul>
<li>
<p>when all the file is parsed - that is, at the END in awk, print all rows in <code>info</code></p>
</li>
<li>
<p>parse the resulting file once again, and simply compute, for each row, the sum of columns $5-$9, and put it in $4</p>
</li>
<li>
<p>sort the result (descending) according to values of $4</p>
</li>
</ul>
<p>So here is the code</p>
<blockquote>
<div class="highlight"><pre><span></span><code><span class="o">(</span><span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;gene\tgene.id\tsources\tscore.sum\tabc.enhancer.intersect.label\tABC.product.label\tchic.enhancer.intersect.label\tcontact.product.label\tcount\tdatatype&quot;</span><span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span><span class="o">(</span>tail<span class="w"> </span>-n+2<span class="w"> </span>networks_hemochromatosis/results/abc_part.list<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>tail<span class="w"> </span>-n+2<span class="w"> </span>promoter_capture_hic/results/chic_part.list<span class="o">)</span><span class="w"> </span><span class="p">|</span>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;BEGIN{OFS=&quot;\t&quot;} {G=$1; if(!info[G]){info[G]=$0} else {split(info[G],parts,&quot;\t&quot;); parts[2] = $2; parts[3] = parts[3]&quot;,&quot;$3; parts[6] = $6; parts[7] = $7; parts[8] = parts[8]+$8; parts[9] = &quot;ABC,CHiC&quot;; n = length(parts); new=parts[1]; for(i=2;i&lt;=n;i++){new=new&quot;&quot;parts[i]}; info[G] = new}} END{for(g in info){print info[g]}}&#39;</span><span class="o">)</span><span class="w"> </span><span class="p">|</span>wc<span class="w"> </span>-l
</code></pre></div>
<p>454</p>
<p>Instead of 444+42=486 inferred genes, we have 453 distinct inferred genes, which means we have 33 redundancies (in other words: 78% of the genes inferred with CHiC data, are inferred using ABC data too). Which is good!</p>
</blockquote>
<div class="highlight"><pre><span></span><code><span class="o">(</span><span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;gene\tgene.id\tscore.adj\tscore.tot\tabc.enhancer.intersect.label\tABC.product.label\tchic.enhancer.intersect.label\tcontact.product.label\tcount\tdatatype\tsources&quot;</span><span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span><span class="o">((</span>tail<span class="w"> </span>-n+2<span class="w"> </span>networks_hemochromatosis/results/abc_part.list<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>tail<span class="w"> </span>-n+2<span class="w"> </span>promoter_capture_hic/results/chic_part.list<span class="o">)</span><span class="w"> </span><span class="p">|</span>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;BEGIN{OFS=&quot;\t&quot;} {G=$1; if(!info[G]){info[G]=$0} else {split(info[G],parts,&quot;\t&quot;); parts[2]=$2; parts[3] = parts[3]&quot;,&quot;$3; parts[6] = $6; parts[7] = $7; parts[8] = parts[8]+$8; parts[9] = &quot;ABC,CHiC&quot;; n = length(parts); new=parts[1]; for(i=2;i&lt;=n;i++){new=new&quot;\t&quot;parts[i]}; info[G] = new}} END{for(g in info){print info[g]}}&#39;</span><span class="w"> </span><span class="p">|</span>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;BEGIN{OFS=&quot;\t&quot;} {print $1, $2, $4+$5+$6+$7+int($8/4), $4+$5+$6+$7+$8, $4, $5, $6, $7, $8, $9, $3}&#39;</span><span class="w"> </span><span class="p">|</span>sort<span class="w"> </span>-nrk3,3<span class="o">))</span><span class="w"> </span>&gt;<span class="w"> </span>merged_inferred_genes_v1.list
</code></pre></div>
<p>Now we already have a good looking file. Lastly, we will add, whenever possible, the ensembl id corresponding to the gene symbol, using the table computed above for that purpose.</p>
<p>Here is the distribution of the scores obtained:</p>
<blockquote>
<div class="highlight"><pre><span></span><code>tail<span class="w"> </span>-n+2<span class="w"> </span>merged_inferred_genes_v1.list<span class="w"> </span><span class="p">|</span>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;{nb_genes[$3]++} END{for(u in nb_genes){print u, nb_genes[u]}}&#39;</span><span class="w"> </span><span class="p">|</span>sort<span class="w"> </span>-rnk1,1
</code></pre></div>
<p>15 1</p>
<p>12 1</p>
<p>11 6</p>
<p>10 9</p>
<p>9 6</p>
<p>8 10</p>
<p>7 11</p>
<p>6 44</p>
<p>5 60</p>
<p>4 97</p>
<p>3 118</p>
<p>2 90</p>
</blockquote>
<p>Ok. Note that as I thought it would be better to give less importance to the number of enhancers, in the total score, I can divide it by 4 for the total score provided in the column $3 <code>score.adj</code>, whereas I kept it untouched for the total score provided in the column $4 <code>score.tot</code>. This is the only difference between these 2 columns.</p>
<h3 id="add-gene-ids-whenever-possible">Add gene ids whenever possible<a class="headerlink" href="#add-gene-ids-whenever-possible" title="Permanent link">&para;</a></h3>
<p>We use the table <code>/work2/project/regenet/results/multi/abc.model/Nasser2021/gname_gnid_full.tsv</code></p>
<blockquote>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;BEGIN{OFS=&quot;\t&quot;} {if(NR==FNR){id[$1]=$2; next} split($0,parts,&quot;\t&quot;); n = length(parts); if(id[$1]){parts[2] = id[$1]} new=parts[1]; for(i=2;i&lt;=n;i++){new=new&quot;\t&quot;parts[i]} print new}&#39;</span><span class="w"> </span>/work2/project/regenet/results/multi/abc.model/Nasser2021/gname_gnid_full.tsv<span class="w"> </span>merged_inferred_genes_v1.list<span class="w"> </span><span class="p">|</span>head
</code></pre></div>
</blockquote>
<div class="highlight"><pre><span></span><code>mv<span class="w"> </span>merged_inferred_genes_v1.list<span class="w"> </span>backup.merged_inferred_genes_v1.list
awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;BEGIN{OFS=&quot;\t&quot;} {if(NR==FNR){id[$1]=$2; next} split($0,parts,&quot;\t&quot;); n = length(parts); if(id[$1]){parts[2] = id[$1]} new=parts[1]; for(i=2;i&lt;=n;i++){new=new&quot;\t&quot;parts[i]} print new}&#39;</span><span class="w"> </span>/work2/project/regenet/results/multi/abc.model/Nasser2021/gname_gnid_full.tsv<span class="w"> </span>backup.merged_inferred_genes_v1.list<span class="w"> </span>&gt;<span class="w"> </span>merged_inferred_genes_v1.list
mv<span class="w"> </span>merged_inferred_genes_v1.list<span class="w"> </span>merged_inferred_genes_v1.csv
</code></pre></div>
<p>OK.</p>
<p>Now we modify the csv by hand, to add ensembl id where missing.</p>
<h3 id="add-source-gene-symbol">Add source gene symbol<a class="headerlink" href="#add-source-gene-symbol" title="Permanent link">&para;</a></h3>
<p>One could also keep this information all along. I had to add it at the end because at first I did not think it would be useful.</p>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span><span class="w"> </span>.../2022/
</code></pre></div>
<blockquote>
<div class="highlight"><pre><span></span><code><span class="o">(</span><span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;gene\tgene.id\tscore.adj\tgene.source\tscore.tot\tabc.enhancer.intersect.label\tABC.product.label\tchic.enhancer.intersect.label\tcontact.product.label\tcount\tdatatype\tsources&quot;</span><span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;BEGIN{OFS=&quot;\t&quot;} {if(NR==FNR){from[$1]=$5; next} source=&quot;.&quot;; if(from[$1]){source=from[$1]}; print $1, $2, $3, source, $4, $5, $6, $7, $8, $9, $10, $11}&#39;</span><span class="w"> </span>networks_hemochromatosis/results/new_genes_abc_v9_more_info.list<span class="w"> </span>&lt;<span class="o">(</span>tail<span class="w"> </span>-n+2<span class="w"> </span>merged_inferred_genes_v1.csv<span class="o">))</span><span class="w"> </span><span class="p">|</span>head
</code></pre></div>
<div class="highlight"><pre><span></span><code><span class="o">(</span><span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;gene\tgene.id\tscore.adj\tgene.source\tscore.tot\tabc.enhancer.intersect.label\tABC.product.label\tchic.enhancer.intersect.label\tcontact.product.label\tcount\tdatatype\tsources&quot;</span><span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;BEGIN{OFS=&quot;\t&quot;} {if(NR==FNR){from[$1]=$6; next} source=&quot;.&quot;; if(from[$1]){if($4==&quot;.&quot; || from[$1]==$4){source=from[$1]} else{source=$4&quot;,&quot;from[$1]}} else {source=$4}; print $1, $2, $3, source, $5, $6, $7, $8, $9, $10, $11, $12}&#39;</span><span class="w"> </span>promoter_capture_hic/results/new_genes_chic_v2_more_info.list<span class="w"> </span>&lt;<span class="o">(</span>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;BEGIN{OFS=&quot;\t&quot;} {if(NR==FNR){from[$1]=$5; next} source=&quot;.&quot;; if(from[$1]){source=from[$1]}; print $1, $2, $3, source, $4, $5, $6, $7, $8, $9, $10, $11}&#39;</span><span class="w"> </span>networks_hemochromatosis/results/new_genes_abc_v9_more_info.list<span class="w"> </span>&lt;<span class="o">(</span>tail<span class="w"> </span>-n+2<span class="w"> </span>merged_inferred_genes_v1.csv<span class="o">)))</span><span class="w"> </span><span class="p">|</span>head
</code></pre></div>
<div class="highlight"><pre><span></span><code><span class="o">(</span><span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;gene\tgene.id\tscore.adj\tgene.source\tscore.tot\tabc.enhancer.intersect.label\tABC.product.label\tchic.enhancer.intersect.label\tcontact.product.label\tcount\tdatatype\tsources&quot;</span><span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;BEGIN{OFS=&quot;\t&quot;} {if(NR==FNR){from[$1]=$6; next} source=&quot;.&quot;; if(from[$1]){if($4==&quot;.&quot; || from[$1]==$4){source=from[$1]} else{source=$4&quot;,&quot;from[$1]}} else {source=$4}; print $1, $2, $3, source, $5, $6, $7, $8, $9, $10, $11, $12}&#39;</span><span class="w"> </span>promoter_capture_hic/results/new_genes_chic_v2_more_info.list<span class="w"> </span>&lt;<span class="o">(</span>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;BEGIN{OFS=&quot;\t&quot;} {if(NR==FNR){from[$1]=$5; next} source=&quot;.&quot;; if(from[$1]){source=from[$1]}; print $1, $2, $3, source, $4, $5, $6, $7, $8, $9, $10, $11}&#39;</span><span class="w"> </span>networks_hemochromatosis/results/new_genes_abc_v9_more_info.list<span class="w"> </span>&lt;<span class="o">(</span>tail<span class="w"> </span>-n+2<span class="w"> </span>merged_inferred_genes_v1.csv<span class="o">)))</span><span class="w"> </span><span class="p">|</span>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;BEGIN{OFS=&quot;\t&quot;} {source[$4]++} END{for(u in source){print u, source[u]}}&#39;</span><span class="w"> </span><span class="p">|</span>sort<span class="w"> </span>-nrk2,2
</code></pre></div>
</blockquote>
<div class="highlight"><pre><span></span><code>mv<span class="w"> </span>merged_inferred_genes_v1.csv<span class="w"> </span>temp<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span><span class="o">(</span><span class="nb">echo</span><span class="w"> </span>-e<span class="w"> </span><span class="s2">&quot;gene\tgene.id\tscore.adj\tgene.source\tscore.tot\tabc.enhancer.intersect.label\tABC.product.label\tchic.enhancer.intersect.label\tcontact.product.label\tcount\tdatatype\tsources&quot;</span><span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;BEGIN{OFS=&quot;\t&quot;} {if(NR==FNR){from[$1]=$6; next} source=&quot;.&quot;; if(from[$1]){if($4==&quot;.&quot; || from[$1]==$4){source=from[$1]} else{source=$4&quot;,&quot;from[$1]}} else {source=$4}; print $1, $2, $3, source, $5, $6, $7, $8, $9, $10, $11, $12}&#39;</span><span class="w"> </span>promoter_capture_hic/results/new_genes_chic_v2_more_info.list<span class="w"> </span>&lt;<span class="o">(</span>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;BEGIN{OFS=&quot;\t&quot;} {if(NR==FNR){from[$1]=$5; next} source=&quot;.&quot;; if(from[$1]){source=from[$1]}; print $1, $2, $3, source, $4, $5, $6, $7, $8, $9, $10, $11}&#39;</span><span class="w"> </span>networks_hemochromatosis/results/new_genes_abc_v9_more_info.list<span class="w"> </span>&lt;<span class="o">(</span>tail<span class="w"> </span>-n+2<span class="w"> </span>temp<span class="o">)))</span><span class="w"> </span>&gt;<span class="w"> </span>merged_inferred_genes_v1.csv<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>rm<span class="w"> </span>temp
</code></pre></div>





                
              </article>
            </div>
          
          
        </div>
        
          <a href="#" class="md-top md-icon" data-md-component="top" hidden>
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg>
            Back to top
          </a>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      2020 - 2022 Tristan Hoellinger
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
        <div class="md-social">
  
    
    
    
    
      
      
    
    <a href="https://twitter.com/TristanHoellin1" target="_blank" rel="noopener" title="twitter.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"/></svg>
    </a>
  
    
    
    
    
      
      
    
    <a href="https://fr.linkedin.com/in/tristan-hoellinger" target="_blank" rel="noopener" title="fr.linkedin.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"/></svg>
    </a>
  
    
    
    
    
    <a href="mailto:sarah.djebali@inserm.fr" target="_blank" rel="noopener" title="" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M498.1 5.6c10.1 7 15.4 19.1 13.5 31.2l-64 416c-1.5 9.7-7.4 18.2-16 23s-18.9 5.4-28 1.6L284 427.7l-68.5 74.1c-8.9 9.7-22.9 12.9-35.2 8.1S160 493.2 160 480v-83.6c0-4 1.5-7.8 4.2-10.7l167.6-182.9c5.8-6.3 5.6-16-.4-22s-15.7-6.4-22-.7L106 360.8l-88.3-44.2C7.1 311.3.3 300.7 0 288.9s5.9-22.8 16.1-28.7l448-256c10.7-6.1 23.9-5.5 34 1.4z"/></svg>
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    <script id="__config" type="application/json">{"base": "..", "features": ["navigation.instant", "navigation.tabs", "navigation.sections", "navigation.top"], "search": "../assets/javascripts/workers/search.db81ec45.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../assets/javascripts/bundle.6df46069.min.js"></script>
      
    
  </body>
</html>