
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
        <meta name="author" content="Tristan Hoellinger">
      
      
        <link rel="canonical" href="https://eg.github.io/data_and_results/haemochromatosis/gwas.html">
      
      
      
      <link rel="icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.4.2, mkdocs-material-9.0.11">
    
    
      
        <title>Intersect enhancers suspected to regulate the 13 initial genes (involved in hemochromatosis or iron metabolism regulation), with the NHGRI-EBI GWAS Catalog - Genome-wide identification of E/G interactions</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.0d440cfe.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.2505c338.min.css">
      
      

    
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../stylesheets/extra.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="black" data-md-color-accent="">
  
    
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#intersect-enhancers-suspected-to-regulate-the-13-initial-genes-involved-in-hemochromatosis-or-iron-metabolism-regulation-with-the-nhgri-ebi-gwas-catalog" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Genome-wide identification of E/G interactions" class="md-header__button md-logo" aria-label="Genome-wide identification of E/G interactions" data-md-component="logo">
      
  <img src="../../logo/logo-inserm-blc.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Genome-wide identification of E/G interactions
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Intersect enhancers suspected to regulate the 13 initial genes (involved in hemochromatosis or iron metabolism regulation), with the NHGRI-EBI GWAS Catalog
            
          </span>
        </div>
      </div>
    </div>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/hoellin/eg" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
  </div>
  <div class="md-source__repository">
    hoellin/eg
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../guidebooks/introduction.html" class="md-tabs__link">
        Guidebooks
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../haemochromatosis/introduction.html" class="md-tabs__link">
        Haemochromatosis
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../notes_BENGI/introduction.html" class="md-tabs__link">
        BENGI
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../notes_ABC/introduction.html" class="md-tabs__link">
        ABC
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../index.html" class="md-tabs__link">
        About
      </a>
    </li>
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Genome-wide identification of E/G interactions" class="md-nav__button md-logo" aria-label="Genome-wide identification of E/G interactions" data-md-component="logo">
      
  <img src="../../logo/logo-inserm-blc.png" alt="logo">

    </a>
    Genome-wide identification of E/G interactions
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/hoellin/eg" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
  </div>
  <div class="md-source__repository">
    hoellin/eg
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1" >
      
      
      
        <label class="md-nav__link" for="__nav_1" id="__nav_1_label" tabindex="0">
          Guidebooks
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_1_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_1">
          <span class="md-nav__icon md-icon"></span>
          Guidebooks
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../guidebooks/introduction.html" class="md-nav__link">
        Introduction
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../guidebooks/script_bedpe_summary/script_bedpe_summary.html" class="md-nav__link">
        Compute summary stats of a bedpe
      </a>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_3" >
      
      
      
        <label class="md-nav__link" for="__nav_1_3" id="__nav_1_3_label" tabindex="0">
          Basics
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_1_3">
          <span class="md-nav__icon md-icon"></span>
          Basics
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../guidebooks/shell.html" class="md-nav__link">
        Shell
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../guidebooks/conda_environments.html" class="md-nav__link">
        Conda basics
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../guidebooks/git.html" class="md-nav__link">
        Git
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../guidebooks/genotoul.html" class="md-nav__link">
        Genotoul cluster
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../guidebooks/notebooks_ssh.html" class="md-nav__link">
        How to use Jupyter through ssh tunneling?
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_4" >
      
      
      
        <label class="md-nav__link" for="__nav_1_4" id="__nav_1_4_label" tabindex="0">
          Misc
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_4_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_1_4">
          <span class="md-nav__icon md-icon"></span>
          Misc
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../guidebooks/compute_correlations.html" class="md-nav__link">
        Compute correlations
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" >
      
      
      
        <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
          Haemochromatosis
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_2">
          <span class="md-nav__icon md-icon"></span>
          Haemochromatosis
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../haemochromatosis/introduction.html" class="md-nav__link">
        Introduction
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../haemochromatosis/data_format.html" class="md-nav__link">
        Data format
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../haemochromatosis/ABC/custom_go.html" class="md-nav__link">
        Custom script GO
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../haemochromatosis/main_results.html" class="md-nav__link">
        Main results
      </a>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2_5" >
      
      
      
        <label class="md-nav__link" for="__nav_2_5" id="__nav_2_5_label" tabindex="0">
          CHiC data
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_2_5_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_2_5">
          <span class="md-nav__icon md-icon"></span>
          CHiC data
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../haemochromatosis/CHiC/preprocess_data.html" class="md-nav__link">
        Preprocessing
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../haemochromatosis/CHiC/preliminary_analysis_chic_v2.html" class="md-nav__link">
        Networks
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../haemochromatosis/CHiC/go_enrichment.html" class="md-nav__link">
        GO enrichment
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2_6" >
      
      
      
        <label class="md-nav__link" for="__nav_2_6" id="__nav_2_6_label" tabindex="0">
          ABC data
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_2_6_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_2_6">
          <span class="md-nav__icon md-icon"></span>
          ABC data
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../haemochromatosis/ABC/data_processing.html" class="md-nav__link">
        Preprocessing
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../haemochromatosis/ABC/preliminary_analysis_v9.html" class="md-nav__link">
        Networks
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../haemochromatosis/ABC/summary_statistics_eg_pairs.html" class="md-nav__link">
        Summary statistics enhancers
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../haemochromatosis/ABC/go_enrichment.html" class="md-nav__link">
        GO enrichment
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../haemochromatosis/ABC/gorilla.html" class="md-nav__link">
        GOrilla
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../haemochromatosis/ABC/compute_avg_nb_of_connections_per_gene.html" class="md-nav__link">
        Avg nb connections per gene
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../haemochromatosis/ABC/gwas.html" class="md-nav__link">
        GWAS (13 initial genes)
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../haemochromatosis/ABC/gwas_10.html" class="md-nav__link">
        GWAS (10 initial genes)
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" >
      
      
      
        <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
          BENGI
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3">
          <span class="md-nav__icon md-icon"></span>
          BENGI
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/introduction.html" class="md-nav__link">
        Introduction
      </a>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_2" >
      
      
      
        <label class="md-nav__link" for="__nav_3_2" id="__nav_3_2_label" tabindex="0">
          Notebooks
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_2_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_2">
          <span class="md-nav__icon md-icon"></span>
          Notebooks
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/BENGI_to_custom_bedpe.html" class="md-nav__link">
        Convert BENGI datasets to custom bedpe
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_3" >
      
      
      
        <label class="md-nav__link" for="__nav_3_3" id="__nav_3_3_label" tabindex="0">
          Summary statistics
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_3">
          <span class="md-nav__icon md-icon"></span>
          Summary statistics
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/summary_statistics/summary_statistics.html" class="md-nav__link">
        Summary statistics on BENGI GM12878 benchmarks
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/summary_statistics_BENGI/summary_BENGI_bedpe.html" class="md-nav__link">
        (R code) Compute summary statistics over BENGI
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_4" >
      
      
      
        <label class="md-nav__link" for="__nav_3_4" id="__nav_3_4_label" tabindex="0">
          Evaluation of a few methods over BENGI GM12878
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_4_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_4">
          <span class="md-nav__icon md-icon"></span>
          Evaluation of a few methods over BENGI GM12878
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_4_1" >
      
      
      
        <label class="md-nav__link" for="__nav_3_4_1" id="__nav_3_4_1_label" tabindex="0">
          Distance method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_4_1_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_4_1">
          <span class="md-nav__icon md-icon"></span>
          Distance method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/distance_method/distance_evaluation_with_code.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/distance_method/summary_distance_evaluation.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_4_2" >
      
      
      
        <label class="md-nav__link" for="__nav_3_4_2" id="__nav_3_4_2_label" tabindex="0">
          DNase-expression correlation method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_4_2_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_4_2">
          <span class="md-nav__icon md-icon"></span>
          DNase-expression correlation method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/dnase_expression_correlation/correlation_method_with_code.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/dnase_expression_correlation/summary_correlation_method.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_4_3" >
      
      
      
        <label class="md-nav__link" for="__nav_3_4_3" id="__nav_3_4_3_label" tabindex="0">
          Average rank method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_4_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_4_3">
          <span class="md-nav__icon md-icon"></span>
          Average rank method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/avg_rank_method/avg_rank_method_with_code.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/avg_rank_method/summary_avg_rank_method.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_5" >
      
      
      
        <label class="md-nav__link" for="__nav_3_5" id="__nav_3_5_label" tabindex="0">
          Evaluation of a few methods over Fulco et al 2019 CRISPRi-FlowFISH
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_5_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_5">
          <span class="md-nav__icon md-icon"></span>
          Evaluation of a few methods over Fulco et al 2019 CRISPRi-FlowFISH
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_5_1" >
      
      
      
        <label class="md-nav__link" for="__nav_3_5_1" id="__nav_3_5_1_label" tabindex="0">
          Distance method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_5_1_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_5_1">
          <span class="md-nav__icon md-icon"></span>
          Distance method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/CRISPRi_FlowFISH/distance_method/distance_over_fulco_et_al_crispri.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/CRISPRi_FlowFISH/distance_method/summary_distance_over_fulco_et_al_crispri.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_5_2" >
      
      
      
        <label class="md-nav__link" for="__nav_3_5_2" id="__nav_3_5_2_label" tabindex="0">
          DNase-expression correlation method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_5_2_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_5_2">
          <span class="md-nav__icon md-icon"></span>
          DNase-expression correlation method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/CRISPRi_FlowFISH/dnase_expression_correlation/correlation_method_with_code.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/CRISPRi_FlowFISH/dnase_expression_correlation/summary_correlation_method.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_5_3" >
      
      
      
        <label class="md-nav__link" for="__nav_3_5_3" id="__nav_3_5_3_label" tabindex="0">
          Average rank method
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_3_5_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_3_5_3">
          <span class="md-nav__icon md-icon"></span>
          Average rank method
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/CRISPRi_FlowFISH/avg_rank_method/avg_rank_method_with_code.html" class="md-nav__link">
        Full code
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_BENGI/CRISPRi_FlowFISH/avg_rank_method/summary_avg_rank.html" class="md-nav__link">
        Results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" >
      
      
      
        <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
          ABC
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_4">
          <span class="md-nav__icon md-icon"></span>
          ABC
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/introduction.html" class="md-nav__link">
        Introduction
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/BENGI/ABC_BENGI_results.html" class="md-nav__link">
        ABC over BENGI for GM12878, main results
      </a>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_3" >
      
      
      
        <label class="md-nav__link" for="__nav_4_3" id="__nav_4_3_label" tabindex="0">
          CRISPRi-FlowFISH
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_3_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_4_3">
          <span class="md-nav__icon md-icon"></span>
          CRISPRi-FlowFISH
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/K562/ABC_K562_CRISPRi_FlowFISH.html" class="md-nav__link">
        Original ABC predictions over K562
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/K562/recap_investigations_ABC_over_CRiFF.html" class="md-nav__link">
        Summary ABC over CRiFF
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_4" >
      
      
      
        <label class="md-nav__link" for="__nav_4_4" id="__nav_4_4_label" tabindex="0">
          Notebooks
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_4_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_4_4">
          <span class="md-nav__icon md-icon"></span>
          Notebooks
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/K562/april_K562_56_genes/april_K562_56_genes.html" class="md-nav__link">
        ABC over K562 cCRiFF
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/K562/april_K562_59_genes/april_K562_59_genes.html" class="md-nav__link">
        ABC over K562 CRiFF
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/K562/april_K562_candidates_not_in_whitelist_56_genes/april_K562_candidates_NOT_in_whitelist.html" class="md-nav__link">
        ABC over K562 cCRiFF, no whitelist
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/generic_notebooks/turnkey_notebook_to_run_ABC_with_example_over_GM12878.html" class="md-nav__link">
        Generic notebook to run ABC method & Example over GM12878
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/BENGI/notebook_ABC_over_BENGI_GM12878.html" class="md-nav__link">
        ABC over GM12878 with same accessions as for BENGI
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/BENGI/notebook_ABC_over_BENGI_GM12878_from_ccRE_ELSs.html" class="md-nav__link">
        ABC over GM12878 with same accessions as for BENGI, starting with ccRE-dELS
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4_5" >
      
      
      
        <label class="md-nav__link" for="__nav_4_5" id="__nav_4_5_label" tabindex="0">
          Full codes
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_4_5_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_4_5">
          <span class="md-nav__icon md-icon"></span>
          Full codes
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/test_abc_pipeline.html" class="md-nav__link">
        Test of ABC pipeline over chr22
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/BENGI/ABC_over_BENGI_GM12878.html" class="md-nav__link">
        ABC over GM12878 with same accessions as for BENGI
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../notes_ABC/differences_example_chr22.html" class="md-nav__link">
        Meld of results over chr22 vs authors' results
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" >
      
      
      
        <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
          About
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_5">
          <span class="md-nav__icon md-icon"></span>
          About
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../index.html" class="md-nav__link">
        Getting started
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../about_me.html" class="md-nav__link">
        About me
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#data-acquisition" class="md-nav__link">
    Data acquisition
  </a>
  
    <nav class="md-nav" aria-label="Data acquisition">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#download-the-latest-gwas-catalog" class="md-nav__link">
    Download the latest GWAS catalog
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#list-of-enhancers-suspected-to-regulate-genes-involved-in-hemochromatosis" class="md-nav__link">
    List of enhancers suspected to regulate genes involved in hemochromatosis
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#list-of-enhancers-suspected-to-regulate-the-457-originalnew-genes" class="md-nav__link">
    List of enhancers suspected to regulate the 457 original+new genes
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#data-pre-processing" class="md-nav__link">
    Data pre-processing
  </a>
  
    <nav class="md-nav" aria-label="Data pre-processing">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#enhancer-lists" class="md-nav__link">
    Enhancer lists
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#snp-related-to-hemochromatosis" class="md-nav__link">
    SNP related to hemochromatosis
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#snp-related-to-iron-in-general" class="md-nav__link">
    SNP related to iron in general
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#perform-intersection-will-enhancers-regulating-the-10-initial-genes" class="md-nav__link">
    Perform intersection (will enhancers regulating the 10 initial genes)
  </a>
  
    <nav class="md-nav" aria-label="Perform intersection (will enhancers regulating the 10 initial genes)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tsv-to-bed" class="md-nav__link">
    TSV to bed
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#liftover" class="md-nav__link">
    Liftover
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#intersect" class="md-nav__link">
    Intersect
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#perform-intersection-with-enhancers-regulating-the-457-original-newly-found-genes" class="md-nav__link">
    Perform intersection (with enhancers regulating the 457 original + newly found genes)
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#are-the-genes-regulated-by-the-enhancers-found-to-intersect-a-gwas-snp-the-same-as-those-given-in-gwas-as-associated-to-the-snp" class="md-nav__link">
    Are the genes regulated by the enhancers found to intersect a GWAS SNP, the same as those given in GWAS as associated to the SNP?
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  

  
  



<h1 id="intersect-enhancers-suspected-to-regulate-the-13-initial-genes-involved-in-hemochromatosis-or-iron-metabolism-regulation-with-the-nhgri-ebi-gwas-catalog">Intersect enhancers suspected to regulate the 13 initial genes (involved in hemochromatosis or iron metabolism regulation), with the NHGRI-EBI GWAS Catalog<a class="headerlink" href="#intersect-enhancers-suspected-to-regulate-the-13-initial-genes-involved-in-hemochromatosis-or-iron-metabolism-regulation-with-the-nhgri-ebi-gwas-catalog" title="Permanent link">&para;</a></h1>
<h2 id="data-acquisition">Data acquisition<a class="headerlink" href="#data-acquisition" title="Permanent link">&para;</a></h2>
<h3 id="download-the-latest-gwas-catalog">Download the latest GWAS catalog<a class="headerlink" href="#download-the-latest-gwas-catalog" title="Permanent link">&para;</a></h3>
<p>The version information on the GWAS website were the following when I downloaded the catalog:</p>
<blockquote>
<p>As of 2021-10-22, the GWAS Catalog contains 5419 publications and 316782 associations. GWAS Catalog data is currently mapped to Genome Assembly GRCh38.p13 and dbSNP Build 154.</p>
</blockquote>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span><span class="w"> </span>/work2/project/regenet/results/gwas/gwas.catalog
</code></pre></div>
<div class="highlight"><pre><span></span><code>wget<span class="w"> </span>https://www.ebi.ac.uk/gwas/api/search/downloads/alternative<span class="w"> </span>-O<span class="w"> </span>/work2/project/regenet/results/gwas/gwas.catalog/v1.0.2_r2021-10-02/gwas_catalog_v1.0.2-associations_r2021-10-02.tsv
</code></pre></div>
<h3 id="list-of-enhancers-suspected-to-regulate-genes-involved-in-hemochromatosis">List of enhancers suspected to regulate genes involved in hemochromatosis<a class="headerlink" href="#list-of-enhancers-suspected-to-regulate-genes-involved-in-hemochromatosis" title="Permanent link">&para;</a></h3>
<p>We obtained it while computing networks with the R markdown.</p>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span><span class="w"> </span>/work2/project/regenet/results/multi/abc.model/Nasser2021/
mv<span class="w"> </span>GWAS/<span class="w"> </span>GWAS_10_initial_genes_fall_2021/
mkdir<span class="w"> </span>GWAS_13_initial_genes/<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span><span class="nb">cd</span><span class="w"> </span>GWAS_13_initial_genes/
</code></pre></div>
<div class="highlight"><pre><span></span><code>cp<span class="w"> </span>/work2/project/regenet/workspace/thoellinger/shared/2022/networks_hemochromatosis/results/enhancers.list<span class="w"> </span>.
</code></pre></div>
<h3 id="list-of-enhancers-suspected-to-regulate-the-457-originalnew-genes">List of enhancers suspected to regulate the 457 original+new genes<a class="headerlink" href="#list-of-enhancers-suspected-to-regulate-the-457-originalnew-genes" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span><span class="w"> </span>/work2/project/regenet/results/multi/abc.model/Nasser2021/GWAS_13_initial_genes/
cp<span class="w"> </span>/work2/project/regenet/workspace/thoellinger/shared/2022/networks_hemochromatosis/results/enhancers.new_genes.list<span class="w"> </span>.
</code></pre></div>
<h2 id="data-pre-processing">Data pre-processing<a class="headerlink" href="#data-pre-processing" title="Permanent link">&para;</a></h2>
<h3 id="enhancer-lists">Enhancer lists<a class="headerlink" href="#enhancer-lists" title="Permanent link">&para;</a></h3>
<p>We want to convert them in the bed format so we do the following:</p>
<div class="highlight"><pre><span></span><code>conda<span class="w"> </span>activate<span class="w"> </span>base<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>module<span class="w"> </span>load<span class="w"> </span>bioinfo/bedtools-2.27.1
</code></pre></div>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span><span class="s1">&#39;BEGIN{FS=&quot;\t&quot;; OFS=&quot;\t&quot;}; {split($1,l1,&quot;:&quot;); split(l1[2],l2,&quot;-&quot;); print l1[1], l2[1], l2[2]}&#39;</span><span class="w"> </span>enhancers.list<span class="w"> </span><span class="p">|</span>bedtools<span class="w"> </span>sort<span class="w"> </span>-faidx<span class="w"> </span>../chr_sizes<span class="w"> </span>-i<span class="w"> </span>stdin<span class="w"> </span>&gt;<span class="w"> </span>enhancers.sorted.bed
</code></pre></div>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span><span class="s1">&#39;BEGIN{FS=&quot;\t&quot;; OFS=&quot;\t&quot;}; {split($1,l1,&quot;:&quot;); split(l1[2],l2,&quot;-&quot;); print l1[1], l2[1], l2[2]}&#39;</span><span class="w"> </span>enhancers.new_genes.list<span class="w"> </span><span class="p">|</span>bedtools<span class="w"> </span>sort<span class="w"> </span>-faidx<span class="w"> </span>../chr_sizes<span class="w"> </span>-i<span class="w"> </span>stdin<span class="w"> </span>&gt;<span class="w"> </span>enhancers.new_genes.sorted.bed
</code></pre></div>
<h3 id="snp-related-to-hemochromatosis">SNP related to hemochromatosis<a class="headerlink" href="#snp-related-to-hemochromatosis" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span><span class="w"> </span>/work2/project/regenet/results/gwas/gwas.catalog/v1.0.2_r2021-10-02/
</code></pre></div>
<p>We want to extract from the GWAS catalog only the SNP related to hemochromatosis. Let us first investigate the list of all diseases / traits in the GWAS catalog.</p>
<blockquote>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span><span class="s1">&#39;BEGIN{FS=&quot;\t&quot;; OFS=&quot;\t&quot;} {traits[$8]++} END{for(u in traits){print u, traits[u]}}&#39;</span><span class="w"> </span>gwas_catalog_v1.0.2-associations_r2021-10-02.tsv<span class="w"> </span><span class="p">|</span>wc<span class="w"> </span>-l
</code></pre></div>
<p>5845</p>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span><span class="s1">&#39;BEGIN{FS=&quot;\t&quot;; OFS=&quot;\t&quot;} {traits[$8]++} END{for(u in traits){print u, traits[u]}}&#39;</span><span class="w"> </span>gwas_catalog_v1.0.2-associations_r2021-10-02.tsv<span class="w"> </span><span class="p">|</span>sort<span class="w"> </span>-nrk2,2<span class="w"> </span><span class="p">|</span>head<span class="w"> </span>-n<span class="w"> </span><span class="m">30</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>Height<span class="w">    </span><span class="m">6144</span>
Schizophrenia<span class="w"> </span><span class="m">2161</span>
Hematocrit<span class="w">    </span><span class="m">1913</span>
Triglycerides<span class="w"> </span><span class="m">1487</span>
Neuroticism<span class="w">   </span><span class="m">1171</span>
Asthma<span class="w">    </span><span class="m">1168</span>
Weight<span class="w">    </span><span class="m">1037</span>
Plateletcrit<span class="w">  </span><span class="m">1014</span>
Hemoglobin<span class="w">    </span><span class="m">973</span>
Intelligence<span class="w">  </span><span class="m">581</span>
Chronotype<span class="w">    </span><span class="m">479</span>
Psoriasis<span class="w"> </span><span class="m">422</span>
Insomnia<span class="w">  </span><span class="m">422</span>
Hypertension<span class="w">  </span><span class="m">308</span>
Hypothyroidism<span class="w">    </span><span class="m">291</span>
Depression<span class="w">    </span><span class="m">290</span>
Morningness<span class="w">   </span><span class="m">274</span>
Tonsillectomy<span class="w"> </span><span class="m">241</span>
Eczema<span class="w">    </span><span class="m">214</span>
FEV1<span class="w">  </span><span class="m">206</span>
Neurociticism<span class="w"> </span><span class="m">181</span>
Longevity<span class="w"> </span><span class="m">169</span>
Adventurousness<span class="w">   </span><span class="m">167</span>
Glaucoma<span class="w">  </span><span class="m">162</span>
Snoring<span class="w">   </span><span class="m">158</span>
Cataracts<span class="w"> </span><span class="m">154</span>
Melanoma<span class="w">  </span><span class="m">149</span>
Endometriosis<span class="w"> </span><span class="m">148</span>
Worry<span class="w"> </span><span class="m">147</span>
Migraine<span class="w">  </span><span class="m">146</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;$8~/([Hh]emochromatosis)/&#39;</span><span class="w"> </span>gwas_catalog_v1.0.2-associations_r2021-10-02.tsv<span class="w"> </span><span class="p">|</span>wc<span class="w"> </span>-l
</code></pre></div>
<p>2</p>
</blockquote>
<div class="highlight"><pre><span></span><code>grep<span class="w"> </span>emochromatos<span class="w"> </span>gwas_catalog_v1.0.2-associations_r2021-10-02.tsv<span class="w"> </span><span class="p">|</span>awk<span class="w"> </span><span class="s1">&#39;BEGIN{FS=&quot;\t&quot;; OFS=&quot;\t&quot;} {traits[$8]++} END{for(u in traits){print u, traits[u]}}&#39;</span><span class="w"> </span><span class="p">|</span>sort<span class="w"> </span>-t<span class="w"> </span><span class="s1">$&#39;\t&#39;</span><span class="w"> </span>-nrk2,2
</code></pre></div>
<blockquote>
<div class="highlight"><pre><span></span><code>Iron<span class="w"> </span>status<span class="w"> </span>biomarkers<span class="w"> </span><span class="o">(</span>transferrin<span class="w"> </span>levels<span class="o">)</span><span class="w">   </span><span class="m">9</span>
Iron<span class="w"> </span>status<span class="w"> </span>biomarkers<span class="w"> </span><span class="o">(</span>transferrin<span class="w"> </span>saturation<span class="o">)</span><span class="w">   </span><span class="m">6</span>
Iron<span class="w"> </span>status<span class="w"> </span>biomarkers<span class="w"> </span><span class="o">(</span>iron<span class="w"> </span>levels<span class="o">)</span><span class="w">  </span><span class="m">5</span>
Iron<span class="w"> </span>status<span class="w"> </span>biomarkers<span class="w"> </span><span class="o">(</span>ferritin<span class="w"> </span>levels<span class="o">)</span><span class="w">  </span><span class="m">5</span>
Hereditary<span class="w"> </span>hemochromatosis-related<span class="w"> </span>traits<span class="w"> </span><span class="o">(</span>HFE<span class="w"> </span>mutation<span class="w"> </span>homozygotes<span class="o">)</span><span class="w">  </span><span class="m">2</span>
</code></pre></div>
</blockquote>
<p>Only 2 entries involve hemochromatosis directly, and a few others (25) involve studies about hemochromatosis but not SNP directly related to hemochromatosis. We check that no entries correspond to the other name of hemochromatosis, namely iron overload:</p>
<blockquote>
<div class="highlight"><pre><span></span><code>grep<span class="w"> </span><span class="s1">&#39;iron.*overload&#39;</span><span class="w"> </span>gwas_catalog_v1.0.2-associations_r2021-10-02.tsv<span class="w"> </span><span class="p">|</span>wc<span class="w"> </span>-l
</code></pre></div>
<p>0</p>
</blockquote>
<p>We conclude that there are too few data to focus on those involving hemochromatosis. We will rather focus on those for which the trait is <code>Iron status biomarkers</code></p>
<h3 id="snp-related-to-iron-in-general">SNP related to iron in general<a class="headerlink" href="#snp-related-to-iron-in-general" title="Permanent link">&para;</a></h3>
<blockquote>
<div class="highlight"><pre><span></span><code>grep<span class="w"> </span><span class="s1">&#39;[iI]ron.*biomarkers&#39;</span><span class="w"> </span>gwas_catalog_v1.0.2-associations_r2021-10-02.tsv<span class="w"> </span><span class="p">|</span>wc<span class="w"> </span>-l
</code></pre></div>
<p>250</p>
</blockquote>
<p>Well we use the following filter to be sure not to miss any trait involving iron, then we will filter out by hand those which do not actually correspond to iron (such as "env<strong>iron</strong>ment") or that do not interest us:</p>
<blockquote>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;$8~/([iI]ron)/&#39;</span><span class="w"> </span>gwas_catalog_v1.0.2-associations_r2021-10-02.tsv<span class="w"> </span><span class="p">|</span>wc<span class="w"> </span>-l
</code></pre></div>
<p>378</p>
</blockquote>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;if($8~/([iI]ron)/ {traits[$8]++}; END{for(u in traits){print u, traits[u]}}&#39;</span><span class="w"> </span>gwas_catalog_v1.0.2-associations_r2021-10-02.tsv<span class="w"> </span><span class="p">|</span>sort<span class="w"> </span>-t<span class="w"> </span><span class="s1">$&#39;\t&#39;</span><span class="w"> </span>-nrk2,2
</code></pre></div>
<blockquote>
<div class="highlight"><pre><span></span><code>Iron<span class="w"> </span>status<span class="w"> </span>biomarkers<span class="w"> </span><span class="o">(</span>ferritin<span class="w"> </span>levels<span class="o">)</span><span class="w">  </span><span class="m">55</span>
Iron<span class="w"> </span>status<span class="w"> </span>biomarkers<span class="w">    </span><span class="m">54</span>
Iron<span class="w"> </span>status<span class="w"> </span>biomarkers<span class="w"> </span><span class="o">(</span>total<span class="w"> </span>iron<span class="w"> </span>binding<span class="w"> </span>capacity<span class="o">)</span><span class="w">  </span><span class="m">48</span>
Sensitivity<span class="w"> </span>to<span class="w"> </span>environmental<span class="w"> </span>stress<span class="w"> </span>and<span class="w"> </span>adversity<span class="w"> </span><span class="m">47</span>
Iron<span class="w"> </span>status<span class="w"> </span>biomarkers<span class="w"> </span><span class="o">(</span>transferrin<span class="w"> </span>saturation<span class="o">)</span><span class="w">   </span><span class="m">47</span>
Iron<span class="w"> </span>status<span class="w"> </span>biomarkers<span class="w"> </span><span class="o">(</span>iron<span class="w"> </span>levels<span class="o">)</span><span class="w">  </span><span class="m">37</span>
Liver<span class="w"> </span>iron<span class="w"> </span>content<span class="w">    </span><span class="m">23</span>
Economic<span class="w"> </span>and<span class="w"> </span>political<span class="w"> </span>preferences<span class="w"> </span><span class="o">(</span>environmentalism<span class="o">)</span><span class="w"> </span><span class="m">17</span>
BMI<span class="w"> </span>x<span class="w">  </span>environmental<span class="w"> </span>factors<span class="w"> </span><span class="o">(</span>excluding<span class="w"> </span>physical<span class="w"> </span>activity<span class="o">)</span><span class="w"> </span>interaction<span class="w">    </span><span class="m">11</span>
Iron<span class="w"> </span>status<span class="w"> </span>biomarkers<span class="w"> </span><span class="o">(</span>transferrin<span class="w"> </span>levels<span class="o">)</span><span class="w">   </span><span class="m">9</span>
Forced<span class="w"> </span>expiratory<span class="w"> </span>volume<span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="m">1</span><span class="w"> </span>second<span class="w"> </span><span class="o">(</span>occupational<span class="w"> </span>environmental<span class="w"> </span>exposures<span class="w"> </span>interaction<span class="o">)</span><span class="w">   </span><span class="m">7</span>
BMI<span class="w"> </span>x<span class="w"> </span>environmental<span class="w"> </span>factors<span class="w"> </span><span class="o">(</span>including<span class="w"> </span>physical<span class="w"> </span>activity<span class="o">)</span><span class="w"> </span>interaction<span class="w"> </span><span class="m">7</span>
Iron<span class="w"> </span>deficiency<span class="w"> </span>anemia<span class="w">    </span><span class="m">5</span>
Type<span class="w"> </span><span class="m">2</span><span class="w"> </span>diabetes<span class="w"> </span><span class="o">(</span>dietary<span class="w"> </span>heme<span class="w"> </span>iron<span class="w"> </span>intake<span class="w"> </span>interaction<span class="o">)</span><span class="w">    </span><span class="m">4</span>
Forced<span class="w"> </span>expiratory<span class="w"> </span>volume<span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="m">1</span><span class="w"> </span>second<span class="w"> </span><span class="o">(</span>environmental<span class="w"> </span>tobacco<span class="w"> </span>smoke<span class="w"> </span>interaction<span class="o">)</span><span class="w">    </span><span class="m">3</span>
Pancreas<span class="w"> </span>iron<span class="w"> </span>content<span class="w"> </span><span class="m">1</span>
Iron<span class="w"> </span>levels<span class="w">   </span><span class="m">1</span>
Iron<span class="w"> </span>deficiency<span class="w">   </span><span class="m">1</span>
Cleft<span class="w"> </span>plate<span class="w"> </span><span class="o">(</span>environmental<span class="w"> </span>tobacco<span class="w"> </span>smoke<span class="w"> </span>interaction<span class="o">)</span><span class="w"> </span><span class="m">1</span>
</code></pre></div>
</blockquote>
<p>So the only ones that could interest us are the following:</p>
<blockquote>
<div class="highlight"><pre><span></span><code>Iron<span class="w"> </span>status<span class="w"> </span>biomarkers<span class="w"> </span><span class="o">(</span>ferritin<span class="w"> </span>levels<span class="o">)</span><span class="w">    </span><span class="m">55</span>
Iron<span class="w"> </span>status<span class="w"> </span>biomarkers<span class="w">    </span><span class="m">54</span>
Iron<span class="w"> </span>status<span class="w"> </span>biomarkers<span class="w"> </span><span class="o">(</span>total<span class="w"> </span>iron<span class="w"> </span>binding<span class="w"> </span>capacity<span class="o">)</span><span class="w">    </span><span class="m">48</span>
Iron<span class="w"> </span>status<span class="w"> </span>biomarkers<span class="w"> </span><span class="o">(</span>transferrin<span class="w"> </span>saturation<span class="o">)</span><span class="w">    </span><span class="m">47</span>
Iron<span class="w"> </span>status<span class="w"> </span>biomarkers<span class="w"> </span><span class="o">(</span>iron<span class="w"> </span>levels<span class="o">)</span><span class="w">    </span><span class="m">37</span>
Liver<span class="w"> </span>iron<span class="w"> </span>content<span class="w">    </span><span class="m">23</span>
Iron<span class="w"> </span>status<span class="w"> </span>biomarkers<span class="w"> </span><span class="o">(</span>transferrin<span class="w"> </span>levels<span class="o">)</span><span class="w">    </span><span class="m">9</span>
Iron<span class="w"> </span>deficiency<span class="w"> </span>anemia<span class="w">    </span><span class="m">5</span>
Type<span class="w"> </span><span class="m">2</span><span class="w"> </span>diabetes<span class="w"> </span><span class="o">(</span>dietary<span class="w"> </span>heme<span class="w"> </span>iron<span class="w"> </span>intake<span class="w"> </span>interaction<span class="o">)</span><span class="w">    </span><span class="m">4</span>
Pancreas<span class="w"> </span>iron<span class="w"> </span>content<span class="w">    </span><span class="m">1</span>
Iron<span class="w"> </span>levels<span class="w">    </span><span class="m">1</span>
Iron<span class="w"> </span>deficiency<span class="w">    </span><span class="m">1</span>
Hereditary<span class="w"> </span>hemochromatosis-related<span class="w"> </span>traits<span class="w"> </span><span class="o">(</span>HFE<span class="w"> </span>mutation<span class="w"> </span>homozygotes<span class="o">)</span><span class="w"> </span><span class="m">2</span>
</code></pre></div>
</blockquote>
<p>We remove those involved in diabetes and pancreas, leading to the following list:</p>
<blockquote>
<div class="highlight"><pre><span></span><code>Iron<span class="w"> </span>status<span class="w"> </span>biomarkers<span class="w"> </span><span class="o">(</span>ferritin<span class="w"> </span>levels<span class="o">)</span>
Iron<span class="w"> </span>status<span class="w"> </span>biomarkers
Iron<span class="w"> </span>status<span class="w"> </span>biomarkers<span class="w"> </span><span class="o">(</span>total<span class="w"> </span>iron<span class="w"> </span>binding<span class="w"> </span>capacity<span class="o">)</span>
Iron<span class="w"> </span>status<span class="w"> </span>biomarkers<span class="w"> </span><span class="o">(</span>transferrin<span class="w"> </span>saturation<span class="o">)</span>
Iron<span class="w"> </span>status<span class="w"> </span>biomarkers<span class="w"> </span><span class="o">(</span>iron<span class="w"> </span>levels<span class="o">)</span>
Liver<span class="w"> </span>iron<span class="w"> </span>content
Iron<span class="w"> </span>status<span class="w"> </span>biomarkers<span class="w"> </span><span class="o">(</span>transferrin<span class="w"> </span>levels<span class="o">)</span>
Iron<span class="w"> </span>deficiency<span class="w"> </span>anemia
Iron<span class="w"> </span>levels
Iron<span class="w"> </span>deficiency
Hereditary<span class="w"> </span>hemochromatosis-related<span class="w"> </span>traits<span class="w"> </span><span class="o">(</span>HFE<span class="w"> </span>mutation<span class="w"> </span>homozygotes<span class="o">)</span>
</code></pre></div>
</blockquote>
<p>So now let's go back to our working directory and extract only the lines of interest:</p>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span><span class="w"> </span>/work2/project/regenet/results/multi/abc.model/Nasser2021/GWAS_13_initial_genes
<span class="nv">gwas</span><span class="o">=</span><span class="s2">&quot;/work2/project/regenet/results/gwas/gwas.catalog/v1.0.2_r2021-10-02/gwas_catalog_v1.0.2-associations_r2021-10-02.tsv&quot;</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span>-F<span class="w"> </span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;$8~/(Iron status biomarkers)|(Liver iron content)|(Iron deficiency anemia)|(Iron levels)|(Iron deficiency)|(Hereditary hemochromatosis-related traits (HFE mutation homozygotes))/&#39;</span><span class="w"> </span><span class="nv">$gwas</span><span class="w"> </span>&gt;<span class="w"> </span>associations_full.tsv
</code></pre></div>
<p><code>associations_full.tsv</code> contains 280 rows. Remark: we see in the following section that only 269 of those lines are valid.</p>
<h2 id="perform-intersection-will-enhancers-regulating-the-10-initial-genes">Perform intersection (will enhancers regulating the 10 initial genes)<a class="headerlink" href="#perform-intersection-will-enhancers-regulating-the-10-initial-genes" title="Permanent link">&para;</a></h2>
<h3 id="tsv-to-bed">TSV to bed<a class="headerlink" href="#tsv-to-bed" title="Permanent link">&para;</a></h3>
<p>Fields $12 and $13 in <code>associations_full.tsv</code> are <chr id> and <chr pos> of the SNP. The 3 first columns of the bed file shall be the position of the SNP, and in column 4, as feature name, we concatenate all the columns of <code>associations_full.tsv</code>  with a separator like "::", so that the output is strictly in bed format.</p>
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span><span class="s1">&#39;BEGIN{FS=&quot;\t&quot;; OFS=&quot;\t&quot;} {if($13 &amp;&amp; $14){ORS=&quot;\t&quot;; print &quot;chr&quot;$12, $13, $13+1; ORS=&quot;\n&quot;; gsub(/\t/,&quot;::&quot;,$0); print}}&#39;</span><span class="w"> </span>associations_full.tsv<span class="w"> </span><span class="p">|</span>bedtools<span class="w"> </span>sort<span class="w"> </span>-faidx<span class="w"> </span>../chr_sizes<span class="w"> </span>-i<span class="w"> </span>stdin<span class="w"> </span>&gt;<span class="w"> </span>associations_full.bed
</code></pre></div>
<h3 id="liftover">Liftover<a class="headerlink" href="#liftover" title="Permanent link">&para;</a></h3>
<p>The Genome Assembly used in our GWAS catalog is GRCh38.p13 whereas it's GRCh37/hg19 in the Nasser2021 predictions, so we need to start with a liftover.</p>
<p>In order to do this we need several modules, that should contain the keywords <code>kent</code>and <code>utils</code>. Note that the output of <code>module av</code> is actually directed to stderr so we use <code>2&gt;&amp;1</code> to redirect stderr to stdout, then <code>&gt;/dev/null</code> to redirect stdout to <code>/dev/null</code>.</p>
<div class="highlight"><pre><span></span><code><span class="o">(</span>module<span class="w"> </span>av<span class="o">)</span><span class="w"> </span><span class="m">2</span>&gt;<span class="p">&amp;</span><span class="m">1</span><span class="w"> </span>&gt;/dev/null<span class="w"> </span><span class="p">|</span>grep<span class="w"> </span>kent
</code></pre></div>
<blockquote>
<div class="highlight"><pre><span></span><code>bioinfo/kentUtils-302.1.0
bioinfo/kentUtils-v370
</code></pre></div>
</blockquote>
<p>So we can finally load those modules:</p>
<div class="highlight"><pre><span></span><code>module<span class="w"> </span>load<span class="w"> </span>bioinfo/kentUtils-302.1.0<span class="w"> </span>bioinfo/kentUtils-v370
</code></pre></div>
<p>(actually <code>bioinfo/kentUtils-v370</code> should suffice, cf mail from Sarah 07/12/2020, 14:34).</p>
<p>Now we can simply use <code>liftOver</code>:</p>
<blockquote>
<div class="highlight"><pre><span></span><code>liftOver
</code></pre></div>
<div class="highlight"><pre><span></span><code>liftOver - Move annotations from one assembly to another
usage:
   liftOver oldFile map.chain newFile unMapped
oldFile and newFile are in bed format by default, but can be in GFF and
maybe eventually others with the appropriate flags below.
The map.chain file has the old genome as the target and the new genome
as the query.
</code></pre></div>
</blockquote>
<p><code>unlifted.bed</code> file is not an optional argument and will contain all genome positions that cannot be lifted in case it happens for some. Full documentation here: https://genome.sph.umich.edu/wiki/LiftOver</p>
<p>One can find the <code>chain</code>file that goes from hg38 to hg19 here: http://hgdownload.soe.ucsc.edu/goldenPath/hg38/liftOver/. Let's download it:</p>
<div class="highlight"><pre><span></span><code>rsync<span class="w"> </span>-avzP<span class="w"> </span>rsync://hgdownload.cse.ucsc.edu/goldenPath/hg38/liftOver/hg38ToHg19.over.chain.gz<span class="w"> </span>.
</code></pre></div>
<blockquote>
<p>Not this chain file was last modified at the date: 2013-12-31 17:24</p>
</blockquote>
<p>So now we can perform the liftover. <code>enhancers.sorted.bed</code> is already mapped over the hg19 assembly, so all we need to do is to convert <code>associations_full.bed</code> to hg19.</p>
<div class="highlight"><pre><span></span><code>liftOver<span class="w"> </span>-bedPlus<span class="o">=</span><span class="m">1</span><span class="w"> </span>-tab<span class="w"> </span>associations_full.bed<span class="w"> </span>hg38ToHg19.over.chain<span class="w"> </span>associations_full.hg19_liftover.bed<span class="w"> </span>unlifted.bed
</code></pre></div>
<p>OK.</p>
<p>Note that using <code>meld</code> we found that all rows are different between <code>associations_full.bed</code> and <code>associations_full.hg19_liftover.bed</code>.</p>
<h3 id="intersect">Intersect<a class="headerlink" href="#intersect" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code>srun<span class="w"> </span>--mem<span class="o">=</span>32G<span class="w"> </span>--pty<span class="w"> </span>bash
conda<span class="w"> </span>activate<span class="w"> </span>base<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>module<span class="w"> </span>load<span class="w"> </span>bioinfo/bedtools-2.27.1
</code></pre></div>
<div class="highlight"><pre><span></span><code>bedtools<span class="w"> </span>intersect<span class="w"> </span>-wo<span class="w"> </span>-a<span class="w"> </span>enhancers.sorted.bed<span class="w"> </span>-b<span class="w"> </span>associations_full.hg19_liftover.bed<span class="w"> </span>&gt;<span class="w"> </span>enhancers.overlap_GWAS.bedpe
</code></pre></div>
<blockquote>
<div class="highlight"><pre><span></span><code>wc<span class="w"> </span>-l<span class="w"> </span>enhancers.overlap_GWAS.bedpe
</code></pre></div>
<p>4</p>
<div class="highlight"><pre><span></span><code>cat<span class="w"> </span>enhancers.overlap_GWAS.bedpe
</code></pre></div>
<div class="highlight"><pre><span></span><code>chr2<span class="w">  </span><span class="m">190405339</span><span class="w">   </span><span class="m">190407623</span><span class="w">   </span>chr2<span class="w">    </span><span class="m">190407500</span><span class="w">   </span><span class="m">190407501</span><span class="w">   </span><span class="m">2021</span>-08-13::34128465::Liu<span class="w"> </span>Y::2021-06-15::Elife::www.ncbi.nlm.nih.gov/pubmed/34128465::Genetic<span class="w"> </span>architecture<span class="w"> </span>of<span class="w"> </span><span class="m">11</span><span class="w"> </span>organ<span class="w"> </span>traits<span class="w"> </span>derived<span class="w"> </span>from<span class="w"> </span>abdominal<span class="w"> </span>MRI<span class="w"> </span>using<span class="w"> </span>deep<span class="w"> </span>learning.::Liver<span class="w"> </span>iron<span class="w"> </span>content::32,858<span class="w"> </span>European<span class="w"> </span>ancestry<span class="w"> </span>individuals::NA::2q32.2::2::189542774::SLC40A1::KDM3AP1<span class="w"> </span>-<span class="w"> </span>SLC40A1::ENSG00000233996::ENSG00000138449::::54782::17816::rs115380467-?::rs115380467::0::115380467::regulatory_region_variant::1::NR::3E-12::11.522878745280337::<span class="o">(</span>conditional<span class="o">)</span>::0.181::NR<span class="w"> </span>mg/g<span class="w"> </span>increase::Affymetrix<span class="w"> </span><span class="o">[</span><span class="m">9390170</span><span class="o">]</span><span class="w"> </span><span class="o">(</span>imputed<span class="o">)</span>::N::liver<span class="w"> </span>iron<span class="w"> </span>measurement::http://www.ebi.ac.uk/efo/EFO_0010056::GCST90016674::Genome-wide<span class="w"> </span>genotyping<span class="w"> </span>array<span class="w"> </span><span class="m">1</span>
chr3<span class="w">  </span><span class="m">195799723</span><span class="w">   </span><span class="m">195801054</span><span class="w">   </span>chr3<span class="w">    </span><span class="m">195800811</span><span class="w">   </span><span class="m">195800812</span><span class="w">   </span><span class="m">2021</span>-03-22::33536631::Bell<span class="w"> </span>S::2021-02-03::Commun<span class="w"> </span>Biol::www.ncbi.nlm.nih.gov/pubmed/33536631::A<span class="w"> </span>genome-wide<span class="w"> </span>meta-analysis<span class="w"> </span>yields<span class="w"> </span><span class="m">46</span><span class="w"> </span>new<span class="w"> </span>loci<span class="w"> </span>associating<span class="w"> </span>with<span class="w"> </span>biomarkers<span class="w"> </span>of<span class="w"> </span>iron<span class="w"> </span>homeostasis.::Iron<span class="w"> </span>status<span class="w"> </span>biomarkers<span class="w"> </span><span class="o">(</span>total<span class="w"> </span>iron<span class="w"> </span>binding<span class="w"> </span>capacity<span class="o">)</span>::135,430<span class="w"> </span>European<span class="w"> </span>ancestry<span class="w"> </span>individuals::NA::3q29::3::196073940::TFRC::TFRC::::::ENSG00000072274::::::rs3817672-C::rs3817672::0::3817672::missense_variant::0::0.44::6E-11::10.221848749616356::::0.031::<span class="o">[</span><span class="m">0</span>.04-0.022<span class="o">]</span><span class="w"> </span>SD<span class="w"> </span>decrease::Affymetrix,<span class="w"> </span>Illumina<span class="w"> </span><span class="o">[</span><span class="m">40000000</span><span class="o">]</span><span class="w"> </span><span class="o">(</span>imputed<span class="o">)</span>::N::total<span class="w"> </span>iron<span class="w"> </span>binding<span class="w"> </span>capacity::http://www.ebi.ac.uk/efo/EFO_0006334::GCST011368::Genome-wide<span class="w"> </span>genotyping<span class="w"> </span>array<span class="w">  </span><span class="m">1</span>
chr3<span class="w">  </span><span class="m">195799723</span><span class="w">   </span><span class="m">195801054</span><span class="w">   </span>chr3<span class="w">    </span><span class="m">195800811</span><span class="w">   </span><span class="m">195800812</span><span class="w">   </span><span class="m">2021</span>-03-22::33536631::Bell<span class="w"> </span>S::2021-02-03::Commun<span class="w"> </span>Biol::www.ncbi.nlm.nih.gov/pubmed/33536631::A<span class="w"> </span>genome-wide<span class="w"> </span>meta-analysis<span class="w"> </span>yields<span class="w"> </span><span class="m">46</span><span class="w"> </span>new<span class="w"> </span>loci<span class="w"> </span>associating<span class="w"> </span>with<span class="w"> </span>biomarkers<span class="w"> </span>of<span class="w"> </span>iron<span class="w"> </span>homeostasis.::Iron<span class="w"> </span>status<span class="w"> </span>biomarkers<span class="w"> </span><span class="o">(</span>transferrin<span class="w"> </span>saturation<span class="o">)</span>::131,471<span class="w"> </span>European<span class="w"> </span>ancestry<span class="w"> </span>individuals::NA::3q29::3::196073940::TFRC::TFRC::::::ENSG00000072274::::::rs3817672-C::rs3817672::0::3817672::missense_variant::0::0.44::2E-8::7.698970004336019::::0.026::<span class="o">[</span><span class="m">0</span>.017-0.034<span class="o">]</span><span class="w"> </span>SD<span class="w"> </span>increase::Affymetrix,<span class="w"> </span>Illumina<span class="w"> </span><span class="o">[</span><span class="m">40000000</span><span class="o">]</span><span class="w"> </span><span class="o">(</span>imputed<span class="o">)</span>::N::transferrin<span class="w"> </span>saturation<span class="w"> </span>measurement::http://www.ebi.ac.uk/efo/EFO_0006333::GCST011366::Genome-wide<span class="w"> </span>genotyping<span class="w"> </span>array<span class="w"> </span><span class="m">1</span>
chr22<span class="w"> </span><span class="m">37505175</span><span class="w">    </span><span class="m">37505880</span><span class="w">    </span>chr22<span class="w">   </span><span class="m">37505552</span><span class="w">    </span><span class="m">37505553</span><span class="w">    </span><span class="m">2015</span>-06-14::25352340::Benyamin<span class="w"> </span>B::2014-10-29::Nat<span class="w"> </span>Commun::www.ncbi.nlm.nih.gov/pubmed/25352340::Novel<span class="w"> </span>loci<span class="w"> </span>affecting<span class="w"> </span>iron<span class="w"> </span>homeostasis<span class="w"> </span>and<span class="w"> </span>their<span class="w"> </span>effects<span class="w"> </span><span class="k">in</span><span class="w"> </span>individuals<span class="w"> </span>at<span class="w"> </span>risk<span class="w"> </span><span class="k">for</span><span class="w"> </span>hemochromatosis.::Iron<span class="w"> </span>status<span class="w"> </span>biomarkers<span class="w"> </span><span class="o">(</span>iron<span class="w"> </span>levels<span class="o">)</span>::23,986<span class="w"> </span>European<span class="w"> </span>ancestry<span class="w"> </span>individuals::Up<span class="w"> </span>to<span class="w"> </span><span class="m">24</span>,986<span class="w"> </span>European<span class="w"> </span>ancestry<span class="w"> </span>individuals::22q12.3::22::37109512::TMPRSS6::TMPRSS6::::::ENSG00000187045::::::rs228916-T::rs228916::0::228916::5_prime_UTR_variant::0::0.875::3E-8::7.522878745280337::::0.086::<span class="o">[</span><span class="m">0</span>.055-0.117<span class="o">]</span><span class="w"> </span>unit<span class="w"> </span>decrease::Affymetrix,<span class="w"> </span>Illumina<span class="w"> </span><span class="o">[</span>~<span class="w"> </span><span class="m">2100000</span><span class="o">]</span><span class="w"> </span><span class="o">(</span>imputed<span class="o">)</span>::N::iron<span class="w"> </span>biomarker<span class="w"> </span>measurement,<span class="w"> </span>serum<span class="w"> </span>iron<span class="w"> </span>measurement::http://www.ebi.ac.uk/efo/EFO_0004461,<span class="w"> </span>http://www.ebi.ac.uk/efo/EFO_0006332::GCST002679::Genome-wide<span class="w"> </span>genotyping<span class="w"> </span>array<span class="w">   </span><span class="m">1</span>
</code></pre></div>
</blockquote>
<p>Well, there are only 4 overlap, involving only 3 unique enhancers. This is expected, as we are searching for overlap between a list of only 137 enhancer over the whole genome, and a list of only 269 SNP over the whole genome.</p>
<h2 id="perform-intersection-with-enhancers-regulating-the-457-original-newly-found-genes">Perform intersection (with enhancers regulating the 457 original + newly found genes)<a class="headerlink" href="#perform-intersection-with-enhancers-regulating-the-457-original-newly-found-genes" title="Permanent link">&para;</a></h2>
<div class="highlight"><pre><span></span><code>srun<span class="w"> </span>--mem<span class="o">=</span>32G<span class="w"> </span>--pty<span class="w"> </span>bash
conda<span class="w"> </span>activate<span class="w"> </span>base<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>module<span class="w"> </span>load<span class="w"> </span>bioinfo/bedtools-2.27.1
</code></pre></div>
<div class="highlight"><pre><span></span><code>bedtools<span class="w"> </span>intersect<span class="w"> </span>-wo<span class="w"> </span>-a<span class="w"> </span>enhancers.new_genes.sorted.bed<span class="w"> </span>-b<span class="w"> </span>associations_full.hg19_liftover.bed<span class="w"> </span>&gt;<span class="w"> </span>enhancers.new_genes.overlap_GWAS.bedpe
</code></pre></div>
<p>There is only one more overlap, namely:</p>
<blockquote>
<div class="highlight"><pre><span></span><code>chr6<span class="w">  </span><span class="m">26103318</span><span class="w">    </span><span class="m">26105346</span><span class="w">    </span>chr6<span class="w">    </span><span class="m">26104632</span><span class="w">    </span><span class="m">26104633</span><span class="w">    </span><span class="m">2021</span>-08-13::34128465::Liu<span class="w"> </span>Y::2021-06-15::Elife::www.ncbi.nlm.nih.gov/pubmed/34128465::Genetic<span class="w"> </span>architecture<span class="w"> </span>of<span class="w"> </span><span class="m">11</span><span class="w"> </span>organ<span class="w"> </span>traits<span class="w"> </span>derived<span class="w"> </span>from<span class="w"> </span>abdominal<span class="w"> </span>MRI<span class="w"> </span>using<span class="w"> </span>deep<span class="w"> </span>learning.::Liver<span class="w"> </span>iron<span class="w"> </span>content::32,858<span class="w"> </span>European<span class="w"> </span>ancestry<span class="w"> </span>individuals::NA::6p22.2::6::26104404::HIST1H4C::H4C3<span class="w"> </span>-<span class="w"> </span>H1-6::ENSG00000197061::ENSG00000187475::::67::3008::rs198851-?::rs198851::0::198851::TF_binding_site_variant::1::NR::3E-73::72.52287874528034::<span class="o">(</span>conditional<span class="o">)</span>::0.194::NR<span class="w"> </span>mg/g<span class="w"> </span>decrease::Affymetrix<span class="w"> </span><span class="o">[</span><span class="m">9390170</span><span class="o">]</span><span class="w"> </span><span class="o">(</span>imputed<span class="o">)</span>::N::liver<span class="w"> </span>iron<span class="w"> </span>measurement::http://www.ebi.ac.uk/efo/EFO_0010056::GCST90016674::Genome-wide<span class="w"> </span>genotyping<span class="w"> </span>array<span class="w">   </span><span class="m">1</span>
</code></pre></div>
</blockquote>
<h2 id="are-the-genes-regulated-by-the-enhancers-found-to-intersect-a-gwas-snp-the-same-as-those-given-in-gwas-as-associated-to-the-snp">Are the genes regulated by the enhancers found to intersect a GWAS SNP, the same as those given in GWAS as associated to the SNP?<a class="headerlink" href="#are-the-genes-regulated-by-the-enhancers-found-to-intersect-a-gwas-snp-the-same-as-those-given-in-gwas-as-associated-to-the-snp" title="Permanent link">&para;</a></h2>
<p>We did this analysis starting from 10 instead of 13 genes, but as it conduces to only 1 more overlap.</p>





                
              </article>
            </div>
          
          
        </div>
        
          <a href="#" class="md-top md-icon" data-md-component="top" hidden>
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg>
            Back to top
          </a>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      2020 - 2022 Tristan Hoellinger
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
        <div class="md-social">
  
    
    
    
    
      
      
    
    <a href="https://twitter.com/TristanHoellin1" target="_blank" rel="noopener" title="twitter.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"/></svg>
    </a>
  
    
    
    
    
      
      
    
    <a href="https://fr.linkedin.com/in/tristan-hoellinger" target="_blank" rel="noopener" title="fr.linkedin.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"/></svg>
    </a>
  
    
    
    
    
    <a href="mailto:sarah.djebali@inserm.fr" target="_blank" rel="noopener" title="" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M498.1 5.6c10.1 7 15.4 19.1 13.5 31.2l-64 416c-1.5 9.7-7.4 18.2-16 23s-18.9 5.4-28 1.6L284 427.7l-68.5 74.1c-8.9 9.7-22.9 12.9-35.2 8.1S160 493.2 160 480v-83.6c0-4 1.5-7.8 4.2-10.7l167.6-182.9c5.8-6.3 5.6-16-.4-22s-15.7-6.4-22-.7L106 360.8l-88.3-44.2C7.1 311.3.3 300.7 0 288.9s5.9-22.8 16.1-28.7l448-256c10.7-6.1 23.9-5.5 34 1.4z"/></svg>
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    <script id="__config" type="application/json">{"base": "../..", "features": ["navigation.instant", "navigation.tabs", "navigation.sections", "navigation.top"], "search": "../../assets/javascripts/workers/search.db81ec45.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../../assets/javascripts/bundle.6df46069.min.js"></script>
      
    
  </body>
</html>